{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 45,
   "metadata": {},
   "outputs": [],
   "source": [
    "import os\n",
    "import re\n",
    "import gzip\n",
    "import pandas as pd\n",
    "from collections import Counter\n",
    "from Bio import SeqIO\n",
    "import math\n",
    "import requests"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 46,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Downloaded assembly report for Escherichia coli K-12\n",
      "Downloaded assembly report for Halobacterium salinarum NRC-1\n",
      "Downloaded assembly report for Plasmodium falciparum\n",
      "Downloaded assembly report for Arabidopsis thaliana\n",
      "Downloaded assembly report for Saccharomyces cerevisiae\n"
     ]
    }
   ],
   "source": [
    "# directory for downloaded assembly reports and sequences\n",
    "assembly_reports_dir = \"assembly_reports\"\n",
    "genome_sequences_dir = \"genome_sequences\"\n",
    "combined_assembly_summary_path = \"combined_assembly_summary.txt\"\n",
    "\n",
    "os.makedirs(assembly_reports_dir, exist_ok=True)\n",
    "os.makedirs(genome_sequences_dir, exist_ok=True)\n",
    "\n",
    "# Define organisms and their assembly report URLs\n",
    "organisms = {\n",
    "    \"Escherichia coli K-12\": \"https://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/000/005/845/GCF_000005845.2_ASM584v2/GCF_000005845.2_ASM584v2_assembly_report.txt\",\n",
    "    \"Halobacterium salinarum NRC-1\": \"https://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/000/006/805/GCF_000006805.1_ASM680v1/GCF_000006805.1_ASM680v1_assembly_report.txt\",\n",
    "    \"Plasmodium falciparum\": \"https://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/000/002/765/GCF_000002765.6_GCA_000002765/GCF_000002765.6_GCA_000002765_assembly_report.txt\",\n",
    "    \"Arabidopsis thaliana\": \"https://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/000/001/735/GCF_000001735.4_TAIR10.1/GCF_000001735.4_TAIR10.1_assembly_report.txt\",\n",
    "    \"Saccharomyces cerevisiae\": \"https://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/000/146/045/GCF_000146045.2_R64/GCF_000146045.2_R64_assembly_report.txt\"\n",
    "}\n",
    "\n",
    "def download_assembly_report(organism_name, url):\n",
    "    response = requests.get(url)\n",
    "    if response.status_code == 200:\n",
    "        file_path = os.path.join(assembly_reports_dir, f\"{organism_name.replace(' ', '_')}_assembly_report.txt\")\n",
    "        with open(file_path, 'wb') as file:\n",
    "            file.write(response.content)\n",
    "        print(f\"Downloaded assembly report for {organism_name}\")\n",
    "    else:\n",
    "        print(f\"Failed to download assembly report for {organism_name}. Status code: {response.status_code}\")\n",
    "        print(f\"URL: {url}\")\n",
    "\n",
    "for organism_name, url in organisms.items():\n",
    "    download_assembly_report(organism_name, url)\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 47,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "excel file created: updated_assembly_summary_w_lengths_and_entropy.xlsx\n"
     ]
    }
   ],
   "source": [
    "# define cols for the assembly summary\n",
    "columns = [\n",
    "    'assembly_accession', 'bioproject', 'biosample', 'wgs_master',\n",
    "    'refseq_category', 'taxid', 'species_taxid', 'organism_name',\n",
    "    'infraspecific_name', 'isolate', 'version_status', 'assembly_level',\n",
    "    'release_type', 'genome_rep', 'seq_rel_date', 'asm_name', 'asm_submitter',\n",
    "    'gbrs_paired_asm', 'paired_asm_comp', 'ftp_path', 'excluded_from_refseq',\n",
    "    'relation_to_type_material', 'asm_not_live_date', 'assembly_type', 'group',\n",
    "    'genome_size', 'genome_size_ungapped', 'gc_percent', 'replicon_count',\n",
    "    'scaffold_count', 'contig_count', 'annotation_provider', 'annotation_name',\n",
    "    'annotation_date', 'total_gene_count', 'protein_coding_gene_count',\n",
    "    'non_coding_gene_count', 'pubmed_id', 'length', 'Shannon'\n",
    "]\n",
    "\n",
    "data = [\n",
    "    {\n",
    "        'assembly_accession': 'GCF_000005845.2', 'bioproject': 'PRJNA225',\n",
    "        'biosample': 'SAMN02604091', 'wgs_master': 'na',\n",
    "        'refseq_category': 'Reference Genome', 'taxid': 511145,\n",
    "        'species_taxid': 511145, 'organism_name': 'Escherichia coli K-12',\n",
    "        'infraspecific_name': 'strain=K-12 substr. MG1655', 'isolate': 'na',\n",
    "        'version_status': 'latest', 'assembly_level': 'Complete Genome',\n",
    "        'release_type': 'Major', 'genome_rep': 'Full',\n",
    "        'seq_rel_date': '2013/09/26', 'asm_name': 'ASM584v2',\n",
    "        'asm_submitter': 'Univ. Wisconsin', 'gbrs_paired_asm': 'GCA_000005845.2',\n",
    "        'paired_asm_comp': 'identical',\n",
    "        'ftp_path': 'https://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/000/005/845/GCF_000005845.2_ASM584v2',\n",
    "        'excluded_from_refseq': 'na', 'relation_to_type_material': 'na',\n",
    "        'asm_not_live_date': 'na', 'assembly_type': 'haploid', 'group': 'bacteria',\n",
    "        'genome_size': 4641652, 'genome_size_ungapped': 4641652,\n",
    "        'gc_percent': 50.79, 'replicon_count': 1, 'scaffold_count': 1,\n",
    "        'contig_count': 1, 'annotation_provider': 'NCBI RefSeq',\n",
    "        'annotation_name': 'NCBI Prokaryotic Genome Annotation Pipeline (PGAP)',\n",
    "        'annotation_date': '2023/04/26', 'total_gene_count': 4200,\n",
    "        'protein_coding_gene_count': 4000, 'non_coding_gene_count': 200,\n",
    "        'pubmed_id': 'na', 'length': 4641652, 'Shannon': 1.9998190013576105\n",
    "    },\n",
    "    {\n",
    "        'assembly_accession': 'GCF_000001735.4', 'bioproject': 'PRJNA10719',\n",
    "        'biosample': 'SAMN03081427', 'wgs_master': 'na',\n",
    "        'refseq_category': 'Reference Genome', 'taxid': 3702,\n",
    "        'species_taxid': 3702, 'organism_name': 'Arabidopsis thaliana',\n",
    "        'infraspecific_name': 'ecotype=Columbia', 'isolate': 'na',\n",
    "        'version_status': 'latest', 'assembly_level': 'Chromosome',\n",
    "        'release_type': 'Minor', 'genome_rep': 'Full',\n",
    "        'seq_rel_date': '2018/03/15', 'asm_name': 'TAIR10.1',\n",
    "        'asm_submitter': 'The Arabidopsis Information Resource (TAIR)', 'gbrs_paired_asm': 'GCA_000001735.2',\n",
    "        'paired_asm_comp': 'identical',\n",
    "        'ftp_path': 'https://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/000/001/735/GCF_000001735.4_TAIR10.1',\n",
    "        'excluded_from_refseq': 'na', 'relation_to_type_material': 'na',\n",
    "        'asm_not_live_date': 'na', 'assembly_type': 'haploid', 'group': 'plants',\n",
    "        'genome_size': 135650600, 'genome_size_ungapped': 135650600,\n",
    "        'gc_percent': 36.1, 'replicon_count': 5, 'scaffold_count': 5,\n",
    "        'contig_count': 7, 'annotation_provider': 'TAIR',\n",
    "        'annotation_name': 'TAIR10.1', 'annotation_date': '2018/03/15',\n",
    "        'total_gene_count': 27415, 'protein_coding_gene_count': 27415,\n",
    "        'non_coding_gene_count': 'na', 'pubmed_id': 'na', 'length': 30263312, 'Shannon': 1.9416279304861543\n",
    "    },\n",
    "    {\n",
    "        'assembly_accession': 'GCF_000006805.1', 'bioproject': 'PRJNA412908',\n",
    "        'biosample': 'SAMN02604216', 'wgs_master': 'na',\n",
    "        'refseq_category': 'Reference Genome', 'taxid': 64091,\n",
    "        'species_taxid': 64091, 'organism_name': 'Halobacterium salinarum NRC-1',\n",
    "        'infraspecific_name': 'strain=NRC-1; ATCC 700922', 'isolate': 'na',\n",
    "        'version_status': 'latest', 'assembly_level': 'Complete Genome',\n",
    "        'release_type': 'Major', 'genome_rep': 'Full',\n",
    "        'seq_rel_date': '2001/01/09', 'asm_name': 'ASM680v1',\n",
    "        'asm_submitter': 'University of Massachusetts-Amherst, University of Washington', 'gbrs_paired_asm': 'GCA_000006805.1',\n",
    "        'paired_asm_comp': 'identical',\n",
    "        'ftp_path': 'https://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/000/006/805/GCF_000006805.1_ASM680v1',\n",
    "        'excluded_from_refseq': 'na', 'relation_to_type_material': 'na',\n",
    "        'asm_not_live_date': 'na', 'assembly_type': 'haploid', 'group': 'archaea',\n",
    "        'genome_size': 2014239, 'genome_size_ungapped': 2014239,\n",
    "        'gc_percent': 68.0, 'replicon_count': 3, 'scaffold_count': 3,\n",
    "        'contig_count': 3, 'annotation_provider': 'NCBI RefSeq',\n",
    "        'annotation_name': 'NCBI Prokaryotic Genome Annotation Pipeline (PGAP)',\n",
    "        'annotation_date': '2001/01/09', 'total_gene_count': 2700,\n",
    "        'protein_coding_gene_count': 2600, 'non_coding_gene_count': 100,\n",
    "        'pubmed_id': 'na', 'length': 2014239, 'Shannon': 1.905323387374254\n",
    "    },\n",
    "    {\n",
    "    'assembly_accession': 'GCF_000002765.6', 'bioproject': 'PRJNA13173',\n",
    "    'biosample': 'SAMN00102897', 'wgs_master': 'na',\n",
    "    'refseq_category': 'Representative Genome', 'taxid': 36329,\n",
    "    'species_taxid': 36329, 'organism_name': 'Plasmodium falciparum 3D7',\n",
    "    'infraspecific_name': 'Isolate=3D7', 'isolate': '3D7',\n",
    "    'version_status': 'latest', 'assembly_level': 'Complete Genome',\n",
    "    'release_type': 'Minor', 'genome_rep': 'Full',\n",
    "    'seq_rel_date': '2016/04/07', 'asm_name': 'GCA_000002765',\n",
    "    'asm_submitter': 'Plasmodium falciparum Genome Sequencing Consortium',\n",
    "    'gbrs_paired_asm': 'GCA_000002765.3', 'paired_asm_comp': 'identical',\n",
    "    'ftp_path': 'https://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/000/002/765/GCF_000002765.6_GCA_000002765',\n",
    "    'excluded_from_refseq': 'na', 'relation_to_type_material': 'na',\n",
    "    'asm_not_live_date': 'na', 'assembly_type': 'haploid', 'group': 'protozoa',\n",
    "    'genome_size': 23313821, 'genome_size_ungapped': 23313821,\n",
    "    'gc_percent': 19.4, 'replicon_count': 14, 'scaffold_count': 14,\n",
    "    'contig_count': 14, 'annotation_provider': 'Plasmodium falciparum Genome Sequencing Consortium',\n",
    "    'annotation_name': 'GCA_000002765', 'annotation_date': '2016/04/07',\n",
    "    'total_gene_count': 5400, 'protein_coding_gene_count': 5200, 'non_coding_gene_count': 200,\n",
    "    'pubmed_id': 'na', 'length': 2925236, 'Shannon': 1.700437913566233\n",
    "},\n",
    "{\n",
    "    'assembly_accession': 'GCF_000146045.2', 'bioproject': 'PRJNA43747',\n",
    "    'biosample': 'na', 'wgs_master': 'na',\n",
    "    'refseq_category': 'Reference Genome', 'taxid': 559292,\n",
    "    'species_taxid': 559292, 'organism_name': 'Saccharomyces cerevisiae S288C',\n",
    "    'infraspecific_name': 'strain=S288C', 'isolate': 'na',\n",
    "    'version_status': 'latest', 'assembly_level': 'Complete Genome',\n",
    "    'release_type': 'Major', 'genome_rep': 'Full',\n",
    "    'seq_rel_date': '2014/12/17', 'asm_name': 'R64',\n",
    "    'asm_submitter': 'Saccharomyces Genome Database', 'gbrs_paired_asm': 'GCA_000146045.2',\n",
    "    'paired_asm_comp': 'identical',\n",
    "    'ftp_path': 'https://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/000/146/045/GCF_000146045.2_R64',\n",
    "    'excluded_from_refseq': 'na', 'relation_to_type_material': 'na',\n",
    "    'asm_not_live_date': 'na', 'assembly_type': 'haploid', 'group': 'fungi',\n",
    "    'genome_size': 12157105, 'genome_size_ungapped': 12157105,\n",
    "    'gc_percent': 38.3, 'replicon_count': 17, 'scaffold_count': 17,\n",
    "    'contig_count': 17, 'annotation_provider': 'NCBI RefSeq',\n",
    "    'annotation_name': 'NCBI Prokaryotic Genome Annotation Pipeline (PGAP)',\n",
    "    'annotation_date': '2014/12/17', 'total_gene_count': 6700,\n",
    "    'protein_coding_gene_count': 6600, 'non_coding_gene_count': 100,\n",
    "    'pubmed_id': 'na', 'length': 230218, 'Shannon': 1.9664766384439507\n",
    "}\n",
    "]\n",
    "\n",
    "df = pd.DataFrame(data, columns = columns)\n",
    "\n",
    "output_file_path = 'updated_assembly_summary_w_lengths_and_entropy.xlsx'\n",
    "df.to_excel(output_file_path, index=False)\n",
    "\n",
    "print(f\"excel file created: {output_file_path}\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 48,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Cleaned sequence for Escherichia coli K-12 already exists. Skipping extraction.\n",
      "Cleaned sequence for Halobacterium salinarum NRC-1 already exists. Skipping extraction.\n",
      "Cleaned sequence for Plasmodium falciparum already exists. Skipping extraction.\n",
      "Cleaned sequence for Arabidopsis thaliana already exists. Skipping extraction.\n",
      "Cleaned sequence for Saccharomyces cerevisiae already exists. Skipping extraction.\n"
     ]
    }
   ],
   "source": [
    "\n",
    "gcf_files_path = \"/Users/tiananoll-walker/Documents/biotokens/GCF_files\"\n",
    "genome_sequences_dir = \"genome_sequences\"\n",
    "if not os.path.exists(genome_sequences_dir):\n",
    "    os.makedirs(genome_sequences_dir)\n",
    "\n",
    "# define organisms and their corresponding GCF files\n",
    "organisms = {\n",
    "    \"Escherichia coli K-12\": \"GCF_000005845.2_ASM584v2_genomic.fna.gz\",\n",
    "    \"Halobacterium salinarum NRC-1\": \"GCF_000006805.1_ASM680v1_genomic.fna.gz\",\n",
    "    \"Plasmodium falciparum\": \"GCF_000002765.6_GCA_000002765_genomic.fna.gz\",\n",
    "    \"Arabidopsis thaliana\": \"GCF_000001735.4_TAIR10.1_genomic.fna.gz\",\n",
    "    \"Saccharomyces cerevisiae\": \"GCF_000146045.2_R64_genomic.fna.gz\"\n",
    "}\n",
    "\n",
    "def extract_and_clean_sequence(file_path, organism_name):\n",
    "    base_filename = f\"{organism_name.replace(' ', '_')}\"\n",
    "    cleaned_filename = f\"{genome_sequences_dir}/{base_filename}_cleaned.txt\"\n",
    "    \n",
    "    if os.path.exists(cleaned_filename):\n",
    "        print(f\"cleaned sequence for {organism_name} already exists. Skipping extraction.\")\n",
    "        return cleaned_filename\n",
    "    \n",
    "    with gzip.open(file_path, 'rt') as infile:\n",
    "        lines = infile.readlines()\n",
    "    \n",
    "    sequence = ''.join([line.strip() for line in lines if not line.startswith(\">\")])\n",
    "    cleaned_sequence = re.sub(r'[^ACGTacgt]', '', sequence).upper()\n",
    "    \n",
    "    with open(cleaned_filename, 'w') as outfile:\n",
    "        outfile.write(cleaned_sequence)\n",
    "    \n",
    "    print(f\"Cleaned sequence saved to {cleaned_filename}\")\n",
    "    return cleaned_filename\n",
    "\n",
    "for organism_name, gcf_filename in organisms.items():\n",
    "    file_path = os.path.join(gcf_files_path, gcf_filename)\n",
    "    extract_and_clean_sequence(file_path, organism_name)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 49,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Plasmodium_falciparum_cleaned.txt: Length = 2925236, Shannon Entropy = 1.700437913566233\n",
      "Arabidopsis_thaliana_cleaned.txt: Length = 119482427, Shannon Entropy = 1.9431467139424183\n",
      "Escherichia_coli_K-12_cleaned.txt: Length = 4641652, Shannon Entropy = 1.9998190013576105\n",
      "Saccharomyces_cerevisiae_cleaned.txt: Length = 230218, Shannon Entropy = 1.9664766384439507\n",
      "Halobacterium_salinarum_NRC-1_cleaned.txt: Length = 2014239, Shannon Entropy = 1.905323387374254\n",
      "Results saved to genome_sequences_lengths_and_entropy.xlsx\n"
     ]
    }
   ],
   "source": [
    "\n",
    "def shannon_entropy(s):\n",
    "    frequency = Counter(s)\n",
    "    probabilities = [freq / len(s) for freq in frequency.values()]\n",
    "    entropy = -sum(p * math.log2(p) for p in probabilities)\n",
    "    return entropy\n",
    "\n",
    "lengths = []\n",
    "entropies = []\n",
    "\n",
    "# calculate length and entropy for cleaned sequences\n",
    "for filename in os.listdir(genome_sequences_dir):\n",
    "    if filename.endswith(\"_cleaned.txt\"):\n",
    "        filepath = os.path.join(genome_sequences_dir, filename)\n",
    "        with open(filepath, 'r') as f:\n",
    "            sequence = f.read()\n",
    "        genome_length = len(sequence)\n",
    "        genome_entropy = shannon_entropy(sequence)\n",
    "        lengths.append(genome_length)\n",
    "        entropies.append(genome_entropy)\n",
    "        print(f'{filename}: Length = {genome_length}, Shannon Entropy = {genome_entropy}')\n",
    "\n",
    "results_df = pd.DataFrame({\n",
    "    \"Filename\": [filename.replace(\"_cleaned.txt\", \"\") for filename in os.listdir(genome_sequences_dir) if filename.endswith(\"_cleaned.txt\")],\n",
    "    \"Length\": lengths,\n",
    "    \"Shannon Entropy\": entropies\n",
    "})\n",
    "\n",
    "results_filename = \"genome_sequences_lengths_and_entropy.xlsx\"\n",
    "results_df.to_excel(results_filename, index=False)\n",
    "print(f\"results saved to {results_filename}\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "base",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.11.7"
  },
  "orig_nbformat": 4
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
