{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Compressing biotypes from Ensembl\n",
    "# You don't need to run this notebook! It takes a while and the output is always the same (in csv files)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import os\n",
    "import concurrent.futures\n",
    "import threading\n",
    "import csv\n",
    "import queue\n",
    "import gzip\n",
    "import shutil\n",
    "import ftplib\n",
    "from ftplib import FTP\n",
    "from Bio import SeqIO\n",
    "import subprocess\n",
    "from typing import List, Dict, Optional\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "def find_fasta_file(ftp_server, ftp_directory, file_suffix):\n",
    "    \"\"\"\n",
    "    Connects to an FTP server, navigates to the specified directory, and finds the correct FASTA file based on suffix.\n",
    "    \"\"\"\n",
    "    with FTP(ftp_server) as ftp:\n",
    "        ftp.login()  # Login anonymously\n",
    "        ftp.cwd(ftp_directory)\n",
    "        files = ftp.nlst()  # List all files in the directory\n",
    "        \n",
    "        # Find the file that ends with the specified suffix\n",
    "        for file in files:\n",
    "            if file.endswith(file_suffix):\n",
    "                return file\n",
    "        raise FileNotFoundError(f\"No file ending with '{file_suffix}' found in the directory.\")\n",
    "\n",
    "def find_download_and_decompress_from_ftp(\n",
    "        ftp_server: str, \n",
    "        ftp_directory: str, \n",
    "        file_suffix: str):\n",
    "    with FTP(ftp_server) as ftp:\n",
    "        ftp.login()  # Login anonymously\n",
    "        ftp.cwd(ftp_directory)\n",
    "        files = ftp.nlst()  # List all files in the directory\n",
    "        \n",
    "        # Find the file that ends with the specified suffix\n",
    "        for file in files:\n",
    "            if file.endswith(file_suffix):\n",
    "                local_filename = file\n",
    "                with open(local_filename, 'wb') as local_file:\n",
    "                    ftp.retrbinary(f\"RETR {file}\", local_file.write)\n",
    "                decompressed_file = local_filename.rstrip('.gz')\n",
    "                with gzip.open(local_filename, 'rb') as f_in:\n",
    "                    with open(decompressed_file, 'wb') as f_out:\n",
    "                        shutil.copyfileobj(f_in, f_out)\n",
    "                os.remove(local_filename)  # Optionally remove the gzipped file\n",
    "                return decompressed_file\n",
    "        raise FileNotFoundError(f\"No file ending with '{file_suffix}' found in the directory.\")\n",
    "\n",
    "def decompress_gz_file(gz_file):\n",
    "    \"\"\"\n",
    "    Decompresses a .gz file.\n",
    "    \"\"\"\n",
    "    decompressed_file = gz_file.rstrip('.gz')\n",
    "    with gzip.open(gz_file, 'rb') as f_in:\n",
    "        with open(decompressed_file, 'wb') as f_out:\n",
    "            shutil.copyfileobj(f_in, f_out)\n",
    "    os.remove(gz_file)  # Optionally remove the gzipped file\n",
    "    return decompressed_file\n",
    "\n",
    "def calculate_compression_ratio(fasta_file):\n",
    "    \"\"\"\n",
    "    Compresses the text file and calculates the compression ratio.\n",
    "    \"\"\"\n",
    "    \n",
    "    compressed_file_name = fasta_file+'.co'\n",
    "    uncompressed_file_name = fasta_file+'.de'\n",
    "    gz_file_name = fasta_file+'.gz'\n",
    "    # Compress the fasta file\n",
    "\n",
    "    # print(f'GECO3 Compressing fasta file {fasta_file}')\n",
    "    command = f'/Users/celia/VSCode/dnacomp/geco3/src/GeCo3 {fasta_file}'\n",
    "    subprocess.run(command, shell=True, stdout=subprocess.DEVNULL, stderr=subprocess.DEVNULL)\n",
    "    compressed_size = os.path.getsize(compressed_file_name)\n",
    "\n",
    "    # print(f'Uncompressing compressed fasta file {compressed_file_name}')\n",
    "    # uncompress the compressed file\n",
    "    command = f'/Users/celia/VSCode/dnacomp/geco3/src/GeDe3 {compressed_file_name}'\n",
    "    subprocess.run(command, shell=True, stdout=subprocess.DEVNULL, stderr=subprocess.DEVNULL)\n",
    "    original_size = os.path.getsize(uncompressed_file_name)\n",
    "    tot_len = 0\n",
    "    with open(uncompressed_file_name, 'r') as file:\n",
    "        content = file.read()  # Read the entire content of the file\n",
    "        tot_len = len(content)\n",
    "\n",
    "    # print(f'GZIP Compressing fasta file {fasta_file}')\n",
    "    # Calculate compression ratio\n",
    "    ratio = (compressed_size / original_size) * 100\n",
    "    # print(f'GECO3 compression ratio: {ratio}')\n",
    "\n",
    "    # Calculate gz compression of decompressed file\n",
    "    with open(uncompressed_file_name, 'rb') as f_in:  # Use 'rb' to read in binary mode\n",
    "        with gzip.open(gz_file_name, 'wb') as f_out:\n",
    "            f_out.writelines(f_in)\n",
    "    compressed_size_gz = os.path.getsize(gz_file_name)\n",
    "    ratio_gz = (compressed_size_gz / original_size) * 100\n",
    "    # print(f'GZIP compression ratio: {ratio_gz}')\n",
    "\n",
    "    os.remove(compressed_file_name)\n",
    "    os.remove(uncompressed_file_name)\n",
    "    os.remove(gz_file_name)\n",
    "    \n",
    "    return ratio, ratio_gz, tot_len\n",
    "\n",
    "def list_organisms(ftp_server: str, ftp_directory: str, starting_organism: Optional[str] = None) -> List[str]:\n",
    "    \"\"\"\n",
    "    Connects to an FTP server, navigates to the specified directory, and lists all subdirectories (organisms).\n",
    "    Optionally starts the list from a given organism.\n",
    "\n",
    "    Parameters:\n",
    "    - ftp_server: The FTP server address.\n",
    "    - ftp_directory: The directory on the FTP server to list organisms from.\n",
    "    - starting_organism: An optional organism name to start the list from.\n",
    "\n",
    "    Returns:\n",
    "    - A list of organism names, ordered starting from the specified organism if provided.\n",
    "    \"\"\"\n",
    "    with FTP(ftp_server) as ftp:\n",
    "        ftp.login()  # Login anonymously\n",
    "        ftp.cwd(ftp_directory)\n",
    "        organisms = sorted(ftp.nlst())  # List and sort all directories in the specified directory\n",
    "\n",
    "    if starting_organism and starting_organism in organisms:\n",
    "        start_index = organisms.index(starting_organism)\n",
    "        return organisms[start_index:]\n",
    "\n",
    "    return organisms\n",
    "\n",
    "def q_write_result_to_csv(q: queue.Queue[Optional[Dict[str, int]]], fieldnames: List, output_csv_file: str) -> None:\n",
    "    \"\"\"\n",
    "    Function to write results from a queue to a CSV file.\n",
    "\n",
    "    Parameters:\n",
    "    - q: A thread-safe queue containing dictionaries to write to the CSV file.\n",
    "    - output_csv_file: The path to the CSV file where results should be written.\n",
    "\n",
    "    Returns:\n",
    "    - None\n",
    "    \"\"\"\n",
    "    with open(output_csv_file, 'a', newline='') as csvfile:\n",
    "        writer = csv.DictWriter(csvfile, fieldnames=fieldnames)\n",
    "        \n",
    "        # Write the header only once\n",
    "        if csvfile.tell() == 0:\n",
    "            writer.writeheader()\n",
    "        \n",
    "        while True:\n",
    "            result = q.get()\n",
    "            if result is None:  # Stop signal\n",
    "                break\n",
    "            writer.writerow(result)\n",
    "            csvfile.flush()\n",
    "            q.task_done()\n",
    "\n",
    "def worker(organism: str, ftp_server: str, base_ftp_directory: str, q: queue.Queue[Optional[Dict[str, int]]]) -> None:\n",
    "    result = process_species(organism, ftp_server, base_ftp_directory)\n",
    "    q.put(result)\n",
    "\n",
    "def process_all_organisms(\n",
    "        ftp_server: str,\n",
    "        base_ftp_directory: str,\n",
    "        output_csv_file: str,\n",
    "        starting_organism: Optional[str] = None):\n",
    "    \"\"\"\n",
    "    Processes all organisms in the specified base FTP directory.\n",
    "    \"\"\"\n",
    "    organisms = list_organisms(ftp_server, base_ftp_directory, starting_organism)\n",
    "    fieldnames = [\n",
    "        \"species_name\", \"cds_num_sequences\", \"cds_tot_len\", \"cds_compression_ratio_geco3\", \n",
    "        \"cds_compression_ratio_gz\", \"ncrna_num_sequences\", \"ncrna_tot_len\", \n",
    "        \"ncrna_compression_ratio_geco3\", \"ncrna_compression_ratio_gz\"\n",
    "    ]\n",
    "    q = queue.Queue()\n",
    "\n",
    "    # Start the CSV writer thread\n",
    "    writer_thread = threading.Thread(target=q_write_result_to_csv, args=(q, fieldnames, output_csv_file))\n",
    "    writer_thread.start()\n",
    "\n",
    "    # Assuming organisms is a list of organism names\n",
    "    with concurrent.futures.ThreadPoolExecutor(max_workers=8) as executor:\n",
    "        futures = {executor.submit(worker, organism, ftp_server, base_ftp_directory, q): organism for organism in organisms}\n",
    "        \n",
    "        # Wait for all the workers to complete\n",
    "        for future in concurrent.futures.as_completed(futures):\n",
    "            organism_name = futures[future]  # Get the organism name from the dictionary\n",
    "            try:\n",
    "                future.result()  # This raises any exception encountered during execution\n",
    "                print(f'{organism_name} completed successfully!')\n",
    "            except Exception as exc:\n",
    "                print(f'{organism_name} generated an exception: {exc}')\n",
    "    \n",
    "    # Stop the writer thread\n",
    "    q.put(None)\n",
    "    writer_thread.join()    \n",
    "    \n",
    "def process_species(species_name, ftp_server, base_ftp_directory):\n",
    "    \"\"\"\n",
    "    Processes a species: downloads the FASTA files, decompresses, calculates the number of sequences and compression ratio.\n",
    "    \"\"\"\n",
    "    results = {\n",
    "        \"species_name\": species_name,\n",
    "        \"cds_num_sequences\": 0,\n",
    "        \"cds_tot_len\": 0,\n",
    "        \"cds_compression_ratio_geco3\": 0,\n",
    "        \"cds_compression_ratio_gz\": 0,\n",
    "        \"ncrna_num_sequences\": 0,\n",
    "        \"ncrna_tot_len\": 0,\n",
    "        \"ncrna_compression_ratio_geco3\": 0,\n",
    "        \"ncrna_compression_ratio_gz\": 0,\n",
    "    }\n",
    "\n",
    "    # Directories and file suffixes for cds, ncrna\n",
    "    data_types = [(\"cds\", \"cds.all.fa.gz\"), (\"ncrna\", \"ncrna.fa.gz\")]\n",
    "\n",
    "    for data_type, file_suffix in data_types:\n",
    "        ftp_directory = f\"{base_ftp_directory}/{species_name}/{data_type}/\"\n",
    "        \n",
    "        try:\n",
    "            # Attempt to find, download, and decompress the FASTA file\n",
    "            decompressed_fasta = find_download_and_decompress_from_ftp(ftp_server, ftp_directory, file_suffix)\n",
    "        except (FileNotFoundError, ftplib.error_perm) as e:\n",
    "            print(f\"Directory or file not found for {species_name} in {data_type} folder: {e}\")\n",
    "            # Skip processing if the directory or file is not found, fields remain at 0\n",
    "            continue\n",
    "        \n",
    "        # Calculate the compression ratio\n",
    "        cr, crg, tl = calculate_compression_ratio(decompressed_fasta)\n",
    "        \n",
    "        num_seqs = sum(1 for _ in SeqIO.parse(decompressed_fasta, \"fasta\"))\n",
    "        # Store results in the appropriate fields\n",
    "        results[f\"{data_type}_num_sequences\"] = num_seqs\n",
    "        results[f\"{data_type}_compression_ratio_geco3\"] = cr\n",
    "        results[f\"{data_type}_compression_ratio_gz\"] = crg\n",
    "        results[f\"{data_type}_tot_len\"] = tl\n",
    "        \n",
    "        # Print the species name and compression ratios\n",
    "        print(data_type, num_seqs, \", total = \", tl)\n",
    "        print(f\"The GECO3 compression ratio for {decompressed_fasta} is {cr:.2f}%\")\n",
    "        print(f\"The GZIP compression ratio for {decompressed_fasta} is {crg:.2f}%\")\n",
    "\n",
    "        os.remove(decompressed_fasta)\n",
    "    return results\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Processing acinetobacter_baumannii_aye_gca_000069245...\n",
      "cds 3712 , total =  3499542\n",
      "The GECO3 compression ratio for Acinetobacter_baumannii_aye_gca_000069245.ASM6924v1.cds.all.fa is 23.53%\n",
      "The GZIP compression ratio for Acinetobacter_baumannii_aye_gca_000069245.ASM6924v1.cds.all.fa is 27.82%\n",
      "ncrna 91 , total =  32793\n",
      "The GECO3 compression ratio for Acinetobacter_baumannii_aye_gca_000069245.ASM6924v1.ncrna.fa is 6.65%\n",
      "The GZIP compression ratio for Acinetobacter_baumannii_aye_gca_000069245.ASM6924v1.ncrna.fa is 8.54%\n",
      "Results for acinetobacter_baumannii_aye_gca_000069245 written to species_compression_ratios_bact_looptest.csv\n",
      "Processing actinobacillus_pleuropneumoniae_serovar_5b_str_l20_gca_000015885...\n",
      "cds 2012 , total =  1964559\n",
      "The GECO3 compression ratio for Actinobacillus_pleuropneumoniae_serovar_5b_str_l20_gca_000015885.ASM1588v1.cds.all.fa is 23.31%\n",
      "The GZIP compression ratio for Actinobacillus_pleuropneumoniae_serovar_5b_str_l20_gca_000015885.ASM1588v1.cds.all.fa is 27.55%\n",
      "ncrna 80 , total =  33008\n",
      "The GECO3 compression ratio for Actinobacillus_pleuropneumoniae_serovar_5b_str_l20_gca_000015885.ASM1588v1.ncrna.fa is 6.77%\n",
      "The GZIP compression ratio for Actinobacillus_pleuropneumoniae_serovar_5b_str_l20_gca_000015885.ASM1588v1.ncrna.fa is 8.81%\n",
      "Results for actinobacillus_pleuropneumoniae_serovar_5b_str_l20_gca_000015885 written to species_compression_ratios_bact_looptest.csv\n",
      "Processing aeromonas_hydrophila_subsp_hydrophila_atcc_7966_gca_000014805...\n",
      "cds 4121 , total =  4151889\n",
      "The GECO3 compression ratio for Aeromonas_hydrophila_subsp_hydrophila_atcc_7966_gca_000014805.ASM1480v1.cds.all.fa is 22.57%\n",
      "The GZIP compression ratio for Aeromonas_hydrophila_subsp_hydrophila_atcc_7966_gca_000014805.ASM1480v1.cds.all.fa is 27.05%\n",
      "ncrna 156 , total =  55368\n",
      "The GECO3 compression ratio for Aeromonas_hydrophila_subsp_hydrophila_atcc_7966_gca_000014805.ASM1480v1.ncrna.fa is 4.49%\n",
      "The GZIP compression ratio for Aeromonas_hydrophila_subsp_hydrophila_atcc_7966_gca_000014805.ASM1480v1.ncrna.fa is 6.15%\n",
      "Results for aeromonas_hydrophila_subsp_hydrophila_atcc_7966_gca_000014805 written to species_compression_ratios_bact_looptest.csv\n",
      "Processing aeropyrum_pernix_k1_gca_000011125...\n",
      "cds 1700 , total =  1481337\n",
      "The GECO3 compression ratio for Aeropyrum_pernix_k1_gca_000011125.ASM1112v1.cds.all.fa is 23.59%\n",
      "The GZIP compression ratio for Aeropyrum_pernix_k1_gca_000011125.ASM1112v1.cds.all.fa is 27.89%\n",
      "ncrna 50 , total =  8437\n",
      "The GECO3 compression ratio for Aeropyrum_pernix_k1_gca_000011125.ASM1112v1.ncrna.fa is 20.78%\n",
      "The GZIP compression ratio for Aeropyrum_pernix_k1_gca_000011125.ASM1112v1.ncrna.fa is 25.44%\n",
      "Results for aeropyrum_pernix_k1_gca_000011125 written to species_compression_ratios_bact_looptest.csv\n",
      "Processing agrobacterium_fabrum_str_c58_gca_000092025...\n",
      "cds 5355 , total =  5004648\n",
      "The GECO3 compression ratio for Agrobacterium_fabrum_str_c58_gca_000092025.ASM9202v1.cds.all.fa is 23.32%\n",
      "The GZIP compression ratio for Agrobacterium_fabrum_str_c58_gca_000092025.ASM9202v1.cds.all.fa is 27.46%\n",
      "ncrna 70 , total =  20603\n",
      "The GECO3 compression ratio for Agrobacterium_fabrum_str_c58_gca_000092025.ASM9202v1.ncrna.fa is 10.57%\n",
      "The GZIP compression ratio for Agrobacterium_fabrum_str_c58_gca_000092025.ASM9202v1.ncrna.fa is 13.45%\n",
      "Results for agrobacterium_fabrum_str_c58_gca_000092025 written to species_compression_ratios_bact_looptest.csv\n",
      "Processing aliivibrio_fischeri_es114_gca_000011805...\n",
      "cds 3818 , total =  3725679\n",
      "The GECO3 compression ratio for Aliivibrio_fischeri_es114_gca_000011805.ASM1180v1.cds.all.fa is 23.82%\n",
      "The GZIP compression ratio for Aliivibrio_fischeri_es114_gca_000011805.ASM1180v1.cds.all.fa is 27.86%\n",
      "ncrna 156 , total =  64517\n",
      "The GECO3 compression ratio for Aliivibrio_fischeri_es114_gca_000011805.ASM1180v1.ncrna.fa is 3.77%\n",
      "The GZIP compression ratio for Aliivibrio_fischeri_es114_gca_000011805.ASM1180v1.ncrna.fa is 5.19%\n",
      "Results for aliivibrio_fischeri_es114_gca_000011805 written to species_compression_ratios_bact_looptest.csv\n",
      "Processing anaplasma_phagocytophilum_str_hz_gca_000013125...\n",
      "cds 1352 , total =  1056654\n",
      "The GECO3 compression ratio for Anaplasma_phagocytophilum_str_hz_gca_000013125.ASM1312v1.cds.all.fa is 22.48%\n",
      "The GZIP compression ratio for Anaplasma_phagocytophilum_str_hz_gca_000013125.ASM1312v1.cds.all.fa is 27.64%\n",
      "ncrna 42 , total =  7722\n",
      "The GECO3 compression ratio for Anaplasma_phagocytophilum_str_hz_gca_000013125.ASM1312v1.ncrna.fa is 25.73%\n",
      "The GZIP compression ratio for Anaplasma_phagocytophilum_str_hz_gca_000013125.ASM1312v1.ncrna.fa is 32.18%\n",
      "Results for anaplasma_phagocytophilum_str_hz_gca_000013125 written to species_compression_ratios_bact_looptest.csv\n",
      "Processing aquifex_aeolicus_vf5_gca_000008625...\n",
      "cds 1552 , total =  1470174\n",
      "The GECO3 compression ratio for Aquifex_aeolicus_vf5_gca_000008625.ASM862v1.cds.all.fa is 23.32%\n",
      "The GZIP compression ratio for Aquifex_aeolicus_vf5_gca_000008625.ASM862v1.cds.all.fa is 27.67%\n",
      "ncrna 51 , total =  13057\n",
      "The GECO3 compression ratio for Aquifex_aeolicus_vf5_gca_000008625.ASM862v1.ncrna.fa is 15.34%\n",
      "The GZIP compression ratio for Aquifex_aeolicus_vf5_gca_000008625.ASM862v1.ncrna.fa is 19.06%\n",
      "Results for aquifex_aeolicus_vf5_gca_000008625 written to species_compression_ratios_bact_looptest.csv\n",
      "Processing archaeoglobus_fulgidus_dsm_4304_gca_000008665...\n",
      "cds 2407 , total =  1995819\n",
      "The GECO3 compression ratio for Archaeoglobus_fulgidus_dsm_4304_gca_000008665.ASM866v1.cds.all.fa is 23.50%\n",
      "The GZIP compression ratio for Archaeoglobus_fulgidus_dsm_4304_gca_000008665.ASM866v1.cds.all.fa is 27.90%\n",
      "ncrna 49 , total =  8158\n",
      "The GECO3 compression ratio for Archaeoglobus_fulgidus_dsm_4304_gca_000008665.ASM866v1.ncrna.fa is 23.88%\n",
      "The GZIP compression ratio for Archaeoglobus_fulgidus_dsm_4304_gca_000008665.ASM866v1.ncrna.fa is 29.39%\n",
      "Results for archaeoglobus_fulgidus_dsm_4304_gca_000008665 written to species_compression_ratios_bact_looptest.csv\n",
      "Processing azotobacter_vinelandii_dj_gca_000021045...\n",
      "cds 5051 , total =  4610838\n",
      "The GECO3 compression ratio for Azotobacter_vinelandii_dj_gca_000021045.ASM2104v1.cds.all.fa is 21.70%\n",
      "The GZIP compression ratio for Azotobacter_vinelandii_dj_gca_000021045.ASM2104v1.cds.all.fa is 26.71%\n",
      "ncrna 108 , total =  51614\n",
      "The GECO3 compression ratio for Azotobacter_vinelandii_dj_gca_000021045.ASM2104v1.ncrna.fa is 10.79%\n",
      "The GZIP compression ratio for Azotobacter_vinelandii_dj_gca_000021045.ASM2104v1.ncrna.fa is 14.78%\n",
      "Results for azotobacter_vinelandii_dj_gca_000021045 written to species_compression_ratios_bact_looptest.csv\n",
      "Processing bacillus_subtilis_subsp_subtilis_str_168_gca_000009045...\n",
      "cds 4237 , total =  3704493\n",
      "The GECO3 compression ratio for Bacillus_subtilis_subsp_subtilis_str_168_gca_000009045.ASM904v1.cds.all.fa is 23.98%\n",
      "The GZIP compression ratio for Bacillus_subtilis_subsp_subtilis_str_168_gca_000009045.ASM904v1.cds.all.fa is 28.03%\n",
      "ncrna 118 , total =  53063\n",
      "The GECO3 compression ratio for Bacillus_subtilis_subsp_subtilis_str_168_gca_000009045.ASM904v1.ncrna.fa is 4.62%\n",
      "The GZIP compression ratio for Bacillus_subtilis_subsp_subtilis_str_168_gca_000009045.ASM904v1.ncrna.fa is 6.23%\n",
      "Results for bacillus_subtilis_subsp_subtilis_str_168_gca_000009045 written to species_compression_ratios_bact_looptest.csv\n",
      "Processing bacteroides_thetaiotaomicron_vpi_5482_gca_000011065...\n",
      "cds 4825 , total =  5638077\n",
      "The GECO3 compression ratio for Bacteroides_thetaiotaomicron_vpi_5482_gca_000011065.ASM1106v1.cds.all.fa is 23.90%\n",
      "The GZIP compression ratio for Bacteroides_thetaiotaomicron_vpi_5482_gca_000011065.ASM1106v1.cds.all.fa is 28.09%\n",
      "ncrna 86 , total =  26205\n",
      "The GECO3 compression ratio for Bacteroides_thetaiotaomicron_vpi_5482_gca_000011065.ASM1106v1.ncrna.fa is 8.40%\n",
      "The GZIP compression ratio for Bacteroides_thetaiotaomicron_vpi_5482_gca_000011065.ASM1106v1.ncrna.fa is 10.87%\n",
      "Results for bacteroides_thetaiotaomicron_vpi_5482_gca_000011065 written to species_compression_ratios_bact_looptest.csv\n",
      "Processing bartonella_henselae_str_houston_1_gca_000046705...\n",
      "cds 1488 , total =  1406688\n",
      "The GECO3 compression ratio for Bartonella_henselae_str_houston_1_gca_000046705.ASM4670v1.cds.all.fa is 22.83%\n",
      "The GZIP compression ratio for Bartonella_henselae_str_houston_1_gca_000046705.ASM4670v1.cds.all.fa is 27.66%\n",
      "ncrna 51 , total =  12433\n",
      "The GECO3 compression ratio for Bartonella_henselae_str_houston_1_gca_000046705.ASM4670v1.ncrna.fa is 15.97%\n",
      "The GZIP compression ratio for Bartonella_henselae_str_houston_1_gca_000046705.ASM4670v1.ncrna.fa is 19.75%\n",
      "Results for bartonella_henselae_str_houston_1_gca_000046705 written to species_compression_ratios_bact_looptest.csv\n",
      "Processing bifidobacterium_longum_ncc2705_gca_000007525...\n",
      "cds 1729 , total =  1929312\n",
      "The GECO3 compression ratio for Bifidobacterium_longum_ncc2705_gca_000007525.ASM752v1.cds.all.fa is 23.22%\n",
      "The GZIP compression ratio for Bifidobacterium_longum_ncc2705_gca_000007525.ASM752v1.cds.all.fa is 27.51%\n",
      "ncrna 69 , total =  23027\n",
      "The GECO3 compression ratio for Bifidobacterium_longum_ncc2705_gca_000007525.ASM752v1.ncrna.fa is 10.17%\n",
      "The GZIP compression ratio for Bifidobacterium_longum_ncc2705_gca_000007525.ASM752v1.ncrna.fa is 12.80%\n",
      "Results for bifidobacterium_longum_ncc2705_gca_000007525 written to species_compression_ratios_bact_looptest.csv\n",
      "Processing borreliella_burgdorferi_b31_gca_000008685...\n",
      "cds 1339 , total =  1150588\n",
      "The GECO3 compression ratio for Borreliella_burgdorferi_b31_gca_000008685.ASM868v2.cds.all.fa is 19.64%\n",
      "The GZIP compression ratio for Borreliella_burgdorferi_b31_gca_000008685.ASM868v2.cds.all.fa is 25.36%\n",
      "Directory or file not found for borreliella_burgdorferi_b31_gca_000008685 in ncrna folder: 550 Failed to change directory.\n",
      "Results for borreliella_burgdorferi_b31_gca_000008685 written to species_compression_ratios_bact_looptest.csv\n",
      "Processing bradyrhizobium_diazoefficiens_usda_110_gca_000011365...\n",
      "cds 8317 , total =  7927989\n",
      "The GECO3 compression ratio for Bradyrhizobium_diazoefficiens_usda_110_gca_000011365.ASM1136v1.cds.all.fa is 22.39%\n",
      "The GZIP compression ratio for Bradyrhizobium_diazoefficiens_usda_110_gca_000011365.ASM1136v1.cds.all.fa is 27.01%\n",
      "ncrna 56 , total =  9110\n",
      "The GECO3 compression ratio for Bradyrhizobium_diazoefficiens_usda_110_gca_000011365.ASM1136v1.ncrna.fa is 24.59%\n",
      "The GZIP compression ratio for Bradyrhizobium_diazoefficiens_usda_110_gca_000011365.ASM1136v1.ncrna.fa is 30.41%\n",
      "Results for bradyrhizobium_diazoefficiens_usda_110_gca_000011365 written to species_compression_ratios_bact_looptest.csv\n",
      "Processing brucella_abortus_bv_1_str_9_941_gca_000008145...\n",
      "cds 3085 , total =  2675535\n",
      "The GECO3 compression ratio for Brucella_abortus_bv_1_str_9_941_gca_000008145.ASM814v1.cds.all.fa is 23.33%\n",
      "The GZIP compression ratio for Brucella_abortus_bv_1_str_9_941_gca_000008145.ASM814v1.cds.all.fa is 27.49%\n",
      "ncrna 64 , total =  17644\n",
      "The GECO3 compression ratio for Brucella_abortus_bv_1_str_9_941_gca_000008145.ASM814v1.ncrna.fa is 11.83%\n",
      "The GZIP compression ratio for Brucella_abortus_bv_1_str_9_941_gca_000008145.ASM814v1.ncrna.fa is 14.96%\n",
      "Results for brucella_abortus_bv_1_str_9_941_gca_000008145 written to species_compression_ratios_bact_looptest.csv\n",
      "Processing buchnera_aphidicola_str_aps_acyrthosiphon_pisum__gca_000009605...\n",
      "cds 574 , total =  566574\n",
      "The GECO3 compression ratio for Buchnera_aphidicola_str_aps_acyrthosiphon_pisum__gca_000009605.ASM960v1.cds.all.fa is 22.54%\n",
      "The GZIP compression ratio for Buchnera_aphidicola_str_aps_acyrthosiphon_pisum__gca_000009605.ASM960v1.cds.all.fa is 26.95%\n",
      "ncrna 35 , total =  6945\n",
      "The GECO3 compression ratio for Buchnera_aphidicola_str_aps_acyrthosiphon_pisum__gca_000009605.ASM960v1.ncrna.fa is 26.00%\n",
      "The GZIP compression ratio for Buchnera_aphidicola_str_aps_acyrthosiphon_pisum__gca_000009605.ASM960v1.ncrna.fa is 32.58%\n",
      "Results for buchnera_aphidicola_str_aps_acyrthosiphon_pisum__gca_000009605 written to species_compression_ratios_bact_looptest.csv\n",
      "Processing burkholderia_pseudomallei_1710b_gca_000012785...\n",
      "cds 6347 , total =  7392630\n",
      "The GECO3 compression ratio for Burkholderia_pseudomallei_1710b_gca_000012785.ASM1278v1.cds.all.fa is 19.54%\n",
      "The GZIP compression ratio for Burkholderia_pseudomallei_1710b_gca_000012785.ASM1278v1.cds.all.fa is 25.92%\n",
      "ncrna 71 , total =  22566\n",
      "The GECO3 compression ratio for Burkholderia_pseudomallei_1710b_gca_000012785.ASM1278v1.ncrna.fa is 9.45%\n",
      "The GZIP compression ratio for Burkholderia_pseudomallei_1710b_gca_000012785.ASM1278v1.ncrna.fa is 12.00%\n",
      "Results for burkholderia_pseudomallei_1710b_gca_000012785 written to species_compression_ratios_bact_looptest.csv\n",
      "Processing campylobacter_jejuni_subsp_jejuni_nctc_11168_atcc_700819_gca_000009085...\n",
      "cds 1622 , total =  1524876\n",
      "The GECO3 compression ratio for Campylobacter_jejuni_subsp_jejuni_nctc_11168_atcc_700819_gca_000009085.ASM908v1.cds.all.fa is 22.27%\n",
      "The GZIP compression ratio for Campylobacter_jejuni_subsp_jejuni_nctc_11168_atcc_700819_gca_000009085.ASM908v1.cds.all.fa is 26.71%\n",
      "ncrna 52 , total =  17001\n",
      "The GECO3 compression ratio for Campylobacter_jejuni_subsp_jejuni_nctc_11168_atcc_700819_gca_000009085.ASM908v1.ncrna.fa is 11.79%\n",
      "The GZIP compression ratio for Campylobacter_jejuni_subsp_jejuni_nctc_11168_atcc_700819_gca_000009085.ASM908v1.ncrna.fa is 15.02%\n",
      "Results for campylobacter_jejuni_subsp_jejuni_nctc_11168_atcc_700819_gca_000009085 written to species_compression_ratios_bact_looptest.csv\n",
      "Processing candidatus_korarchaeum_cryptofilum_opf8_gca_000019605...\n",
      "cds 1602 , total =  1429791\n",
      "The GECO3 compression ratio for Candidatus_korarchaeum_cryptofilum_opf8_gca_000019605.ASM1960v1.cds.all.fa is 23.70%\n",
      "The GZIP compression ratio for Candidatus_korarchaeum_cryptofilum_opf8_gca_000019605.ASM1960v1.cds.all.fa is 27.85%\n",
      "ncrna 49 , total =  8671\n",
      "The GECO3 compression ratio for Candidatus_korarchaeum_cryptofilum_opf8_gca_000019605.ASM1960v1.ncrna.fa is 22.85%\n",
      "The GZIP compression ratio for Candidatus_korarchaeum_cryptofilum_opf8_gca_000019605.ASM1960v1.ncrna.fa is 28.49%\n",
      "Results for candidatus_korarchaeum_cryptofilum_opf8_gca_000019605 written to species_compression_ratios_bact_looptest.csv\n",
      "Processing candidatus_koribacter_versatilis_ellin345_gca_000014005...\n",
      "cds 4777 , total =  5052252\n",
      "The GECO3 compression ratio for Candidatus_koribacter_versatilis_ellin345_gca_000014005.ASM1400v1.cds.all.fa is 23.79%\n",
      "The GZIP compression ratio for Candidatus_koribacter_versatilis_ellin345_gca_000014005.ASM1400v1.cds.all.fa is 27.86%\n",
      "ncrna 49 , total =  8204\n",
      "The GECO3 compression ratio for Candidatus_koribacter_versatilis_ellin345_gca_000014005.ASM1400v1.ncrna.fa is 25.05%\n",
      "The GZIP compression ratio for Candidatus_koribacter_versatilis_ellin345_gca_000014005.ASM1400v1.ncrna.fa is 31.22%\n",
      "Results for candidatus_koribacter_versatilis_ellin345_gca_000014005 written to species_compression_ratios_bact_looptest.csv\n",
      "Processing caulobacter_vibrioides_cb15_gca_000006905...\n",
      "cds 3737 , total =  3638394\n",
      "The GECO3 compression ratio for Caulobacter_vibrioides_cb15_gca_000006905.ASM690v1.cds.all.fa is 22.05%\n",
      "The GZIP compression ratio for Caulobacter_vibrioides_cb15_gca_000006905.ASM690v1.cds.all.fa is 26.58%\n",
      "ncrna 57 , total =  12673\n",
      "The GECO3 compression ratio for Caulobacter_vibrioides_cb15_gca_000006905.ASM690v1.ncrna.fa is 16.03%\n",
      "The GZIP compression ratio for Caulobacter_vibrioides_cb15_gca_000006905.ASM690v1.ncrna.fa is 19.71%\n",
      "Results for caulobacter_vibrioides_cb15_gca_000006905 written to species_compression_ratios_bact_looptest.csv\n",
      "Processing cenarchaeum_symbiosum_a_gca_000200715...\n",
      "cds 2017 , total =  1868698\n",
      "The GECO3 compression ratio for Cenarchaeum_symbiosum_a_gca_000200715.ASM20071v1.cds.all.fa is 22.91%\n",
      "The GZIP compression ratio for Cenarchaeum_symbiosum_a_gca_000200715.ASM20071v1.cds.all.fa is 27.53%\n",
      "ncrna 48 , total =  8388\n",
      "The GECO3 compression ratio for Cenarchaeum_symbiosum_a_gca_000200715.ASM20071v1.ncrna.fa is 25.54%\n",
      "The GZIP compression ratio for Cenarchaeum_symbiosum_a_gca_000200715.ASM20071v1.ncrna.fa is 31.66%\n",
      "Results for cenarchaeum_symbiosum_a_gca_000200715 written to species_compression_ratios_bact_looptest.csv\n",
      "Processing chlamydia_trachomatis_d_uw_3_cx_gca_000008725...\n",
      "cds 894 , total =  939210\n",
      "The GECO3 compression ratio for Chlamydia_trachomatis_d_uw_3_cx_gca_000008725.ASM872v1.cds.all.fa is 24.19%\n",
      "The GZIP compression ratio for Chlamydia_trachomatis_d_uw_3_cx_gca_000008725.ASM872v1.cds.all.fa is 28.18%\n",
      "ncrna 43 , total =  11877\n",
      "The GECO3 compression ratio for Chlamydia_trachomatis_d_uw_3_cx_gca_000008725.ASM872v1.ncrna.fa is 16.44%\n",
      "The GZIP compression ratio for Chlamydia_trachomatis_d_uw_3_cx_gca_000008725.ASM872v1.ncrna.fa is 20.45%\n",
      "Results for chlamydia_trachomatis_d_uw_3_cx_gca_000008725 written to species_compression_ratios_bact_looptest.csv\n",
      "Processing chlorobaculum_tepidum_tls_gca_000006985...\n",
      "cds 2255 , total =  1904213\n",
      "The GECO3 compression ratio for Chlorobaculum_tepidum_tls_gca_000006985.ASM698v1.cds.all.fa is 23.40%\n",
      "The GZIP compression ratio for Chlorobaculum_tepidum_tls_gca_000006985.ASM698v1.cds.all.fa is 27.67%\n",
      "Directory or file not found for chlorobaculum_tepidum_tls_gca_000006985 in ncrna folder: 550 Failed to change directory.\n",
      "Results for chlorobaculum_tepidum_tls_gca_000006985 written to species_compression_ratios_bact_looptest.csv\n",
      "Processing chloroflexus_aurantiacus_j_10_fl_gca_000018865...\n",
      "cds 3852 , total =  4230195\n",
      "The GECO3 compression ratio for Chloroflexus_aurantiacus_j_10_fl_gca_000018865.ASM1886v1.cds.all.fa is 23.64%\n",
      "The GZIP compression ratio for Chloroflexus_aurantiacus_j_10_fl_gca_000018865.ASM1886v1.cds.all.fa is 27.78%\n",
      "ncrna 61 , total =  18246\n",
      "The GECO3 compression ratio for Chloroflexus_aurantiacus_j_10_fl_gca_000018865.ASM1886v1.ncrna.fa is 13.02%\n",
      "The GZIP compression ratio for Chloroflexus_aurantiacus_j_10_fl_gca_000018865.ASM1886v1.ncrna.fa is 16.29%\n",
      "Results for chloroflexus_aurantiacus_j_10_fl_gca_000018865 written to species_compression_ratios_bact_looptest.csv\n",
      "Processing clostridioides_difficile_630_gca_000009205...\n",
      "cds 3803 , total =  3526770\n",
      "The GECO3 compression ratio for Clostridioides_difficile_630_gca_000009205.ASM920v2.cds.all.fa is 22.29%\n",
      "The GZIP compression ratio for Clostridioides_difficile_630_gca_000009205.ASM920v2.cds.all.fa is 26.84%\n",
      "ncrna 119 , total =  56227\n",
      "The GECO3 compression ratio for Clostridioides_difficile_630_gca_000009205.ASM920v2.ncrna.fa is 3.99%\n",
      "The GZIP compression ratio for Clostridioides_difficile_630_gca_000009205.ASM920v2.ncrna.fa is 5.44%\n",
      "Results for clostridioides_difficile_630_gca_000009205 written to species_compression_ratios_bact_looptest.csv\n",
      "Processing coxiella_burnetii_rsa_493_gca_000007765...\n",
      "cds 1850 , total =  1575942\n",
      "The GECO3 compression ratio for Coxiella_burnetii_rsa_493_gca_000007765.ASM776v2.cds.all.fa is 23.62%\n",
      "The GZIP compression ratio for Coxiella_burnetii_rsa_493_gca_000007765.ASM776v2.cds.all.fa is 27.97%\n",
      "ncrna 45 , total =  5809\n",
      "The GECO3 compression ratio for Coxiella_burnetii_rsa_493_gca_000007765.ASM776v2.ncrna.fa is 25.37%\n",
      "The GZIP compression ratio for Coxiella_burnetii_rsa_493_gca_000007765.ASM776v2.ncrna.fa is 31.19%\n",
      "Results for coxiella_burnetii_rsa_493_gca_000007765 written to species_compression_ratios_bact_looptest.csv\n",
      "Processing cutibacterium_acnes_kpa171202_gca_000008345...\n",
      "cds 2297 , total =  2294694\n",
      "The GECO3 compression ratio for Cutibacterium_acnes_kpa171202_gca_000008345.ASM834v1.cds.all.fa is 23.88%\n",
      "The GZIP compression ratio for Cutibacterium_acnes_kpa171202_gca_000008345.ASM834v1.cds.all.fa is 27.97%\n",
      "ncrna 54 , total =  17209\n",
      "The GECO3 compression ratio for Cutibacterium_acnes_kpa171202_gca_000008345.ASM834v1.ncrna.fa is 12.59%\n",
      "The GZIP compression ratio for Cutibacterium_acnes_kpa171202_gca_000008345.ASM834v1.ncrna.fa is 15.78%\n",
      "Results for cutibacterium_acnes_kpa171202_gca_000008345 written to species_compression_ratios_bact_looptest.csv\n",
      "Processing deinococcus_radiodurans_r1_gca_000008565...\n",
      "cds 3101 , total =  2879501\n",
      "The GECO3 compression ratio for Deinococcus_radiodurans_r1_gca_000008565.ASM856v1.cds.all.fa is 22.10%\n",
      "The GZIP compression ratio for Deinococcus_radiodurans_r1_gca_000008565.ASM856v1.cds.all.fa is 26.75%\n",
      "ncrna 58 , total =  12279\n",
      "The GECO3 compression ratio for Deinococcus_radiodurans_r1_gca_000008565.ASM856v1.ncrna.fa is 15.31%\n",
      "The GZIP compression ratio for Deinococcus_radiodurans_r1_gca_000008565.ASM856v1.ncrna.fa is 19.20%\n",
      "Results for deinococcus_radiodurans_r1_gca_000008565 written to species_compression_ratios_bact_looptest.csv\n",
      "Processing desulfovibrio_vulgaris_str_hildenborough_gca_000195755...\n",
      "cds 3523 , total =  3230640\n",
      "The GECO3 compression ratio for Desulfovibrio_vulgaris_str_hildenborough_gca_000195755.ASM19575v1.cds.all.fa is 22.86%\n",
      "The GZIP compression ratio for Desulfovibrio_vulgaris_str_hildenborough_gca_000195755.ASM19575v1.cds.all.fa is 27.27%\n",
      "ncrna 83 , total =  28073\n",
      "The GECO3 compression ratio for Desulfovibrio_vulgaris_str_hildenborough_gca_000195755.ASM19575v1.ncrna.fa is 8.12%\n",
      "The GZIP compression ratio for Desulfovibrio_vulgaris_str_hildenborough_gca_000195755.ASM19575v1.ncrna.fa is 10.49%\n",
      "Results for desulfovibrio_vulgaris_str_hildenborough_gca_000195755 written to species_compression_ratios_bact_looptest.csv\n",
      "Processing dictyoglomus_turgidum_dsm_6724_gca_000021645...\n",
      "cds 1744 , total =  1721508\n",
      "The GECO3 compression ratio for Dictyoglomus_turgidum_dsm_6724_gca_000021645.ASM2164v1.cds.all.fa is 22.92%\n",
      "The GZIP compression ratio for Dictyoglomus_turgidum_dsm_6724_gca_000021645.ASM2164v1.cds.all.fa is 27.15%\n",
      "ncrna 53 , total =  13050\n",
      "The GECO3 compression ratio for Dictyoglomus_turgidum_dsm_6724_gca_000021645.ASM2164v1.ncrna.fa is 16.19%\n",
      "The GZIP compression ratio for Dictyoglomus_turgidum_dsm_6724_gca_000021645.ASM2164v1.ncrna.fa is 20.08%\n",
      "Results for dictyoglomus_turgidum_dsm_6724_gca_000021645 written to species_compression_ratios_bact_looptest.csv\n",
      "Processing enterobacter_cloacae_subsp_cloacae_atcc_13047_gca_000025565...\n",
      "cds 5518 , total =  4919459\n",
      "The GECO3 compression ratio for Enterobacter_cloacae_subsp_cloacae_atcc_13047_gca_000025565.ASM2556v1.cds.all.fa is 23.28%\n",
      "The GZIP compression ratio for Enterobacter_cloacae_subsp_cloacae_atcc_13047_gca_000025565.ASM2556v1.cds.all.fa is 27.83%\n",
      "ncrna 109 , total =  43305\n",
      "The GECO3 compression ratio for Enterobacter_cloacae_subsp_cloacae_atcc_13047_gca_000025565.ASM2556v1.ncrna.fa is 5.77%\n",
      "The GZIP compression ratio for Enterobacter_cloacae_subsp_cloacae_atcc_13047_gca_000025565.ASM2556v1.ncrna.fa is 7.46%\n",
      "Results for enterobacter_cloacae_subsp_cloacae_atcc_13047_gca_000025565 written to species_compression_ratios_bact_looptest.csv\n",
      "Processing enterococcus_faecalis_v583_gca_000007785...\n",
      "cds 3265 , total =  2890887\n",
      "The GECO3 compression ratio for Enterococcus_faecalis_v583_gca_000007785.ASM778v1.cds.all.fa is 23.20%\n",
      "The GZIP compression ratio for Enterococcus_faecalis_v583_gca_000007785.ASM778v1.cds.all.fa is 27.72%\n",
      "ncrna 79 , total =  23275\n",
      "The GECO3 compression ratio for Enterococcus_faecalis_v583_gca_000007785.ASM778v1.ncrna.fa is 9.40%\n",
      "The GZIP compression ratio for Enterococcus_faecalis_v583_gca_000007785.ASM778v1.ncrna.fa is 11.85%\n",
      "Results for enterococcus_faecalis_v583_gca_000007785 written to species_compression_ratios_bact_looptest.csv\n",
      "Processing escherichia_coli_str_k_12_substr_mg1655_gca_000005845...\n",
      "cds 4239 , total =  3978528\n",
      "The GECO3 compression ratio for Escherichia_coli_str_k_12_substr_mg1655_gca_000005845.ASM584v2.cds.all.fa is 23.66%\n",
      "The GZIP compression ratio for Escherichia_coli_str_k_12_substr_mg1655_gca_000005845.ASM584v2.cds.all.fa is 27.92%\n",
      "ncrna 179 , total =  48300\n",
      "The GECO3 compression ratio for Escherichia_coli_str_k_12_substr_mg1655_gca_000005845.ASM584v2.ncrna.fa is 9.71%\n",
      "The GZIP compression ratio for Escherichia_coli_str_k_12_substr_mg1655_gca_000005845.ASM584v2.ncrna.fa is 12.51%\n",
      "Results for escherichia_coli_str_k_12_substr_mg1655_gca_000005845 written to species_compression_ratios_bact_looptest.csv\n",
      "Processing flavobacterium_psychrophilum_jip02_86_gca_000064305...\n",
      "cds 2446 , total =  2447952\n",
      "The GECO3 compression ratio for Flavobacterium_psychrophilum_jip02_86_gca_000064305.ASM6430v2.cds.all.fa is 22.73%\n",
      "The GZIP compression ratio for Flavobacterium_psychrophilum_jip02_86_gca_000064305.ASM6430v2.cds.all.fa is 27.19%\n",
      "ncrna 67 , total =  30761\n",
      "The GECO3 compression ratio for Flavobacterium_psychrophilum_jip02_86_gca_000064305.ASM6430v2.ncrna.fa is 6.47%\n",
      "The GZIP compression ratio for Flavobacterium_psychrophilum_jip02_86_gca_000064305.ASM6430v2.ncrna.fa is 8.60%\n",
      "Results for flavobacterium_psychrophilum_jip02_86_gca_000064305 written to species_compression_ratios_bact_looptest.csv\n",
      "Processing francisella_tularensis_subsp_tularensis_schu_s4_gca_000008985...\n",
      "cds 1605 , total =  1499085\n",
      "The GECO3 compression ratio for Francisella_tularensis_subsp_tularensis_schu_s4_gca_000008985.ASM898v1.cds.all.fa is 22.26%\n",
      "The GZIP compression ratio for Francisella_tularensis_subsp_tularensis_schu_s4_gca_000008985.ASM898v1.cds.all.fa is 27.06%\n",
      "ncrna 48 , total =  16548\n",
      "The GECO3 compression ratio for Francisella_tularensis_subsp_tularensis_schu_s4_gca_000008985.ASM898v1.ncrna.fa is 11.63%\n",
      "The GZIP compression ratio for Francisella_tularensis_subsp_tularensis_schu_s4_gca_000008985.ASM898v1.ncrna.fa is 14.85%\n",
      "Results for francisella_tularensis_subsp_tularensis_schu_s4_gca_000008985 written to species_compression_ratios_bact_looptest.csv\n",
      "Processing fusobacterium_nucleatum_subsp_nucleatum_atcc_25586_gca_000007325...\n",
      "cds 2067 , total =  1937724\n",
      "The GECO3 compression ratio for Fusobacterium_nucleatum_subsp_nucleatum_atcc_25586_gca_000007325.ASM732v1.cds.all.fa is 21.55%\n",
      "The GZIP compression ratio for Fusobacterium_nucleatum_subsp_nucleatum_atcc_25586_gca_000007325.ASM732v1.cds.all.fa is 26.09%\n",
      "ncrna 62 , total =  26216\n",
      "The GECO3 compression ratio for Fusobacterium_nucleatum_subsp_nucleatum_atcc_25586_gca_000007325.ASM732v1.ncrna.fa is 7.81%\n",
      "The GZIP compression ratio for Fusobacterium_nucleatum_subsp_nucleatum_atcc_25586_gca_000007325.ASM732v1.ncrna.fa is 10.25%\n",
      "Results for fusobacterium_nucleatum_subsp_nucleatum_atcc_25586_gca_000007325 written to species_compression_ratios_bact_looptest.csv\n",
      "Processing gardnerella_vaginalis_0288e_gca_000263555...\n",
      "cds 1356 , total =  1486542\n",
      "The GECO3 compression ratio for Gardnerella_vaginalis_0288e_gca_000263555.ASM26355v1.cds.all.fa is 23.78%\n",
      "The GZIP compression ratio for Gardnerella_vaginalis_0288e_gca_000263555.ASM26355v1.cds.all.fa is 27.90%\n",
      "ncrna 47 , total =  5121\n",
      "The GECO3 compression ratio for Gardnerella_vaginalis_0288e_gca_000263555.ASM26355v1.ncrna.fa is 25.64%\n",
      "The GZIP compression ratio for Gardnerella_vaginalis_0288e_gca_000263555.ASM26355v1.ncrna.fa is 31.79%\n",
      "Results for gardnerella_vaginalis_0288e_gca_000263555 written to species_compression_ratios_bact_looptest.csv\n",
      "Processing geobacter_sulfurreducens_pca_gca_000007985...\n",
      "cds 3423 , total =  3457608\n",
      "The GECO3 compression ratio for Geobacter_sulfurreducens_pca_gca_000007985.ASM798v2.cds.all.fa is 23.29%\n",
      "The GZIP compression ratio for Geobacter_sulfurreducens_pca_gca_000007985.ASM798v2.cds.all.fa is 27.66%\n",
      "ncrna 240 , total =  20092\n",
      "The GECO3 compression ratio for Geobacter_sulfurreducens_pca_gca_000007985.ASM798v2.ncrna.fa is 19.15%\n",
      "The GZIP compression ratio for Geobacter_sulfurreducens_pca_gca_000007985.ASM798v2.ncrna.fa is 23.77%\n",
      "Results for geobacter_sulfurreducens_pca_gca_000007985 written to species_compression_ratios_bact_looptest.csv\n",
      "Processing gloeobacter_violaceus_pcc_7421_gca_000011385...\n",
      "cds 4430 , total =  4166481\n",
      "The GECO3 compression ratio for Gloeobacter_violaceus_pcc_7421_gca_000011385.ASM1138v1.cds.all.fa is 23.01%\n",
      "The GZIP compression ratio for Gloeobacter_violaceus_pcc_7421_gca_000011385.ASM1138v1.cds.all.fa is 27.45%\n",
      "ncrna 50 , total =  8446\n",
      "The GECO3 compression ratio for Gloeobacter_violaceus_pcc_7421_gca_000011385.ASM1138v1.ncrna.fa is 25.38%\n",
      "The GZIP compression ratio for Gloeobacter_violaceus_pcc_7421_gca_000011385.ASM1138v1.ncrna.fa is 31.61%\n",
      "Results for gloeobacter_violaceus_pcc_7421_gca_000011385 written to species_compression_ratios_bact_looptest.csv\n",
      "Processing haemophilus_influenzae_rd_kw20_gca_000027305...\n",
      "cds 1709 , total =  1568293\n",
      "The GECO3 compression ratio for Haemophilus_influenzae_rd_kw20_gca_000027305.ASM2730v1.cds.all.fa is 23.41%\n",
      "The GZIP compression ratio for Haemophilus_influenzae_rd_kw20_gca_000027305.ASM2730v1.cds.all.fa is 27.68%\n",
      "ncrna 36 , total =  28673\n",
      "The GECO3 compression ratio for Haemophilus_influenzae_rd_kw20_gca_000027305.ASM2730v1.ncrna.fa is 5.79%\n",
      "The GZIP compression ratio for Haemophilus_influenzae_rd_kw20_gca_000027305.ASM2730v1.ncrna.fa is 7.78%\n",
      "Results for haemophilus_influenzae_rd_kw20_gca_000027305 written to species_compression_ratios_bact_looptest.csv\n",
      "Processing haloarcula_marismortui_atcc_43049_gca_000011085...\n",
      "cds 4240 , total =  3622977\n",
      "The GECO3 compression ratio for Haloarcula_marismortui_atcc_43049_gca_000011085.ASM1108v1.cds.all.fa is 23.02%\n",
      "The GZIP compression ratio for Haloarcula_marismortui_atcc_43049_gca_000011085.ASM1108v1.cds.all.fa is 27.38%\n",
      "ncrna 60 , total =  17458\n",
      "The GECO3 compression ratio for Haloarcula_marismortui_atcc_43049_gca_000011085.ASM1108v1.ncrna.fa is 13.15%\n",
      "The GZIP compression ratio for Haloarcula_marismortui_atcc_43049_gca_000011085.ASM1108v1.ncrna.fa is 17.08%\n",
      "Results for haloarcula_marismortui_atcc_43049_gca_000011085 written to species_compression_ratios_bact_looptest.csv\n",
      "Processing halobacterium_salinarum_r1_gca_000069025...\n",
      "cds 2704 , total =  2309574\n",
      "The GECO3 compression ratio for Halobacterium_salinarum_r1_gca_000069025.ASM6902v1.cds.all.fa is 20.97%\n",
      "The GZIP compression ratio for Halobacterium_salinarum_r1_gca_000069025.ASM6902v1.cds.all.fa is 26.54%\n",
      "ncrna 50 , total =  8270\n",
      "The GECO3 compression ratio for Halobacterium_salinarum_r1_gca_000069025.ASM6902v1.ncrna.fa is 24.92%\n",
      "The GZIP compression ratio for Halobacterium_salinarum_r1_gca_000069025.ASM6902v1.ncrna.fa is 31.21%\n",
      "Results for halobacterium_salinarum_r1_gca_000069025 written to species_compression_ratios_bact_looptest.csv\n",
      "Processing haloferax_volcanii_ds2_gca_000025685...\n",
      "cds 4027 , total =  3437592\n",
      "The GECO3 compression ratio for Haloferax_volcanii_ds2_gca_000025685.ASM2568v1.cds.all.fa is 21.30%\n",
      "The GZIP compression ratio for Haloferax_volcanii_ds2_gca_000025685.ASM2568v1.cds.all.fa is 26.29%\n",
      "ncrna 49 , total =  12278\n",
      "The GECO3 compression ratio for Haloferax_volcanii_ds2_gca_000025685.ASM2568v1.ncrna.fa is 15.65%\n",
      "The GZIP compression ratio for Haloferax_volcanii_ds2_gca_000025685.ASM2568v1.ncrna.fa is 19.69%\n",
      "Results for haloferax_volcanii_ds2_gca_000025685 written to species_compression_ratios_bact_looptest.csv\n",
      "Processing helicobacter_pylori_26695_gca_000008525...\n",
      "cds 1566 , total =  1494786\n",
      "The GECO3 compression ratio for Helicobacter_pylori_26695_gca_000008525.ASM852v1.cds.all.fa is 22.52%\n",
      "The GZIP compression ratio for Helicobacter_pylori_26695_gca_000008525.ASM852v1.cds.all.fa is 27.15%\n",
      "ncrna 43 , total =  12057\n",
      "The GECO3 compression ratio for Helicobacter_pylori_26695_gca_000008525.ASM852v1.ncrna.fa is 16.54%\n",
      "The GZIP compression ratio for Helicobacter_pylori_26695_gca_000008525.ASM852v1.ncrna.fa is 20.49%\n",
      "Results for helicobacter_pylori_26695_gca_000008525 written to species_compression_ratios_bact_looptest.csv\n",
      "Processing hyperthermus_butylicus_dsm_5456_gca_000015145...\n",
      "cds 1602 , total =  1354200\n",
      "The GECO3 compression ratio for Hyperthermus_butylicus_dsm_5456_gca_000015145.ASM1514v1.cds.all.fa is 23.90%\n",
      "The GZIP compression ratio for Hyperthermus_butylicus_dsm_5456_gca_000015145.ASM1514v1.cds.all.fa is 28.05%\n",
      "ncrna 52 , total =  14808\n",
      "The GECO3 compression ratio for Hyperthermus_butylicus_dsm_5456_gca_000015145.ASM1514v1.ncrna.fa is 19.73%\n",
      "The GZIP compression ratio for Hyperthermus_butylicus_dsm_5456_gca_000015145.ASM1514v1.ncrna.fa is 24.68%\n",
      "Results for hyperthermus_butylicus_dsm_5456_gca_000015145 written to species_compression_ratios_bact_looptest.csv\n",
      "Processing klebsiella_pneumoniae_subsp_pneumoniae_mgh_78578_gca_000016305...\n",
      "cds 5185 , total =  4872147\n",
      "The GECO3 compression ratio for Klebsiella_pneumoniae_subsp_pneumoniae_mgh_78578_gca_000016305.ASM1630v1.cds.all.fa is 22.97%\n",
      "The GZIP compression ratio for Klebsiella_pneumoniae_subsp_pneumoniae_mgh_78578_gca_000016305.ASM1630v1.cds.all.fa is 27.57%\n",
      "ncrna 111 , total =  43434\n",
      "The GECO3 compression ratio for Klebsiella_pneumoniae_subsp_pneumoniae_mgh_78578_gca_000016305.ASM1630v1.ncrna.fa is 5.39%\n",
      "The GZIP compression ratio for Klebsiella_pneumoniae_subsp_pneumoniae_mgh_78578_gca_000016305.ASM1630v1.ncrna.fa is 7.17%\n",
      "Results for klebsiella_pneumoniae_subsp_pneumoniae_mgh_78578_gca_000016305 written to species_compression_ratios_bact_looptest.csv\n",
      "Processing lactobacillus_plantarum_wcfs1_gca_000203855...\n",
      "cds 3108 , total =  2826147\n",
      "The GECO3 compression ratio for Lactobacillus_plantarum_wcfs1_gca_000203855.ASM20385v3.cds.all.fa is 23.98%\n",
      "The GZIP compression ratio for Lactobacillus_plantarum_wcfs1_gca_000203855.ASM20385v3.cds.all.fa is 28.07%\n",
      "ncrna 85 , total =  28379\n",
      "The GECO3 compression ratio for Lactobacillus_plantarum_wcfs1_gca_000203855.ASM20385v3.ncrna.fa is 8.15%\n",
      "The GZIP compression ratio for Lactobacillus_plantarum_wcfs1_gca_000203855.ASM20385v3.ncrna.fa is 10.54%\n",
      "Results for lactobacillus_plantarum_wcfs1_gca_000203855 written to species_compression_ratios_bact_looptest.csv\n",
      "Processing lactococcus_lactis_subsp_lactis_il1403_gca_000006865...\n",
      "cds 2266 , total =  2002833\n",
      "The GECO3 compression ratio for Lactococcus_lactis_subsp_lactis_il1403_gca_000006865.ASM686v1.cds.all.fa is 22.89%\n",
      "The GZIP compression ratio for Lactococcus_lactis_subsp_lactis_il1403_gca_000006865.ASM686v1.cds.all.fa is 27.52%\n",
      "Directory or file not found for lactococcus_lactis_subsp_lactis_il1403_gca_000006865 in ncrna folder: 550 Failed to change directory.\n",
      "Results for lactococcus_lactis_subsp_lactis_il1403_gca_000006865 written to species_compression_ratios_bact_looptest.csv\n",
      "Processing leptospira_interrogans_serovar_lai_str_56601_gca_000092565...\n",
      "cds 3706 , total =  3519372\n",
      "The GECO3 compression ratio for Leptospira_interrogans_serovar_lai_str_56601_gca_000092565.ASM9256v1.cds.all.fa is 22.91%\n",
      "The GZIP compression ratio for Leptospira_interrogans_serovar_lai_str_56601_gca_000092565.ASM9256v1.cds.all.fa is 27.45%\n",
      "ncrna 42 , total =  11778\n",
      "The GECO3 compression ratio for Leptospira_interrogans_serovar_lai_str_56601_gca_000092565.ASM9256v1.ncrna.fa is 16.59%\n",
      "The GZIP compression ratio for Leptospira_interrogans_serovar_lai_str_56601_gca_000092565.ASM9256v1.ncrna.fa is 20.89%\n",
      "Results for leptospira_interrogans_serovar_lai_str_56601_gca_000092565 written to species_compression_ratios_bact_looptest.csv\n",
      "Processing leuconostoc_mesenteroides_subsp_mesenteroides_atcc_8293_gca_000014445...\n",
      "cds 2005 , total =  1826316\n",
      "The GECO3 compression ratio for Leuconostoc_mesenteroides_subsp_mesenteroides_atcc_8293_gca_000014445.ASM1444v1.cds.all.fa is 23.79%\n",
      "The GZIP compression ratio for Leuconostoc_mesenteroides_subsp_mesenteroides_atcc_8293_gca_000014445.ASM1444v1.cds.all.fa is 27.92%\n",
      "ncrna 83 , total =  23938\n",
      "The GECO3 compression ratio for Leuconostoc_mesenteroides_subsp_mesenteroides_atcc_8293_gca_000014445.ASM1444v1.ncrna.fa is 9.38%\n",
      "The GZIP compression ratio for Leuconostoc_mesenteroides_subsp_mesenteroides_atcc_8293_gca_000014445.ASM1444v1.ncrna.fa is 11.89%\n",
      "Results for leuconostoc_mesenteroides_subsp_mesenteroides_atcc_8293_gca_000014445 written to species_compression_ratios_bact_looptest.csv\n",
      "Processing listeria_monocytogenes_egd_e_gca_000196035...\n",
      "cds 2846 , total =  2621172\n",
      "The GECO3 compression ratio for Listeria_monocytogenes_egd_e_gca_000196035.ASM19603v1.cds.all.fa is 23.73%\n",
      "The GZIP compression ratio for Listeria_monocytogenes_egd_e_gca_000196035.ASM19603v1.cds.all.fa is 27.86%\n",
      "ncrna 85 , total =  32572\n",
      "The GECO3 compression ratio for Listeria_monocytogenes_egd_e_gca_000196035.ASM19603v1.ncrna.fa is 6.59%\n",
      "The GZIP compression ratio for Listeria_monocytogenes_egd_e_gca_000196035.ASM19603v1.ncrna.fa is 8.58%\n",
      "Results for listeria_monocytogenes_egd_e_gca_000196035 written to species_compression_ratios_bact_looptest.csv\n",
      "Processing lysinibacillus_sphaericus_c3_41_gca_000017965...\n",
      "cds 4771 , total =  3865005\n",
      "The GECO3 compression ratio for Lysinibacillus_sphaericus_c3_41_gca_000017965.ASM1796v1.cds.all.fa is 23.68%\n",
      "The GZIP compression ratio for Lysinibacillus_sphaericus_c3_41_gca_000017965.ASM1796v1.cds.all.fa is 27.91%\n",
      "ncrna 116 , total =  52322\n",
      "The GECO3 compression ratio for Lysinibacillus_sphaericus_c3_41_gca_000017965.ASM1796v1.ncrna.fa is 4.44%\n",
      "The GZIP compression ratio for Lysinibacillus_sphaericus_c3_41_gca_000017965.ASM1796v1.ncrna.fa is 5.90%\n",
      "Results for lysinibacillus_sphaericus_c3_41_gca_000017965 written to species_compression_ratios_bact_looptest.csv\n",
      "Processing mesoplasma_florum_l1_gca_000008305...\n",
      "cds 683 , total =  737283\n",
      "The GECO3 compression ratio for Mesoplasma_florum_l1_gca_000008305.ASM830v1.cds.all.fa is 21.98%\n",
      "The GZIP compression ratio for Mesoplasma_florum_l1_gca_000008305.ASM830v1.cds.all.fa is 26.26%\n",
      "ncrna 29 , total =  2276\n",
      "The GECO3 compression ratio for Mesoplasma_florum_l1_gca_000008305.ASM830v1.ncrna.fa is 26.80%\n",
      "The GZIP compression ratio for Mesoplasma_florum_l1_gca_000008305.ASM830v1.ncrna.fa is 32.86%\n",
      "Results for mesoplasma_florum_l1_gca_000008305 written to species_compression_ratios_bact_looptest.csv\n",
      "Processing methanobrevibacter_smithii_atcc_35061_gca_000016525...\n",
      "cds 1795 , total =  1673511\n",
      "The GECO3 compression ratio for Methanobrevibacter_smithii_atcc_35061_gca_000016525.ASM1652v1.cds.all.fa is 22.43%\n",
      "The GZIP compression ratio for Methanobrevibacter_smithii_atcc_35061_gca_000016525.ASM1652v1.cds.all.fa is 26.85%\n",
      "ncrna 42 , total =  11854\n",
      "The GECO3 compression ratio for Methanobrevibacter_smithii_atcc_35061_gca_000016525.ASM1652v1.ncrna.fa is 16.16%\n",
      "The GZIP compression ratio for Methanobrevibacter_smithii_atcc_35061_gca_000016525.ASM1652v1.ncrna.fa is 20.27%\n",
      "Results for methanobrevibacter_smithii_atcc_35061_gca_000016525 written to species_compression_ratios_bact_looptest.csv\n",
      "Processing methanocaldococcus_jannaschii_dsm_2661_gca_000091665...\n",
      "cds 1770 , total =  1510485\n",
      "The GECO3 compression ratio for Methanocaldococcus_jannaschii_dsm_2661_gca_000091665.ASM9166v1.cds.all.fa is 22.36%\n",
      "The GZIP compression ratio for Methanocaldococcus_jannaschii_dsm_2661_gca_000091665.ASM9166v1.cds.all.fa is 26.70%\n",
      "ncrna 43 , total =  11918\n",
      "The GECO3 compression ratio for Methanocaldococcus_jannaschii_dsm_2661_gca_000091665.ASM9166v1.ncrna.fa is 15.76%\n",
      "The GZIP compression ratio for Methanocaldococcus_jannaschii_dsm_2661_gca_000091665.ASM9166v1.ncrna.fa is 19.74%\n",
      "Results for methanocaldococcus_jannaschii_dsm_2661_gca_000091665 written to species_compression_ratios_bact_looptest.csv\n",
      "Processing methanococcus_maripaludis_s2_gca_000011585...\n",
      "cds 1714 , total =  1466880\n",
      "The GECO3 compression ratio for Methanococcus_maripaludis_s2_gca_000011585.ASM1158v1.cds.all.fa is 22.95%\n",
      "The GZIP compression ratio for Methanococcus_maripaludis_s2_gca_000011585.ASM1158v1.cds.all.fa is 27.22%\n",
      "ncrna 48 , total =  16566\n",
      "The GECO3 compression ratio for Methanococcus_maripaludis_s2_gca_000011585.ASM1158v1.ncrna.fa is 11.73%\n",
      "The GZIP compression ratio for Methanococcus_maripaludis_s2_gca_000011585.ASM1158v1.ncrna.fa is 14.72%\n",
      "Results for methanococcus_maripaludis_s2_gca_000011585 written to species_compression_ratios_bact_looptest.csv\n",
      "Processing methanopyrus_kandleri_av19_gca_000007185...\n",
      "cds 1681 , total =  1500381\n",
      "The GECO3 compression ratio for Methanopyrus_kandleri_av19_gca_000007185.ASM718v1.cds.all.fa is 23.43%\n",
      "The GZIP compression ratio for Methanopyrus_kandleri_av19_gca_000007185.ASM718v1.cds.all.fa is 27.75%\n",
      "ncrna 38 , total =  7546\n",
      "The GECO3 compression ratio for Methanopyrus_kandleri_av19_gca_000007185.ASM718v1.ncrna.fa is 21.79%\n",
      "The GZIP compression ratio for Methanopyrus_kandleri_av19_gca_000007185.ASM718v1.ncrna.fa is 27.23%\n",
      "Results for methanopyrus_kandleri_av19_gca_000007185 written to species_compression_ratios_bact_looptest.csv\n",
      "Processing methanosarcina_acetivorans_c2a_gca_000007345...\n",
      "cds 4527 , total =  4245426\n",
      "The GECO3 compression ratio for Methanosarcina_acetivorans_c2a_gca_000007345.ASM734v1.cds.all.fa is 22.56%\n",
      "The GZIP compression ratio for Methanosarcina_acetivorans_c2a_gca_000007345.ASM734v1.cds.all.fa is 27.87%\n",
      "ncrna 10 , total =  13741\n",
      "The GECO3 compression ratio for Methanosarcina_acetivorans_c2a_gca_000007345.ASM734v1.ncrna.fa is 10.14%\n",
      "The GZIP compression ratio for Methanosarcina_acetivorans_c2a_gca_000007345.ASM734v1.ncrna.fa is 13.24%\n",
      "Results for methanosarcina_acetivorans_c2a_gca_000007345 written to species_compression_ratios_bact_looptest.csv\n",
      "Processing methanospirillum_hungatei_jf_1_gca_000013445...\n",
      "cds 3139 , total =  3051699\n",
      "The GECO3 compression ratio for Methanospirillum_hungatei_jf_1_gca_000013445.ASM1344v1.cds.all.fa is 23.21%\n",
      "The GZIP compression ratio for Methanospirillum_hungatei_jf_1_gca_000013445.ASM1344v1.cds.all.fa is 27.76%\n",
      "ncrna 65 , total =  22069\n",
      "The GECO3 compression ratio for Methanospirillum_hungatei_jf_1_gca_000013445.ASM1344v1.ncrna.fa is 9.74%\n",
      "The GZIP compression ratio for Methanospirillum_hungatei_jf_1_gca_000013445.ASM1344v1.ncrna.fa is 12.43%\n",
      "Results for methanospirillum_hungatei_jf_1_gca_000013445 written to species_compression_ratios_bact_looptest.csv\n",
      "Processing methanothermobacter_thermautotrophicus_str_delta_h_gca_000008645...\n",
      "cds 1869 , total =  1582128\n",
      "The GECO3 compression ratio for Methanothermobacter_thermautotrophicus_str_delta_h_gca_000008645.ASM864v1.cds.all.fa is 23.52%\n",
      "The GZIP compression ratio for Methanothermobacter_thermautotrophicus_str_delta_h_gca_000008645.ASM864v1.cds.all.fa is 27.79%\n",
      "ncrna 45 , total =  12373\n",
      "The GECO3 compression ratio for Methanothermobacter_thermautotrophicus_str_delta_h_gca_000008645.ASM864v1.ncrna.fa is 16.04%\n",
      "The GZIP compression ratio for Methanothermobacter_thermautotrophicus_str_delta_h_gca_000008645.ASM864v1.ncrna.fa is 20.07%\n",
      "Results for methanothermobacter_thermautotrophicus_str_delta_h_gca_000008645 written to species_compression_ratios_bact_looptest.csv\n",
      "Processing micrococcus_luteus_nctc_2665_gca_000023205...\n",
      "cds 2236 , total =  2227542\n",
      "The GECO3 compression ratio for Micrococcus_luteus_nctc_2665_gca_000023205.ASM2320v1.cds.all.fa is 20.71%\n",
      "The GZIP compression ratio for Micrococcus_luteus_nctc_2665_gca_000023205.ASM2320v1.cds.all.fa is 25.53%\n",
      "ncrna 56 , total =  12659\n",
      "The GECO3 compression ratio for Micrococcus_luteus_nctc_2665_gca_000023205.ASM2320v1.ncrna.fa is 18.58%\n",
      "The GZIP compression ratio for Micrococcus_luteus_nctc_2665_gca_000023205.ASM2320v1.ncrna.fa is 23.20%\n",
      "Results for micrococcus_luteus_nctc_2665_gca_000023205 written to species_compression_ratios_bact_looptest.csv\n",
      "Processing microcystis_aeruginosa_nies_843_gca_000010625...\n",
      "cds 6311 , total =  4755657\n",
      "The GECO3 compression ratio for Microcystis_aeruginosa_nies_843_gca_000010625.ASM1062v1.cds.all.fa is 20.78%\n",
      "The GZIP compression ratio for Microcystis_aeruginosa_nies_843_gca_000010625.ASM1062v1.cds.all.fa is 27.47%\n",
      "ncrna 52 , total =  13354\n",
      "The GECO3 compression ratio for Microcystis_aeruginosa_nies_843_gca_000010625.ASM1062v1.ncrna.fa is 17.13%\n",
      "The GZIP compression ratio for Microcystis_aeruginosa_nies_843_gca_000010625.ASM1062v1.ncrna.fa is 21.31%\n",
      "Results for microcystis_aeruginosa_nies_843_gca_000010625 written to species_compression_ratios_bact_looptest.csv\n",
      "Processing moorella_thermoacetica_atcc_39073_gca_000013105...\n",
      "cds 2460 , total =  2264676\n",
      "The GECO3 compression ratio for Moorella_thermoacetica_atcc_39073_gca_000013105.ASM1310v1.cds.all.fa is 23.44%\n",
      "The GZIP compression ratio for Moorella_thermoacetica_atcc_39073_gca_000013105.ASM1310v1.cds.all.fa is 27.81%\n",
      "ncrna 60 , total =  9838\n",
      "The GECO3 compression ratio for Moorella_thermoacetica_atcc_39073_gca_000013105.ASM1310v1.ncrna.fa is 24.40%\n",
      "The GZIP compression ratio for Moorella_thermoacetica_atcc_39073_gca_000013105.ASM1310v1.ncrna.fa is 30.09%\n",
      "Results for moorella_thermoacetica_atcc_39073_gca_000013105 written to species_compression_ratios_bact_looptest.csv\n",
      "Processing mycoplasma_pneumoniae_m129_gca_000027345...\n",
      "cds 688 , total =  721044\n",
      "The GECO3 compression ratio for Mycoplasma_pneumoniae_m129_gca_000027345.ASM2734v1.cds.all.fa is 22.63%\n",
      "The GZIP compression ratio for Mycoplasma_pneumoniae_m129_gca_000027345.ASM2734v1.cds.all.fa is 27.75%\n",
      "ncrna 41 , total =  7524\n",
      "The GECO3 compression ratio for Mycoplasma_pneumoniae_m129_gca_000027345.ASM2734v1.ncrna.fa is 25.88%\n",
      "The GZIP compression ratio for Mycoplasma_pneumoniae_m129_gca_000027345.ASM2734v1.ncrna.fa is 32.03%\n",
      "Results for mycoplasma_pneumoniae_m129_gca_000027345 written to species_compression_ratios_bact_looptest.csv\n",
      "Processing myxococcus_xanthus_dk_1622_gca_000012685...\n",
      "cds 7328 , total =  8360820\n",
      "The GECO3 compression ratio for Myxococcus_xanthus_dk_1622_gca_000012685.ASM1268v1.cds.all.fa is 22.00%\n",
      "The GZIP compression ratio for Myxococcus_xanthus_dk_1622_gca_000012685.ASM1268v1.cds.all.fa is 26.52%\n",
      "ncrna 76 , total =  23420\n",
      "The GECO3 compression ratio for Myxococcus_xanthus_dk_1622_gca_000012685.ASM1268v1.ncrna.fa is 9.97%\n",
      "The GZIP compression ratio for Myxococcus_xanthus_dk_1622_gca_000012685.ASM1268v1.ncrna.fa is 12.74%\n",
      "Results for myxococcus_xanthus_dk_1622_gca_000012685 written to species_compression_ratios_bact_looptest.csv\n",
      "Processing nanoarchaeum_equitans_kin4_m_gca_000008085...\n",
      "cds 536 , total =  456327\n",
      "The GECO3 compression ratio for Nanoarchaeum_equitans_kin4_m_gca_000008085.ASM808v1.cds.all.fa is 22.61%\n",
      "The GZIP compression ratio for Nanoarchaeum_equitans_kin4_m_gca_000008085.ASM808v1.cds.all.fa is 26.94%\n",
      "ncrna 17 , total =  2618\n",
      "The GECO3 compression ratio for Nanoarchaeum_equitans_kin4_m_gca_000008085.ASM808v1.ncrna.fa is 26.43%\n",
      "The GZIP compression ratio for Nanoarchaeum_equitans_kin4_m_gca_000008085.ASM808v1.ncrna.fa is 33.80%\n",
      "Results for nanoarchaeum_equitans_kin4_m_gca_000008085 written to species_compression_ratios_bact_looptest.csv\n",
      "Processing natronomonas_pharaonis_dsm_2160_gca_000026045...\n",
      "cds 2802 , total =  2459706\n",
      "The GECO3 compression ratio for Natronomonas_pharaonis_dsm_2160_gca_000026045.ASM2604v1.cds.all.fa is 22.46%\n",
      "The GZIP compression ratio for Natronomonas_pharaonis_dsm_2160_gca_000026045.ASM2604v1.cds.all.fa is 27.08%\n",
      "ncrna 49 , total =  8195\n",
      "The GECO3 compression ratio for Natronomonas_pharaonis_dsm_2160_gca_000026045.ASM2604v1.ncrna.fa is 25.08%\n",
      "The GZIP compression ratio for Natronomonas_pharaonis_dsm_2160_gca_000026045.ASM2604v1.ncrna.fa is 31.06%\n",
      "Results for natronomonas_pharaonis_dsm_2160_gca_000026045 written to species_compression_ratios_bact_looptest.csv\n",
      "Processing neisseria_meningitidis_z2491_gca_000009105...\n",
      "cds 1903 , total =  1717641\n",
      "The GECO3 compression ratio for Neisseria_meningitidis_z2491_gca_000009105.ASM910v1.cds.all.fa is 22.79%\n",
      "The GZIP compression ratio for Neisseria_meningitidis_z2491_gca_000009105.ASM910v1.cds.all.fa is 27.46%\n",
      "Directory or file not found for neisseria_meningitidis_z2491_gca_000009105 in ncrna folder: 550 Failed to change directory.\n",
      "Results for neisseria_meningitidis_z2491_gca_000009105 written to species_compression_ratios_bact_looptest.csv\n",
      "Processing nitrosopumilus_maritimus_scm1_gca_000018465...\n",
      "cds 1795 , total =  1486245\n",
      "The GECO3 compression ratio for Nitrosopumilus_maritimus_scm1_gca_000018465.ASM1846v1.cds.all.fa is 22.80%\n",
      "The GZIP compression ratio for Nitrosopumilus_maritimus_scm1_gca_000018465.ASM1846v1.cds.all.fa is 27.05%\n",
      "ncrna 39 , total =  7291\n",
      "The GECO3 compression ratio for Nitrosopumilus_maritimus_scm1_gca_000018465.ASM1846v1.ncrna.fa is 25.81%\n",
      "The GZIP compression ratio for Nitrosopumilus_maritimus_scm1_gca_000018465.ASM1846v1.ncrna.fa is 32.37%\n",
      "Results for nitrosopumilus_maritimus_scm1_gca_000018465 written to species_compression_ratios_bact_looptest.csv\n",
      "Processing nostoc_punctiforme_pcc_73102_gca_000020025...\n",
      "cds 6690 , total =  6995283\n",
      "The GECO3 compression ratio for Nostoc_punctiforme_pcc_73102_gca_000020025.ASM2002v1.cds.all.fa is 23.35%\n",
      "The GZIP compression ratio for Nostoc_punctiforme_pcc_73102_gca_000020025.ASM2002v1.cds.all.fa is 27.89%\n",
      "ncrna 103 , total =  25727\n",
      "The GECO3 compression ratio for Nostoc_punctiforme_pcc_73102_gca_000020025.ASM2002v1.ncrna.fa is 11.88%\n",
      "The GZIP compression ratio for Nostoc_punctiforme_pcc_73102_gca_000020025.ASM2002v1.ncrna.fa is 14.94%\n",
      "Results for nostoc_punctiforme_pcc_73102_gca_000020025 written to species_compression_ratios_bact_looptest.csv\n",
      "Processing paracoccus_denitrificans_pd1222_gca_000203895...\n",
      "cds 5077 , total =  4754025\n",
      "The GECO3 compression ratio for Paracoccus_denitrificans_pd1222_gca_000203895.ASM20389v1.cds.all.fa is 21.67%\n",
      "The GZIP compression ratio for Paracoccus_denitrificans_pd1222_gca_000203895.ASM20389v1.cds.all.fa is 26.44%\n",
      "ncrna 51 , total =  12631\n",
      "The GECO3 compression ratio for Paracoccus_denitrificans_pd1222_gca_000203895.ASM20389v1.ncrna.fa is 15.91%\n",
      "The GZIP compression ratio for Paracoccus_denitrificans_pd1222_gca_000203895.ASM20389v1.ncrna.fa is 19.97%\n",
      "Results for paracoccus_denitrificans_pd1222_gca_000203895 written to species_compression_ratios_bact_looptest.csv\n",
      "Processing pasteurella_multocida_subsp_multocida_str_pm70_gca_000006825...\n",
      "cds 2014 , total =  2010153\n",
      "The GECO3 compression ratio for Pasteurella_multocida_subsp_multocida_str_pm70_gca_000006825.ASM682v1.cds.all.fa is 23.84%\n",
      "The GZIP compression ratio for Pasteurella_multocida_subsp_multocida_str_pm70_gca_000006825.ASM682v1.cds.all.fa is 27.91%\n",
      "Directory or file not found for pasteurella_multocida_subsp_multocida_str_pm70_gca_000006825 in ncrna folder: 550 Failed to change directory.\n",
      "Results for pasteurella_multocida_subsp_multocida_str_pm70_gca_000006825 written to species_compression_ratios_bact_looptest.csv\n",
      "Processing porphyromonas_gingivalis_w83_gca_000007585...\n",
      "cds 1909 , total =  1935270\n",
      "The GECO3 compression ratio for Porphyromonas_gingivalis_w83_gca_000007585.ASM758v1.cds.all.fa is 23.46%\n",
      "The GZIP compression ratio for Porphyromonas_gingivalis_w83_gca_000007585.ASM758v1.cds.all.fa is 28.02%\n",
      "ncrna 64 , total =  21686\n",
      "The GECO3 compression ratio for Porphyromonas_gingivalis_w83_gca_000007585.ASM758v1.ncrna.fa is 9.56%\n",
      "The GZIP compression ratio for Porphyromonas_gingivalis_w83_gca_000007585.ASM758v1.ncrna.fa is 12.20%\n",
      "Results for porphyromonas_gingivalis_w83_gca_000007585 written to species_compression_ratios_bact_looptest.csv\n",
      "Processing prochlorococcus_marinus_subsp_marinus_str_ccmp1375_gca_000007925...\n",
      "cds 1882 , total =  1555914\n",
      "The GECO3 compression ratio for Prochlorococcus_marinus_subsp_marinus_str_ccmp1375_gca_000007925.ASM792v1.cds.all.fa is 23.73%\n",
      "The GZIP compression ratio for Prochlorococcus_marinus_subsp_marinus_str_ccmp1375_gca_000007925.ASM792v1.cds.all.fa is 27.83%\n",
      "ncrna 42 , total =  7410\n",
      "The GECO3 compression ratio for Prochlorococcus_marinus_subsp_marinus_str_ccmp1375_gca_000007925.ASM792v1.ncrna.fa is 25.65%\n",
      "The GZIP compression ratio for Prochlorococcus_marinus_subsp_marinus_str_ccmp1375_gca_000007925.ASM792v1.ncrna.fa is 32.08%\n",
      "Results for prochlorococcus_marinus_subsp_marinus_str_ccmp1375_gca_000007925 written to species_compression_ratios_bact_looptest.csv\n",
      "Processing proteus_mirabilis_hi4320_gca_000069965...\n",
      "cds 3661 , total =  3459756\n",
      "The GECO3 compression ratio for Proteus_mirabilis_hi4320_gca_000069965.ASM6996v1.cds.all.fa is 23.73%\n",
      "The GZIP compression ratio for Proteus_mirabilis_hi4320_gca_000069965.ASM6996v1.cds.all.fa is 27.96%\n",
      "ncrna 105 , total =  34961\n",
      "The GECO3 compression ratio for Proteus_mirabilis_hi4320_gca_000069965.ASM6996v1.ncrna.fa is 6.24%\n",
      "The GZIP compression ratio for Proteus_mirabilis_hi4320_gca_000069965.ASM6996v1.ncrna.fa is 8.13%\n",
      "Results for proteus_mirabilis_hi4320_gca_000069965 written to species_compression_ratios_bact_looptest.csv\n",
      "Processing pyrobaculum_aerophilum_str_im2_gca_000007225...\n",
      "cds 2605 , total =  1975335\n",
      "The GECO3 compression ratio for Pyrobaculum_aerophilum_str_im2_gca_000007225.ASM722v1.cds.all.fa is 23.62%\n",
      "The GZIP compression ratio for Pyrobaculum_aerophilum_str_im2_gca_000007225.ASM722v1.cds.all.fa is 28.01%\n",
      "ncrna 90 , total =  11193\n",
      "The GECO3 compression ratio for Pyrobaculum_aerophilum_str_im2_gca_000007225.ASM722v1.ncrna.fa is 22.80%\n",
      "The GZIP compression ratio for Pyrobaculum_aerophilum_str_im2_gca_000007225.ASM722v1.ncrna.fa is 28.66%\n",
      "Results for pyrobaculum_aerophilum_str_im2_gca_000007225 written to species_compression_ratios_bact_looptest.csv\n",
      "Processing pyrococcus_horikoshii_ot3_gca_000011105...\n",
      "cds 2061 , total =  1711821\n",
      "The GECO3 compression ratio for Pyrococcus_horikoshii_ot3_gca_000011105.ASM1110v1.cds.all.fa is 22.28%\n",
      "The GZIP compression ratio for Pyrococcus_horikoshii_ot3_gca_000011105.ASM1110v1.cds.all.fa is 27.97%\n",
      "ncrna 49 , total =  9132\n",
      "The GECO3 compression ratio for Pyrococcus_horikoshii_ot3_gca_000011105.ASM1110v1.ncrna.fa is 22.16%\n",
      "The GZIP compression ratio for Pyrococcus_horikoshii_ot3_gca_000011105.ASM1110v1.ncrna.fa is 27.80%\n",
      "Results for pyrococcus_horikoshii_ot3_gca_000011105 written to species_compression_ratios_bact_looptest.csv\n",
      "Processing ralstonia_solanacearum_gmi1000_gca_000009125...\n",
      "cds 5120 , total =  5094702\n",
      "The GECO3 compression ratio for Ralstonia_solanacearum_gmi1000_gca_000009125.ASM912v1.cds.all.fa is 21.56%\n",
      "The GZIP compression ratio for Ralstonia_solanacearum_gmi1000_gca_000009125.ASM912v1.cds.all.fa is 26.55%\n",
      "ncrna 69 , total =  22418\n",
      "The GECO3 compression ratio for Ralstonia_solanacearum_gmi1000_gca_000009125.ASM912v1.ncrna.fa is 9.69%\n",
      "The GZIP compression ratio for Ralstonia_solanacearum_gmi1000_gca_000009125.ASM912v1.ncrna.fa is 18.39%\n",
      "Results for ralstonia_solanacearum_gmi1000_gca_000009125 written to species_compression_ratios_bact_looptest.csv\n",
      "Processing rhizobium_leguminosarum_bv_viciae_3841_gca_000009265...\n",
      "cds 7131 , total =  6711624\n",
      "The GECO3 compression ratio for Rhizobium_leguminosarum_bv_viciae_3841_gca_000009265.ASM926v1.cds.all.fa is 22.90%\n",
      "The GZIP compression ratio for Rhizobium_leguminosarum_bv_viciae_3841_gca_000009265.ASM926v1.cds.all.fa is 27.28%\n",
      "Directory or file not found for rhizobium_leguminosarum_bv_viciae_3841_gca_000009265 in ncrna folder: 550 Failed to change directory.\n",
      "Results for rhizobium_leguminosarum_bv_viciae_3841_gca_000009265 written to species_compression_ratios_bact_looptest.csv\n",
      "Processing rhodobacter_sphaeroides_2_4_1_gca_000012905...\n",
      "cds 4287 , total =  4009499\n",
      "The GECO3 compression ratio for Rhodobacter_sphaeroides_2_4_1_gca_000012905.ASM1290v2.cds.all.fa is 21.77%\n",
      "The GZIP compression ratio for Rhodobacter_sphaeroides_2_4_1_gca_000012905.ASM1290v2.cds.all.fa is 26.22%\n",
      "ncrna 67 , total =  18554\n",
      "The GECO3 compression ratio for Rhodobacter_sphaeroides_2_4_1_gca_000012905.ASM1290v2.ncrna.fa is 12.54%\n",
      "The GZIP compression ratio for Rhodobacter_sphaeroides_2_4_1_gca_000012905.ASM1290v2.ncrna.fa is 15.82%\n",
      "Results for rhodobacter_sphaeroides_2_4_1_gca_000012905 written to species_compression_ratios_bact_looptest.csv\n",
      "Processing rhodopirellula_baltica_sh_1_gca_000196115...\n",
      "cds 7325 , total =  6942009\n",
      "The GECO3 compression ratio for Rhodopirellula_baltica_sh_1_gca_000196115.ASM19611v1.cds.all.fa is 23.18%\n",
      "The GZIP compression ratio for Rhodopirellula_baltica_sh_1_gca_000196115.ASM19611v1.cds.all.fa is 27.71%\n",
      "ncrna 73 , total =  9782\n",
      "The GECO3 compression ratio for Rhodopirellula_baltica_sh_1_gca_000196115.ASM19611v1.ncrna.fa is 24.19%\n",
      "The GZIP compression ratio for Rhodopirellula_baltica_sh_1_gca_000196115.ASM19611v1.ncrna.fa is 29.86%\n",
      "Results for rhodopirellula_baltica_sh_1_gca_000196115 written to species_compression_ratios_bact_looptest.csv\n",
      "Processing rhodospirillum_rubrum_atcc_11170_gca_000013085...\n",
      "cds 3841 , total =  3893385\n",
      "The GECO3 compression ratio for Rhodospirillum_rubrum_atcc_11170_gca_000013085.ASM1308v1.cds.all.fa is 22.25%\n",
      "The GZIP compression ratio for Rhodospirillum_rubrum_atcc_11170_gca_000013085.ASM1308v1.cds.all.fa is 26.71%\n",
      "ncrna 67 , total =  21702\n",
      "The GECO3 compression ratio for Rhodospirillum_rubrum_atcc_11170_gca_000013085.ASM1308v1.ncrna.fa is 9.52%\n",
      "The GZIP compression ratio for Rhodospirillum_rubrum_atcc_11170_gca_000013085.ASM1308v1.ncrna.fa is 12.24%\n",
      "Results for rhodospirillum_rubrum_atcc_11170_gca_000013085 written to species_compression_ratios_bact_looptest.csv\n",
      "Processing rickettsia_prowazekii_str_madrid_e_gca_000195735...\n",
      "cds 834 , total =  839742\n",
      "The GECO3 compression ratio for Rickettsia_prowazekii_str_madrid_e_gca_000195735.ASM19573v1.cds.all.fa is 23.09%\n",
      "The GZIP compression ratio for Rickettsia_prowazekii_str_madrid_e_gca_000195735.ASM19573v1.cds.all.fa is 27.34%\n",
      "ncrna 36 , total =  6880\n",
      "The GECO3 compression ratio for Rickettsia_prowazekii_str_madrid_e_gca_000195735.ASM19573v1.ncrna.fa is 25.87%\n",
      "The GZIP compression ratio for Rickettsia_prowazekii_str_madrid_e_gca_000195735.ASM19573v1.ncrna.fa is 32.38%\n",
      "Results for rickettsia_prowazekii_str_madrid_e_gca_000195735 written to species_compression_ratios_bact_looptest.csv\n",
      "Processing saccharolobus_solfataricus_p2_gca_000007005...\n",
      "cds 2994 , total =  2533785\n",
      "The GECO3 compression ratio for Saccharolobus_solfataricus_p2_gca_000007005.ASM700v1.cds.all.fa is 21.87%\n",
      "The GZIP compression ratio for Saccharolobus_solfataricus_p2_gca_000007005.ASM700v1.cds.all.fa is 27.55%\n",
      "ncrna 56 , total =  9297\n",
      "The GECO3 compression ratio for Saccharolobus_solfataricus_p2_gca_000007005.ASM700v1.ncrna.fa is 22.80%\n",
      "The GZIP compression ratio for Saccharolobus_solfataricus_p2_gca_000007005.ASM700v1.ncrna.fa is 28.03%\n",
      "Results for saccharolobus_solfataricus_p2_gca_000007005 written to species_compression_ratios_bact_looptest.csv\n",
      "Processing salinibacter_ruber_dsm_13855_gca_000013045...\n",
      "cds 2833 , total =  3045168\n",
      "The GECO3 compression ratio for Salinibacter_ruber_dsm_13855_gca_000013045.ASM1304v1.cds.all.fa is 22.74%\n",
      "The GZIP compression ratio for Salinibacter_ruber_dsm_13855_gca_000013045.ASM1304v1.cds.all.fa is 27.15%\n",
      "ncrna 47 , total =  8016\n",
      "The GECO3 compression ratio for Salinibacter_ruber_dsm_13855_gca_000013045.ASM1304v1.ncrna.fa is 24.95%\n",
      "The GZIP compression ratio for Salinibacter_ruber_dsm_13855_gca_000013045.ASM1304v1.ncrna.fa is 31.08%\n",
      "Results for salinibacter_ruber_dsm_13855_gca_000013045 written to species_compression_ratios_bact_looptest.csv\n",
      "Processing shigella_dysenteriae_sd197_gca_000012005...\n",
      "cds 4504 , total =  3516828\n",
      "The GECO3 compression ratio for Shigella_dysenteriae_sd197_gca_000012005.ASM1200v1.cds.all.fa is 21.28%\n",
      "The GZIP compression ratio for Shigella_dysenteriae_sd197_gca_000012005.ASM1200v1.cds.all.fa is 26.27%\n",
      "ncrna 106 , total =  38647\n",
      "The GECO3 compression ratio for Shigella_dysenteriae_sd197_gca_000012005.ASM1200v1.ncrna.fa is 6.04%\n",
      "The GZIP compression ratio for Shigella_dysenteriae_sd197_gca_000012005.ASM1200v1.ncrna.fa is 7.91%\n",
      "Results for shigella_dysenteriae_sd197_gca_000012005 written to species_compression_ratios_bact_looptest.csv\n",
      "Processing sinorhizobium_meliloti_1021_gca_000006965...\n",
      "cds 6232 , total =  5786261\n",
      "The GECO3 compression ratio for Sinorhizobium_meliloti_1021_gca_000006965.ASM696v1.cds.all.fa is 22.76%\n",
      "The GZIP compression ratio for Sinorhizobium_meliloti_1021_gca_000006965.ASM696v1.cds.all.fa is 27.32%\n",
      "ncrna 64 , total =  17989\n",
      "The GECO3 compression ratio for Sinorhizobium_meliloti_1021_gca_000006965.ASM696v1.ncrna.fa is 12.39%\n",
      "The GZIP compression ratio for Sinorhizobium_meliloti_1021_gca_000006965.ASM696v1.ncrna.fa is 15.35%\n",
      "Results for sinorhizobium_meliloti_1021_gca_000006965 written to species_compression_ratios_bact_looptest.csv\n",
      "Processing staphylococcus_aureus_subsp_aureus_n315_gca_000009645...\n",
      "cds 2624 , total =  2371482\n",
      "The GECO3 compression ratio for Staphylococcus_aureus_subsp_aureus_n315_gca_000009645.ASM964v1.cds.all.fa is 22.91%\n",
      "The GZIP compression ratio for Staphylococcus_aureus_subsp_aureus_n315_gca_000009645.ASM964v1.cds.all.fa is 27.43%\n",
      "ncrna 78 , total =  27711\n",
      "The GECO3 compression ratio for Staphylococcus_aureus_subsp_aureus_n315_gca_000009645.ASM964v1.ncrna.fa is 7.61%\n",
      "The GZIP compression ratio for Staphylococcus_aureus_subsp_aureus_n315_gca_000009645.ASM964v1.ncrna.fa is 9.93%\n",
      "Results for staphylococcus_aureus_subsp_aureus_n315_gca_000009645 written to species_compression_ratios_bact_looptest.csv\n",
      "Processing stenotrophomonas_maltophilia_k279a_gca_000072485...\n",
      "cds 4385 , total =  4309278\n",
      "The GECO3 compression ratio for Stenotrophomonas_maltophilia_k279a_gca_000072485.ASM7248v1.cds.all.fa is 22.05%\n",
      "The GZIP compression ratio for Stenotrophomonas_maltophilia_k279a_gca_000072485.ASM7248v1.cds.all.fa is 26.56%\n",
      "Directory or file not found for stenotrophomonas_maltophilia_k279a_gca_000072485 in ncrna folder: 550 Failed to change directory.\n",
      "Results for stenotrophomonas_maltophilia_k279a_gca_000072485 written to species_compression_ratios_bact_looptest.csv\n",
      "Processing streptococcus_pneumoniae_tigr4_gca_000006885...\n",
      "cds 2125 , total =  1808970\n",
      "The GECO3 compression ratio for Streptococcus_pneumoniae_tigr4_gca_000006885.ASM688v1.cds.all.fa is 23.28%\n",
      "The GZIP compression ratio for Streptococcus_pneumoniae_tigr4_gca_000006885.ASM688v1.cds.all.fa is 27.73%\n",
      "ncrna 58 , total =  4376\n",
      "The GECO3 compression ratio for Streptococcus_pneumoniae_tigr4_gca_000006885.ASM688v1.ncrna.fa is 18.08%\n",
      "The GZIP compression ratio for Streptococcus_pneumoniae_tigr4_gca_000006885.ASM688v1.ncrna.fa is 21.46%\n",
      "Results for streptococcus_pneumoniae_tigr4_gca_000006885 written to species_compression_ratios_bact_looptest.csv\n",
      "Processing synechocystis_sp_pcc_6803_gca_000009725...\n",
      "cds 3564 , total =  3421566\n",
      "The GECO3 compression ratio for Synechocystis_sp_pcc_6803_gca_000009725.ASM972v1.cds.all.fa is 23.20%\n",
      "The GZIP compression ratio for Synechocystis_sp_pcc_6803_gca_000009725.ASM972v1.cds.all.fa is 27.71%\n",
      "ncrna 50 , total =  12742\n",
      "The GECO3 compression ratio for Synechocystis_sp_pcc_6803_gca_000009725.ASM972v1.ncrna.fa is 16.61%\n",
      "The GZIP compression ratio for Synechocystis_sp_pcc_6803_gca_000009725.ASM972v1.ncrna.fa is 20.78%\n",
      "Results for synechocystis_sp_pcc_6803_gca_000009725 written to species_compression_ratios_bact_looptest.csv\n",
      "Processing thermanaerovibrio_acidaminovorans_dsm_6589_gca_000024905...\n",
      "cds 1734 , total =  1703883\n",
      "The GECO3 compression ratio for Thermanaerovibrio_acidaminovorans_dsm_6589_gca_000024905.ASM2490v1.cds.all.fa is 22.39%\n",
      "The GZIP compression ratio for Thermanaerovibrio_acidaminovorans_dsm_6589_gca_000024905.ASM2490v1.cds.all.fa is 26.90%\n",
      "ncrna 63 , total =  18679\n",
      "The GECO3 compression ratio for Thermanaerovibrio_acidaminovorans_dsm_6589_gca_000024905.ASM2490v1.ncrna.fa is 12.92%\n",
      "The GZIP compression ratio for Thermanaerovibrio_acidaminovorans_dsm_6589_gca_000024905.ASM2490v1.ncrna.fa is 16.37%\n",
      "Results for thermanaerovibrio_acidaminovorans_dsm_6589_gca_000024905 written to species_compression_ratios_bact_looptest.csv\n",
      "Processing thermococcus_kodakarensis_kod1_gca_000009965...\n",
      "cds 2306 , total =  1921467\n",
      "The GECO3 compression ratio for Thermococcus_kodakarensis_kod1_gca_000009965.ASM996v1.cds.all.fa is 23.68%\n",
      "The GZIP compression ratio for Thermococcus_kodakarensis_kod1_gca_000009965.ASM996v1.cds.all.fa is 27.98%\n",
      "ncrna 51 , total =  8844\n",
      "The GECO3 compression ratio for Thermococcus_kodakarensis_kod1_gca_000009965.ASM996v1.ncrna.fa is 22.04%\n",
      "The GZIP compression ratio for Thermococcus_kodakarensis_kod1_gca_000009965.ASM996v1.ncrna.fa is 27.58%\n",
      "Results for thermococcus_kodakarensis_kod1_gca_000009965 written to species_compression_ratios_bact_looptest.csv\n",
      "Processing thermodesulfovibrio_yellowstonii_dsm_11347_gca_000020985...\n",
      "cds 2028 , total =  1873647\n",
      "The GECO3 compression ratio for Thermodesulfovibrio_yellowstonii_dsm_11347_gca_000020985.ASM2098v1.cds.all.fa is 22.44%\n",
      "The GZIP compression ratio for Thermodesulfovibrio_yellowstonii_dsm_11347_gca_000020985.ASM2098v1.cds.all.fa is 27.18%\n",
      "ncrna 54 , total =  9143\n",
      "The GECO3 compression ratio for Thermodesulfovibrio_yellowstonii_dsm_11347_gca_000020985.ASM2098v1.ncrna.fa is 24.81%\n",
      "The GZIP compression ratio for Thermodesulfovibrio_yellowstonii_dsm_11347_gca_000020985.ASM2098v1.ncrna.fa is 30.81%\n",
      "Results for thermodesulfovibrio_yellowstonii_dsm_11347_gca_000020985 written to species_compression_ratios_bact_looptest.csv\n",
      "Processing thermofilum_pendens_hrk_5_gca_000015225...\n",
      "cds 1876 , total =  1637931\n",
      "The GECO3 compression ratio for Thermofilum_pendens_hrk_5_gca_000015225.ASM1522v1.cds.all.fa is 23.46%\n",
      "The GZIP compression ratio for Thermofilum_pendens_hrk_5_gca_000015225.ASM1522v1.cds.all.fa is 27.85%\n",
      "ncrna 30 , total =  7145\n",
      "The GECO3 compression ratio for Thermofilum_pendens_hrk_5_gca_000015225.ASM1522v1.ncrna.fa is 23.36%\n",
      "The GZIP compression ratio for Thermofilum_pendens_hrk_5_gca_000015225.ASM1522v1.ncrna.fa is 29.31%\n",
      "Results for thermofilum_pendens_hrk_5_gca_000015225 written to species_compression_ratios_bact_looptest.csv\n",
      "Processing thermoplasma_acidophilum_dsm_1728_gca_000195915...\n",
      "cds 1478 , total =  1363746\n",
      "The GECO3 compression ratio for Thermoplasma_acidophilum_dsm_1728_gca_000195915.ASM19591v1.cds.all.fa is 24.05%\n",
      "The GZIP compression ratio for Thermoplasma_acidophilum_dsm_1728_gca_000195915.ASM19591v1.cds.all.fa is 28.11%\n",
      "ncrna 48 , total =  7151\n",
      "The GECO3 compression ratio for Thermoplasma_acidophilum_dsm_1728_gca_000195915.ASM19591v1.ncrna.fa is 25.06%\n",
      "The GZIP compression ratio for Thermoplasma_acidophilum_dsm_1728_gca_000195915.ASM19591v1.ncrna.fa is 32.02%\n",
      "Results for thermoplasma_acidophilum_dsm_1728_gca_000195915 written to species_compression_ratios_bact_looptest.csv\n",
      "Processing thermosynechococcus_elongatus_bp_1_gca_000011345...\n",
      "cds 2475 , total =  2330901\n",
      "The GECO3 compression ratio for Thermosynechococcus_elongatus_bp_1_gca_000011345.ASM1134v1.cds.all.fa is 23.02%\n",
      "The GZIP compression ratio for Thermosynechococcus_elongatus_bp_1_gca_000011345.ASM1134v1.cds.all.fa is 27.77%\n",
      "ncrna 49 , total =  8729\n",
      "The GECO3 compression ratio for Thermosynechococcus_elongatus_bp_1_gca_000011345.ASM1134v1.ncrna.fa is 25.42%\n",
      "The GZIP compression ratio for Thermosynechococcus_elongatus_bp_1_gca_000011345.ASM1134v1.ncrna.fa is 31.50%\n",
      "Results for thermosynechococcus_elongatus_bp_1_gca_000011345 written to species_compression_ratios_bact_looptest.csv\n",
      "Processing thermotoga_maritima_msb8_gca_000008545...\n",
      "cds 1846 , total =  1751016\n",
      "The GECO3 compression ratio for Thermotoga_maritima_msb8_gca_000008545.ASM854v1.cds.all.fa is 23.54%\n",
      "The GZIP compression ratio for Thermotoga_maritima_msb8_gca_000008545.ASM854v1.cds.all.fa is 27.83%\n",
      "ncrna 49 , total =  8199\n",
      "The GECO3 compression ratio for Thermotoga_maritima_msb8_gca_000008545.ASM854v1.ncrna.fa is 23.99%\n",
      "The GZIP compression ratio for Thermotoga_maritima_msb8_gca_000008545.ASM854v1.ncrna.fa is 29.76%\n",
      "Results for thermotoga_maritima_msb8_gca_000008545 written to species_compression_ratios_bact_looptest.csv\n",
      "Processing thermus_thermophilus_hb8_gca_000091545...\n",
      "cds 2238 , total =  2009249\n",
      "The GECO3 compression ratio for Thermus_thermophilus_hb8_gca_000091545.ASM9154v1.cds.all.fa is 20.63%\n",
      "The GZIP compression ratio for Thermus_thermophilus_hb8_gca_000091545.ASM9154v1.cds.all.fa is 25.36%\n",
      "ncrna 53 , total =  12788\n",
      "The GECO3 compression ratio for Thermus_thermophilus_hb8_gca_000091545.ASM9154v1.ncrna.fa is 15.88%\n",
      "The GZIP compression ratio for Thermus_thermophilus_hb8_gca_000091545.ASM9154v1.ncrna.fa is 19.78%\n",
      "Results for thermus_thermophilus_hb8_gca_000091545 written to species_compression_ratios_bact_looptest.csv\n",
      "Processing treponema_pallidum_subsp_pallidum_str_nichols_gca_000008605...\n",
      "cds 1031 , total =  1055058\n",
      "The GECO3 compression ratio for Treponema_pallidum_subsp_pallidum_str_nichols_gca_000008605.ASM860v1.cds.all.fa is 24.02%\n",
      "The GZIP compression ratio for Treponema_pallidum_subsp_pallidum_str_nichols_gca_000008605.ASM860v1.cds.all.fa is 28.25%\n",
      "ncrna 51 , total =  12369\n",
      "The GECO3 compression ratio for Treponema_pallidum_subsp_pallidum_str_nichols_gca_000008605.ASM860v1.ncrna.fa is 16.72%\n",
      "The GZIP compression ratio for Treponema_pallidum_subsp_pallidum_str_nichols_gca_000008605.ASM860v1.ncrna.fa is 20.93%\n",
      "Results for treponema_pallidum_subsp_pallidum_str_nichols_gca_000008605 written to species_compression_ratios_bact_looptest.csv\n",
      "Processing ureaplasma_parvum_serovar_3_str_atcc_700970_gca_000006625...\n",
      "cds 611 , total =  684984\n",
      "The GECO3 compression ratio for Ureaplasma_parvum_serovar_3_str_atcc_700970_gca_000006625.ASM662v1.cds.all.fa is 21.87%\n",
      "The GZIP compression ratio for Ureaplasma_parvum_serovar_3_str_atcc_700970_gca_000006625.ASM662v1.cds.all.fa is 26.27%\n",
      "ncrna 39 , total =  12136\n",
      "The GECO3 compression ratio for Ureaplasma_parvum_serovar_3_str_atcc_700970_gca_000006625.ASM662v1.ncrna.fa is 16.88%\n",
      "The GZIP compression ratio for Ureaplasma_parvum_serovar_3_str_atcc_700970_gca_000006625.ASM662v1.ncrna.fa is 21.24%\n",
      "Results for ureaplasma_parvum_serovar_3_str_atcc_700970_gca_000006625 written to species_compression_ratios_bact_looptest.csv\n",
      "Processing vibrio_cholerae_o1_biovar_el_tor_str_n16961_gca_000006745...\n",
      "cds 3828 , total =  3498146\n",
      "The GECO3 compression ratio for Vibrio_cholerae_o1_biovar_el_tor_str_n16961_gca_000006745.ASM674v1.cds.all.fa is 23.94%\n",
      "The GZIP compression ratio for Vibrio_cholerae_o1_biovar_el_tor_str_n16961_gca_000006745.ASM674v1.cds.all.fa is 28.08%\n",
      "ncrna 122 , total =  44018\n",
      "The GECO3 compression ratio for Vibrio_cholerae_o1_biovar_el_tor_str_n16961_gca_000006745.ASM674v1.ncrna.fa is 5.44%\n",
      "The GZIP compression ratio for Vibrio_cholerae_o1_biovar_el_tor_str_n16961_gca_000006745.ASM674v1.ncrna.fa is 7.22%\n",
      "Results for vibrio_cholerae_o1_biovar_el_tor_str_n16961_gca_000006745 written to species_compression_ratios_bact_looptest.csv\n",
      "Processing wolbachia_endosymbiont_of_drosophila_melanogaster_gca_000008025...\n",
      "cds 1195 , total =  1016925\n",
      "The GECO3 compression ratio for Wolbachia_endosymbiont_of_drosophila_melanogaster_gca_000008025.ASM802v1.cds.all.fa is 22.95%\n",
      "The GZIP compression ratio for Wolbachia_endosymbiont_of_drosophila_melanogaster_gca_000008025.ASM802v1.cds.all.fa is 27.63%\n",
      "ncrna 37 , total =  6937\n",
      "The GECO3 compression ratio for Wolbachia_endosymbiont_of_drosophila_melanogaster_gca_000008025.ASM802v1.ncrna.fa is 25.75%\n",
      "The GZIP compression ratio for Wolbachia_endosymbiont_of_drosophila_melanogaster_gca_000008025.ASM802v1.ncrna.fa is 32.35%\n",
      "Results for wolbachia_endosymbiont_of_drosophila_melanogaster_gca_000008025 written to species_compression_ratios_bact_looptest.csv\n",
      "Processing xanthomonas_campestris_pv_campestris_str_atcc_33913_gca_000007145...\n",
      "cds 4181 , total =  4310090\n",
      "The GECO3 compression ratio for Xanthomonas_campestris_pv_campestris_str_atcc_33913_gca_000007145.ASM714v1.cds.all.fa is 22.06%\n",
      "The GZIP compression ratio for Xanthomonas_campestris_pv_campestris_str_atcc_33913_gca_000007145.ASM714v1.cds.all.fa is 26.78%\n",
      "ncrna 61 , total =  14003\n",
      "The GECO3 compression ratio for Xanthomonas_campestris_pv_campestris_str_atcc_33913_gca_000007145.ASM714v1.ncrna.fa is 16.40%\n",
      "The GZIP compression ratio for Xanthomonas_campestris_pv_campestris_str_atcc_33913_gca_000007145.ASM714v1.ncrna.fa is 20.50%\n",
      "Results for xanthomonas_campestris_pv_campestris_str_atcc_33913_gca_000007145 written to species_compression_ratios_bact_looptest.csv\n",
      "Processing yersinia_pestis_biovar_microtus_str_91001_gca_000007885...\n",
      "cds 4016 , total =  3889239\n",
      "The GECO3 compression ratio for Yersinia_pestis_biovar_microtus_str_91001_gca_000007885.ASM788v1.cds.all.fa is 23.36%\n",
      "The GZIP compression ratio for Yersinia_pestis_biovar_microtus_str_91001_gca_000007885.ASM788v1.cds.all.fa is 27.90%\n",
      "ncrna 101 , total =  38967\n",
      "The GECO3 compression ratio for Yersinia_pestis_biovar_microtus_str_91001_gca_000007885.ASM788v1.ncrna.fa is 6.79%\n",
      "The GZIP compression ratio for Yersinia_pestis_biovar_microtus_str_91001_gca_000007885.ASM788v1.ncrna.fa is 9.03%\n",
      "Results for yersinia_pestis_biovar_microtus_str_91001_gca_000007885 written to species_compression_ratios_bact_looptest.csv\n",
      "Results saved to species_compression_ratios_bact_looptest.csv\n"
     ]
    }
   ],
   "source": [
    "# Usage bacteria\n",
    "\n",
    "ftp_server = \"ftp.ensemblgenomes.ebi.ac.uk\"\n",
    "base_ftp_directory = \"/pub/bacteria/current/fasta/bacteria_0_collection\"\n",
    "output_csv_file = \"compression_ratios_bact.csv\"\n",
    "\n",
    "process_all_organisms(ftp_server, base_ftp_directory, output_csv_file)\n",
    "\n",
    "print(f\"Results saved to {output_csv_file}\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "cds 18698 , total =  30229734\n",
      "The GECO3 compression ratio for Cavia_aperea.CavAp1.0.cds.all.fa is 18.17%\n",
      "The GZIP compression ratio for Cavia_aperea.CavAp1.0.cds.all.fa is 20.32%\n",
      "ncrna 3614 , total =  468177\n",
      "The GECO3 compression ratio for Cavia_aperea.CavAp1.0.ncrna.fa is 14.55%\n",
      "The GZIP compression ratio for Cavia_aperea.CavAp1.0.ncrna.fa is 19.30%\n",
      "cavia_aperea completed successfully!\n",
      "cds 25582 , total =  43333286\n",
      "The GECO3 compression ratio for Cavia_porcellus.Cavpor3.0.cds.all.fa is 18.22%\n",
      "The GZIP compression ratio for Cavia_porcellus.Cavpor3.0.cds.all.fa is 20.03%\n",
      "ncrna 9022 , total =  3529522\n",
      "The GECO3 compression ratio for Cavia_porcellus.Cavpor3.0.ncrna.fa is 18.70%\n",
      "The GZIP compression ratio for Cavia_porcellus.Cavpor3.0.ncrna.fa is 23.04%\n",
      "cavia_porcellus completed successfully!\n",
      "cds 31797 , total =  51770540\n",
      "The GECO3 compression ratio for Chelonoidis_abingdonii.ASM359739v1.cds.all.fa is 15.82%\n",
      "The GZIP compression ratio for Chelonoidis_abingdonii.ASM359739v1.cds.all.fa is 16.88%\n",
      "cds 29492 , total =  51616060\n",
      "The GECO3 compression ratio for Chelydra_serpentina.Chelydra_serpentina-1.0.cds.all.fa is 15.62%\n",
      "The GZIP compression ratio for Chelydra_serpentina.Chelydra_serpentina-1.0.cds.all.fa is 17.84%\n",
      "ncrna 2954 , total =  694349\n",
      "The GECO3 compression ratio for Chelonoidis_abingdonii.ASM359739v1.ncrna.fa is 21.29%\n",
      "The GZIP compression ratio for Chelonoidis_abingdonii.ASM359739v1.ncrna.fa is 25.50%\n",
      "chelonoidis_abingdonii completed successfully!\n",
      "ncrna 1036 , total =  271082\n",
      "The GECO3 compression ratio for Chelydra_serpentina.Chelydra_serpentina-1.0.ncrna.fa is 15.78%\n",
      "The GZIP compression ratio for Chelydra_serpentina.Chelydra_serpentina-1.0.ncrna.fa is 19.34%\n",
      "chelydra_serpentina completed successfully!\n",
      "cds 29200 , total =  59452038\n",
      "The GECO3 compression ratio for Catharus_ustulatus.bCatUst1.pri.cds.all.fa is 13.92%\n",
      "The GZIP compression ratio for Catharus_ustulatus.bCatUst1.pri.cds.all.fa is 14.20%\n",
      "ncrna 858 , total =  654933\n",
      "The GECO3 compression ratio for Catharus_ustulatus.bCatUst1.pri.ncrna.fa is 18.74%\n",
      "The GZIP compression ratio for Catharus_ustulatus.bCatUst1.pri.ncrna.fa is 22.98%\n",
      "catharus_ustulatus completed successfully!\n",
      "cds 40677 , total =  66097527\n",
      "The GECO3 compression ratio for Cebus_imitator.Cebus_imitator-1.0.cds.all.fa is 15.31%\n",
      "The GZIP compression ratio for Cebus_imitator.Cebus_imitator-1.0.cds.all.fa is 15.40%\n",
      "ncrna 7157 , total =  1403327\n",
      "The GECO3 compression ratio for Cebus_imitator.Cebus_imitator-1.0.ncrna.fa is 13.67%\n",
      "The GZIP compression ratio for Cebus_imitator.Cebus_imitator-1.0.ncrna.fa is 19.53%\n",
      "cebus_imitator completed successfully!\n",
      "cds 41224 , total =  72590107\n",
      "The GECO3 compression ratio for Cervus_hanglu_yarkandensis.CEY_v1.cds.all.fa is 14.89%\n",
      "The GZIP compression ratio for Cervus_hanglu_yarkandensis.CEY_v1.cds.all.fa is 15.00%\n",
      "cds 45963 , total =  75758979\n",
      "The GECO3 compression ratio for Cercocebus_atys.Caty_1.0.cds.all.fa is 14.62%\n",
      "The GZIP compression ratio for Cercocebus_atys.Caty_1.0.cds.all.fa is 14.06%\n",
      "cds 12435 , total =  16377944\n",
      "The GECO3 compression ratio for Choloepus_hoffmanni.choHof1.cds.all.fa is 23.51%\n",
      "The GZIP compression ratio for Choloepus_hoffmanni.choHof1.cds.all.fa is 27.89%\n",
      "ncrna 4968 , total =  2040731\n",
      "The GECO3 compression ratio for Cervus_hanglu_yarkandensis.CEY_v1.ncrna.fa is 17.28%\n",
      "The GZIP compression ratio for Cervus_hanglu_yarkandensis.CEY_v1.ncrna.fa is 21.51%\n",
      "cervus_hanglu_yarkandensis completed successfully!\n",
      "ncrna 2030 , total =  227623\n",
      "The GECO3 compression ratio for Choloepus_hoffmanni.choHof1.ncrna.fa is 16.99%\n",
      "The GZIP compression ratio for Choloepus_hoffmanni.choHof1.ncrna.fa is 21.68%\n",
      "choloepus_hoffmanni completed successfully!\n",
      "ncrna 7318 , total =  1833093\n",
      "The GECO3 compression ratio for Cercocebus_atys.Caty_1.0.ncrna.fa is 17.74%\n",
      "The GZIP compression ratio for Cercocebus_atys.Caty_1.0.ncrna.fa is 22.43%\n",
      "cercocebus_atys completed successfully!\n",
      "cds 25983 , total =  45719399\n",
      "The GECO3 compression ratio for Chinchilla_lanigera.ChiLan1.0.cds.all.fa is 17.78%\n",
      "The GZIP compression ratio for Chinchilla_lanigera.ChiLan1.0.cds.all.fa is 19.24%\n",
      "cds 19204 , total =  31307734\n",
      "The GECO3 compression ratio for Chlorocebus_sabaeus.ChlSab1.1.cds.all.fa is 23.36%\n",
      "The GZIP compression ratio for Chlorocebus_sabaeus.ChlSab1.1.cds.all.fa is 27.61%\n",
      "cds 17302 , total =  17323469\n",
      "The GECO3 compression ratio for Ciona_intestinalis.KH.cds.all.fa is 22.62%\n",
      "The GZIP compression ratio for Ciona_intestinalis.KH.cds.all.fa is 26.30%\n",
      "ncrna 8248 , total =  1045072\n",
      "The GECO3 compression ratio for Chlorocebus_sabaeus.ChlSab1.1.ncrna.fa is 17.02%\n",
      "The GZIP compression ratio for Chlorocebus_sabaeus.ChlSab1.1.ncrna.fa is 22.94%\n",
      "chlorocebus_sabaeus completed successfully!\n",
      "ncrna 455 , total =  40854\n",
      "The GECO3 compression ratio for Ciona_intestinalis.KH.ncrna.fa is 16.89%\n",
      "The GZIP compression ratio for Ciona_intestinalis.KH.ncrna.fa is 21.45%\n",
      "ciona_intestinalis completed successfully!\n",
      "ncrna 13073 , total =  9826458\n",
      "The GECO3 compression ratio for Chinchilla_lanigera.ChiLan1.0.ncrna.fa is 19.97%\n",
      "The GZIP compression ratio for Chinchilla_lanigera.ChiLan1.0.ncrna.fa is 23.78%\n",
      "chinchilla_lanigera completed successfully!\n",
      "cds 18806 , total =  40882079\n",
      "The GECO3 compression ratio for Chrysolophus_pictus.Chrysolophus_pictus_GenomeV1.0.cds.all.fa is 14.34%\n",
      "The GZIP compression ratio for Chrysolophus_pictus.Chrysolophus_pictus_GenomeV1.0.cds.all.fa is 15.56%\n",
      "cds 20155 , total =  29089779\n",
      "The GECO3 compression ratio for Ciona_savignyi.CSAV2.0.cds.all.fa is 12.70%\n",
      "The GZIP compression ratio for Ciona_savignyi.CSAV2.0.cds.all.fa is 14.70%\n",
      "ncrna 3121 , total =  2701245\n",
      "The GECO3 compression ratio for Chrysolophus_pictus.Chrysolophus_pictus_GenomeV1.0.ncrna.fa is 20.26%\n",
      "The GZIP compression ratio for Chrysolophus_pictus.Chrysolophus_pictus_GenomeV1.0.ncrna.fa is 24.34%\n",
      "chrysolophus_pictus completed successfully!\n",
      "ncrna 340 , total =  42199\n",
      "The GECO3 compression ratio for Ciona_savignyi.CSAV2.0.ncrna.fa is 11.03%\n",
      "The GZIP compression ratio for Ciona_savignyi.CSAV2.0.ncrna.fa is 14.87%\n",
      "ciona_savignyi completed successfully!\n",
      "cds 39084 , total =  75595698\n",
      "The GECO3 compression ratio for Chrysemys_picta_bellii.Chrysemys_picta_bellii-3.0.3.cds.all.fa is 13.84%\n",
      "The GZIP compression ratio for Chrysemys_picta_bellii.Chrysemys_picta_bellii-3.0.3.cds.all.fa is 13.54%\n",
      "ncrna 7038 , total =  4989031\n",
      "The GECO3 compression ratio for Chrysemys_picta_bellii.Chrysemys_picta_bellii-3.0.3.ncrna.fa is 18.30%\n",
      "The GZIP compression ratio for Chrysemys_picta_bellii.Chrysemys_picta_bellii-3.0.3.ncrna.fa is 23.00%\n",
      "chrysemys_picta_bellii completed successfully!\n",
      "cds 40319 , total =  63867026\n",
      "The GECO3 compression ratio for Colobus_angolensis_palliatus.Cang.pa_1.0.cds.all.fa is 15.48%\n",
      "The GZIP compression ratio for Colobus_angolensis_palliatus.Cang.pa_1.0.cds.all.fa is 15.78%\n",
      "ncrna 6138 , total =  815703\n",
      "The GECO3 compression ratio for Colobus_angolensis_palliatus.Cang.pa_1.0.ncrna.fa is 16.57%\n",
      "The GZIP compression ratio for Colobus_angolensis_palliatus.Cang.pa_1.0.ncrna.fa is 22.40%\n",
      "colobus_angolensis_palliatus completed successfully!\n",
      "cds 28454 , total =  62151053\n",
      "The GECO3 compression ratio for Corvus_moneduloides.bCorMon1.pri.cds.all.fa is 13.97%\n",
      "The GZIP compression ratio for Corvus_moneduloides.bCorMon1.pri.cds.all.fa is 14.04%\n",
      "cds 26985 , total =  42953065\n",
      "The GECO3 compression ratio for Cricetulus_griseus_crigri.CriGri_1.0.cds.all.fa is 18.60%\n",
      "The GZIP compression ratio for Cricetulus_griseus_crigri.CriGri_1.0.cds.all.fa is 20.49%\n",
      "cds 27883 , total =  59510114\n",
      "The GECO3 compression ratio for Coturnix_japonica.Coturnix_japonica_2.0.cds.all.fa is 13.90%\n",
      "The GZIP compression ratio for Coturnix_japonica.Coturnix_japonica_2.0.cds.all.fa is 14.07%\n",
      "ncrna 1857 , total =  2127370\n",
      "The GECO3 compression ratio for Corvus_moneduloides.bCorMon1.pri.ncrna.fa is 18.28%\n",
      "The GZIP compression ratio for Corvus_moneduloides.bCorMon1.pri.ncrna.fa is 22.60%\n",
      "corvus_moneduloides completed successfully!\n",
      "ncrna 7041 , total =  3454577\n",
      "The GECO3 compression ratio for Cricetulus_griseus_crigri.CriGri_1.0.ncrna.fa is 20.36%\n",
      "The GZIP compression ratio for Cricetulus_griseus_crigri.CriGri_1.0.ncrna.fa is 24.46%\n",
      "cricetulus_griseus_crigri completed successfully!\n",
      "cds 28327 , total =  48259665\n",
      "The GECO3 compression ratio for Cricetulus_griseus_chok1gshd.CHOK1GS_HDv1.cds.all.fa is 18.49%\n",
      "The GZIP compression ratio for Cricetulus_griseus_chok1gshd.CHOK1GS_HDv1.cds.all.fa is 20.14%\n",
      "ncrna 4142 , total =  565982\n",
      "The GECO3 compression ratio for Cricetulus_griseus_chok1gshd.CHOK1GS_HDv1.ncrna.fa is 18.44%\n",
      "The GZIP compression ratio for Cricetulus_griseus_chok1gshd.CHOK1GS_HDv1.ncrna.fa is 23.22%\n",
      "cricetulus_griseus_chok1gshd completed successfully!\n",
      "ncrna 9581 , total =  8920625\n",
      "The GECO3 compression ratio for Coturnix_japonica.Coturnix_japonica_2.0.ncrna.fa is 17.66%\n",
      "The GZIP compression ratio for Coturnix_japonica.Coturnix_japonica_2.0.ncrna.fa is 20.94%\n",
      "coturnix_japonica completed successfully!\n",
      "cds 57114 , total =  111310745\n",
      "The GECO3 compression ratio for Cottoperca_gobio.fCotGob3.1.cds.all.fa is 12.52%\n",
      "The GZIP compression ratio for Cottoperca_gobio.fCotGob3.1.cds.all.fa is 10.56%\n",
      "cds 62913 , total =  128272519\n",
      "The GECO3 compression ratio for Clupea_harengus.Ch_v2.0.2.cds.all.fa is 12.90%\n",
      "The GZIP compression ratio for Clupea_harengus.Ch_v2.0.2.cds.all.fa is 10.51%\n",
      "ncrna 3419 , total =  3557151\n",
      "The GECO3 compression ratio for Cottoperca_gobio.fCotGob3.1.ncrna.fa is 17.24%\n",
      "The GZIP compression ratio for Cottoperca_gobio.fCotGob3.1.ncrna.fa is 21.30%\n",
      "cottoperca_gobio completed successfully!\n",
      "ncrna 4572 , total =  5317017\n",
      "The GECO3 compression ratio for Clupea_harengus.Ch_v2.0.2.ncrna.fa is 16.31%\n",
      "The GZIP compression ratio for Clupea_harengus.Ch_v2.0.2.ncrna.fa is 20.55%\n",
      "clupea_harengus completed successfully!\n",
      "cds 27731 , total =  56779316\n",
      "The GECO3 compression ratio for Crocodylus_porosus.CroPor_comp1.cds.all.fa is 13.90%\n",
      "The GZIP compression ratio for Crocodylus_porosus.CroPor_comp1.cds.all.fa is 14.13%\n",
      "ncrna 4618 , total =  2611539\n",
      "The GECO3 compression ratio for Crocodylus_porosus.CroPor_comp1.ncrna.fa is 20.13%\n",
      "The GZIP compression ratio for Crocodylus_porosus.CroPor_comp1.ncrna.fa is 23.99%\n",
      "crocodylus_porosus completed successfully!\n",
      "cds 28345 , total =  57340131\n",
      "The GECO3 compression ratio for Cyanistes_caeruleus.cyaCae2.cds.all.fa is 13.64%\n",
      "The GZIP compression ratio for Cyanistes_caeruleus.cyaCae2.cds.all.fa is 13.66%\n",
      "cds 33996 , total =  58389908\n",
      "The GECO3 compression ratio for Cynoglossus_semilaevis.Cse_v1.0.cds.all.fa is 17.16%\n",
      "The GZIP compression ratio for Cynoglossus_semilaevis.Cse_v1.0.cds.all.fa is 17.82%\n",
      "ncrna 412 , total =  50402\n",
      "The GECO3 compression ratio for Cynoglossus_semilaevis.Cse_v1.0.ncrna.fa is 18.21%\n",
      "The GZIP compression ratio for Cynoglossus_semilaevis.Cse_v1.0.ncrna.fa is 22.43%\n",
      "cynoglossus_semilaevis completed successfully!\n",
      "cds 51113 , total =  86890762\n",
      "The GECO3 compression ratio for Cricetulus_griseus_picr.CriGri-PICRH-1.0.cds.all.fa is 14.18%\n",
      "The GZIP compression ratio for Cricetulus_griseus_picr.CriGri-PICRH-1.0.cds.all.fa is 13.39%\n",
      "cds 32924 , total =  53995863\n",
      "The GECO3 compression ratio for Cyprinodon_variegatus.C_variegatus-1.0.cds.all.fa is 18.13%\n",
      "The GZIP compression ratio for Cyprinodon_variegatus.C_variegatus-1.0.cds.all.fa is 19.42%\n",
      "ncrna 404 , total =  47873\n",
      "The GECO3 compression ratio for Cyprinodon_variegatus.C_variegatus-1.0.ncrna.fa is 20.27%\n",
      "The GZIP compression ratio for Cyprinodon_variegatus.C_variegatus-1.0.ncrna.fa is 25.16%\n",
      "cyprinodon_variegatus completed successfully!\n",
      "ncrna 13386 , total =  12611515\n",
      "The GECO3 compression ratio for Cyanistes_caeruleus.cyaCae2.ncrna.fa is 18.66%\n",
      "The GZIP compression ratio for Cyanistes_caeruleus.cyaCae2.ncrna.fa is 22.05%\n",
      "cyanistes_caeruleus completed successfully!\n",
      "cds 51131 , total =  96988770\n",
      "The GECO3 compression ratio for Cyclopterus_lumpus.fCycLum1.pri.cds.all.fa is 13.11%\n",
      "The GZIP compression ratio for Cyclopterus_lumpus.fCycLum1.pri.cds.all.fa is 11.62%\n",
      "ncrna 1424 , total =  458994\n",
      "The GECO3 compression ratio for Cyclopterus_lumpus.fCycLum1.pri.ncrna.fa is 14.70%\n",
      "The GZIP compression ratio for Cyclopterus_lumpus.fCycLum1.pri.ncrna.fa is 18.13%\n",
      "cyclopterus_lumpus completed successfully!\n",
      "ncrna 15950 , total =  28249888\n",
      "The GECO3 compression ratio for Cricetulus_griseus_picr.CriGri-PICRH-1.0.ncrna.fa is 14.37%\n",
      "The GZIP compression ratio for Cricetulus_griseus_picr.CriGri-PICRH-1.0.ncrna.fa is 17.88%\n",
      "cricetulus_griseus_picr completed successfully!\n",
      "cds 26551 , total =  39997345\n",
      "The GECO3 compression ratio for Dasypus_novemcinctus.Dasnov3.0.cds.all.fa is 20.48%\n",
      "The GZIP compression ratio for Dasypus_novemcinctus.Dasnov3.0.cds.all.fa is 24.09%\n",
      "ncrna 9672 , total =  4031874\n",
      "The GECO3 compression ratio for Dasypus_novemcinctus.Dasnov3.0.ncrna.fa is 20.10%\n",
      "The GZIP compression ratio for Dasypus_novemcinctus.Dasnov3.0.ncrna.fa is 24.73%\n",
      "dasypus_novemcinctus completed successfully!\n",
      "cds 52089 , total =  80549133\n",
      "The GECO3 compression ratio for Danio_rerio.GRCz11.cds.all.fa is 15.86%\n",
      "The GZIP compression ratio for Danio_rerio.GRCz11.cds.all.fa is 18.03%\n",
      "cds 30992 , total =  61555576\n",
      "The GECO3 compression ratio for Delphinapterus_leucas.ASM228892v3.cds.all.fa is 14.92%\n",
      "The GZIP compression ratio for Delphinapterus_leucas.ASM228892v3.cds.all.fa is 15.41%\n",
      "ncrna 8115 , total =  5231392\n",
      "The GECO3 compression ratio for Danio_rerio.GRCz11.ncrna.fa is 15.83%\n",
      "The GZIP compression ratio for Danio_rerio.GRCz11.ncrna.fa is 21.10%\n",
      "danio_rerio completed successfully!\n",
      "ncrna 2559 , total =  1047101\n",
      "The GECO3 compression ratio for Delphinapterus_leucas.ASM228892v3.ncrna.fa is 19.25%\n",
      "The GZIP compression ratio for Delphinapterus_leucas.ASM228892v3.ncrna.fa is 23.60%\n",
      "delphinapterus_leucas completed successfully!\n",
      "cds 23962 , total =  39475984\n",
      "The GECO3 compression ratio for Dipodomys_ordii.Dord_2.0.cds.all.fa is 17.82%\n",
      "The GZIP compression ratio for Dipodomys_ordii.Dord_2.0.cds.all.fa is 19.46%\n",
      "ncrna 3317 , total =  441474\n",
      "The GECO3 compression ratio for Dipodomys_ordii.Dord_2.0.ncrna.fa is 16.07%\n",
      "The GZIP compression ratio for Dipodomys_ordii.Dord_2.0.ncrna.fa is 20.74%\n",
      "dipodomys_ordii completed successfully!\n",
      "cds 27685 , total =  54875047\n",
      "The GECO3 compression ratio for Dromaius_novaehollandiae.droNov1.cds.all.fa is 14.36%\n",
      "The GZIP compression ratio for Dromaius_novaehollandiae.droNov1.cds.all.fa is 14.83%\n",
      "cds 63912 , total =  125508176\n",
      "The GECO3 compression ratio for Denticeps_clupeoides.fDenClu1.1.cds.all.fa is 12.33%\n",
      "The GZIP compression ratio for Denticeps_clupeoides.fDenClu1.1.cds.all.fa is 10.51%\n",
      "ncrna 8304 , total =  3584172\n",
      "The GECO3 compression ratio for Denticeps_clupeoides.fDenClu1.1.ncrna.fa is 12.99%\n",
      "The GZIP compression ratio for Denticeps_clupeoides.fDenClu1.1.ncrna.fa is 17.13%\n",
      "denticeps_clupeoides completed successfully!\n",
      "ncrna 5675 , total =  4298096\n",
      "The GECO3 compression ratio for Dromaius_novaehollandiae.droNov1.ncrna.fa is 18.97%\n",
      "The GZIP compression ratio for Dromaius_novaehollandiae.droNov1.ncrna.fa is 22.62%\n",
      "dromaius_novaehollandiae completed successfully!\n",
      "cds 16575 , total =  22120505\n",
      "The GECO3 compression ratio for Echinops_telfairi.TENREC.cds.all.fa is 23.52%\n",
      "The GZIP compression ratio for Echinops_telfairi.TENREC.cds.all.fa is 27.93%\n",
      "ncrna 5680 , total =  741613\n",
      "The GECO3 compression ratio for Echinops_telfairi.TENREC.ncrna.fa is 14.90%\n",
      "The GZIP compression ratio for Echinops_telfairi.TENREC.ncrna.fa is 20.85%\n",
      "echinops_telfairi completed successfully!\n",
      "cds 30799 , total =  61218290\n",
      "The GECO3 compression ratio for Drosophila_melanogaster.BDGP6.46.cds.all.fa is 11.60%\n",
      "The GZIP compression ratio for Drosophila_melanogaster.BDGP6.46.cds.all.fa is 12.37%\n",
      "cds 67280 , total =  148686109\n",
      "The GECO3 compression ratio for Dicentrarchus_labrax.dlabrax2021.cds.all.fa is 12.61%\n",
      "The GZIP compression ratio for Dicentrarchus_labrax.dlabrax2021.cds.all.fa is 9.55%\n",
      "ncrna 2158 , total =  2229695\n",
      "The GECO3 compression ratio for Dicentrarchus_labrax.dlabrax2021.ncrna.fa is 16.60%\n",
      "The GZIP compression ratio for Dicentrarchus_labrax.dlabrax2021.ncrna.fa is 20.84%\n",
      "dicentrarchus_labrax completed successfully!\n",
      "ncrna 4557 , total =  3469581\n",
      "The GECO3 compression ratio for Drosophila_melanogaster.BDGP6.46.ncrna.fa is 18.15%\n",
      "The GZIP compression ratio for Drosophila_melanogaster.BDGP6.46.ncrna.fa is 21.96%\n",
      "drosophila_melanogaster completed successfully!\n",
      "cds 120282 , total =  236887019\n",
      "The GECO3 compression ratio for Cyprinus_carpio_hebaored.Hebao_red_carp_1.0.cds.all.fa is 13.61%\n",
      "The GZIP compression ratio for Cyprinus_carpio_hebaored.Hebao_red_carp_1.0.cds.all.fa is 10.42%\n",
      "cds 122277 , total =  249444852\n",
      "The GECO3 compression ratio for Cyprinus_carpio_germanmirror.German_Mirror_carp_1.0.cds.all.fa is 13.58%\n",
      "The GZIP compression ratio for Cyprinus_carpio_germanmirror.German_Mirror_carp_1.0.cds.all.fa is 10.22%\n",
      "ncrna 9931 , total =  12127743\n",
      "The GECO3 compression ratio for Cyprinus_carpio_hebaored.Hebao_red_carp_1.0.ncrna.fa is 18.10%\n",
      "The GZIP compression ratio for Cyprinus_carpio_hebaored.Hebao_red_carp_1.0.ncrna.fa is 22.08%\n",
      "cyprinus_carpio_hebaored completed successfully!\n",
      "ncrna 9000 , total =  10849580\n",
      "The GECO3 compression ratio for Cyprinus_carpio_germanmirror.German_Mirror_carp_1.0.ncrna.fa is 18.16%\n",
      "The GZIP compression ratio for Cyprinus_carpio_germanmirror.German_Mirror_carp_1.0.ncrna.fa is 22.39%\n",
      "cyprinus_carpio_germanmirror completed successfully!\n",
      "cds 123157 , total =  249465714\n",
      "The GECO3 compression ratio for Cyprinus_carpio_huanghe.Hunaghe_carp_2.0.cds.all.fa is 13.63%\n",
      "The GZIP compression ratio for Cyprinus_carpio_huanghe.Hunaghe_carp_2.0.cds.all.fa is 10.17%\n",
      "ncrna 3310 , total =  4440580\n",
      "The GECO3 compression ratio for Cyprinus_carpio_huanghe.Hunaghe_carp_2.0.ncrna.fa is 18.27%\n",
      "The GZIP compression ratio for Cyprinus_carpio_huanghe.Hunaghe_carp_2.0.ncrna.fa is 22.16%\n",
      "cyprinus_carpio_huanghe completed successfully!\n",
      "cds 27960 , total =  42227287\n",
      "The GECO3 compression ratio for Eptatretus_burgeri.Eburgeri_3.2.cds.all.fa is 15.27%\n",
      "The GZIP compression ratio for Eptatretus_burgeri.Eburgeri_3.2.cds.all.fa is 16.64%\n",
      "ncrna 1089 , total =  1652653\n",
      "The GECO3 compression ratio for Eptatretus_burgeri.Eburgeri_3.2.ncrna.fa is 15.53%\n",
      "The GZIP compression ratio for Eptatretus_burgeri.Eburgeri_3.2.ncrna.fa is 18.94%\n",
      "eptatretus_burgeri completed successfully!\n",
      "cds 16021 , total =  21731889\n",
      "The GECO3 compression ratio for Equus_caballus.EquCab3.0.cds.all.fa is 9.01%\n",
      "The GZIP compression ratio for Equus_caballus.EquCab3.0.cds.all.fa is 12.65%\n",
      "cds 14605 , total =  20013639\n",
      "The GECO3 compression ratio for Erinaceus_europaeus.HEDGEHOG.cds.all.fa is 23.55%\n",
      "The GZIP compression ratio for Erinaceus_europaeus.HEDGEHOG.cds.all.fa is 27.92%\n",
      "cds 55084 , total =  110312639\n",
      "The GECO3 compression ratio for Echeneis_naucrates.fEcheNa1.1.cds.all.fa is 12.81%\n",
      "The GZIP compression ratio for Echeneis_naucrates.fEcheNa1.1.cds.all.fa is 10.94%\n",
      "ncrna 6456 , total =  1092020\n",
      "The GECO3 compression ratio for Erinaceus_europaeus.HEDGEHOG.ncrna.fa is 14.35%\n",
      "The GZIP compression ratio for Erinaceus_europaeus.HEDGEHOG.ncrna.fa is 20.11%\n",
      "erinaceus_europaeus completed successfully!\n",
      "ncrna 1016 , total =  502868\n",
      "The GECO3 compression ratio for Echeneis_naucrates.fEcheNa1.1.ncrna.fa is 15.40%\n",
      "The GZIP compression ratio for Echeneis_naucrates.fEcheNa1.1.ncrna.fa is 18.75%\n",
      "echeneis_naucrates completed successfully!\n",
      "cds 145576 , total =  304781730\n",
      "The GECO3 compression ratio for Cyprinus_carpio_carpio.Cypcar_WagV4.0.cds.all.fa is 13.68%\n",
      "The GZIP compression ratio for Cyprinus_carpio_carpio.Cypcar_WagV4.0.cds.all.fa is 9.09%\n",
      "cds 50993 , total =  88062387\n",
      "The GECO3 compression ratio for Electrophorus_electricus.Ee_SOAP_WITH_SSPACE.cds.all.fa is 13.51%\n",
      "The GZIP compression ratio for Electrophorus_electricus.Ee_SOAP_WITH_SSPACE.cds.all.fa is 12.69%\n",
      "ncrna 304 , total =  36683\n",
      "The GECO3 compression ratio for Electrophorus_electricus.Ee_SOAP_WITH_SSPACE.ncrna.fa is 20.33%\n",
      "The GZIP compression ratio for Electrophorus_electricus.Ee_SOAP_WITH_SSPACE.ncrna.fa is 25.27%\n",
      "electrophorus_electricus completed successfully!\n",
      "cds 8482 , total =  10863178\n",
      "The GECO3 compression ratio for Felis_catus.Felis_catus_9.0.cds.all.fa is 13.88%\n",
      "The GZIP compression ratio for Felis_catus.Felis_catus_9.0.cds.all.fa is 17.64%\n",
      "cds 51240 , total =  85984997\n",
      "The GECO3 compression ratio for Equus_asinus.ASM1607732v2.cds.all.fa is 13.92%\n",
      "The GZIP compression ratio for Equus_asinus.ASM1607732v2.cds.all.fa is 12.97%\n",
      "ncrna 16536 , total =  24217630\n",
      "The GECO3 compression ratio for Cyprinus_carpio_carpio.Cypcar_WagV4.0.ncrna.fa is 15.07%\n",
      "The GZIP compression ratio for Cyprinus_carpio_carpio.Cypcar_WagV4.0.ncrna.fa is 19.87%\n",
      "cyprinus_carpio_carpio completed successfully!\n",
      "cds 23847 , total =  46523060\n",
      "The GECO3 compression ratio for Erythrura_gouldiae.GouldianFinch.cds.all.fa is 15.59%\n",
      "The GZIP compression ratio for Erythrura_gouldiae.GouldianFinch.cds.all.fa is 16.76%\n",
      "ncrna 13627 , total =  12371298\n",
      "The GECO3 compression ratio for Felis_catus.Felis_catus_9.0.ncrna.fa is 19.22%\n",
      "The GZIP compression ratio for Felis_catus.Felis_catus_9.0.ncrna.fa is 23.08%\n",
      "felis_catus completed successfully!\n",
      "ncrna 3675 , total =  3212428\n",
      "The GECO3 compression ratio for Erythrura_gouldiae.GouldianFinch.ncrna.fa is 18.62%\n",
      "The GZIP compression ratio for Erythrura_gouldiae.GouldianFinch.ncrna.fa is 22.74%\n",
      "erythrura_gouldiae completed successfully!\n",
      "cds 34135 , total =  55636600\n",
      "The GECO3 compression ratio for Erpetoichthys_calabaricus.fErpCal1.1.cds.all.fa is 16.85%\n",
      "The GZIP compression ratio for Erpetoichthys_calabaricus.fErpCal1.1.cds.all.fa is 18.57%\n",
      "ncrna 670 , total =  132829\n",
      "The GECO3 compression ratio for Erpetoichthys_calabaricus.fErpCal1.1.ncrna.fa is 12.76%\n",
      "The GZIP compression ratio for Erpetoichthys_calabaricus.fErpCal1.1.ncrna.fa is 16.61%\n",
      "erpetoichthys_calabaricus completed successfully!\n",
      "cds 25330 , total =  54549405\n",
      "The GECO3 compression ratio for Falco_tinnunculus.FalTin1.0.cds.all.fa is 14.52%\n",
      "The GZIP compression ratio for Falco_tinnunculus.FalTin1.0.cds.all.fa is 15.03%\n",
      "ncrna 18393 , total =  30943062\n",
      "The GECO3 compression ratio for Equus_asinus.ASM1607732v2.ncrna.fa is 15.81%\n",
      "The GZIP compression ratio for Equus_asinus.ASM1607732v2.ncrna.fa is 18.97%\n",
      "equus_asinus completed successfully!\n",
      "ncrna 1002 , total =  819074\n",
      "The GECO3 compression ratio for Falco_tinnunculus.FalTin1.0.ncrna.fa is 20.65%\n",
      "The GZIP compression ratio for Falco_tinnunculus.FalTin1.0.ncrna.fa is 24.99%\n",
      "falco_tinnunculus completed successfully!\n",
      "ncrna 39114 , total =  80443261\n",
      "The GECO3 compression ratio for Equus_caballus.EquCab3.0.ncrna.fa is 16.42%\n",
      "The GZIP compression ratio for Equus_caballus.EquCab3.0.ncrna.fa is 17.64%\n",
      "equus_caballus completed successfully!\n",
      "cds 23413 , total =  38668210\n",
      "The GECO3 compression ratio for Fukomys_damarensis.DMR_v1.0.cds.all.fa is 18.96%\n",
      "The GZIP compression ratio for Fukomys_damarensis.DMR_v1.0.cds.all.fa is 21.29%\n",
      "cds 29825 , total =  62219939\n",
      "The GECO3 compression ratio for Ficedula_albicollis.FicAlb1.5.cds.all.fa is 13.39%\n",
      "The GZIP compression ratio for Ficedula_albicollis.FicAlb1.5.cds.all.fa is 13.04%\n",
      "ncrna 17997 , total =  13059411\n",
      "The GECO3 compression ratio for Fukomys_damarensis.DMR_v1.0.ncrna.fa is 21.27%\n",
      "The GZIP compression ratio for Fukomys_damarensis.DMR_v1.0.ncrna.fa is 25.19%\n",
      "fukomys_damarensis completed successfully!\n",
      "ncrna 3366 , total =  4710743\n",
      "The GECO3 compression ratio for Ficedula_albicollis.FicAlb1.5.ncrna.fa is 16.74%\n",
      "The GZIP compression ratio for Ficedula_albicollis.FicAlb1.5.ncrna.fa is 20.35%\n",
      "ficedula_albicollis completed successfully!\n",
      "cds 35197 , total =  56328899\n",
      "The GECO3 compression ratio for Fundulus_heteroclitus.Fundulus_heteroclitus-3.0.2.cds.all.fa is 17.77%\n",
      "The GZIP compression ratio for Fundulus_heteroclitus.Fundulus_heteroclitus-3.0.2.cds.all.fa is 18.74%\n",
      "ncrna 372 , total =  44996\n",
      "The GECO3 compression ratio for Fundulus_heteroclitus.Fundulus_heteroclitus-3.0.2.ncrna.fa is 20.20%\n",
      "The GZIP compression ratio for Fundulus_heteroclitus.Fundulus_heteroclitus-3.0.2.ncrna.fa is 24.85%\n",
      "fundulus_heteroclitus completed successfully!\n",
      "cds 44876 , total =  86507422\n",
      "The GECO3 compression ratio for Gallus_gallus.bGalGal1.mat.broiler.GRCg7b.cds.all.fa is 12.44%\n",
      "The GZIP compression ratio for Gallus_gallus.bGalGal1.mat.broiler.GRCg7b.cds.all.fa is 11.58%\n",
      "cds 27576 , total =  40771380\n",
      "The GECO3 compression ratio for Gasterosteus_aculeatus.BROADS1.cds.all.fa is 20.17%\n",
      "The GZIP compression ratio for Gasterosteus_aculeatus.BROADS1.cds.all.fa is 22.72%\n",
      "cds 72047 , total =  152071530\n",
      "The GECO3 compression ratio for Esox_lucius.Eluc_v4.cds.all.fa is 12.82%\n",
      "The GZIP compression ratio for Esox_lucius.Eluc_v4.cds.all.fa is 9.83%\n",
      "ncrna 1617 , total =  195070\n",
      "The GECO3 compression ratio for Gasterosteus_aculeatus.BROADS1.ncrna.fa is 8.79%\n",
      "The GZIP compression ratio for Gasterosteus_aculeatus.BROADS1.ncrna.fa is 10.94%\n",
      "gasterosteus_aculeatus completed successfully!\n",
      "ncrna 6897 , total =  3283161\n",
      "The GECO3 compression ratio for Esox_lucius.Eluc_v4.ncrna.fa is 12.70%\n",
      "The GZIP compression ratio for Esox_lucius.Eluc_v4.ncrna.fa is 15.88%\n",
      "esox_lucius completed successfully!\n",
      "cds 32756 , total =  54713167\n",
      "The GECO3 compression ratio for Gambusia_affinis.ASM309773v1.cds.all.fa is 17.84%\n",
      "The GZIP compression ratio for Gambusia_affinis.ASM309773v1.cds.all.fa is 18.77%\n",
      "cds 44826 , total =  85510191\n",
      "The GECO3 compression ratio for Gallus_gallus_gca000002315v5.GRCg6a.cds.all.fa is 12.49%\n",
      "The GZIP compression ratio for Gallus_gallus_gca000002315v5.GRCg6a.cds.all.fa is 11.68%\n",
      "ncrna 306 , total =  35176\n",
      "The GECO3 compression ratio for Gambusia_affinis.ASM309773v1.ncrna.fa is 20.72%\n",
      "The GZIP compression ratio for Gambusia_affinis.ASM309773v1.ncrna.fa is 25.52%\n",
      "gambusia_affinis completed successfully!\n",
      "cds 62247 , total =  128018545\n",
      "The GECO3 compression ratio for Gadus_morhua.gadMor3.0.cds.all.fa is 13.48%\n",
      "The GZIP compression ratio for Gadus_morhua.gadMor3.0.cds.all.fa is 10.43%\n",
      "ncrna 6397 , total =  6996814\n",
      "The GECO3 compression ratio for Gadus_morhua.gadMor3.0.ncrna.fa is 14.30%\n",
      "The GZIP compression ratio for Gadus_morhua.gadMor3.0.ncrna.fa is 20.72%\n",
      "gadus_morhua completed successfully!\n",
      "cds 44628 , total =  85779936\n",
      "The GECO3 compression ratio for Gallus_gallus_gca016700215v2.bGalGal1.pat.whiteleghornlayer.GRCg7w.cds.all.fa is 12.45%\n",
      "The GZIP compression ratio for Gallus_gallus_gca016700215v2.bGalGal1.pat.whiteleghornlayer.GRCg7w.cds.all.fa is 11.59%\n",
      "cds 68845 , total =  131098845\n",
      "The GECO3 compression ratio for Gadus_morhua_gca010882105v1.gadMor_Celtic.cds.all.fa is 13.46%\n",
      "The GZIP compression ratio for Gadus_morhua_gca010882105v1.gadMor_Celtic.cds.all.fa is 9.99%\n",
      "ncrna 16925 , total =  21490003\n",
      "The GECO3 compression ratio for Gadus_morhua_gca010882105v1.gadMor_Celtic.ncrna.fa is 12.67%\n",
      "The GZIP compression ratio for Gadus_morhua_gca010882105v1.gadMor_Celtic.ncrna.fa is 17.22%\n",
      "gadus_morhua_gca010882105v1 completed successfully!\n",
      "ncrna 27752 , total =  67091595\n",
      "The GECO3 compression ratio for Gallus_gallus.bGalGal1.mat.broiler.GRCg7b.ncrna.fa is 16.05%\n",
      "The GZIP compression ratio for Gallus_gallus.bGalGal1.mat.broiler.GRCg7b.ncrna.fa is 18.13%\n",
      "gallus_gallus completed successfully!\n",
      "cds 22488 , total =  43556437\n",
      "The GECO3 compression ratio for Geospiza_fortis.GeoFor_1.0.cds.all.fa is 15.18%\n",
      "The GZIP compression ratio for Geospiza_fortis.GeoFor_1.0.cds.all.fa is 16.04%\n",
      "ncrna 605 , total =  355144\n",
      "The GECO3 compression ratio for Geospiza_fortis.GeoFor_1.0.ncrna.fa is 20.20%\n",
      "The GZIP compression ratio for Geospiza_fortis.GeoFor_1.0.ncrna.fa is 24.45%\n",
      "geospiza_fortis completed successfully!\n",
      "ncrna 29414 , total =  69768266\n",
      "The GECO3 compression ratio for Gallus_gallus_gca000002315v5.GRCg6a.ncrna.fa is 16.03%\n",
      "The GZIP compression ratio for Gallus_gallus_gca000002315v5.GRCg6a.ncrna.fa is 17.91%\n",
      "gallus_gallus_gca000002315v5 completed successfully!\n",
      "cds 50125 , total =  89128478\n",
      "The GECO3 compression ratio for Gasterosteus_aculeatus_gca006232265v1.BAM.cds.all.fa is 12.73%\n",
      "The GZIP compression ratio for Gasterosteus_aculeatus_gca006232265v1.BAM.cds.all.fa is 11.40%\n",
      "ncrna 868 , total =  1217067\n",
      "The GECO3 compression ratio for Gasterosteus_aculeatus_gca006232265v1.BAM.ncrna.fa is 18.40%\n",
      "The GZIP compression ratio for Gasterosteus_aculeatus_gca006232265v1.BAM.ncrna.fa is 22.25%\n",
      "gasterosteus_aculeatus_gca006232265v1 completed successfully!\n",
      "cds 53810 , total =  104704693\n",
      "The GECO3 compression ratio for Gasterosteus_aculeatus_gca006229185v1.BOT.cds.all.fa is 12.67%\n",
      "The GZIP compression ratio for Gasterosteus_aculeatus_gca006229185v1.BOT.cds.all.fa is 10.67%\n",
      "ncrna 748 , total =  941803\n",
      "The GECO3 compression ratio for Gasterosteus_aculeatus_gca006229185v1.BOT.ncrna.fa is 18.17%\n",
      "The GZIP compression ratio for Gasterosteus_aculeatus_gca006229185v1.BOT.ncrna.fa is 22.08%\n",
      "gasterosteus_aculeatus_gca006229185v1 completed successfully!\n",
      "ncrna 28370 , total =  69110896\n",
      "The GECO3 compression ratio for Gallus_gallus_gca016700215v2.bGalGal1.pat.whiteleghornlayer.GRCg7w.ncrna.fa is 16.00%\n",
      "The GZIP compression ratio for Gallus_gallus_gca016700215v2.bGalGal1.pat.whiteleghornlayer.GRCg7w.ncrna.fa is 17.98%\n",
      "gallus_gallus_gca016700215v2 completed successfully!\n",
      "cds 55534 , total =  111745127\n",
      "The GECO3 compression ratio for Gasterosteus_aculeatus_gca006232285v1.SYL.cds.all.fa is 12.67%\n",
      "The GZIP compression ratio for Gasterosteus_aculeatus_gca006232285v1.SYL.cds.all.fa is 10.42%\n",
      "ncrna 748 , total =  978552\n",
      "The GECO3 compression ratio for Gasterosteus_aculeatus_gca006232285v1.SYL.ncrna.fa is 17.37%\n",
      "The GZIP compression ratio for Gasterosteus_aculeatus_gca006232285v1.SYL.ncrna.fa is 21.16%\n",
      "gasterosteus_aculeatus_gca006232285v1 completed successfully!\n",
      "cds 30453 , total =  54087048\n",
      "The GECO3 compression ratio for Gopherus_evgoodei.rGopEvg1_v1.p.cds.all.fa is 15.91%\n",
      "The GZIP compression ratio for Gopherus_evgoodei.rGopEvg1_v1.p.cds.all.fa is 17.09%\n",
      "ncrna 1351 , total =  525255\n",
      "The GECO3 compression ratio for Gopherus_evgoodei.rGopEvg1_v1.p.ncrna.fa is 18.38%\n",
      "The GZIP compression ratio for Gopherus_evgoodei.rGopEvg1_v1.p.ncrna.fa is 22.55%\n",
      "gopherus_evgoodei completed successfully!\n",
      "cds 34329 , total =  64782592\n",
      "The GECO3 compression ratio for Gopherus_agassizii.ASM289641v1.cds.all.fa is 14.28%\n",
      "The GZIP compression ratio for Gopherus_agassizii.ASM289641v1.cds.all.fa is 14.42%\n",
      "ncrna 4719 , total =  2623664\n",
      "The GECO3 compression ratio for Gopherus_agassizii.ASM289641v1.ncrna.fa is 18.26%\n",
      "The GZIP compression ratio for Gopherus_agassizii.ASM289641v1.ncrna.fa is 22.48%\n",
      "gopherus_agassizii completed successfully!\n",
      "cds 45194 , total =  70344475\n",
      "The GECO3 compression ratio for Gorilla_gorilla.gorGor4.cds.all.fa is 15.21%\n",
      "The GZIP compression ratio for Gorilla_gorilla.gorGor4.cds.all.fa is 15.11%\n",
      "ncrna 7989 , total =  1300289\n",
      "The GECO3 compression ratio for Gorilla_gorilla.gorGor4.ncrna.fa is 17.18%\n",
      "The GZIP compression ratio for Gorilla_gorilla.gorGor4.ncrna.fa is 22.37%\n",
      "gorilla_gorilla completed successfully!\n",
      "cds 35619 , total =  61862018\n",
      "The GECO3 compression ratio for Haplochromis_burtoni.AstBur1.0.cds.all.fa is 17.47%\n",
      "The GZIP compression ratio for Haplochromis_burtoni.AstBur1.0.cds.all.fa is 18.12%\n",
      "ncrna 781 , total =  71213\n",
      "The GECO3 compression ratio for Haplochromis_burtoni.AstBur1.0.ncrna.fa is 20.06%\n",
      "The GZIP compression ratio for Haplochromis_burtoni.AstBur1.0.ncrna.fa is 24.53%\n",
      "haplochromis_burtoni completed successfully!\n",
      "cds 28435 , total =  43353887\n",
      "The GECO3 compression ratio for Hippocampus_comes.H_comes_QL1_v1.cds.all.fa is 19.15%\n",
      "The GZIP compression ratio for Hippocampus_comes.H_comes_QL1_v1.cds.all.fa is 21.24%\n",
      "ncrna 353 , total =  43357\n",
      "The GECO3 compression ratio for Hippocampus_comes.H_comes_QL1_v1.ncrna.fa is 18.83%\n",
      "The GZIP compression ratio for Hippocampus_comes.H_comes_QL1_v1.ncrna.fa is 23.52%\n",
      "hippocampus_comes completed successfully!\n",
      "cds 56693 , total =  105637458\n",
      "The GECO3 compression ratio for Gouania_willdenowi.fGouWil2.1.cds.all.fa is 12.49%\n",
      "The GZIP compression ratio for Gouania_willdenowi.fGouWil2.1.cds.all.fa is 11.29%\n",
      "cds 50671 , total =  83062968\n",
      "The GECO3 compression ratio for Heterocephalus_glaber_female.Naked_mole-rat_maternal.cds.all.fa is 13.58%\n",
      "The GZIP compression ratio for Heterocephalus_glaber_female.Naked_mole-rat_maternal.cds.all.fa is 13.06%\n",
      "cds 51112 , total =  83047479\n",
      "The GECO3 compression ratio for Heterocephalus_glaber_male.Naked_mole-rat_paternal.cds.all.fa is 13.69%\n",
      "The GZIP compression ratio for Heterocephalus_glaber_male.Naked_mole-rat_paternal.cds.all.fa is 12.97%\n",
      "ncrna 3637 , total =  3328631\n",
      "The GECO3 compression ratio for Gouania_willdenowi.fGouWil2.1.ncrna.fa is 15.08%\n",
      "The GZIP compression ratio for Gouania_willdenowi.fGouWil2.1.ncrna.fa is 20.27%\n",
      "gouania_willdenowi completed successfully!\n",
      "cds 25958 , total =  44683051\n",
      "The GECO3 compression ratio for Ictidomys_tridecemlineatus.SpeTri2.0.cds.all.fa is 18.04%\n",
      "The GZIP compression ratio for Ictidomys_tridecemlineatus.SpeTri2.0.cds.all.fa is 19.73%\n",
      "ncrna 7106 , total =  3301606\n",
      "The GECO3 compression ratio for Ictidomys_tridecemlineatus.SpeTri2.0.ncrna.fa is 20.37%\n",
      "The GZIP compression ratio for Ictidomys_tridecemlineatus.SpeTri2.0.ncrna.fa is 24.29%\n",
      "ictidomys_tridecemlineatus completed successfully!\n",
      "cds 25180 , total =  41630765\n",
      "The GECO3 compression ratio for Jaculus_jaculus.JacJac1.0.cds.all.fa is 17.82%\n",
      "The GZIP compression ratio for Jaculus_jaculus.JacJac1.0.cds.all.fa is 19.45%\n",
      "ncrna 6267 , total =  932913\n",
      "The GECO3 compression ratio for Jaculus_jaculus.JacJac1.0.ncrna.fa is 15.65%\n",
      "The GZIP compression ratio for Jaculus_jaculus.JacJac1.0.ncrna.fa is 20.99%\n",
      "jaculus_jaculus completed successfully!\n",
      "ncrna 20000 , total =  32492004\n",
      "The GECO3 compression ratio for Heterocephalus_glaber_female.Naked_mole-rat_maternal.ncrna.fa is 16.36%\n",
      "The GZIP compression ratio for Heterocephalus_glaber_female.Naked_mole-rat_maternal.ncrna.fa is 19.07%\n",
      "heterocephalus_glaber_female completed successfully!\n",
      "ncrna 19489 , total =  31484253\n",
      "The GECO3 compression ratio for Heterocephalus_glaber_male.Naked_mole-rat_paternal.ncrna.fa is 16.20%\n",
      "The GZIP compression ratio for Heterocephalus_glaber_male.Naked_mole-rat_paternal.ncrna.fa is 19.01%\n",
      "heterocephalus_glaber_male completed successfully!\n",
      "cds 23225 , total =  44919639\n",
      "The GECO3 compression ratio for Junco_hyemalis.ASM382977v1.cds.all.fa is 15.45%\n",
      "The GZIP compression ratio for Junco_hyemalis.ASM382977v1.cds.all.fa is 16.24%\n",
      "ncrna 4989 , total =  4422277\n",
      "The GECO3 compression ratio for Junco_hyemalis.ASM382977v1.ncrna.fa is 17.50%\n",
      "The GZIP compression ratio for Junco_hyemalis.ASM382977v1.ncrna.fa is 21.16%\n",
      "junco_hyemalis completed successfully!\n",
      "cds 123495 , total =  148214017\n",
      "The GECO3 compression ratio for Homo_sapiens.GRCh38.cds.all.fa is 11.00%\n",
      "The GZIP compression ratio for Homo_sapiens.GRCh38.cds.all.fa is 9.29%\n",
      "cds 30189 , total =  53285907\n",
      "The GECO3 compression ratio for Kryptolebias_marmoratus.ASM164957v1.cds.all.fa is 18.40%\n",
      "The GZIP compression ratio for Kryptolebias_marmoratus.ASM164957v1.cds.all.fa is 19.87%\n",
      "ncrna 347 , total =  40366\n",
      "The GECO3 compression ratio for Kryptolebias_marmoratus.ASM164957v1.ncrna.fa is 21.35%\n",
      "The GZIP compression ratio for Kryptolebias_marmoratus.ASM164957v1.ncrna.fa is 26.39%\n",
      "kryptolebias_marmoratus completed successfully!\n",
      "cds 91817 , total =  143964654\n",
      "The GECO3 compression ratio for Hucho_hucho.ASM331708v1.cds.all.fa is 15.78%\n",
      "The GZIP compression ratio for Hucho_hucho.ASM331708v1.cds.all.fa is 14.84%\n",
      "ncrna 2215 , total =  251750\n",
      "The GECO3 compression ratio for Hucho_hucho.ASM331708v1.ncrna.fa is 11.89%\n",
      "The GZIP compression ratio for Hucho_hucho.ASM331708v1.ncrna.fa is 15.49%\n",
      "hucho_hucho completed successfully!\n",
      "cds 38878 , total =  60038989\n",
      "The GECO3 compression ratio for Labrus_bergylta.BallGen_V1.cds.all.fa is 17.38%\n",
      "The GZIP compression ratio for Labrus_bergylta.BallGen_V1.cds.all.fa is 19.55%\n",
      "ncrna 1650 , total =  205004\n",
      "The GECO3 compression ratio for Labrus_bergylta.BallGen_V1.ncrna.fa is 6.82%\n",
      "The GZIP compression ratio for Labrus_bergylta.BallGen_V1.ncrna.fa is 8.81%\n",
      "labrus_bergylta completed successfully!\n",
      "cds 25227 , total =  38966512\n",
      "The GECO3 compression ratio for Laticauda_laticaudata.latLat_1.0.cds.all.fa is 16.46%\n",
      "The GZIP compression ratio for Laticauda_laticaudata.latLat_1.0.cds.all.fa is 18.13%\n",
      "ncrna 810 , total =  453653\n",
      "The GECO3 compression ratio for Laticauda_laticaudata.latLat_1.0.ncrna.fa is 20.25%\n",
      "The GZIP compression ratio for Laticauda_laticaudata.latLat_1.0.ncrna.fa is 24.34%\n",
      "laticauda_laticaudata completed successfully!\n",
      "cds 84434 , total =  163073736\n",
      "The GECO3 compression ratio for Ictalurus_punctatus.ASM400665v3.cds.all.fa is 11.76%\n",
      "The GZIP compression ratio for Ictalurus_punctatus.ASM400665v3.cds.all.fa is 9.73%\n",
      "cds 23601 , total =  36696347\n",
      "The GECO3 compression ratio for Latimeria_chalumnae.LatCha1.cds.all.fa is 21.07%\n",
      "The GZIP compression ratio for Latimeria_chalumnae.LatCha1.cds.all.fa is 24.09%\n",
      "ncrna 2918 , total =  338142\n",
      "The GECO3 compression ratio for Latimeria_chalumnae.LatCha1.ncrna.fa is 11.60%\n",
      "The GZIP compression ratio for Latimeria_chalumnae.LatCha1.ncrna.fa is 16.35%\n",
      "latimeria_chalumnae completed successfully!\n",
      "cds 22483 , total =  40288174\n",
      "The GECO3 compression ratio for Lepisosteus_oculatus.LepOcu1.cds.all.fa is 20.09%\n",
      "The GZIP compression ratio for Lepisosteus_oculatus.LepOcu1.cds.all.fa is 22.38%\n",
      "cds 23570 , total =  46266135\n",
      "The GECO3 compression ratio for Lepidothrix_coronata.Lepidothrix_coronata-1.0.cds.all.fa is 15.05%\n",
      "The GZIP compression ratio for Lepidothrix_coronata.Lepidothrix_coronata-1.0.cds.all.fa is 16.08%\n",
      "ncrna 16303 , total =  26777358\n",
      "The GECO3 compression ratio for Ictalurus_punctatus.ASM400665v3.ncrna.fa is 13.80%\n",
      "The GZIP compression ratio for Ictalurus_punctatus.ASM400665v3.ncrna.fa is 17.85%\n",
      "ictalurus_punctatus completed successfully!\n",
      "ncrna 1256 , total =  684453\n",
      "The GECO3 compression ratio for Lepidothrix_coronata.Lepidothrix_coronata-1.0.ncrna.fa is 21.26%\n",
      "The GZIP compression ratio for Lepidothrix_coronata.Lepidothrix_coronata-1.0.ncrna.fa is 25.38%\n",
      "lepidothrix_coronata completed successfully!\n",
      "ncrna 5362 , total =  3544014\n",
      "The GECO3 compression ratio for Lepisosteus_oculatus.LepOcu1.ncrna.fa is 19.47%\n",
      "The GZIP compression ratio for Lepisosteus_oculatus.LepOcu1.ncrna.fa is 23.45%\n",
      "lepisosteus_oculatus completed successfully!\n",
      "cds 58216 , total =  112719229\n",
      "The GECO3 compression ratio for Lates_calcarifer.ASB_HGAPassembly_v1.cds.all.fa is 13.29%\n",
      "The GZIP compression ratio for Lates_calcarifer.ASB_HGAPassembly_v1.cds.all.fa is 11.50%\n",
      "ncrna 1306 , total =  171942\n",
      "The GECO3 compression ratio for Lates_calcarifer.ASB_HGAPassembly_v1.ncrna.fa is 7.19%\n",
      "The GZIP compression ratio for Lates_calcarifer.ASB_HGAPassembly_v1.ncrna.fa is 9.01%\n",
      "lates_calcarifer completed successfully!\n",
      "cds 10411 , total =  12763251\n",
      "The GECO3 compression ratio for Macaca_mulatta.Mmul_10.cds.all.fa is 10.52%\n",
      "The GZIP compression ratio for Macaca_mulatta.Mmul_10.cds.all.fa is 15.57%\n",
      "cds 65776 , total =  136125759\n",
      "The GECO3 compression ratio for Larimichthys_crocea.L_crocea_2.0.cds.all.fa is 12.78%\n",
      "The GZIP compression ratio for Larimichthys_crocea.L_crocea_2.0.cds.all.fa is 9.97%\n",
      "ncrna 70339 , total =  82728451\n",
      "The GECO3 compression ratio for Homo_sapiens.GRCh38.ncrna.fa is 13.57%\n",
      "The GZIP compression ratio for Homo_sapiens.GRCh38.ncrna.fa is 15.44%\n",
      "homo_sapiens completed successfully!\n",
      "ncrna 771 , total =  103238\n",
      "The GECO3 compression ratio for Larimichthys_crocea.L_crocea_2.0.ncrna.fa is 13.58%\n",
      "The GZIP compression ratio for Larimichthys_crocea.L_crocea_2.0.ncrna.fa is 16.62%\n",
      "larimichthys_crocea completed successfully!\n",
      "ncrna 14688 , total =  11769246\n",
      "The GECO3 compression ratio for Macaca_mulatta.Mmul_10.ncrna.fa is 19.57%\n",
      "The GZIP compression ratio for Macaca_mulatta.Mmul_10.ncrna.fa is 24.09%\n",
      "macaca_mulatta completed successfully!\n",
      "cds 26162 , total =  52295249\n",
      "The GECO3 compression ratio for Lonchura_striata_domestica.LonStrDom1.cds.all.fa is 14.64%\n",
      "The GZIP compression ratio for Lonchura_striata_domestica.LonStrDom1.cds.all.fa is 15.10%\n",
      "ncrna 3164 , total =  2803902\n",
      "The GECO3 compression ratio for Lonchura_striata_domestica.LonStrDom1.ncrna.fa is 15.90%\n",
      "The GZIP compression ratio for Lonchura_striata_domestica.LonStrDom1.ncrna.fa is 19.71%\n",
      "lonchura_striata_domestica completed successfully!\n",
      "cds 49453 , total =  87063240\n",
      "The GECO3 compression ratio for Leptobrachium_leishanense.ASM966780v1.cds.all.fa is 13.62%\n",
      "The GZIP compression ratio for Leptobrachium_leishanense.ASM966780v1.cds.all.fa is 16.25%\n",
      "ncrna 912 , total =  147894\n",
      "The GECO3 compression ratio for Leptobrachium_leishanense.ASM966780v1.ncrna.fa is 10.94%\n",
      "The GZIP compression ratio for Leptobrachium_leishanense.ASM966780v1.ncrna.fa is 14.48%\n",
      "leptobrachium_leishanense completed successfully!\n",
      "cds 25635 , total =  41611076\n",
      "The GECO3 compression ratio for Loxodonta_africana.loxAfr3.cds.all.fa is 19.33%\n",
      "The GZIP compression ratio for Loxodonta_africana.loxAfr3.cds.all.fa is 21.57%\n",
      "ncrna 2644 , total =  284140\n",
      "The GECO3 compression ratio for Loxodonta_africana.loxAfr3.ncrna.fa is 16.55%\n",
      "The GZIP compression ratio for Loxodonta_africana.loxAfr3.ncrna.fa is 21.23%\n",
      "loxodonta_africana completed successfully!\n",
      "cds 35180 , total =  62157179\n",
      "The GECO3 compression ratio for Lynx_canadensis.mLynCan4_v1.p.cds.all.fa is 15.11%\n",
      "The GZIP compression ratio for Lynx_canadensis.mLynCan4_v1.p.cds.all.fa is 15.42%\n",
      "cds 23133 , total =  40028088\n",
      "The GECO3 compression ratio for Malurus_cyaneus_samueli.mCya_1.0.cds.all.fa is 15.86%\n",
      "The GZIP compression ratio for Malurus_cyaneus_samueli.mCya_1.0.cds.all.fa is 17.36%\n",
      "ncrna 446 , total =  101350\n",
      "The GECO3 compression ratio for Malurus_cyaneus_samueli.mCya_1.0.ncrna.fa is 21.66%\n",
      "The GZIP compression ratio for Malurus_cyaneus_samueli.mCya_1.0.ncrna.fa is 26.27%\n",
      "malurus_cyaneus_samueli completed successfully!\n",
      "ncrna 3426 , total =  1381957\n",
      "The GECO3 compression ratio for Lynx_canadensis.mLynCan4_v1.p.ncrna.fa is 19.03%\n",
      "The GZIP compression ratio for Lynx_canadensis.mLynCan4_v1.p.ncrna.fa is 23.31%\n",
      "lynx_canadensis completed successfully!\n",
      "cds 49810 , total =  81229699\n",
      "The GECO3 compression ratio for Macaca_fascicularis.Macaca_fascicularis_6.0.cds.all.fa is 13.57%\n",
      "The GZIP compression ratio for Macaca_fascicularis.Macaca_fascicularis_6.0.cds.all.fa is 12.91%\n",
      "cds 26964 , total =  51595396\n",
      "The GECO3 compression ratio for Manacus_vitellinus.ASM171598v2.cds.all.fa is 14.04%\n",
      "The GZIP compression ratio for Manacus_vitellinus.ASM171598v2.cds.all.fa is 14.89%\n",
      "ncrna 3464 , total =  2671135\n",
      "The GECO3 compression ratio for Manacus_vitellinus.ASM171598v2.ncrna.fa is 18.64%\n",
      "The GZIP compression ratio for Manacus_vitellinus.ASM171598v2.ncrna.fa is 22.96%\n",
      "manacus_vitellinus completed successfully!\n",
      "cds 27615 , total =  44164811\n",
      "The GECO3 compression ratio for Marmota_marmota_marmota.marMar2.1.cds.all.fa is 18.31%\n",
      "The GZIP compression ratio for Marmota_marmota_marmota.marMar2.1.cds.all.fa is 20.28%\n",
      "ncrna 3113 , total =  650202\n",
      "The GECO3 compression ratio for Marmota_marmota_marmota.marMar2.1.ncrna.fa is 19.25%\n",
      "The GZIP compression ratio for Marmota_marmota_marmota.marMar2.1.ncrna.fa is 23.76%\n",
      "marmota_marmota_marmota completed successfully!\n",
      "cds 46148 , total =  75423975\n",
      "The GECO3 compression ratio for Macaca_nemestrina.Mnem_1.0.cds.all.fa is 14.66%\n",
      "The GZIP compression ratio for Macaca_nemestrina.Mnem_1.0.cds.all.fa is 13.99%\n",
      "ncrna 15243 , total =  14917525\n",
      "The GECO3 compression ratio for Macaca_fascicularis.Macaca_fascicularis_6.0.ncrna.fa is 19.26%\n",
      "The GZIP compression ratio for Macaca_fascicularis.Macaca_fascicularis_6.0.ncrna.fa is 25.74%\n",
      "macaca_fascicularis completed successfully!\n",
      "ncrna 7631 , total =  2153203\n",
      "The GECO3 compression ratio for Macaca_nemestrina.Mnem_1.0.ncrna.fa is 17.57%\n",
      "The GZIP compression ratio for Macaca_nemestrina.Mnem_1.0.ncrna.fa is 22.09%\n",
      "macaca_nemestrina completed successfully!\n",
      "cds 40903 , total =  64076262\n",
      "The GECO3 compression ratio for Mandrillus_leucophaeus.Mleu.le_1.0.cds.all.fa is 15.44%\n",
      "The GZIP compression ratio for Mandrillus_leucophaeus.Mleu.le_1.0.cds.all.fa is 15.59%\n",
      "ncrna 6448 , total =  860433\n",
      "The GECO3 compression ratio for Mandrillus_leucophaeus.Mleu.le_1.0.ncrna.fa is 16.54%\n",
      "The GZIP compression ratio for Mandrillus_leucophaeus.Mleu.le_1.0.ncrna.fa is 22.41%\n",
      "mandrillus_leucophaeus completed successfully!\n",
      "cds 28295 , total =  52083794\n",
      "The GECO3 compression ratio for Meleagris_gallopavo.Turkey_5.1.cds.all.fa is 13.98%\n",
      "The GZIP compression ratio for Meleagris_gallopavo.Turkey_5.1.cds.all.fa is 14.44%\n",
      "ncrna 2254 , total =  2799674\n",
      "The GECO3 compression ratio for Meleagris_gallopavo.Turkey_5.1.ncrna.fa is 18.90%\n",
      "The GZIP compression ratio for Meleagris_gallopavo.Turkey_5.1.ncrna.fa is 22.76%\n",
      "meleagris_gallopavo completed successfully!\n",
      "cds 38246 , total =  62303113\n",
      "The GECO3 compression ratio for Maylandia_zebra.M_zebra_UMD2a.cds.all.fa is 17.84%\n",
      "The GZIP compression ratio for Maylandia_zebra.M_zebra_UMD2a.cds.all.fa is 19.62%\n",
      "ncrna 1386 , total =  155851\n",
      "The GECO3 compression ratio for Maylandia_zebra.M_zebra_UMD2a.ncrna.fa is 11.84%\n",
      "The GZIP compression ratio for Maylandia_zebra.M_zebra_UMD2a.ncrna.fa is 14.88%\n",
      "maylandia_zebra completed successfully!\n",
      "cds 25210 , total =  49440701\n",
      "The GECO3 compression ratio for Melopsittacus_undulatus.bMelUnd1.mat.Z.cds.all.fa is 15.34%\n",
      "The GZIP compression ratio for Melopsittacus_undulatus.bMelUnd1.mat.Z.cds.all.fa is 16.41%\n",
      "ncrna 1099 , total =  781290\n",
      "The GECO3 compression ratio for Melopsittacus_undulatus.bMelUnd1.mat.Z.ncrna.fa is 15.77%\n",
      "The GZIP compression ratio for Melopsittacus_undulatus.bMelUnd1.mat.Z.ncrna.fa is 19.31%\n",
      "melopsittacus_undulatus completed successfully!\n",
      "cds 25910 , total =  41663629\n",
      "The GECO3 compression ratio for Mesocricetus_auratus.MesAur1.0.cds.all.fa is 18.16%\n",
      "The GZIP compression ratio for Mesocricetus_auratus.MesAur1.0.cds.all.fa is 19.70%\n",
      "ncrna 3720 , total =  508286\n",
      "The GECO3 compression ratio for Mesocricetus_auratus.MesAur1.0.ncrna.fa is 18.83%\n",
      "The GZIP compression ratio for Mesocricetus_auratus.MesAur1.0.ncrna.fa is 23.66%\n",
      "mesocricetus_auratus completed successfully!\n",
      "cds 31063 , total =  51478834\n",
      "The GECO3 compression ratio for Meriones_unguiculatus.MunDraft-v1.0.cds.all.fa is 16.62%\n",
      "The GZIP compression ratio for Meriones_unguiculatus.MunDraft-v1.0.cds.all.fa is 18.24%\n",
      "ncrna 4166 , total =  1031951\n",
      "The GECO3 compression ratio for Meriones_unguiculatus.MunDraft-v1.0.ncrna.fa is 19.59%\n",
      "The GZIP compression ratio for Meriones_unguiculatus.MunDraft-v1.0.ncrna.fa is 23.89%\n",
      "meriones_unguiculatus completed successfully!\n",
      "cds 27611 , total =  45953734\n",
      "The GECO3 compression ratio for Microtus_ochrogaster.MicOch1.0.cds.all.fa is 17.97%\n",
      "The GZIP compression ratio for Microtus_ochrogaster.MicOch1.0.cds.all.fa is 19.41%\n",
      "ncrna 3379 , total =  451440\n",
      "The GECO3 compression ratio for Microtus_ochrogaster.MicOch1.0.ncrna.fa is 19.00%\n",
      "The GZIP compression ratio for Microtus_ochrogaster.MicOch1.0.ncrna.fa is 23.63%\n",
      "microtus_ochrogaster completed successfully!\n",
      "cds 59463 , total =  121682011\n",
      "The GECO3 compression ratio for Mastacembelus_armatus.fMasArm1.2.cds.all.fa is 12.77%\n",
      "The GZIP compression ratio for Mastacembelus_armatus.fMasArm1.2.cds.all.fa is 10.34%\n",
      "ncrna 931 , total =  725103\n",
      "The GECO3 compression ratio for Mastacembelus_armatus.fMasArm1.2.ncrna.fa is 19.33%\n",
      "The GZIP compression ratio for Mastacembelus_armatus.fMasArm1.2.ncrna.fa is 23.44%\n",
      "mastacembelus_armatus completed successfully!\n",
      "cds 38003 , total =  64398742\n",
      "The GECO3 compression ratio for Microcebus_murinus.Mmur_3.0.cds.all.fa is 15.05%\n",
      "The GZIP compression ratio for Microcebus_murinus.Mmur_3.0.cds.all.fa is 15.14%\n",
      "cds 29015 , total =  42232139\n",
      "The GECO3 compression ratio for Mola_mola.ASM169857v1.cds.all.fa is 19.62%\n",
      "The GZIP compression ratio for Mola_mola.ASM169857v1.cds.all.fa is 21.96%\n",
      "ncrna 444 , total =  50274\n",
      "The GECO3 compression ratio for Mola_mola.ASM169857v1.ncrna.fa is 18.81%\n",
      "The GZIP compression ratio for Mola_mola.ASM169857v1.ncrna.fa is 23.74%\n",
      "mola_mola completed successfully!\n",
      "ncrna 7848 , total =  2233361\n",
      "The GECO3 compression ratio for Microcebus_murinus.Mmur_3.0.ncrna.fa is 16.32%\n",
      "The GZIP compression ratio for Microcebus_murinus.Mmur_3.0.ncrna.fa is 20.55%\n",
      "microcebus_murinus completed successfully!\n",
      "cds 30379 , total =  51692186\n",
      "The GECO3 compression ratio for Monodon_monoceros.NGI_Narwhal_1.cds.all.fa is 16.88%\n",
      "The GZIP compression ratio for Monodon_monoceros.NGI_Narwhal_1.cds.all.fa is 18.24%\n",
      "ncrna 2271 , total =  539958\n",
      "The GECO3 compression ratio for Monodon_monoceros.NGI_Narwhal_1.ncrna.fa is 18.90%\n",
      "The GZIP compression ratio for Monodon_monoceros.NGI_Narwhal_1.ncrna.fa is 23.56%\n",
      "monodon_monoceros completed successfully!\n",
      "cds 36557 , total =  67748702\n",
      "The GECO3 compression ratio for Monodelphis_domestica.ASM229v1.cds.all.fa is 14.51%\n",
      "The GZIP compression ratio for Monodelphis_domestica.ASM229v1.cds.all.fa is 15.37%\n",
      "cds 33002 , total =  54202793\n",
      "The GECO3 compression ratio for Monopterus_albus.M_albus_1.0.cds.all.fa is 18.00%\n",
      "The GZIP compression ratio for Monopterus_albus.M_albus_1.0.cds.all.fa is 19.17%\n",
      "ncrna 527 , total =  61045\n",
      "The GECO3 compression ratio for Monopterus_albus.M_albus_1.0.ncrna.fa is 19.25%\n",
      "The GZIP compression ratio for Monopterus_albus.M_albus_1.0.ncrna.fa is 23.76%\n",
      "monopterus_albus completed successfully!\n",
      "cds 33292 , total =  54927363\n",
      "The GECO3 compression ratio for Moschus_moschiferus.MosMos_v2_BIUU_UCD.cds.all.fa is 16.81%\n",
      "The GZIP compression ratio for Moschus_moschiferus.MosMos_v2_BIUU_UCD.cds.all.fa is 17.95%\n",
      "ncrna 2548 , total =  419039\n",
      "The GECO3 compression ratio for Moschus_moschiferus.MosMos_v2_BIUU_UCD.ncrna.fa is 17.58%\n",
      "The GZIP compression ratio for Moschus_moschiferus.MosMos_v2_BIUU_UCD.ncrna.fa is 22.14%\n",
      "moschus_moschiferus completed successfully!\n",
      "ncrna 21460 , total =  18374839\n",
      "The GECO3 compression ratio for Monodelphis_domestica.ASM229v1.ncrna.fa is 17.83%\n",
      "The GZIP compression ratio for Monodelphis_domestica.ASM229v1.ncrna.fa is 21.78%\n",
      "monodelphis_domestica completed successfully!\n",
      "cds 50492 , total =  69878968\n",
      "The GECO3 compression ratio for Mus_caroli.CAROLI_EIJ_v1.1.cds.all.fa is 14.88%\n",
      "The GZIP compression ratio for Mus_caroli.CAROLI_EIJ_v1.1.cds.all.fa is 14.79%\n",
      "cds 52191 , total =  71594255\n",
      "The GECO3 compression ratio for Mus_musculus_129s1svimj.129S1_SvImJ_v1.cds.all.fa is 14.94%\n",
      "The GZIP compression ratio for Mus_musculus_129s1svimj.129S1_SvImJ_v1.cds.all.fa is 14.78%\n",
      "ncrna 12513 , total =  10415111\n",
      "The GECO3 compression ratio for Mus_caroli.CAROLI_EIJ_v1.1.ncrna.fa is 20.59%\n",
      "The GZIP compression ratio for Mus_caroli.CAROLI_EIJ_v1.1.ncrna.fa is 24.33%\n",
      "mus_caroli completed successfully!\n",
      "cds 52196 , total =  71325647\n",
      "The GECO3 compression ratio for Mus_musculus_aj.A_J_v1.cds.all.fa is 14.94%\n",
      "The GZIP compression ratio for Mus_musculus_aj.A_J_v1.cds.all.fa is 14.83%\n",
      "cds 66464 , total =  84822263\n",
      "The GECO3 compression ratio for Mus_musculus.GRCm39.cds.all.fa is 14.19%\n",
      "The GZIP compression ratio for Mus_musculus.GRCm39.cds.all.fa is 13.21%\n",
      "ncrna 13865 , total =  11339035\n",
      "The GECO3 compression ratio for Mus_musculus_129s1svimj.129S1_SvImJ_v1.ncrna.fa is 20.46%\n",
      "The GZIP compression ratio for Mus_musculus_129s1svimj.129S1_SvImJ_v1.ncrna.fa is 24.20%\n",
      "mus_musculus_129s1svimj completed successfully!\n",
      "ncrna 13888 , total =  11380747\n",
      "The GECO3 compression ratio for Mus_musculus_aj.A_J_v1.ncrna.fa is 20.48%\n",
      "The GZIP compression ratio for Mus_musculus_aj.A_J_v1.ncrna.fa is 24.23%\n",
      "mus_musculus_aj completed successfully!\n",
      "cds 52193 , total =  71719862\n",
      "The GECO3 compression ratio for Mus_musculus_akrj.AKR_J_v1.cds.all.fa is 14.94%\n",
      "The GZIP compression ratio for Mus_musculus_akrj.AKR_J_v1.cds.all.fa is 14.75%\n",
      "cds 52145 , total =  71313748\n",
      "The GECO3 compression ratio for Mus_musculus_balbcj.BALB_cJ_v1.cds.all.fa is 14.95%\n",
      "The GZIP compression ratio for Mus_musculus_balbcj.BALB_cJ_v1.cds.all.fa is 14.84%\n",
      "cds 51971 , total =  71238264\n",
      "The GECO3 compression ratio for Mus_musculus_c3hhej.C3H_HeJ_v1.cds.all.fa is 14.93%\n",
      "The GZIP compression ratio for Mus_musculus_c3hhej.C3H_HeJ_v1.cds.all.fa is 14.80%\n",
      "ncrna 29930 , total =  29886426\n",
      "The GECO3 compression ratio for Mus_musculus.GRCm39.ncrna.fa is 15.90%\n",
      "The GZIP compression ratio for Mus_musculus.GRCm39.ncrna.fa is 18.64%\n",
      "mus_musculus completed successfully!\n",
      "ncrna 13878 , total =  11361142\n",
      "The GECO3 compression ratio for Mus_musculus_akrj.AKR_J_v1.ncrna.fa is 20.48%\n",
      "The GZIP compression ratio for Mus_musculus_akrj.AKR_J_v1.ncrna.fa is 24.24%\n",
      "mus_musculus_akrj completed successfully!\n",
      "ncrna 13864 , total =  11333029\n",
      "The GECO3 compression ratio for Mus_musculus_balbcj.BALB_cJ_v1.ncrna.fa is 20.49%\n",
      "The GZIP compression ratio for Mus_musculus_balbcj.BALB_cJ_v1.ncrna.fa is 24.23%\n",
      "mus_musculus_balbcj completed successfully!\n",
      "cds 52383 , total =  71618484\n",
      "The GECO3 compression ratio for Mus_musculus_c57bl6nj.C57BL_6NJ_v1.cds.all.fa is 14.97%\n",
      "The GZIP compression ratio for Mus_musculus_c57bl6nj.C57BL_6NJ_v1.cds.all.fa is 14.86%\n",
      "ncrna 13813 , total =  11319649\n",
      "The GECO3 compression ratio for Mus_musculus_c3hhej.C3H_HeJ_v1.ncrna.fa is 20.51%\n",
      "The GZIP compression ratio for Mus_musculus_c3hhej.C3H_HeJ_v1.ncrna.fa is 24.25%\n",
      "mus_musculus_c3hhej completed successfully!\n",
      "ncrna 14023 , total =  11439658\n",
      "The GECO3 compression ratio for Mus_musculus_c57bl6nj.C57BL_6NJ_v1.ncrna.fa is 20.47%\n",
      "The GZIP compression ratio for Mus_musculus_c57bl6nj.C57BL_6NJ_v1.ncrna.fa is 24.21%\n",
      "mus_musculus_c57bl6nj completed successfully!\n",
      "cds 53281 , total =  73628422\n",
      "The GECO3 compression ratio for Mus_musculus_casteij.CAST_EiJ_v1.cds.all.fa is 14.70%\n",
      "The GZIP compression ratio for Mus_musculus_casteij.CAST_EiJ_v1.cds.all.fa is 14.26%\n",
      "cds 51976 , total =  71327696\n",
      "The GECO3 compression ratio for Mus_musculus_cbaj.CBA_J_v1.cds.all.fa is 14.92%\n",
      "The GZIP compression ratio for Mus_musculus_cbaj.CBA_J_v1.cds.all.fa is 14.77%\n",
      "cds 52057 , total =  71320353\n",
      "The GECO3 compression ratio for Mus_musculus_dba2j.DBA_2J_v1.cds.all.fa is 14.93%\n",
      "The GZIP compression ratio for Mus_musculus_dba2j.DBA_2J_v1.cds.all.fa is 14.80%\n",
      "ncrna 13526 , total =  11099272\n",
      "The GECO3 compression ratio for Mus_musculus_casteij.CAST_EiJ_v1.ncrna.fa is 20.53%\n",
      "The GZIP compression ratio for Mus_musculus_casteij.CAST_EiJ_v1.ncrna.fa is 24.28%\n",
      "mus_musculus_casteij completed successfully!\n",
      "ncrna 13774 , total =  11293667\n",
      "The GECO3 compression ratio for Mus_musculus_cbaj.CBA_J_v1.ncrna.fa is 20.51%\n",
      "The GZIP compression ratio for Mus_musculus_cbaj.CBA_J_v1.ncrna.fa is 24.25%\n",
      "mus_musculus_cbaj completed successfully!\n",
      "ncrna 13819 , total =  11303384\n",
      "The GECO3 compression ratio for Mus_musculus_dba2j.DBA_2J_v1.ncrna.fa is 20.50%\n",
      "The GZIP compression ratio for Mus_musculus_dba2j.DBA_2J_v1.ncrna.fa is 24.23%\n",
      "mus_musculus_dba2j completed successfully!\n",
      "cds 52060 , total =  71393217\n",
      "The GECO3 compression ratio for Mus_musculus_lpj.LP_J_v1.cds.all.fa is 14.91%\n",
      "The GZIP compression ratio for Mus_musculus_lpj.LP_J_v1.cds.all.fa is 14.80%\n",
      "cds 51898 , total =  71203736\n",
      "The GECO3 compression ratio for Mus_musculus_fvbnj.FVB_NJ_v1.cds.all.fa is 14.92%\n",
      "The GZIP compression ratio for Mus_musculus_fvbnj.FVB_NJ_v1.cds.all.fa is 14.73%\n",
      "cds 51974 , total =  71167850\n",
      "The GECO3 compression ratio for Mus_musculus_nodshiltj.NOD_ShiLtJ_v1.cds.all.fa is 14.95%\n",
      "The GZIP compression ratio for Mus_musculus_nodshiltj.NOD_ShiLtJ_v1.cds.all.fa is 14.76%\n",
      "cds 52538 , total =  71864972\n",
      "The GECO3 compression ratio for Mus_musculus_nzohlltj.NZO_HlLtJ_v1.cds.all.fa is 14.96%\n",
      "The GZIP compression ratio for Mus_musculus_nzohlltj.NZO_HlLtJ_v1.cds.all.fa is 14.81%\n",
      "ncrna 13860 , total =  11350904\n",
      "The GECO3 compression ratio for Mus_musculus_lpj.LP_J_v1.ncrna.fa is 20.50%\n",
      "The GZIP compression ratio for Mus_musculus_lpj.LP_J_v1.ncrna.fa is 24.24%\n",
      "mus_musculus_lpj completed successfully!\n",
      "ncrna 13790 , total =  11283056\n",
      "The GECO3 compression ratio for Mus_musculus_fvbnj.FVB_NJ_v1.ncrna.fa is 20.51%\n",
      "The GZIP compression ratio for Mus_musculus_fvbnj.FVB_NJ_v1.ncrna.fa is 24.25%\n",
      "mus_musculus_fvbnj completed successfully!\n",
      "ncrna 13747 , total =  11210296\n",
      "The GECO3 compression ratio for Mus_musculus_nodshiltj.NOD_ShiLtJ_v1.ncrna.fa is 20.51%\n",
      "The GZIP compression ratio for Mus_musculus_nodshiltj.NOD_ShiLtJ_v1.ncrna.fa is 24.27%\n",
      "mus_musculus_nodshiltj completed successfully!\n",
      "cds 52971 , total =  73329810\n",
      "The GECO3 compression ratio for Mus_musculus_pwkphj.PWK_PhJ_v1.cds.all.fa is 14.66%\n",
      "The GZIP compression ratio for Mus_musculus_pwkphj.PWK_PhJ_v1.cds.all.fa is 14.27%\n",
      "ncrna 13994 , total =  11430289\n",
      "The GECO3 compression ratio for Mus_musculus_nzohlltj.NZO_HlLtJ_v1.ncrna.fa is 20.47%\n",
      "The GZIP compression ratio for Mus_musculus_nzohlltj.NZO_HlLtJ_v1.ncrna.fa is 24.22%\n",
      "mus_musculus_nzohlltj completed successfully!\n",
      "ncrna 13439 , total =  11018077\n",
      "The GECO3 compression ratio for Mus_musculus_pwkphj.PWK_PhJ_v1.ncrna.fa is 20.55%\n",
      "The GZIP compression ratio for Mus_musculus_pwkphj.PWK_PhJ_v1.ncrna.fa is 24.28%\n",
      "mus_musculus_pwkphj completed successfully!\n",
      "cds 51502 , total =  70534455\n",
      "The GECO3 compression ratio for Mus_musculus_wsbeij.WSB_EiJ_v1.cds.all.fa is 14.93%\n",
      "The GZIP compression ratio for Mus_musculus_wsbeij.WSB_EiJ_v1.cds.all.fa is 14.70%\n",
      "cds 20062 , total =  32200188\n",
      "The GECO3 compression ratio for Mustela_putorius_furo.MusPutFur1.0.cds.all.fa is 23.35%\n",
      "The GZIP compression ratio for Mustela_putorius_furo.MusPutFur1.0.cds.all.fa is 27.60%\n",
      "cds 37002 , total =  63459346\n",
      "The GECO3 compression ratio for Mus_spicilegus.MUSP714.cds.all.fa is 16.04%\n",
      "The GZIP compression ratio for Mus_spicilegus.MUSP714.cds.all.fa is 16.59%\n",
      "cds 50002 , total =  69078273\n",
      "The GECO3 compression ratio for Mus_pahari.PAHARI_EIJ_v1.1.cds.all.fa is 14.97%\n",
      "The GZIP compression ratio for Mus_pahari.PAHARI_EIJ_v1.1.cds.all.fa is 14.74%\n",
      "cds 20719 , total =  31811562\n",
      "The GECO3 compression ratio for Myotis_lucifugus.Myoluc2.0.cds.all.fa is 22.12%\n",
      "The GZIP compression ratio for Myotis_lucifugus.Myoluc2.0.cds.all.fa is 26.72%\n",
      "ncrna 13650 , total =  11212848\n",
      "The GECO3 compression ratio for Mus_musculus_wsbeij.WSB_EiJ_v1.ncrna.fa is 20.50%\n",
      "The GZIP compression ratio for Mus_musculus_wsbeij.WSB_EiJ_v1.ncrna.fa is 24.24%\n",
      "mus_musculus_wsbeij completed successfully!\n",
      "ncrna 9367 , total =  4837754\n",
      "The GECO3 compression ratio for Mus_spicilegus.MUSP714.ncrna.fa is 19.10%\n",
      "The GZIP compression ratio for Mus_spicilegus.MUSP714.ncrna.fa is 22.73%\n",
      "mus_spicilegus completed successfully!\n",
      "ncrna 4408 , total =  541807\n",
      "The GECO3 compression ratio for Myotis_lucifugus.Myoluc2.0.ncrna.fa is 14.27%\n",
      "The GZIP compression ratio for Myotis_lucifugus.Myoluc2.0.ncrna.fa is 19.62%\n",
      "myotis_lucifugus completed successfully!\n",
      "ncrna 13218 , total =  7984085\n",
      "The GECO3 compression ratio for Mustela_putorius_furo.MusPutFur1.0.ncrna.fa is 21.69%\n",
      "The GZIP compression ratio for Mustela_putorius_furo.MusPutFur1.0.ncrna.fa is 25.83%\n",
      "mustela_putorius_furo completed successfully!\n",
      "ncrna 11622 , total =  9689384\n",
      "The GECO3 compression ratio for Mus_pahari.PAHARI_EIJ_v1.1.ncrna.fa is 20.71%\n",
      "The GZIP compression ratio for Mus_pahari.PAHARI_EIJ_v1.1.ncrna.fa is 24.40%\n",
      "mus_pahari completed successfully!\n",
      "cds 52631 , total =  72833534\n",
      "The GECO3 compression ratio for Mus_spretus.SPRET_EiJ_v1.cds.all.fa is 14.69%\n",
      "The GZIP compression ratio for Mus_spretus.SPRET_EiJ_v1.cds.all.fa is 14.38%\n",
      "cds 29965 , total =  53507748\n",
      "The GECO3 compression ratio for Naja_naja.Nana_v5.cds.all.fa is 14.78%\n",
      "The GZIP compression ratio for Naja_naja.Nana_v5.cds.all.fa is 15.64%\n",
      "ncrna 1345 , total =  898989\n",
      "The GECO3 compression ratio for Naja_naja.Nana_v5.ncrna.fa is 17.87%\n",
      "The GZIP compression ratio for Naja_naja.Nana_v5.ncrna.fa is 21.80%\n",
      "naja_naja completed successfully!\n",
      "ncrna 13239 , total =  10916590\n",
      "The GECO3 compression ratio for Mus_spretus.SPRET_EiJ_v1.ncrna.fa is 20.56%\n",
      "The GZIP compression ratio for Mus_spretus.SPRET_EiJ_v1.ncrna.fa is 24.30%\n",
      "mus_spretus completed successfully!\n",
      "cds 26872 , total =  44177206\n",
      "The GECO3 compression ratio for Nannospalax_galili.S.galili_v1.0.cds.all.fa is 17.90%\n",
      "The GZIP compression ratio for Nannospalax_galili.S.galili_v1.0.cds.all.fa is 19.33%\n",
      "ncrna 5623 , total =  1042162\n",
      "The GECO3 compression ratio for Nannospalax_galili.S.galili_v1.0.ncrna.fa is 18.78%\n",
      "The GZIP compression ratio for Nannospalax_galili.S.galili_v1.0.ncrna.fa is 23.73%\n",
      "nannospalax_galili completed successfully!\n",
      "cds 32146 , total =  45226817\n",
      "The GECO3 compression ratio for Neolamprologus_brichardi.NeoBri1.0.cds.all.fa is 19.47%\n",
      "The GZIP compression ratio for Neolamprologus_brichardi.NeoBri1.0.cds.all.fa is 21.70%\n",
      "ncrna 776 , total =  72987\n",
      "The GECO3 compression ratio for Neolamprologus_brichardi.NeoBri1.0.ncrna.fa is 20.33%\n",
      "The GZIP compression ratio for Neolamprologus_brichardi.NeoBri1.0.ncrna.fa is 24.87%\n",
      "neolamprologus_brichardi completed successfully!\n",
      "cds 31888 , total =  53450482\n",
      "The GECO3 compression ratio for Neovison_vison.NNQGG.v01.cds.all.fa is 16.57%\n",
      "The GZIP compression ratio for Neovison_vison.NNQGG.v01.cds.all.fa is 17.54%\n",
      "ncrna 4709 , total =  1177803\n",
      "The GECO3 compression ratio for Neovison_vison.NNQGG.v01.ncrna.fa is 18.42%\n",
      "The GZIP compression ratio for Neovison_vison.NNQGG.v01.ncrna.fa is 22.73%\n",
      "neovison_vison completed successfully!\n",
      "cds 15342 , total =  21849677\n",
      "The GECO3 compression ratio for Notamacropus_eugenii.Meug_1.0.cds.all.fa is 23.67%\n",
      "The GZIP compression ratio for Notamacropus_eugenii.Meug_1.0.cds.all.fa is 27.85%\n",
      "ncrna 1472 , total =  158340\n",
      "The GECO3 compression ratio for Notamacropus_eugenii.Meug_1.0.ncrna.fa is 18.61%\n",
      "The GZIP compression ratio for Notamacropus_eugenii.Meug_1.0.ncrna.fa is 23.22%\n",
      "notamacropus_eugenii completed successfully!\n",
      "cds 45100 , total =  66011748\n",
      "The GECO3 compression ratio for Neogobius_melanostomus.RGoby_Basel_V2.cds.all.fa is 13.64%\n",
      "The GZIP compression ratio for Neogobius_melanostomus.RGoby_Basel_V2.cds.all.fa is 14.05%\n",
      "cds 56108 , total =  103228310\n",
      "The GECO3 compression ratio for Myripristis_murdjan.fMyrMur1.1.cds.all.fa is 13.65%\n",
      "The GZIP compression ratio for Myripristis_murdjan.fMyrMur1.1.cds.all.fa is 11.93%\n",
      "ncrna 3181 , total =  1794795\n",
      "The GECO3 compression ratio for Neogobius_melanostomus.RGoby_Basel_V2.ncrna.fa is 17.17%\n",
      "The GZIP compression ratio for Neogobius_melanostomus.RGoby_Basel_V2.ncrna.fa is 21.50%\n",
      "neogobius_melanostomus completed successfully!\n",
      "ncrna 871 , total =  297365\n",
      "The GECO3 compression ratio for Myripristis_murdjan.fMyrMur1.1.ncrna.fa is 18.70%\n",
      "The GZIP compression ratio for Myripristis_murdjan.fMyrMur1.1.ncrna.fa is 22.44%\n",
      "myripristis_murdjan completed successfully!\n",
      "cds 40527 , total =  62299050\n",
      "The GECO3 compression ratio for Nomascus_leucogenys.Nleu_3.0.cds.all.fa is 15.71%\n",
      "The GZIP compression ratio for Nomascus_leucogenys.Nleu_3.0.cds.all.fa is 16.01%\n",
      "ncrna 6465 , total =  869623\n",
      "The GECO3 compression ratio for Nomascus_leucogenys.Nleu_3.0.ncrna.fa is 16.30%\n",
      "The GZIP compression ratio for Nomascus_leucogenys.Nleu_3.0.ncrna.fa is 22.15%\n",
      "nomascus_leucogenys completed successfully!\n",
      "cds 21931 , total =  33658517\n",
      "The GECO3 compression ratio for Nothoprocta_perdicaria.notPer1.cds.all.fa is 18.16%\n",
      "The GZIP compression ratio for Nothoprocta_perdicaria.notPer1.cds.all.fa is 20.40%\n",
      "ncrna 461 , total =  56402\n",
      "The GECO3 compression ratio for Nothoprocta_perdicaria.notPer1.ncrna.fa is 23.18%\n",
      "The GZIP compression ratio for Nothoprocta_perdicaria.notPer1.ncrna.fa is 28.01%\n",
      "nothoprocta_perdicaria completed successfully!\n",
      "cds 29097 , total =  46718070\n",
      "The GECO3 compression ratio for Notechis_scutatus.TS10Xv2-PRI.cds.all.fa is 15.50%\n",
      "The GZIP compression ratio for Notechis_scutatus.TS10Xv2-PRI.cds.all.fa is 17.02%\n",
      "ncrna 1691 , total =  803465\n",
      "The GECO3 compression ratio for Notechis_scutatus.TS10Xv2-PRI.ncrna.fa is 20.13%\n",
      "The GZIP compression ratio for Notechis_scutatus.TS10Xv2-PRI.ncrna.fa is 24.39%\n",
      "notechis_scutatus completed successfully!\n",
      "cds 16006 , total =  23425776\n",
      "The GECO3 compression ratio for Ochotona_princeps.OchPri2.0-Ens.cds.all.fa is 23.47%\n",
      "The GZIP compression ratio for Ochotona_princeps.OchPri2.0-Ens.cds.all.fa is 27.90%\n",
      "ncrna 5732 , total =  697107\n",
      "The GECO3 compression ratio for Ochotona_princeps.OchPri2.0-Ens.ncrna.fa is 11.76%\n",
      "The GZIP compression ratio for Ochotona_princeps.OchPri2.0-Ens.ncrna.fa is 17.94%\n",
      "ochotona_princeps completed successfully!\n",
      "cds 27047 , total =  42137549\n",
      "The GECO3 compression ratio for Octodon_degus.OctDeg1.0.cds.all.fa is 17.99%\n",
      "The GZIP compression ratio for Octodon_degus.OctDeg1.0.cds.all.fa is 20.12%\n",
      "ncrna 5340 , total =  769137\n",
      "The GECO3 compression ratio for Octodon_degus.OctDeg1.0.ncrna.fa is 15.66%\n",
      "The GZIP compression ratio for Octodon_degus.OctDeg1.0.ncrna.fa is 20.80%\n",
      "octodon_degus completed successfully!\n",
      "cds 27847 , total =  59451877\n",
      "The GECO3 compression ratio for Numida_meleagris.NumMel1.0.cds.all.fa is 13.94%\n",
      "The GZIP compression ratio for Numida_meleagris.NumMel1.0.cds.all.fa is 14.17%\n",
      "ncrna 12181 , total =  11503054\n",
      "The GECO3 compression ratio for Numida_meleagris.NumMel1.0.ncrna.fa is 18.23%\n",
      "The GZIP compression ratio for Numida_meleagris.NumMel1.0.ncrna.fa is 21.60%\n",
      "numida_meleagris completed successfully!\n",
      "cds 54965 , total =  105052098\n",
      "The GECO3 compression ratio for Nothobranchius_furzeri.Nfu_20140520.cds.all.fa is 12.94%\n",
      "The GZIP compression ratio for Nothobranchius_furzeri.Nfu_20140520.cds.all.fa is 12.01%\n",
      "ncrna 1671 , total =  2608373\n",
      "The GECO3 compression ratio for Nothobranchius_furzeri.Nfu_20140520.ncrna.fa is 17.64%\n",
      "The GZIP compression ratio for Nothobranchius_furzeri.Nfu_20140520.ncrna.fa is 23.20%\n",
      "nothobranchius_furzeri completed successfully!\n",
      "cds 32939 , total =  67423057\n",
      "The GECO3 compression ratio for Ornithorhynchus_anatinus.mOrnAna1.p.v1.cds.all.fa is 14.10%\n",
      "The GZIP compression ratio for Ornithorhynchus_anatinus.mOrnAna1.p.v1.cds.all.fa is 14.08%\n",
      "ncrna 9746 , total =  4488273\n",
      "The GECO3 compression ratio for Ornithorhynchus_anatinus.mOrnAna1.p.v1.ncrna.fa is 15.57%\n",
      "The GZIP compression ratio for Ornithorhynchus_anatinus.mOrnAna1.p.v1.ncrna.fa is 20.31%\n",
      "ornithorhynchus_anatinus completed successfully!\n",
      "cds 59303 , total =  112548826\n",
      "The GECO3 compression ratio for Oreochromis_aureus.ASM587006v1.cds.all.fa is 13.00%\n",
      "The GZIP compression ratio for Oreochromis_aureus.ASM587006v1.cds.all.fa is 11.20%\n",
      "ncrna 1215 , total =  351281\n",
      "The GECO3 compression ratio for Oreochromis_aureus.ASM587006v1.ncrna.fa is 19.50%\n",
      "The GZIP compression ratio for Oreochromis_aureus.ASM587006v1.ncrna.fa is 23.37%\n",
      "oreochromis_aureus completed successfully!\n",
      "cds 41055 , total =  73750978\n",
      "The GECO3 compression ratio for Oryctolagus_cuniculus.OryCun2.0.cds.all.fa is 13.77%\n",
      "The GZIP compression ratio for Oryctolagus_cuniculus.OryCun2.0.cds.all.fa is 14.01%\n",
      "ncrna 10142 , total =  9849549\n",
      "The GECO3 compression ratio for Oryctolagus_cuniculus.OryCun2.0.ncrna.fa is 19.43%\n",
      "The GZIP compression ratio for Oryctolagus_cuniculus.OryCun2.0.ncrna.fa is 24.05%\n",
      "oryctolagus_cuniculus completed successfully!\n",
      "cds 75555 , total =  152206382\n",
      "The GECO3 compression ratio for Oreochromis_niloticus.O_niloticus_UMD_NMBU.cds.all.fa is 12.68%\n",
      "The GZIP compression ratio for Oreochromis_niloticus.O_niloticus_UMD_NMBU.cds.all.fa is 10.63%\n",
      "cds 50290 , total =  89054261\n",
      "The GECO3 compression ratio for Oryzias_javanicus.OJAV_1.1.cds.all.fa is 13.20%\n",
      "The GZIP compression ratio for Oryzias_javanicus.OJAV_1.1.cds.all.fa is 12.21%\n",
      "ncrna 2072 , total =  2365404\n",
      "The GECO3 compression ratio for Oryzias_javanicus.OJAV_1.1.ncrna.fa is 16.50%\n",
      "The GZIP compression ratio for Oryzias_javanicus.OJAV_1.1.ncrna.fa is 20.22%\n",
      "oryzias_javanicus completed successfully!\n",
      "ncrna 6269 , total =  8354063\n",
      "The GECO3 compression ratio for Oreochromis_niloticus.O_niloticus_UMD_NMBU.ncrna.fa is 14.40%\n",
      "The GZIP compression ratio for Oreochromis_niloticus.O_niloticus_UMD_NMBU.ncrna.fa is 19.57%\n",
      "oreochromis_niloticus completed successfully!\n",
      "cds 37468 , total =  67838709\n",
      "The GECO3 compression ratio for Oryzias_latipes.ASM223467v1.cds.all.fa is 16.48%\n",
      "The GZIP compression ratio for Oryzias_latipes.ASM223467v1.cds.all.fa is 17.44%\n",
      "ncrna 716 , total =  87223\n",
      "The GECO3 compression ratio for Oryzias_latipes.ASM223467v1.ncrna.fa is 15.53%\n",
      "The GZIP compression ratio for Oryzias_latipes.ASM223467v1.ncrna.fa is 19.16%\n",
      "oryzias_latipes completed successfully!\n",
      "cds 35451 , total =  63276545\n",
      "The GECO3 compression ratio for Oryzias_latipes_hni.ASM223471v1.cds.all.fa is 16.58%\n",
      "The GZIP compression ratio for Oryzias_latipes_hni.ASM223471v1.cds.all.fa is 17.32%\n",
      "ncrna 528 , total =  62116\n",
      "The GECO3 compression ratio for Oryzias_latipes_hni.ASM223471v1.ncrna.fa is 17.50%\n",
      "The GZIP compression ratio for Oryzias_latipes_hni.ASM223471v1.ncrna.fa is 21.64%\n",
      "oryzias_latipes_hni completed successfully!\n",
      "cds 105189 , total =  199567247\n",
      "The GECO3 compression ratio for Oncorhynchus_tshawytscha.Otsh_v1.0.cds.all.fa is 13.57%\n",
      "The GZIP compression ratio for Oncorhynchus_tshawytscha.Otsh_v1.0.cds.all.fa is 11.02%\n",
      "cds 36044 , total =  62857825\n",
      "The GECO3 compression ratio for Oryzias_latipes_hsok.ASM223469v1.cds.all.fa is 16.78%\n",
      "The GZIP compression ratio for Oryzias_latipes_hsok.ASM223469v1.cds.all.fa is 17.61%\n",
      "ncrna 499 , total =  59542\n",
      "The GECO3 compression ratio for Oryzias_latipes_hsok.ASM223469v1.ncrna.fa is 17.60%\n",
      "The GZIP compression ratio for Oryzias_latipes_hsok.ASM223469v1.ncrna.fa is 21.68%\n",
      "oryzias_latipes_hsok completed successfully!\n",
      "ncrna 7704 , total =  8562496\n",
      "The GECO3 compression ratio for Oncorhynchus_tshawytscha.Otsh_v1.0.ncrna.fa is 15.03%\n",
      "The GZIP compression ratio for Oncorhynchus_tshawytscha.Otsh_v1.0.ncrna.fa is 18.75%\n",
      "oncorhynchus_tshawytscha completed successfully!\n",
      "cds 118040 , total =  243325112\n",
      "The GECO3 compression ratio for Oncorhynchus_kisutch.Okis_V2.cds.all.fa is 13.81%\n",
      "The GZIP compression ratio for Oncorhynchus_kisutch.Okis_V2.cds.all.fa is 10.29%\n",
      "cds 19926 , total =  32265600\n",
      "The GECO3 compression ratio for Otolemur_garnettii.OtoGar3.cds.all.fa is 23.10%\n",
      "The GZIP compression ratio for Otolemur_garnettii.OtoGar3.cds.all.fa is 27.19%\n",
      "ncrna 7276 , total =  944804\n",
      "The GECO3 compression ratio for Otolemur_garnettii.OtoGar3.ncrna.fa is 14.85%\n",
      "The GZIP compression ratio for Otolemur_garnettii.OtoGar3.ncrna.fa is 20.64%\n",
      "otolemur_garnettii completed successfully!\n",
      "ncrna 7575 , total =  6628355\n",
      "The GECO3 compression ratio for Oncorhynchus_kisutch.Okis_V2.ncrna.fa is 11.96%\n",
      "The GZIP compression ratio for Oncorhynchus_kisutch.Okis_V2.ncrna.fa is 15.95%\n",
      "oncorhynchus_kisutch completed successfully!\n",
      "cds 36345 , total =  61442655\n",
      "The GECO3 compression ratio for Oryzias_melastigma.Om_v0.7.RACA.cds.all.fa is 17.21%\n",
      "The GZIP compression ratio for Oryzias_melastigma.Om_v0.7.RACA.cds.all.fa is 18.05%\n",
      "ncrna 450 , total =  57340\n",
      "The GECO3 compression ratio for Oryzias_melastigma.Om_v0.7.RACA.ncrna.fa is 16.43%\n",
      "The GZIP compression ratio for Oryzias_melastigma.Om_v0.7.RACA.ncrna.fa is 20.68%\n",
      "oryzias_melastigma completed successfully!\n",
      "cds 24055 , total =  48183936\n",
      "The GECO3 compression ratio for Otus_sunia.OtuSun1.0.cds.all.fa is 15.15%\n",
      "The GZIP compression ratio for Otus_sunia.OtuSun1.0.cds.all.fa is 16.26%\n",
      "cds 121548 , total =  256587121\n",
      "The GECO3 compression ratio for Oncorhynchus_mykiss.USDA_OmykA_1.1.cds.all.fa is 13.51%\n",
      "The GZIP compression ratio for Oncorhynchus_mykiss.USDA_OmykA_1.1.cds.all.fa is 10.58%\n",
      "ncrna 628 , total =  287738\n",
      "The GECO3 compression ratio for Otus_sunia.OtuSun1.0.ncrna.fa is 21.44%\n",
      "The GZIP compression ratio for Otus_sunia.OtuSun1.0.ncrna.fa is 25.56%\n",
      "otus_sunia completed successfully!\n",
      "ncrna 16287 , total =  9507944\n",
      "The GECO3 compression ratio for Oncorhynchus_mykiss.USDA_OmykA_1.1.ncrna.fa is 8.22%\n",
      "The GZIP compression ratio for Oncorhynchus_mykiss.USDA_OmykA_1.1.ncrna.fa is 11.57%\n",
      "oncorhynchus_mykiss completed successfully!\n",
      "cds 22823 , total =  36423291\n",
      "The GECO3 compression ratio for Ovis_aries.Oar_v3.1.cds.all.fa is 21.60%\n",
      "The GZIP compression ratio for Ovis_aries.Oar_v3.1.cds.all.fa is 25.29%\n",
      "ncrna 6005 , total =  4072937\n",
      "The GECO3 compression ratio for Ovis_aries.Oar_v3.1.ncrna.fa is 20.53%\n",
      "The GZIP compression ratio for Ovis_aries.Oar_v3.1.ncrna.fa is 25.73%\n",
      "ovis_aries completed successfully!\n",
      "cds 51968 , total =  92862884\n",
      "The GECO3 compression ratio for Oryzias_sinensis.ASM858656v1.cds.all.fa is 12.88%\n",
      "The GZIP compression ratio for Oryzias_sinensis.ASM858656v1.cds.all.fa is 11.65%\n",
      "ncrna 1802 , total =  2637748\n",
      "The GECO3 compression ratio for Oryzias_sinensis.ASM858656v1.ncrna.fa is 18.04%\n",
      "The GZIP compression ratio for Oryzias_sinensis.ASM858656v1.ncrna.fa is 22.03%\n",
      "oryzias_sinensis completed successfully!\n",
      "cds 31175 , total =  53052292\n",
      "The GECO3 compression ratio for Panthera_leo.PanLeo1.0.cds.all.fa is 16.90%\n",
      "The GZIP compression ratio for Panthera_leo.PanLeo1.0.cds.all.fa is 18.26%\n",
      "ncrna 2741 , total =  585665\n",
      "The GECO3 compression ratio for Panthera_leo.PanLeo1.0.ncrna.fa is 18.83%\n",
      "The GZIP compression ratio for Panthera_leo.PanLeo1.0.ncrna.fa is 23.52%\n",
      "panthera_leo completed successfully!\n",
      "cds 29763 , total =  46235788\n",
      "The GECO3 compression ratio for Panthera_pardus.PanPar1.0.cds.all.fa is 17.93%\n",
      "The GZIP compression ratio for Panthera_pardus.PanPar1.0.cds.all.fa is 19.59%\n",
      "ncrna 3900 , total =  500833\n",
      "The GECO3 compression ratio for Panthera_pardus.PanPar1.0.ncrna.fa is 15.56%\n",
      "The GZIP compression ratio for Panthera_pardus.PanPar1.0.ncrna.fa is 20.57%\n",
      "panthera_pardus completed successfully!\n",
      "cds 26795 , total =  40737347\n",
      "The GECO3 compression ratio for Panthera_tigris_altaica.PanTig1.0.cds.all.fa is 17.81%\n",
      "The GZIP compression ratio for Panthera_tigris_altaica.PanTig1.0.cds.all.fa is 19.47%\n",
      "cds 43232 , total =  68488732\n",
      "The GECO3 compression ratio for Pan_paniscus.panpan1.1.cds.all.fa is 15.31%\n",
      "The GZIP compression ratio for Pan_paniscus.panpan1.1.cds.all.fa is 15.14%\n",
      "ncrna 3602 , total =  468030\n",
      "The GECO3 compression ratio for Panthera_tigris_altaica.PanTig1.0.ncrna.fa is 15.78%\n",
      "The GZIP compression ratio for Panthera_tigris_altaica.PanTig1.0.ncrna.fa is 20.94%\n",
      "panthera_tigris_altaica completed successfully!\n",
      "cds 52652 , total =  88584162\n",
      "The GECO3 compression ratio for Ovis_aries_rambouillet.ARS-UI_Ramb_v2.0.cds.all.fa is 13.61%\n",
      "The GZIP compression ratio for Ovis_aries_rambouillet.ARS-UI_Ramb_v2.0.cds.all.fa is 12.96%\n",
      "ncrna 9579 , total =  3744325\n",
      "The GECO3 compression ratio for Pan_paniscus.panpan1.1.ncrna.fa is 17.46%\n",
      "The GZIP compression ratio for Pan_paniscus.panpan1.1.ncrna.fa is 21.58%\n",
      "pan_paniscus completed successfully!\n",
      "cds 49949 , total =  79771423\n",
      "The GECO3 compression ratio for Pan_troglodytes.Pan_tro_3.0.cds.all.fa is 14.71%\n",
      "The GZIP compression ratio for Pan_troglodytes.Pan_tro_3.0.cds.all.fa is 14.47%\n",
      "ncrna 11023 , total =  4426035\n",
      "The GECO3 compression ratio for Pan_troglodytes.Pan_tro_3.0.ncrna.fa is 16.98%\n",
      "The GZIP compression ratio for Pan_troglodytes.Pan_tro_3.0.ncrna.fa is 21.20%\n",
      "pan_troglodytes completed successfully!\n",
      "ncrna 14550 , total =  21158983\n",
      "The GECO3 compression ratio for Ovis_aries_rambouillet.ARS-UI_Ramb_v2.0.ncrna.fa is 16.45%\n",
      "The GZIP compression ratio for Ovis_aries_rambouillet.ARS-UI_Ramb_v2.0.ncrna.fa is 20.06%\n",
      "ovis_aries_rambouillet completed successfully!\n",
      "cds 43861 , total =  78595786\n",
      "The GECO3 compression ratio for Papio_anubis.Panubis1.0.cds.all.fa is 13.73%\n",
      "The GZIP compression ratio for Papio_anubis.Panubis1.0.cds.all.fa is 14.03%\n",
      "cds 20669 , total =  32805035\n",
      "The GECO3 compression ratio for Pelodiscus_sinensis.PelSin_1.0.cds.all.fa is 20.35%\n",
      "The GZIP compression ratio for Pelodiscus_sinensis.PelSin_1.0.cds.all.fa is 23.88%\n",
      "ncrna 1042 , total =  121411\n",
      "The GECO3 compression ratio for Pelodiscus_sinensis.PelSin_1.0.ncrna.fa is 19.25%\n",
      "The GZIP compression ratio for Pelodiscus_sinensis.PelSin_1.0.ncrna.fa is 23.59%\n",
      "pelodiscus_sinensis completed successfully!\n",
      "cds 26506 , total =  41885913\n",
      "The GECO3 compression ratio for Pavo_cristatus.AIIM_Pcri_1.0.cds.all.fa is 14.97%\n",
      "The GZIP compression ratio for Pavo_cristatus.AIIM_Pcri_1.0.cds.all.fa is 17.24%\n",
      "ncrna 1163 , total =  1054344\n",
      "The GECO3 compression ratio for Pavo_cristatus.AIIM_Pcri_1.0.ncrna.fa is 21.44%\n",
      "The GZIP compression ratio for Pavo_cristatus.AIIM_Pcri_1.0.ncrna.fa is 25.70%\n",
      "pavo_cristatus completed successfully!\n",
      "ncrna 12730 , total =  8675915\n",
      "The GECO3 compression ratio for Papio_anubis.Panubis1.0.ncrna.fa is 18.72%\n",
      "The GZIP compression ratio for Papio_anubis.Panubis1.0.ncrna.fa is 23.31%\n",
      "papio_anubis completed successfully!\n",
      "cds 27488 , total =  57835170\n",
      "The GECO3 compression ratio for Parus_major.Parus_major1.1.cds.all.fa is 14.24%\n",
      "The GZIP compression ratio for Parus_major.Parus_major1.1.cds.all.fa is 14.12%\n",
      "cds 11442 , total =  13836153\n",
      "The GECO3 compression ratio for Petromyzon_marinus.Pmarinus_7.0.cds.all.fa is 21.34%\n",
      "The GZIP compression ratio for Petromyzon_marinus.Pmarinus_7.0.cds.all.fa is 25.25%\n",
      "ncrna 2652 , total =  322994\n",
      "The GECO3 compression ratio for Petromyzon_marinus.Pmarinus_7.0.ncrna.fa is 10.00%\n",
      "The GZIP compression ratio for Petromyzon_marinus.Pmarinus_7.0.ncrna.fa is 13.54%\n",
      "petromyzon_marinus completed successfully!\n",
      "ncrna 7788 , total =  6817822\n",
      "The GECO3 compression ratio for Parus_major.Parus_major1.1.ncrna.fa is 18.34%\n",
      "The GZIP compression ratio for Parus_major.Parus_major1.1.ncrna.fa is 21.80%\n",
      "parus_major completed successfully!\n",
      "cds 30071 , total =  39112243\n",
      "The GECO3 compression ratio for Periophthalmus_magnuspinnatus.PM.fa.cds.all.fa is 19.45%\n",
      "The GZIP compression ratio for Periophthalmus_magnuspinnatus.PM.fa.cds.all.fa is 21.86%\n",
      "cds 42077 , total =  73314470\n",
      "The GECO3 compression ratio for Paramormyrops_kingsleyae.PKINGS_0.1.cds.all.fa is 16.69%\n",
      "The GZIP compression ratio for Paramormyrops_kingsleyae.PKINGS_0.1.cds.all.fa is 16.68%\n",
      "cds 57281 , total =  105879843\n",
      "The GECO3 compression ratio for Parambassis_ranga.fParRan2.1.cds.all.fa is 13.11%\n",
      "The GZIP compression ratio for Parambassis_ranga.fParRan2.1.cds.all.fa is 11.52%\n",
      "ncrna 1901 , total =  215390\n",
      "The GECO3 compression ratio for Periophthalmus_magnuspinnatus.PM.fa.ncrna.fa is 15.80%\n",
      "The GZIP compression ratio for Periophthalmus_magnuspinnatus.PM.fa.ncrna.fa is 21.42%\n",
      "periophthalmus_magnuspinnatus completed successfully!\n",
      "cds 27057 , total =  48633937\n",
      "The GECO3 compression ratio for Pelusios_castaneus.Pelusios_castaneus-1.0.cds.all.fa is 16.88%\n",
      "The GZIP compression ratio for Pelusios_castaneus.Pelusios_castaneus-1.0.cds.all.fa is 18.57%\n",
      "ncrna 860 , total =  102469\n",
      "The GECO3 compression ratio for Paramormyrops_kingsleyae.PKINGS_0.1.ncrna.fa is 16.00%\n",
      "The GZIP compression ratio for Paramormyrops_kingsleyae.PKINGS_0.1.ncrna.fa is 21.37%\n",
      "paramormyrops_kingsleyae completed successfully!\n",
      "ncrna 804 , total =  286403\n",
      "The GECO3 compression ratio for Pelusios_castaneus.Pelusios_castaneus-1.0.ncrna.fa is 18.44%\n",
      "The GZIP compression ratio for Pelusios_castaneus.Pelusios_castaneus-1.0.ncrna.fa is 22.42%\n",
      "pelusios_castaneus completed successfully!\n",
      "ncrna 2541 , total =  2077907\n",
      "The GECO3 compression ratio for Parambassis_ranga.fParRan2.1.ncrna.fa is 16.59%\n",
      "The GZIP compression ratio for Parambassis_ranga.fParRan2.1.ncrna.fa is 21.61%\n",
      "parambassis_ranga completed successfully!\n",
      "cds 28171 , total =  44724010\n",
      "The GECO3 compression ratio for Peromyscus_maniculatus_bairdii.HU_Pman_2.1.cds.all.fa is 19.23%\n",
      "The GZIP compression ratio for Peromyscus_maniculatus_bairdii.HU_Pman_2.1.cds.all.fa is 21.46%\n",
      "ncrna 2987 , total =  413615\n",
      "The GECO3 compression ratio for Peromyscus_maniculatus_bairdii.HU_Pman_2.1.ncrna.fa is 18.35%\n",
      "The GZIP compression ratio for Peromyscus_maniculatus_bairdii.HU_Pman_2.1.ncrna.fa is 23.16%\n",
      "peromyscus_maniculatus_bairdii completed successfully!\n",
      "cds 25525 , total =  53800082\n",
      "The GECO3 compression ratio for Phasianus_colchicus.ASM414374v1.cds.all.fa is 14.66%\n",
      "The GZIP compression ratio for Phasianus_colchicus.ASM414374v1.cds.all.fa is 15.28%\n",
      "cds 31556 , total =  52989461\n",
      "The GECO3 compression ratio for Phocoena_sinus.mPhoSin1.pri.cds.all.fa is 16.49%\n",
      "The GZIP compression ratio for Phocoena_sinus.mPhoSin1.pri.cds.all.fa is 17.45%\n",
      "cds 37921 , total =  71674219\n",
      "The GECO3 compression ratio for Phascolarctos_cinereus.phaCin_unsw_v4.1.cds.all.fa is 14.09%\n",
      "The GZIP compression ratio for Phascolarctos_cinereus.phaCin_unsw_v4.1.cds.all.fa is 13.80%\n",
      "ncrna 3097 , total =  2690451\n",
      "The GECO3 compression ratio for Phocoena_sinus.mPhoSin1.pri.ncrna.fa is 21.05%\n",
      "The GZIP compression ratio for Phocoena_sinus.mPhoSin1.pri.ncrna.fa is 25.96%\n",
      "phocoena_sinus completed successfully!\n",
      "cds 32206 , total =  53901178\n",
      "The GECO3 compression ratio for Physeter_catodon.ASM283717v2.cds.all.fa is 16.07%\n",
      "The GZIP compression ratio for Physeter_catodon.ASM283717v2.cds.all.fa is 17.11%\n",
      "ncrna 7261 , total =  5212343\n",
      "The GECO3 compression ratio for Phascolarctos_cinereus.phaCin_unsw_v4.1.ncrna.fa is 18.39%\n",
      "The GZIP compression ratio for Phascolarctos_cinereus.phaCin_unsw_v4.1.ncrna.fa is 22.21%\n",
      "phascolarctos_cinereus completed successfully!\n",
      "ncrna 2380 , total =  436623\n",
      "The GECO3 compression ratio for Physeter_catodon.ASM283717v2.ncrna.fa is 17.47%\n",
      "The GZIP compression ratio for Physeter_catodon.ASM283717v2.ncrna.fa is 21.98%\n",
      "physeter_catodon completed successfully!\n",
      "ncrna 9683 , total =  9350873\n",
      "The GECO3 compression ratio for Phasianus_colchicus.ASM414374v1.ncrna.fa is 18.37%\n",
      "The GZIP compression ratio for Phasianus_colchicus.ASM414374v1.ncrna.fa is 21.76%\n",
      "phasianus_colchicus completed successfully!\n",
      "cds 30898 , total =  52906151\n",
      "The GECO3 compression ratio for Poecilia_formosa.PoeFor_5.1.2.cds.all.fa is 19.76%\n",
      "The GZIP compression ratio for Poecilia_formosa.PoeFor_5.1.2.cds.all.fa is 21.97%\n",
      "ncrna 679 , total =  71907\n",
      "The GECO3 compression ratio for Poecilia_formosa.PoeFor_5.1.2.ncrna.fa is 19.21%\n",
      "The GZIP compression ratio for Poecilia_formosa.PoeFor_5.1.2.ncrna.fa is 23.27%\n",
      "poecilia_formosa completed successfully!\n",
      "cds 41245 , total =  64911133\n",
      "The GECO3 compression ratio for Piliocolobus_tephrosceles.ASM277652v2.cds.all.fa is 16.02%\n",
      "The GZIP compression ratio for Piliocolobus_tephrosceles.ASM277652v2.cds.all.fa is 17.19%\n",
      "cds 31774 , total =  49172446\n",
      "The GECO3 compression ratio for Poecilia_latipinna.P_latipinna-1.0.cds.all.fa is 19.01%\n",
      "The GZIP compression ratio for Poecilia_latipinna.P_latipinna-1.0.cds.all.fa is 21.11%\n",
      "ncrna 355 , total =  40251\n",
      "The GECO3 compression ratio for Poecilia_latipinna.P_latipinna-1.0.ncrna.fa is 18.74%\n",
      "The GZIP compression ratio for Poecilia_latipinna.P_latipinna-1.0.ncrna.fa is 22.97%\n",
      "poecilia_latipinna completed successfully!\n",
      "cds 37240 , total =  75422373\n",
      "The GECO3 compression ratio for Podarcis_muralis.PodMur_1.0.cds.all.fa is 13.69%\n",
      "The GZIP compression ratio for Podarcis_muralis.PodMur_1.0.cds.all.fa is 14.84%\n",
      "ncrna 1920 , total =  2065513\n",
      "The GECO3 compression ratio for Podarcis_muralis.PodMur_1.0.ncrna.fa is 18.85%\n",
      "The GZIP compression ratio for Podarcis_muralis.PodMur_1.0.ncrna.fa is 24.28%\n",
      "podarcis_muralis completed successfully!\n",
      "ncrna 13641 , total =  13401838\n",
      "The GECO3 compression ratio for Piliocolobus_tephrosceles.ASM277652v2.ncrna.fa is 18.63%\n",
      "The GZIP compression ratio for Piliocolobus_tephrosceles.ASM277652v2.ncrna.fa is 22.46%\n",
      "piliocolobus_tephrosceles completed successfully!\n",
      "cds 16114 , total =  23706028\n",
      "The GECO3 compression ratio for Procavia_capensis.proCap1.cds.all.fa is 23.62%\n",
      "The GZIP compression ratio for Procavia_capensis.proCap1.cds.all.fa is 27.84%\n",
      "ncrna 1720 , total =  181978\n",
      "The GECO3 compression ratio for Procavia_capensis.proCap1.ncrna.fa is 18.60%\n",
      "The GZIP compression ratio for Procavia_capensis.proCap1.ncrna.fa is 23.27%\n",
      "procavia_capensis completed successfully!\n",
      "cds 34964 , total =  56601308\n",
      "The GECO3 compression ratio for Poecilia_mexicana.P_mexicana-1.0.cds.all.fa is 17.99%\n",
      "The GZIP compression ratio for Poecilia_mexicana.P_mexicana-1.0.cds.all.fa is 19.36%\n",
      "ncrna 359 , total =  40548\n",
      "The GECO3 compression ratio for Poecilia_mexicana.P_mexicana-1.0.ncrna.fa is 18.76%\n",
      "The GZIP compression ratio for Poecilia_mexicana.P_mexicana-1.0.ncrna.fa is 22.90%\n",
      "poecilia_mexicana completed successfully!\n",
      "cds 34347 , total =  58829153\n",
      "The GECO3 compression ratio for Poecilia_reticulata.Guppy_female_1.0_MT.cds.all.fa is 17.54%\n",
      "The GZIP compression ratio for Poecilia_reticulata.Guppy_female_1.0_MT.cds.all.fa is 18.44%\n",
      "ncrna 358 , total =  43070\n",
      "The GECO3 compression ratio for Poecilia_reticulata.Guppy_female_1.0_MT.ncrna.fa is 19.89%\n",
      "The GZIP compression ratio for Poecilia_reticulata.Guppy_female_1.0_MT.ncrna.fa is 24.38%\n",
      "poecilia_reticulata completed successfully!\n",
      "cds 38110 , total =  62190488\n",
      "The GECO3 compression ratio for Pongo_abelii.Susie_PABv2.cds.all.fa is 16.00%\n",
      "The GZIP compression ratio for Pongo_abelii.Susie_PABv2.cds.all.fa is 16.54%\n",
      "cds 17053 , total =  27378438\n",
      "The GECO3 compression ratio for Pteropus_vampyrus.pteVam1.cds.all.fa is 23.58%\n",
      "The GZIP compression ratio for Pteropus_vampyrus.pteVam1.cds.all.fa is 27.85%\n",
      "ncrna 4171 , total =  489549\n",
      "The GECO3 compression ratio for Pteropus_vampyrus.pteVam1.ncrna.fa is 14.33%\n",
      "The GZIP compression ratio for Pteropus_vampyrus.pteVam1.ncrna.fa is 19.64%\n",
      "pteropus_vampyrus completed successfully!\n",
      "cds 35191 , total =  68903790\n",
      "The GECO3 compression ratio for Pogona_vitticeps.pvi1.1.cds.all.fa is 14.13%\n",
      "The GZIP compression ratio for Pogona_vitticeps.pvi1.1.cds.all.fa is 14.17%\n",
      "ncrna 10782 , total =  2458743\n",
      "The GECO3 compression ratio for Pongo_abelii.Susie_PABv2.ncrna.fa is 15.61%\n",
      "The GZIP compression ratio for Pongo_abelii.Susie_PABv2.ncrna.fa is 21.20%\n",
      "pongo_abelii completed successfully!\n",
      "cds 32120 , total =  51865682\n",
      "The GECO3 compression ratio for Propithecus_coquereli.Pcoq_1.0.cds.all.fa is 16.30%\n",
      "The GZIP compression ratio for Propithecus_coquereli.Pcoq_1.0.cds.all.fa is 17.15%\n",
      "ncrna 7867 , total =  6722770\n",
      "The GECO3 compression ratio for Pogona_vitticeps.pvi1.1.ncrna.fa is 19.54%\n",
      "The GZIP compression ratio for Pogona_vitticeps.pvi1.1.ncrna.fa is 23.34%\n",
      "pogona_vitticeps completed successfully!\n",
      "ncrna 5294 , total =  657332\n",
      "The GECO3 compression ratio for Propithecus_coquereli.Pcoq_1.0.ncrna.fa is 14.21%\n",
      "The GZIP compression ratio for Propithecus_coquereli.Pcoq_1.0.ncrna.fa is 19.58%\n",
      "propithecus_coquereli completed successfully!\n",
      "cds 38056 , total =  60662101\n",
      "The GECO3 compression ratio for Prolemur_simus.Prosim_1.0.cds.all.fa is 15.66%\n",
      "The GZIP compression ratio for Prolemur_simus.Prosim_1.0.cds.all.fa is 16.00%\n",
      "cds 27955 , total =  51362991\n",
      "The GECO3 compression ratio for Pseudonaja_textilis.EBS10Xv2-PRI.cds.all.fa is 15.08%\n",
      "The GZIP compression ratio for Pseudonaja_textilis.EBS10Xv2-PRI.cds.all.fa is 16.33%\n",
      "ncrna 5290 , total =  661743\n",
      "The GECO3 compression ratio for Prolemur_simus.Prosim_1.0.ncrna.fa is 13.93%\n",
      "The GZIP compression ratio for Prolemur_simus.Prosim_1.0.ncrna.fa is 19.31%\n",
      "prolemur_simus completed successfully!\n",
      "ncrna 751 , total =  317786\n",
      "The GECO3 compression ratio for Pseudonaja_textilis.EBS10Xv2-PRI.ncrna.fa is 20.20%\n",
      "The GZIP compression ratio for Pseudonaja_textilis.EBS10Xv2-PRI.ncrna.fa is 24.22%\n",
      "pseudonaja_textilis completed successfully!\n",
      "cds 32153 , total =  49655097\n",
      "The GECO3 compression ratio for Pundamilia_nyererei.PunNye1.0.cds.all.fa is 19.22%\n",
      "The GZIP compression ratio for Pundamilia_nyererei.PunNye1.0.cds.all.fa is 21.09%\n",
      "ncrna 744 , total =  68248\n",
      "The GECO3 compression ratio for Pundamilia_nyererei.PunNye1.0.ncrna.fa is 20.46%\n",
      "The GZIP compression ratio for Pundamilia_nyererei.PunNye1.0.ncrna.fa is 24.92%\n",
      "pundamilia_nyererei completed successfully!\n",
      "cds 37995 , total =  64930610\n",
      "The GECO3 compression ratio for Pygocentrus_nattereri.Pygocentrus_nattereri-1.0.2.cds.all.fa is 17.95%\n",
      "The GZIP compression ratio for Pygocentrus_nattereri.Pygocentrus_nattereri-1.0.2.cds.all.fa is 19.07%\n",
      "ncrna 3319 , total =  435449\n",
      "The GECO3 compression ratio for Pygocentrus_nattereri.Pygocentrus_nattereri-1.0.2.ncrna.fa is 4.93%\n",
      "The GZIP compression ratio for Pygocentrus_nattereri.Pygocentrus_nattereri-1.0.2.ncrna.fa is 7.42%\n",
      "pygocentrus_nattereri completed successfully!\n",
      "cds 33717 , total =  59014606\n",
      "The GECO3 compression ratio for Rhinolophus_ferrumequinum.mRhiFer1_v1.p.cds.all.fa is 15.90%\n",
      "The GZIP compression ratio for Rhinolophus_ferrumequinum.mRhiFer1_v1.p.cds.all.fa is 16.72%\n",
      "ncrna 2180 , total =  698251\n",
      "The GECO3 compression ratio for Rhinolophus_ferrumequinum.mRhiFer1_v1.p.ncrna.fa is 17.99%\n",
      "The GZIP compression ratio for Rhinolophus_ferrumequinum.mRhiFer1_v1.p.ncrna.fa is 22.14%\n",
      "rhinolophus_ferrumequinum completed successfully!\n",
      "cds 6600 , total =  8762001\n",
      "The GECO3 compression ratio for Saccharomyces_cerevisiae.R64-1-1.cds.all.fa is 23.25%\n",
      "The GZIP compression ratio for Saccharomyces_cerevisiae.R64-1-1.cds.all.fa is 28.04%\n",
      "ncrna 424 , total =  84277\n",
      "The GECO3 compression ratio for Saccharomyces_cerevisiae.R64-1-1.ncrna.fa is 14.53%\n",
      "The GZIP compression ratio for Saccharomyces_cerevisiae.R64-1-1.ncrna.fa is 18.22%\n",
      "saccharomyces_cerevisiae completed successfully!\n",
      "cds 45938 , total =  78477732\n",
      "The GECO3 compression ratio for Rattus_norvegicus.mRatBN7.2.cds.all.fa is 14.98%\n",
      "The GZIP compression ratio for Rattus_norvegicus.mRatBN7.2.cds.all.fa is 14.56%\n",
      "cds 47739 , total =  82471908\n",
      "The GECO3 compression ratio for Rattus_norvegicus_shrspbbbutx.UTH_Rnor_SHRSP_BbbUtx_1.0.cds.all.fa is 14.56%\n",
      "The GZIP compression ratio for Rattus_norvegicus_shrspbbbutx.UTH_Rnor_SHRSP_BbbUtx_1.0.cds.all.fa is 14.32%\n",
      "cds 43656 , total =  68085772\n",
      "The GECO3 compression ratio for Rhinopithecus_bieti.ASM169854v1.cds.all.fa is 15.04%\n",
      "The GZIP compression ratio for Rhinopithecus_bieti.ASM169854v1.cds.all.fa is 14.76%\n",
      "ncrna 8137 , total =  8211374\n",
      "The GECO3 compression ratio for Rattus_norvegicus.mRatBN7.2.ncrna.fa is 16.61%\n",
      "The GZIP compression ratio for Rattus_norvegicus.mRatBN7.2.ncrna.fa is 20.72%\n",
      "rattus_norvegicus completed successfully!\n",
      "ncrna 9659 , total =  3972575\n",
      "The GECO3 compression ratio for Rhinopithecus_bieti.ASM169854v1.ncrna.fa is 17.43%\n",
      "The GZIP compression ratio for Rhinopithecus_bieti.ASM169854v1.ncrna.fa is 21.39%\n",
      "rhinopithecus_bieti completed successfully!\n",
      "cds 47306 , total =  82045278\n",
      "The GECO3 compression ratio for Rattus_norvegicus_shrutx.UTH_Rnor_SHR_Utx.cds.all.fa is 14.59%\n",
      "The GZIP compression ratio for Rattus_norvegicus_shrutx.UTH_Rnor_SHR_Utx.cds.all.fa is 14.36%\n",
      "ncrna 8671 , total =  7301024\n",
      "The GECO3 compression ratio for Rattus_norvegicus_shrspbbbutx.UTH_Rnor_SHRSP_BbbUtx_1.0.ncrna.fa is 15.67%\n",
      "The GZIP compression ratio for Rattus_norvegicus_shrspbbbutx.UTH_Rnor_SHRSP_BbbUtx_1.0.ncrna.fa is 19.38%\n",
      "rattus_norvegicus_shrspbbbutx completed successfully!\n",
      "cds 48005 , total =  82619171\n",
      "The GECO3 compression ratio for Rattus_norvegicus_wkybbb.UTH_Rnor_WKY_Bbb_1.0.cds.all.fa is 14.60%\n",
      "The GZIP compression ratio for Rattus_norvegicus_wkybbb.UTH_Rnor_WKY_Bbb_1.0.cds.all.fa is 14.31%\n",
      "ncrna 8583 , total =  7072157\n",
      "The GECO3 compression ratio for Rattus_norvegicus_shrutx.UTH_Rnor_SHR_Utx.ncrna.fa is 16.16%\n",
      "The GZIP compression ratio for Rattus_norvegicus_shrutx.UTH_Rnor_SHR_Utx.ncrna.fa is 19.90%\n",
      "rattus_norvegicus_shrutx completed successfully!\n",
      "ncrna 8682 , total =  7421004\n",
      "The GECO3 compression ratio for Rattus_norvegicus_wkybbb.UTH_Rnor_WKY_Bbb_1.0.ncrna.fa is 15.38%\n",
      "The GZIP compression ratio for Rattus_norvegicus_wkybbb.UTH_Rnor_WKY_Bbb_1.0.ncrna.fa is 19.07%\n",
      "rattus_norvegicus_wkybbb completed successfully!\n",
      "cds 45897 , total =  74109074\n",
      "The GECO3 compression ratio for Rhinopithecus_roxellana.Rrox_v1.cds.all.fa is 14.61%\n",
      "The GZIP compression ratio for Rhinopithecus_roxellana.Rrox_v1.cds.all.fa is 14.16%\n",
      "ncrna 7047 , total =  1207328\n",
      "The GECO3 compression ratio for Rhinopithecus_roxellana.Rrox_v1.ncrna.fa is 17.42%\n",
      "The GZIP compression ratio for Rhinopithecus_roxellana.Rrox_v1.ncrna.fa is 22.62%\n",
      "rhinopithecus_roxellana completed successfully!\n",
      "cds 40616 , total =  68193423\n",
      "The GECO3 compression ratio for Saimiri_boliviensis_boliviensis.SaiBol1.0.cds.all.fa is 14.79%\n",
      "The GZIP compression ratio for Saimiri_boliviensis_boliviensis.SaiBol1.0.cds.all.fa is 14.57%\n",
      "ncrna 7937 , total =  1886336\n",
      "The GECO3 compression ratio for Saimiri_boliviensis_boliviensis.SaiBol1.0.ncrna.fa is 14.66%\n",
      "The GZIP compression ratio for Saimiri_boliviensis_boliviensis.SaiBol1.0.ncrna.fa is 20.05%\n",
      "saimiri_boliviensis_boliviensis completed successfully!\n",
      "cds 56799 , total =  101374085\n",
      "The GECO3 compression ratio for Salarias_fasciatus.fSalaFa1.1.cds.all.fa is 13.56%\n",
      "The GZIP compression ratio for Salarias_fasciatus.fSalaFa1.1.cds.all.fa is 12.17%\n",
      "ncrna 1196 , total =  611464\n",
      "The GECO3 compression ratio for Salarias_fasciatus.fSalaFa1.1.ncrna.fa is 17.14%\n",
      "The GZIP compression ratio for Salarias_fasciatus.fSalaFa1.1.ncrna.fa is 21.29%\n",
      "salarias_fasciatus completed successfully!\n",
      "cds 30893 , total =  60280633\n",
      "The GECO3 compression ratio for Salvator_merianae.HLtupMer3.cds.all.fa is 15.35%\n",
      "The GZIP compression ratio for Salvator_merianae.HLtupMer3.cds.all.fa is 15.93%\n",
      "ncrna 5116 , total =  2902376\n",
      "The GECO3 compression ratio for Salvator_merianae.HLtupMer3.ncrna.fa is 18.58%\n",
      "The GZIP compression ratio for Salvator_merianae.HLtupMer3.ncrna.fa is 22.82%\n",
      "salvator_merianae completed successfully!\n",
      "cds 70155 , total =  108709352\n",
      "The GECO3 compression ratio for Salmo_salar_gca931346935v2.Ssal_ALTA.cds.all.fa is 16.57%\n",
      "The GZIP compression ratio for Salmo_salar_gca931346935v2.Ssal_ALTA.cds.all.fa is 17.12%\n",
      "ncrna 2496 , total =  356672\n",
      "The GECO3 compression ratio for Salmo_salar_gca931346935v2.Ssal_ALTA.ncrna.fa is 8.28%\n",
      "The GZIP compression ratio for Salmo_salar_gca931346935v2.Ssal_ALTA.ncrna.fa is 10.99%\n",
      "salmo_salar_gca931346935v2 completed successfully!\n",
      "cds 33234 , total =  52365848\n",
      "The GECO3 compression ratio for Sciurus_vulgaris.mSciVul1.1.cds.all.fa is 17.25%\n",
      "The GZIP compression ratio for Sciurus_vulgaris.mSciVul1.1.cds.all.fa is 19.17%\n",
      "ncrna 2415 , total =  307890\n",
      "The GECO3 compression ratio for Sciurus_vulgaris.mSciVul1.1.ncrna.fa is 16.92%\n",
      "The GZIP compression ratio for Sciurus_vulgaris.mSciVul1.1.ncrna.fa is 21.74%\n",
      "sciurus_vulgaris completed successfully!\n",
      "cds 37215 , total =  72200421\n",
      "The GECO3 compression ratio for Sarcophilus_harrisii.mSarHar1.11.cds.all.fa is 14.33%\n",
      "The GZIP compression ratio for Sarcophilus_harrisii.mSarHar1.11.cds.all.fa is 14.08%\n",
      "cds 61350 , total =  119499774\n",
      "The GECO3 compression ratio for Sander_lucioperca.SLUC_FBN_1.cds.all.fa is 12.78%\n",
      "The GZIP compression ratio for Sander_lucioperca.SLUC_FBN_1.cds.all.fa is 10.83%\n",
      "ncrna 1402 , total =  890153\n",
      "The GECO3 compression ratio for Sander_lucioperca.SLUC_FBN_1.ncrna.fa is 16.40%\n",
      "The GZIP compression ratio for Sander_lucioperca.SLUC_FBN_1.ncrna.fa is 19.93%\n",
      "sander_lucioperca completed successfully!\n",
      "cds 104286 , total =  173990592\n",
      "The GECO3 compression ratio for Salmo_salar_gca923944775v1.Ssal_Brian_v1.0.cds.all.fa is 15.42%\n",
      "The GZIP compression ratio for Salmo_salar_gca923944775v1.Ssal_Brian_v1.0.cds.all.fa is 13.49%\n",
      "ncrna 6041 , total =  12651460\n",
      "The GECO3 compression ratio for Sarcophilus_harrisii.mSarHar1.11.ncrna.fa is 17.63%\n",
      "The GZIP compression ratio for Sarcophilus_harrisii.mSarHar1.11.ncrna.fa is 22.10%\n",
      "sarcophilus_harrisii completed successfully!\n",
      "cds 24281 , total =  46655483\n",
      "The GECO3 compression ratio for Serinus_canaria.SCA1.cds.all.fa is 15.21%\n",
      "The GZIP compression ratio for Serinus_canaria.SCA1.cds.all.fa is 15.98%\n",
      "ncrna 2604 , total =  1972102\n",
      "The GECO3 compression ratio for Serinus_canaria.SCA1.ncrna.fa is 18.80%\n",
      "The GZIP compression ratio for Serinus_canaria.SCA1.ncrna.fa is 22.98%\n",
      "serinus_canaria completed successfully!\n",
      "ncrna 37249 , total =  55238107\n",
      "The GECO3 compression ratio for Salmo_salar_gca923944775v1.Ssal_Brian_v1.0.ncrna.fa is 11.99%\n",
      "The GZIP compression ratio for Salmo_salar_gca923944775v1.Ssal_Brian_v1.0.ncrna.fa is 15.48%\n",
      "salmo_salar_gca923944775v1 completed successfully!\n",
      "cds 33187 , total =  56584553\n",
      "The GECO3 compression ratio for Seriola_dumerili.Sdu_1.0.cds.all.fa is 18.65%\n",
      "The GZIP compression ratio for Seriola_dumerili.Sdu_1.0.cds.all.fa is 19.85%\n",
      "ncrna 526 , total =  60040\n",
      "The GECO3 compression ratio for Seriola_dumerili.Sdu_1.0.ncrna.fa is 18.24%\n",
      "The GZIP compression ratio for Seriola_dumerili.Sdu_1.0.ncrna.fa is 22.18%\n",
      "seriola_dumerili completed successfully!\n",
      "cds 116557 , total =  239846645\n",
      "The GECO3 compression ratio for Salmo_trutta.fSalTru1.1.cds.all.fa is 13.88%\n",
      "The GZIP compression ratio for Salmo_trutta.fSalTru1.1.cds.all.fa is 10.31%\n",
      "ncrna 5243 , total =  4453891\n",
      "The GECO3 compression ratio for Salmo_trutta.fSalTru1.1.ncrna.fa is 13.10%\n",
      "The GZIP compression ratio for Salmo_trutta.fSalTru1.1.ncrna.fa is 17.23%\n",
      "salmo_trutta completed successfully!\n",
      "cds 68558 , total =  143833664\n",
      "The GECO3 compression ratio for Scleropages_formosus.fSclFor1.1.cds.all.fa is 12.68%\n",
      "The GZIP compression ratio for Scleropages_formosus.fSclFor1.1.cds.all.fa is 9.93%\n",
      "ncrna 3904 , total =  4317355\n",
      "The GECO3 compression ratio for Scleropages_formosus.fSclFor1.1.ncrna.fa is 18.03%\n",
      "The GZIP compression ratio for Scleropages_formosus.fSclFor1.1.ncrna.fa is 22.22%\n",
      "scleropages_formosus completed successfully!\n",
      "cds 146172 , total =  294737280\n",
      "The GECO3 compression ratio for Salmo_salar.Ssal_v3.1.cds.all.fa is 13.84%\n",
      "The GZIP compression ratio for Salmo_salar.Ssal_v3.1.cds.all.fa is 9.21%\n",
      "cds 62641 , total =  131265321\n",
      "The GECO3 compression ratio for Scophthalmus_maximus.ASM1334776v1.cds.all.fa is 12.56%\n",
      "The GZIP compression ratio for Scophthalmus_maximus.ASM1334776v1.cds.all.fa is 9.79%\n",
      "ncrna 13487 , total =  4010266\n",
      "The GECO3 compression ratio for Scophthalmus_maximus.ASM1334776v1.ncrna.fa is 11.81%\n",
      "The GZIP compression ratio for Scophthalmus_maximus.ASM1334776v1.ncrna.fa is 14.69%\n",
      "scophthalmus_maximus completed successfully!\n",
      "cds 33582 , total =  54166112\n",
      "The GECO3 compression ratio for Seriola_lalandi_dorsalis.Sedor1.cds.all.fa is 18.81%\n",
      "The GZIP compression ratio for Seriola_lalandi_dorsalis.Sedor1.cds.all.fa is 20.46%\n",
      "ncrna 865 , total =  97717\n",
      "The GECO3 compression ratio for Seriola_lalandi_dorsalis.Sedor1.ncrna.fa is 14.21%\n",
      "The GZIP compression ratio for Seriola_lalandi_dorsalis.Sedor1.ncrna.fa is 17.41%\n",
      "seriola_lalandi_dorsalis completed successfully!\n",
      "cds 13192 , total =  17609732\n",
      "The GECO3 compression ratio for Sorex_araneus.COMMON_SHREW1.cds.all.fa is 23.39%\n",
      "The GZIP compression ratio for Sorex_araneus.COMMON_SHREW1.cds.all.fa is 27.93%\n",
      "cds 151347 , total =  308538825\n",
      "The GECO3 compression ratio for Salmo_salar_gca021399835v1.USDA_NASsal_1.1.cds.all.fa is 13.95%\n",
      "The GZIP compression ratio for Salmo_salar_gca021399835v1.USDA_NASsal_1.1.cds.all.fa is 8.98%\n",
      "ncrna 4558 , total =  632578\n",
      "The GECO3 compression ratio for Sorex_araneus.COMMON_SHREW1.ncrna.fa is 17.09%\n",
      "The GZIP compression ratio for Sorex_araneus.COMMON_SHREW1.ncrna.fa is 22.57%\n",
      "sorex_araneus completed successfully!\n",
      "ncrna 37429 , total =  52521510\n",
      "The GECO3 compression ratio for Salmo_salar.Ssal_v3.1.ncrna.fa is 10.25%\n",
      "The GZIP compression ratio for Salmo_salar.Ssal_v3.1.ncrna.fa is 14.22%\n",
      "salmo_salar completed successfully!\n",
      "cds 27607 , total =  40388247\n",
      "The GECO3 compression ratio for Spermophilus_dauricus.ASM240643v1.cds.all.fa is 18.42%\n",
      "The GZIP compression ratio for Spermophilus_dauricus.ASM240643v1.cds.all.fa is 20.77%\n",
      "ncrna 3058 , total =  401742\n",
      "The GECO3 compression ratio for Spermophilus_dauricus.ASM240643v1.ncrna.fa is 16.87%\n",
      "The GZIP compression ratio for Spermophilus_dauricus.ASM240643v1.ncrna.fa is 21.72%\n",
      "spermophilus_dauricus completed successfully!\n",
      "ncrna 50188 , total =  51471774\n",
      "The GECO3 compression ratio for Salmo_salar_gca021399835v1.USDA_NASsal_1.1.ncrna.fa is 11.26%\n",
      "The GZIP compression ratio for Salmo_salar_gca021399835v1.USDA_NASsal_1.1.ncrna.fa is 14.89%\n",
      "salmo_salar_gca021399835v1 completed successfully!\n",
      "cds 25114 , total =  37408931\n",
      "The GECO3 compression ratio for Sphenodon_punctatus.ASM311381v1.cds.all.fa is 17.76%\n",
      "The GZIP compression ratio for Sphenodon_punctatus.ASM311381v1.cds.all.fa is 19.78%\n",
      "ncrna 910 , total =  124162\n",
      "The GECO3 compression ratio for Sphenodon_punctatus.ASM311381v1.ncrna.fa is 19.44%\n",
      "The GZIP compression ratio for Sphenodon_punctatus.ASM311381v1.ncrna.fa is 24.09%\n",
      "sphenodon_punctatus completed successfully!\n",
      "cds 21620 , total =  43240034\n",
      "The GECO3 compression ratio for Stachyris_ruficeps.ASM869450v1.cds.all.fa is 16.50%\n",
      "The GZIP compression ratio for Stachyris_ruficeps.ASM869450v1.cds.all.fa is 17.87%\n",
      "ncrna 642 , total =  384832\n",
      "The GECO3 compression ratio for Stachyris_ruficeps.ASM869450v1.ncrna.fa is 20.53%\n",
      "The GZIP compression ratio for Stachyris_ruficeps.ASM869450v1.ncrna.fa is 24.87%\n",
      "stachyris_ruficeps completed successfully!\n",
      "cds 58677 , total =  110527454\n",
      "The GECO3 compression ratio for Sphaeramia_orbicularis.fSphaOr1.1.cds.all.fa is 12.87%\n",
      "The GZIP compression ratio for Sphaeramia_orbicularis.fSphaOr1.1.cds.all.fa is 11.17%\n",
      "cds 31466 , total =  50229730\n",
      "The GECO3 compression ratio for Stegastes_partitus.Stegastes_partitus-1.0.2.cds.all.fa is 19.19%\n",
      "The GZIP compression ratio for Stegastes_partitus.Stegastes_partitus-1.0.2.cds.all.fa is 20.94%\n",
      "ncrna 993 , total =  508584\n",
      "The GECO3 compression ratio for Sphaeramia_orbicularis.fSphaOr1.1.ncrna.fa is 17.46%\n",
      "The GZIP compression ratio for Sphaeramia_orbicularis.fSphaOr1.1.ncrna.fa is 21.29%\n",
      "sphaeramia_orbicularis completed successfully!\n",
      "ncrna 461 , total =  47636\n",
      "The GECO3 compression ratio for Stegastes_partitus.Stegastes_partitus-1.0.2.ncrna.fa is 18.77%\n",
      "The GZIP compression ratio for Stegastes_partitus.Stegastes_partitus-1.0.2.ncrna.fa is 23.18%\n",
      "stegastes_partitus completed successfully!\n",
      "cds 102785 , total =  190250695\n",
      "The GECO3 compression ratio for Sinocyclocheilus_anshuiensis.SAMN03320099.WGS_v1.1.cds.all.fa is 13.75%\n",
      "The GZIP compression ratio for Sinocyclocheilus_anshuiensis.SAMN03320099.WGS_v1.1.cds.all.fa is 11.61%\n",
      "cds 70047 , total =  144540325\n",
      "The GECO3 compression ratio for Sparus_aurata.fSpaAur1.1.cds.all.fa is 12.81%\n",
      "The GZIP compression ratio for Sparus_aurata.fSpaAur1.1.cds.all.fa is 9.95%\n",
      "ncrna 5875 , total =  3963691\n",
      "The GECO3 compression ratio for Sinocyclocheilus_anshuiensis.SAMN03320099.WGS_v1.1.ncrna.fa is 17.48%\n",
      "The GZIP compression ratio for Sinocyclocheilus_anshuiensis.SAMN03320099.WGS_v1.1.ncrna.fa is 21.52%\n",
      "sinocyclocheilus_anshuiensis completed successfully!\n",
      "cds 23602 , total =  36237036\n",
      "The GECO3 compression ratio for Strix_occidentalis_caurina.Soccid_v01.cds.all.fa is 17.74%\n",
      "The GZIP compression ratio for Strix_occidentalis_caurina.Soccid_v01.cds.all.fa is 19.91%\n",
      "cds 99938 , total =  179656902\n",
      "The GECO3 compression ratio for Sinocyclocheilus_rhinocerous.SAMN03320098_v1.1.cds.all.fa is 13.88%\n",
      "The GZIP compression ratio for Sinocyclocheilus_rhinocerous.SAMN03320098_v1.1.cds.all.fa is 12.14%\n",
      "cds 24520 , total =  50302316\n",
      "The GECO3 compression ratio for Strigops_habroptila.bStrHab1_v1.p.cds.all.fa is 14.89%\n",
      "The GZIP compression ratio for Strigops_habroptila.bStrHab1_v1.p.cds.all.fa is 15.98%\n",
      "ncrna 479 , total =  136769\n",
      "The GECO3 compression ratio for Strix_occidentalis_caurina.Soccid_v01.ncrna.fa is 22.73%\n",
      "The GZIP compression ratio for Strix_occidentalis_caurina.Soccid_v01.ncrna.fa is 27.47%\n",
      "strix_occidentalis_caurina completed successfully!\n",
      "ncrna 3051 , total =  4273853\n",
      "The GECO3 compression ratio for Sparus_aurata.fSpaAur1.1.ncrna.fa is 16.05%\n",
      "The GZIP compression ratio for Sparus_aurata.fSpaAur1.1.ncrna.fa is 20.25%\n",
      "sparus_aurata completed successfully!\n",
      "cds 105618 , total =  193553035\n",
      "The GECO3 compression ratio for Sinocyclocheilus_grahami.SAMN03320097.WGS_v1.1.cds.all.fa is 13.55%\n",
      "The GZIP compression ratio for Sinocyclocheilus_grahami.SAMN03320097.WGS_v1.1.cds.all.fa is 11.22%\n",
      "ncrna 2200 , total =  357751\n",
      "The GECO3 compression ratio for Sinocyclocheilus_rhinocerous.SAMN03320098_v1.1.ncrna.fa is 13.50%\n",
      "The GZIP compression ratio for Sinocyclocheilus_rhinocerous.SAMN03320098_v1.1.ncrna.fa is 17.06%\n",
      "sinocyclocheilus_rhinocerous completed successfully!\n",
      "ncrna 6320 , total =  4956252\n",
      "The GECO3 compression ratio for Sinocyclocheilus_grahami.SAMN03320097.WGS_v1.1.ncrna.fa is 17.16%\n",
      "The GZIP compression ratio for Sinocyclocheilus_grahami.SAMN03320097.WGS_v1.1.ncrna.fa is 21.21%\n",
      "sinocyclocheilus_grahami completed successfully!\n",
      "ncrna 4495 , total =  7783189\n",
      "The GECO3 compression ratio for Strigops_habroptila.bStrHab1_v1.p.ncrna.fa is 19.78%\n",
      "The GZIP compression ratio for Strigops_habroptila.bStrHab1_v1.p.ncrna.fa is 26.49%\n",
      "strigops_habroptila completed successfully!\n",
      "cds 26224 , total =  54966604\n",
      "The GECO3 compression ratio for Struthio_camelus_australis.ASM69896v1.cds.all.fa is 13.49%\n",
      "The GZIP compression ratio for Struthio_camelus_australis.ASM69896v1.cds.all.fa is 13.74%\n",
      "ncrna 8431 , total =  7541144\n",
      "The GECO3 compression ratio for Struthio_camelus_australis.ASM69896v1.ncrna.fa is 18.85%\n",
      "The GZIP compression ratio for Struthio_camelus_australis.ASM69896v1.ncrna.fa is 22.21%\n",
      "struthio_camelus_australis completed successfully!\n",
      "cds 37290 , total =  65284801\n",
      "The GECO3 compression ratio for Suricata_suricatta.meerkat_22Aug2017_6uvM2_HiC.cds.all.fa is 14.98%\n",
      "The GZIP compression ratio for Suricata_suricatta.meerkat_22Aug2017_6uvM2_HiC.cds.all.fa is 15.17%\n",
      "ncrna 4656 , total =  2402105\n",
      "The GECO3 compression ratio for Suricata_suricatta.meerkat_22Aug2017_6uvM2_HiC.ncrna.fa is 18.33%\n",
      "The GZIP compression ratio for Suricata_suricatta.meerkat_22Aug2017_6uvM2_HiC.ncrna.fa is 22.35%\n",
      "suricata_suricatta completed successfully!\n",
      "cds 45916 , total =  79664828\n",
      "The GECO3 compression ratio for Sus_scrofa.Sscrofa11.1.cds.all.fa is 13.97%\n",
      "The GZIP compression ratio for Sus_scrofa.Sscrofa11.1.cds.all.fa is 13.55%\n",
      "cds 48297 , total =  86694197\n",
      "The GECO3 compression ratio for Sus_scrofa_bamei.Bamei_pig_v1.cds.all.fa is 13.49%\n",
      "The GZIP compression ratio for Sus_scrofa_bamei.Bamei_pig_v1.cds.all.fa is 12.46%\n",
      "cds 48735 , total =  87333020\n",
      "The GECO3 compression ratio for Sus_scrofa_hampshire.Hampshire_pig_v1.cds.all.fa is 13.51%\n",
      "The GZIP compression ratio for Sus_scrofa_hampshire.Hampshire_pig_v1.cds.all.fa is 12.43%\n",
      "cds 49300 , total =  89672756\n",
      "The GECO3 compression ratio for Sus_scrofa_berkshire.Berkshire_pig_v1.cds.all.fa is 13.26%\n",
      "The GZIP compression ratio for Sus_scrofa_berkshire.Berkshire_pig_v1.cds.all.fa is 12.12%\n",
      "cds 48533 , total =  87143924\n",
      "The GECO3 compression ratio for Sus_scrofa_jinhua.Jinhua_pig_v1.cds.all.fa is 13.45%\n",
      "The GZIP compression ratio for Sus_scrofa_jinhua.Jinhua_pig_v1.cds.all.fa is 12.35%\n",
      "cds 48803 , total =  87246334\n",
      "The GECO3 compression ratio for Sus_scrofa_landrace.Landrace_pig_v1.cds.all.fa is 13.44%\n",
      "The GZIP compression ratio for Sus_scrofa_landrace.Landrace_pig_v1.cds.all.fa is 12.42%\n",
      "ncrna 6671 , total =  14681340\n",
      "The GECO3 compression ratio for Sus_scrofa_landrace.Landrace_pig_v1.ncrna.fa is 19.35%\n",
      "The GZIP compression ratio for Sus_scrofa_landrace.Landrace_pig_v1.ncrna.fa is 24.45%\n",
      "sus_scrofa_landrace completed successfully!\n",
      "ncrna 10782 , total =  27410656\n",
      "The GECO3 compression ratio for Sus_scrofa_bamei.Bamei_pig_v1.ncrna.fa is 20.08%\n",
      "The GZIP compression ratio for Sus_scrofa_bamei.Bamei_pig_v1.ncrna.fa is 24.82%\n",
      "sus_scrofa_bamei completed successfully!\n",
      "ncrna 10891 , total =  27340031\n",
      "The GECO3 compression ratio for Sus_scrofa_hampshire.Hampshire_pig_v1.ncrna.fa is 19.92%\n",
      "The GZIP compression ratio for Sus_scrofa_hampshire.Hampshire_pig_v1.ncrna.fa is 24.56%\n",
      "sus_scrofa_hampshire completed successfully!\n",
      "ncrna 10745 , total =  27034148\n",
      "The GECO3 compression ratio for Sus_scrofa_berkshire.Berkshire_pig_v1.ncrna.fa is 20.00%\n",
      "The GZIP compression ratio for Sus_scrofa_berkshire.Berkshire_pig_v1.ncrna.fa is 24.82%\n",
      "sus_scrofa_berkshire completed successfully!\n",
      "ncrna 10913 , total =  27393922\n",
      "The GECO3 compression ratio for Sus_scrofa_jinhua.Jinhua_pig_v1.ncrna.fa is 20.09%\n",
      "The GZIP compression ratio for Sus_scrofa_jinhua.Jinhua_pig_v1.ncrna.fa is 24.89%\n",
      "sus_scrofa_jinhua completed successfully!\n",
      "cds 48700 , total =  88105073\n",
      "The GECO3 compression ratio for Sus_scrofa_largewhite.Large_White_v1.cds.all.fa is 13.42%\n",
      "The GZIP compression ratio for Sus_scrofa_largewhite.Large_White_v1.cds.all.fa is 12.35%\n",
      "cds 48401 , total =  86485748\n",
      "The GECO3 compression ratio for Sus_scrofa_meishan.Meishan_pig_v1.cds.all.fa is 13.47%\n",
      "The GZIP compression ratio for Sus_scrofa_meishan.Meishan_pig_v1.cds.all.fa is 12.47%\n",
      "ncrna 13904 , total =  49897369\n",
      "The GECO3 compression ratio for Sus_scrofa.Sscrofa11.1.ncrna.fa is 21.23%\n",
      "The GZIP compression ratio for Sus_scrofa.Sscrofa11.1.ncrna.fa is 26.82%\n",
      "sus_scrofa completed successfully!\n",
      "ncrna 10747 , total =  26722043\n",
      "The GECO3 compression ratio for Sus_scrofa_largewhite.Large_White_v1.ncrna.fa is 20.01%\n",
      "The GZIP compression ratio for Sus_scrofa_largewhite.Large_White_v1.ncrna.fa is 24.80%\n",
      "sus_scrofa_largewhite completed successfully!\n",
      "ncrna 10490 , total =  26834122\n",
      "The GECO3 compression ratio for Sus_scrofa_meishan.Meishan_pig_v1.ncrna.fa is 20.15%\n",
      "The GZIP compression ratio for Sus_scrofa_meishan.Meishan_pig_v1.ncrna.fa is 25.01%\n",
      "sus_scrofa_meishan completed successfully!\n",
      "cds 49184 , total =  88605933\n",
      "The GECO3 compression ratio for Sus_scrofa_pietrain.Pietrain_pig_v1.cds.all.fa is 13.31%\n",
      "The GZIP compression ratio for Sus_scrofa_pietrain.Pietrain_pig_v1.cds.all.fa is 12.20%\n",
      "cds 32098 , total =  64932945\n",
      "The GECO3 compression ratio for Taeniopygia_guttata.bTaeGut1_v1.p.cds.all.fa is 14.21%\n",
      "The GZIP compression ratio for Taeniopygia_guttata.bTaeGut1_v1.p.cds.all.fa is 14.20%\n",
      "cds 47030 , total =  83561054\n",
      "The GECO3 compression ratio for Sus_scrofa_tibetan.Tibetan_Pig_v2.cds.all.fa is 13.16%\n",
      "The GZIP compression ratio for Sus_scrofa_tibetan.Tibetan_Pig_v2.cds.all.fa is 12.38%\n",
      "cds 48738 , total =  88259778\n",
      "The GECO3 compression ratio for Sus_scrofa_rongchang.Rongchang_pig_v1.cds.all.fa is 13.40%\n",
      "The GZIP compression ratio for Sus_scrofa_rongchang.Rongchang_pig_v1.cds.all.fa is 12.33%\n",
      "cds 46747 , total =  81054042\n",
      "The GECO3 compression ratio for Sus_scrofa_wuzhishan.minipig_v1.0.cds.all.fa is 12.90%\n",
      "The GZIP compression ratio for Sus_scrofa_wuzhishan.minipig_v1.0.cds.all.fa is 12.27%\n",
      "cds 49601 , total =  88462139\n",
      "The GECO3 compression ratio for Sus_scrofa_usmarc.USMARCv1.0.cds.all.fa is 13.15%\n",
      "The GZIP compression ratio for Sus_scrofa_usmarc.USMARCv1.0.cds.all.fa is 12.36%\n",
      "ncrna 8417 , total =  6488152\n",
      "The GECO3 compression ratio for Sus_scrofa_usmarc.USMARCv1.0.ncrna.fa is 17.47%\n",
      "The GZIP compression ratio for Sus_scrofa_usmarc.USMARCv1.0.ncrna.fa is 22.27%\n",
      "sus_scrofa_usmarc completed successfully!\n",
      "ncrna 6639 , total =  17254935\n",
      "The GECO3 compression ratio for Taeniopygia_guttata.bTaeGut1_v1.p.ncrna.fa is 18.61%\n",
      "The GZIP compression ratio for Taeniopygia_guttata.bTaeGut1_v1.p.ncrna.fa is 25.19%\n",
      "taeniopygia_guttata completed successfully!\n",
      "cds 32509 , total =  57831005\n",
      "The GECO3 compression ratio for Terrapene_carolina_triunguis.T_m_triunguis-2.0.cds.all.fa is 14.97%\n",
      "The GZIP compression ratio for Terrapene_carolina_triunguis.T_m_triunguis-2.0.cds.all.fa is 17.01%\n",
      "ncrna 1029 , total =  287854\n",
      "The GECO3 compression ratio for Terrapene_carolina_triunguis.T_m_triunguis-2.0.ncrna.fa is 17.48%\n",
      "The GZIP compression ratio for Terrapene_carolina_triunguis.T_m_triunguis-2.0.ncrna.fa is 21.43%\n",
      "terrapene_carolina_triunguis completed successfully!\n",
      "ncrna 10789 , total =  27005973\n",
      "The GECO3 compression ratio for Sus_scrofa_pietrain.Pietrain_pig_v1.ncrna.fa is 20.02%\n",
      "The GZIP compression ratio for Sus_scrofa_pietrain.Pietrain_pig_v1.ncrna.fa is 24.85%\n",
      "sus_scrofa_pietrain completed successfully!\n",
      "ncrna 10442 , total =  25801126\n",
      "The GECO3 compression ratio for Sus_scrofa_rongchang.Rongchang_pig_v1.ncrna.fa is 20.02%\n",
      "The GZIP compression ratio for Sus_scrofa_rongchang.Rongchang_pig_v1.ncrna.fa is 24.84%\n",
      "sus_scrofa_rongchang completed successfully!\n",
      "ncrna 10259 , total =  26843532\n",
      "The GECO3 compression ratio for Sus_scrofa_wuzhishan.minipig_v1.0.ncrna.fa is 19.53%\n",
      "The GZIP compression ratio for Sus_scrofa_wuzhishan.minipig_v1.0.ncrna.fa is 24.82%\n",
      "sus_scrofa_wuzhishan completed successfully!\n",
      "ncrna 13865 , total =  38548626\n",
      "The GECO3 compression ratio for Sus_scrofa_tibetan.Tibetan_Pig_v2.ncrna.fa is 20.35%\n",
      "The GZIP compression ratio for Sus_scrofa_tibetan.Tibetan_Pig_v2.ncrna.fa is 25.48%\n",
      "sus_scrofa_tibetan completed successfully!\n",
      "cds 15475 , total =  20473324\n",
      "The GECO3 compression ratio for Tupaia_belangeri.TREESHREW.cds.all.fa is 23.49%\n",
      "The GZIP compression ratio for Tupaia_belangeri.TREESHREW.cds.all.fa is 27.96%\n",
      "ncrna 3073 , total =  354700\n",
      "The GECO3 compression ratio for Tupaia_belangeri.TREESHREW.ncrna.fa is 16.10%\n",
      "The GZIP compression ratio for Tupaia_belangeri.TREESHREW.ncrna.fa is 20.97%\n",
      "tupaia_belangeri completed successfully!\n",
      "cds 52243 , total =  98892606\n",
      "The GECO3 compression ratio for Takifugu_rubripes.fTakRub1.2.cds.all.fa is 12.80%\n",
      "The GZIP compression ratio for Takifugu_rubripes.fTakRub1.2.cds.all.fa is 11.70%\n",
      "ncrna 2681 , total =  3017523\n",
      "The GECO3 compression ratio for Takifugu_rubripes.fTakRub1.2.ncrna.fa is 14.88%\n",
      "The GZIP compression ratio for Takifugu_rubripes.fTakRub1.2.ncrna.fa is 19.68%\n",
      "takifugu_rubripes completed successfully!\n",
      "cds 23118 , total =  37893630\n",
      "The GECO3 compression ratio for Tetraodon_nigroviridis.TETRAODON8.cds.all.fa is 19.98%\n",
      "The GZIP compression ratio for Tetraodon_nigroviridis.TETRAODON8.cds.all.fa is 22.53%\n",
      "ncrna 813 , total =  87955\n",
      "The GECO3 compression ratio for Tetraodon_nigroviridis.TETRAODON8.ncrna.fa is 15.61%\n",
      "The GZIP compression ratio for Tetraodon_nigroviridis.TETRAODON8.ncrna.fa is 18.99%\n",
      "tetraodon_nigroviridis completed successfully!\n",
      "cds 16611 , total =  27127090\n",
      "The GECO3 compression ratio for Tursiops_truncatus.turTru1.cds.all.fa is 23.53%\n",
      "The GZIP compression ratio for Tursiops_truncatus.turTru1.cds.all.fa is 27.82%\n",
      "ncrna 3790 , total =  415935\n",
      "The GECO3 compression ratio for Tursiops_truncatus.turTru1.ncrna.fa is 13.88%\n",
      "The GZIP compression ratio for Tursiops_truncatus.turTru1.ncrna.fa is 18.15%\n",
      "tursiops_truncatus completed successfully!\n",
      "cds 11806 , total =  16548597\n",
      "The GECO3 compression ratio for Vicugna_pacos.vicPac1.cds.all.fa is 23.71%\n",
      "The GZIP compression ratio for Vicugna_pacos.vicPac1.cds.all.fa is 27.90%\n",
      "ncrna 2532 , total =  296642\n",
      "The GECO3 compression ratio for Vicugna_pacos.vicPac1.ncrna.fa is 16.58%\n",
      "The GZIP compression ratio for Vicugna_pacos.vicPac1.ncrna.fa is 21.41%\n",
      "vicugna_pacos completed successfully!\n",
      "cds 36976 , total =  58198754\n",
      "The GECO3 compression ratio for Theropithecus_gelada.Tgel_1.0.cds.all.fa is 16.64%\n",
      "The GZIP compression ratio for Theropithecus_gelada.Tgel_1.0.cds.all.fa is 17.56%\n",
      "ncrna 6181 , total =  1060890\n",
      "The GECO3 compression ratio for Theropithecus_gelada.Tgel_1.0.ncrna.fa is 16.98%\n",
      "The GZIP compression ratio for Theropithecus_gelada.Tgel_1.0.ncrna.fa is 22.56%\n",
      "theropithecus_gelada completed successfully!\n",
      "cds 30023 , total =  52180063\n",
      "The GECO3 compression ratio for Urocitellus_parryii.ASM342692v1.cds.all.fa is 16.84%\n",
      "The GZIP compression ratio for Urocitellus_parryii.ASM342692v1.cds.all.fa is 17.85%\n",
      "ncrna 3717 , total =  1257741\n",
      "The GECO3 compression ratio for Urocitellus_parryii.ASM342692v1.ncrna.fa is 19.66%\n",
      "The GZIP compression ratio for Urocitellus_parryii.ASM342692v1.ncrna.fa is 23.78%\n",
      "urocitellus_parryii completed successfully!\n",
      "cds 33197 , total =  56938591\n",
      "The GECO3 compression ratio for Ursus_americanus.ASM334442v1.cds.all.fa is 14.54%\n",
      "The GZIP compression ratio for Ursus_americanus.ASM334442v1.cds.all.fa is 15.30%\n",
      "ncrna 3899 , total =  1264381\n",
      "The GECO3 compression ratio for Ursus_americanus.ASM334442v1.ncrna.fa is 19.98%\n",
      "The GZIP compression ratio for Ursus_americanus.ASM334442v1.ncrna.fa is 24.47%\n",
      "ursus_americanus completed successfully!\n",
      "cds 34052 , total =  62994956\n",
      "The GECO3 compression ratio for Ursus_maritimus.UrsMar_1.0.cds.all.fa is 13.85%\n",
      "The GZIP compression ratio for Ursus_maritimus.UrsMar_1.0.cds.all.fa is 14.06%\n",
      "cds 29181 , total =  56753973\n",
      "The GECO3 compression ratio for Varanus_komodoensis.ASM479886v1.cds.all.fa is 14.80%\n",
      "The GZIP compression ratio for Varanus_komodoensis.ASM479886v1.cds.all.fa is 15.58%\n",
      "ncrna 6215 , total =  3627355\n",
      "The GECO3 compression ratio for Ursus_maritimus.UrsMar_1.0.ncrna.fa is 20.00%\n",
      "The GZIP compression ratio for Ursus_maritimus.UrsMar_1.0.ncrna.fa is 24.01%\n",
      "ursus_maritimus completed successfully!\n",
      "ncrna 610 , total =  265083\n",
      "The GECO3 compression ratio for Varanus_komodoensis.ASM479886v1.ncrna.fa is 22.18%\n",
      "The GZIP compression ratio for Varanus_komodoensis.ASM479886v1.ncrna.fa is 26.47%\n",
      "varanus_komodoensis completed successfully!\n",
      "cds 36405 , total =  66018090\n",
      "The GECO3 compression ratio for Ursus_thibetanus_thibetanus.ASM966005v1.cds.all.fa is 15.20%\n",
      "The GZIP compression ratio for Ursus_thibetanus_thibetanus.ASM966005v1.cds.all.fa is 15.80%\n",
      "ncrna 2809 , total =  1257848\n",
      "The GECO3 compression ratio for Ursus_thibetanus_thibetanus.ASM966005v1.ncrna.fa is 19.24%\n",
      "The GZIP compression ratio for Ursus_thibetanus_thibetanus.ASM966005v1.ncrna.fa is 23.61%\n",
      "ursus_thibetanus_thibetanus completed successfully!\n",
      "cds 32889 , total =  59218507\n",
      "The GECO3 compression ratio for Vombatus_ursinus.bare-nosed_wombat_genome_assembly.cds.all.fa is 15.46%\n",
      "The GZIP compression ratio for Vombatus_ursinus.bare-nosed_wombat_genome_assembly.cds.all.fa is 16.92%\n",
      "ncrna 3987 , total =  2006158\n",
      "The GECO3 compression ratio for Vombatus_ursinus.bare-nosed_wombat_genome_assembly.ncrna.fa is 18.12%\n",
      "The GZIP compression ratio for Vombatus_ursinus.bare-nosed_wombat_genome_assembly.ncrna.fa is 22.37%\n",
      "vombatus_ursinus completed successfully!\n",
      "cds 28029 , total =  36916708\n",
      "The GECO3 compression ratio for Xiphophorus_couchianus.Xiphophorus_couchianus-4.0.1.cds.all.fa is 19.27%\n",
      "The GZIP compression ratio for Xiphophorus_couchianus.Xiphophorus_couchianus-4.0.1.cds.all.fa is 21.65%\n",
      "ncrna 303 , total =  35562\n",
      "The GECO3 compression ratio for Xiphophorus_couchianus.Xiphophorus_couchianus-4.0.1.ncrna.fa is 19.77%\n",
      "The GZIP compression ratio for Xiphophorus_couchianus.Xiphophorus_couchianus-4.0.1.ncrna.fa is 24.26%\n",
      "xiphophorus_couchianus completed successfully!\n",
      "cds 41026 , total =  82461290\n",
      "The GECO3 compression ratio for Vulpes_vulpes.VulVul2.2.cds.all.fa is 13.21%\n",
      "The GZIP compression ratio for Vulpes_vulpes.VulVul2.2.cds.all.fa is 12.50%\n",
      "cds 21605 , total =  39753807\n",
      "The GECO3 compression ratio for Zosterops_lateralis_melanops.ASM128173v1.cds.all.fa is 15.80%\n",
      "The GZIP compression ratio for Zosterops_lateralis_melanops.ASM128173v1.cds.all.fa is 17.03%\n",
      "ncrna 462 , total =  154745\n",
      "The GECO3 compression ratio for Zosterops_lateralis_melanops.ASM128173v1.ncrna.fa is 21.38%\n",
      "The GZIP compression ratio for Zosterops_lateralis_melanops.ASM128173v1.ncrna.fa is 25.67%\n",
      "zosterops_lateralis_melanops completed successfully!\n",
      "cds 36613 , total =  65634600\n",
      "The GECO3 compression ratio for Xiphophorus_maculatus.X_maculatus-5.0-male.cds.all.fa is 16.97%\n",
      "The GZIP compression ratio for Xiphophorus_maculatus.X_maculatus-5.0-male.cds.all.fa is 17.82%\n",
      "cds 23611 , total =  49424454\n",
      "The GECO3 compression ratio for Zonotrichia_albicollis.Zonotrichia_albicollis-1.0.1.cds.all.fa is 14.20%\n",
      "The GZIP compression ratio for Zonotrichia_albicollis.Zonotrichia_albicollis-1.0.1.cds.all.fa is 14.50%\n",
      "ncrna 397 , total =  49661\n",
      "The GECO3 compression ratio for Xiphophorus_maculatus.X_maculatus-5.0-male.ncrna.fa is 17.37%\n",
      "The GZIP compression ratio for Xiphophorus_maculatus.X_maculatus-5.0-male.ncrna.fa is 21.54%\n",
      "xiphophorus_maculatus completed successfully!\n",
      "ncrna 12733 , total =  9731907\n",
      "The GECO3 compression ratio for Vulpes_vulpes.VulVul2.2.ncrna.fa is 18.06%\n",
      "The GZIP compression ratio for Vulpes_vulpes.VulVul2.2.ncrna.fa is 21.44%\n",
      "vulpes_vulpes completed successfully!\n",
      "ncrna 6963 , total =  6015774\n",
      "The GECO3 compression ratio for Zonotrichia_albicollis.Zonotrichia_albicollis-1.0.1.ncrna.fa is 16.93%\n",
      "The GZIP compression ratio for Zonotrichia_albicollis.Zonotrichia_albicollis-1.0.1.ncrna.fa is 20.19%\n",
      "zonotrichia_albicollis completed successfully!\n",
      "cds 49792 , total =  94278950\n",
      "The GECO3 compression ratio for Xenopus_tropicalis.UCB_Xtro_10.0.cds.all.fa is 13.14%\n",
      "The GZIP compression ratio for Xenopus_tropicalis.UCB_Xtro_10.0.cds.all.fa is 12.74%\n",
      "ncrna 2023 , total =  346076\n",
      "The GECO3 compression ratio for Xenopus_tropicalis.UCB_Xtro_10.0.ncrna.fa is 6.30%\n",
      "The GZIP compression ratio for Xenopus_tropicalis.UCB_Xtro_10.0.ncrna.fa is 8.25%\n",
      "xenopus_tropicalis completed successfully!\n",
      "cds 40461 , total =  72603133\n",
      "The GECO3 compression ratio for Zalophus_californianus.mZalCal1.pri.cds.all.fa is 14.43%\n",
      "The GZIP compression ratio for Zalophus_californianus.mZalCal1.pri.cds.all.fa is 14.29%\n",
      "ncrna 4067 , total =  1849904\n",
      "The GECO3 compression ratio for Zalophus_californianus.mZalCal1.pri.ncrna.fa is 18.25%\n",
      "The GZIP compression ratio for Zalophus_californianus.mZalCal1.pri.ncrna.fa is 22.44%\n",
      "zalophus_californianus completed successfully!\n",
      "Results saved to species_compression_ratios_vert_looptest_pt2.csv\n"
     ]
    }
   ],
   "source": [
    "# Usage Vertebrates\n",
    "\n",
    "ftp_server = \"ftp.ensembl.org\" \n",
    "base_ftp_directory = \"/pub/current_fasta\"\n",
    "output_csv_file = \"compression_ratios_vert.csv\"\n",
    "\n",
    "process_all_organisms(ftp_server, base_ftp_directory, output_csv_file)\n",
    "\n",
    "print(f\"Results saved to {output_csv_file}\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "cds 27313 , total =  25807032\n",
      "The GECO3 compression ratio for Amborella_trichopoda.AMTR1.0.cds.all.fa is 23.36%\n",
      "The GZIP compression ratio for Amborella_trichopoda.AMTR1.0.cds.all.fa is 27.81%\n",
      "ncrna 1040 , total =  149397\n",
      "The GECO3 compression ratio for Amborella_trichopoda.AMTR1.0.ncrna.fa is 16.20%\n",
      "The GZIP compression ratio for Amborella_trichopoda.AMTR1.0.ncrna.fa is 21.41%\n",
      "amborella_trichopoda completed successfully!\n",
      "cds 25783 , total =  29489560\n",
      "The GECO3 compression ratio for Ananas_comosus.F153.cds.all.fa is 22.27%\n",
      "The GZIP compression ratio for Ananas_comosus.F153.cds.all.fa is 27.94%\n",
      "Directory or file not found for ananas_comosus in ncrna folder: 550 Failed to change directory.\n",
      "ananas_comosus completed successfully!\n",
      "cds 32667 , total =  35412914\n",
      "The GECO3 compression ratio for Arabidopsis_lyrata.v.1.0.cds.all.fa is 23.01%\n",
      "The GZIP compression ratio for Arabidopsis_lyrata.v.1.0.cds.all.fa is 27.73%\n",
      "ncrna 795 , total =  113407\n",
      "The GECO3 compression ratio for Arabidopsis_lyrata.v.1.0.ncrna.fa is 9.82%\n",
      "The GZIP compression ratio for Arabidopsis_lyrata.v.1.0.ncrna.fa is 12.54%\n",
      "arabidopsis_lyrata completed successfully!\n",
      "cds 34087 , total =  41323905\n",
      "The GECO3 compression ratio for Arabidopsis_halleri.Ahal2.2.cds.all.fa is 21.90%\n",
      "The GZIP compression ratio for Arabidopsis_halleri.Ahal2.2.cds.all.fa is 26.46%\n",
      "cds 33115 , total =  42333820\n",
      "The GECO3 compression ratio for Actinidia_chinensis.Red5_PS1_1.69.0.cds.all.fa is 22.38%\n",
      "The GZIP compression ratio for Actinidia_chinensis.Red5_PS1_1.69.0.cds.all.fa is 27.99%\n",
      "Directory or file not found for actinidia_chinensis in ncrna folder: 550 Failed to change directory.\n",
      "actinidia_chinensis completed successfully!\n",
      "ncrna 670 , total =  75506\n",
      "The GECO3 compression ratio for Arabidopsis_halleri.Ahal2.2.ncrna.fa is 13.20%\n",
      "The GZIP compression ratio for Arabidopsis_halleri.Ahal2.2.ncrna.fa is 16.69%\n",
      "arabidopsis_halleri completed successfully!\n",
      "cds 24141 , total =  24277419\n",
      "The GECO3 compression ratio for Asparagus_officinalis.Aspof.V1.cds.all.fa is 23.06%\n",
      "The GZIP compression ratio for Asparagus_officinalis.Aspof.V1.cds.all.fa is 28.04%\n",
      "cds 23286 , total =  27396077\n",
      "The GECO3 compression ratio for Arabis_alpina.A_alpina_V4.cds.all.fa is 21.42%\n",
      "The GZIP compression ratio for Arabis_alpina.A_alpina_V4.cds.all.fa is 25.87%\n",
      "ncrna 535 , total =  69987\n",
      "The GECO3 compression ratio for Asparagus_officinalis.Aspof.V1.ncrna.fa is 11.71%\n",
      "The GZIP compression ratio for Asparagus_officinalis.Aspof.V1.ncrna.fa is 15.64%\n",
      "asparagus_officinalis completed successfully!\n",
      "ncrna 2400 , total =  256760\n",
      "The GECO3 compression ratio for Arabis_alpina.A_alpina_V4.ncrna.fa is 10.69%\n",
      "The GZIP compression ratio for Arabis_alpina.A_alpina_V4.ncrna.fa is 14.78%\n",
      "arabis_alpina completed successfully!\n",
      "cds 48321 , total =  62644390\n",
      "The GECO3 compression ratio for Arabidopsis_thaliana.TAIR10.cds.all.fa is 15.72%\n",
      "The GZIP compression ratio for Arabidopsis_thaliana.TAIR10.cds.all.fa is 16.39%\n",
      "ncrna 5654 , total =  2748728\n",
      "The GECO3 compression ratio for Arabidopsis_thaliana.TAIR10.ncrna.fa is 17.13%\n",
      "The GZIP compression ratio for Arabidopsis_thaliana.TAIR10.ncrna.fa is 21.06%\n",
      "arabidopsis_thaliana completed successfully!\n",
      "cds 28721 , total =  31014209\n",
      "The GECO3 compression ratio for Beta_vulgaris.RefBeet-1.2.2.cds.all.fa is 22.00%\n",
      "The GZIP compression ratio for Beta_vulgaris.RefBeet-1.2.2.cds.all.fa is 25.75%\n",
      "ncrna 1443 , total =  175251\n",
      "The GECO3 compression ratio for Beta_vulgaris.RefBeet-1.2.2.ncrna.fa is 13.15%\n",
      "The GZIP compression ratio for Beta_vulgaris.RefBeet-1.2.2.ncrna.fa is 17.29%\n",
      "beta_vulgaris completed successfully!\n",
      "cds 52972 , total =  66747974\n",
      "The GECO3 compression ratio for Brachypodium_distachyon.Brachypodium_distachyon_v3.0.cds.all.fa is 15.74%\n",
      "The GZIP compression ratio for Brachypodium_distachyon.Brachypodium_distachyon_v3.0.cds.all.fa is 17.42%\n",
      "ncrna 815 , total =  145592\n",
      "The GECO3 compression ratio for Brachypodium_distachyon.Brachypodium_distachyon_v3.0.ncrna.fa is 8.43%\n",
      "The GZIP compression ratio for Brachypodium_distachyon.Brachypodium_distachyon_v3.0.ncrna.fa is 11.19%\n",
      "brachypodium_distachyon completed successfully!\n",
      "cds 75016 , total =  78130154\n",
      "The GECO3 compression ratio for Brassica_juncea.ASM1870372v1.cds.all.fa is 21.79%\n",
      "The GZIP compression ratio for Brassica_juncea.ASM1870372v1.cds.all.fa is 27.99%\n",
      "Directory or file not found for brassica_juncea in ncrna folder: 550 Failed to change directory.\n",
      "brassica_juncea completed successfully!\n",
      "cds 59220 , total =  61709195\n",
      "The GECO3 compression ratio for Brassica_oleracea.BOL.cds.all.fa is 22.03%\n",
      "The GZIP compression ratio for Brassica_oleracea.BOL.cds.all.fa is 27.96%\n",
      "ncrna 1361 , total =  178032\n",
      "The GECO3 compression ratio for Brassica_oleracea.BOL.ncrna.fa is 11.29%\n",
      "The GZIP compression ratio for Brassica_oleracea.BOL.ncrna.fa is 14.71%\n",
      "brassica_oleracea completed successfully!\n",
      "cds 153277 , total =  189725405\n",
      "The GECO3 compression ratio for Aegilops_umbellulata.TA1851.cds.all.fa is 10.59%\n",
      "The GZIP compression ratio for Aegilops_umbellulata.TA1851.cds.all.fa is 7.83%\n",
      "ncrna 224 , total =  283449\n",
      "The GECO3 compression ratio for Aegilops_umbellulata.TA1851.ncrna.fa is 14.44%\n",
      "The GZIP compression ratio for Aegilops_umbellulata.TA1851.ncrna.fa is 17.82%\n",
      "aegilops_umbellulata completed successfully!\n",
      "cds 101040 , total =  101137907\n",
      "The GECO3 compression ratio for Brassica_napus.AST_PRJEB5043_v1.cds.all.fa is 20.81%\n",
      "The GZIP compression ratio for Brassica_napus.AST_PRJEB5043_v1.cds.all.fa is 27.80%\n",
      "cds 94968 , total =  123087434\n",
      "The GECO3 compression ratio for Avena_sativa_sang.Asativa_sang.v1.1.cds.all.fa is 17.72%\n",
      "The GZIP compression ratio for Avena_sativa_sang.Asativa_sang.v1.1.cds.all.fa is 24.10%\n",
      "Directory or file not found for avena_sativa_sang in ncrna folder: 550 Failed to change directory.\n",
      "avena_sativa_sang completed successfully!\n",
      "ncrna 3533 , total =  606756\n",
      "The GECO3 compression ratio for Brassica_napus.AST_PRJEB5043_v1.ncrna.fa is 8.09%\n",
      "The GZIP compression ratio for Brassica_napus.AST_PRJEB5043_v1.ncrna.fa is 12.57%\n",
      "brassica_napus completed successfully!\n",
      "cds 41025 , total =  48107696\n",
      "The GECO3 compression ratio for Brassica_rapa.Brapa_1.0.cds.all.fa is 23.17%\n",
      "The GZIP compression ratio for Brassica_rapa.Brapa_1.0.cds.all.fa is 28.04%\n",
      "ncrna 1168 , total =  152331\n",
      "The GECO3 compression ratio for Brassica_rapa.Brapa_1.0.ncrna.fa is 10.96%\n",
      "The GZIP compression ratio for Brassica_rapa.Brapa_1.0.ncrna.fa is 13.81%\n",
      "brassica_rapa completed successfully!\n",
      "cds 48618 , total =  46568533\n",
      "The GECO3 compression ratio for Cajanus_cajan.C.cajan_V1.0.cds.all.fa is 20.87%\n",
      "The GZIP compression ratio for Cajanus_cajan.C.cajan_V1.0.cds.all.fa is 27.98%\n",
      "Directory or file not found for cajanus_cajan in ncrna folder: 550 Failed to change directory.\n",
      "cajanus_cajan completed successfully!\n",
      "cds 43129 , total =  52969993\n",
      "The GECO3 compression ratio for Brassica_rapa_ro18.SCU_BraROA_2.3.cds.all.fa is 21.44%\n",
      "The GZIP compression ratio for Brassica_rapa_ro18.SCU_BraROA_2.3.cds.all.fa is 27.69%\n",
      "Directory or file not found for brassica_rapa_ro18 in ncrna folder: 550 Failed to change directory.\n",
      "brassica_rapa_ro18 completed successfully!\n",
      "cds 36241 , total =  39955678\n",
      "The GECO3 compression ratio for Cannabis_sativa_female.cs10.cds.all.fa is 20.12%\n",
      "The GZIP compression ratio for Cannabis_sativa_female.cs10.cds.all.fa is 23.86%\n",
      "Directory or file not found for cannabis_sativa_female in ncrna folder: 550 Failed to change directory.\n",
      "cannabis_sativa_female completed successfully!\n",
      "cds 35845 , total =  39100104\n",
      "The GECO3 compression ratio for Capsicum_annuum.ASM51225v2.cds.all.fa is 21.83%\n",
      "The GZIP compression ratio for Capsicum_annuum.ASM51225v2.cds.all.fa is 27.70%\n",
      "Directory or file not found for capsicum_annuum in ncrna folder: 550 Failed to change directory.\n",
      "capsicum_annuum completed successfully!\n",
      "cds 258680 , total =  264015221\n",
      "The GECO3 compression ratio for Aegilops_tauschii.Aet_v4.0.cds.all.fa is 10.50%\n",
      "The GZIP compression ratio for Aegilops_tauschii.Aet_v4.0.cds.all.fa is 7.13%\n",
      "ncrna 3732 , total =  513802\n",
      "The GECO3 compression ratio for Aegilops_tauschii.Aet_v4.0.ncrna.fa is 7.02%\n",
      "The GZIP compression ratio for Aegilops_tauschii.Aet_v4.0.ncrna.fa is 10.85%\n",
      "aegilops_tauschii completed successfully!\n",
      "cds 9807 , total =  10272816\n",
      "The GECO3 compression ratio for Chondrus_crispus.ASM35022v2.cds.all.fa is 22.06%\n",
      "The GZIP compression ratio for Chondrus_crispus.ASM35022v2.cds.all.fa is 27.75%\n",
      "ncrna 34 , total =  7260\n",
      "The GECO3 compression ratio for Chondrus_crispus.ASM35022v2.ncrna.fa is 25.96%\n",
      "The GZIP compression ratio for Chondrus_crispus.ASM35022v2.ncrna.fa is 31.96%\n",
      "chondrus_crispus completed successfully!\n",
      "cds 22541 , total =  26302942\n",
      "The GECO3 compression ratio for Citrullus_lanatus.Cla97_v1.cds.all.fa is 23.86%\n",
      "The GZIP compression ratio for Citrullus_lanatus.Cla97_v1.cds.all.fa is 28.07%\n",
      "Directory or file not found for citrullus_lanatus in ncrna folder: 550 Failed to change directory.\n",
      "citrullus_lanatus completed successfully!\n",
      "cds 34426 , total =  43820271\n",
      "The GECO3 compression ratio for Chenopodium_quinoa.ASM168347v1.cds.all.fa is 20.82%\n",
      "The GZIP compression ratio for Chenopodium_quinoa.ASM168347v1.cds.all.fa is 27.97%\n",
      "Directory or file not found for chenopodium_quinoa in ncrna folder: 550 Failed to change directory.\n",
      "chenopodium_quinoa completed successfully!\n",
      "cds 19528 , total =  43184075\n",
      "The GECO3 compression ratio for Chlamydomonas_reinhardtii.Chlamydomonas_reinhardtii_v5.5.cds.all.fa is 20.15%\n",
      "The GZIP compression ratio for Chlamydomonas_reinhardtii.Chlamydomonas_reinhardtii_v5.5.cds.all.fa is 23.93%\n",
      "ncrna 200 , total =  28307\n",
      "The GECO3 compression ratio for Chlamydomonas_reinhardtii.Chlamydomonas_reinhardtii_v5.5.ncrna.fa is 10.83%\n",
      "The GZIP compression ratio for Chlamydomonas_reinhardtii.Chlamydomonas_reinhardtii_v5.5.ncrna.fa is 13.57%\n",
      "chlamydomonas_reinhardtii completed successfully!\n",
      "cds 35664 , total =  70515850\n",
      "The GECO3 compression ratio for Chara_braunii.Cbr_1.0.cds.all.fa is 17.02%\n",
      "The GZIP compression ratio for Chara_braunii.Cbr_1.0.cds.all.fa is 26.57%\n",
      "Directory or file not found for chara_braunii in ncrna folder: 550 Failed to change directory.\n",
      "chara_braunii completed successfully!\n",
      "cds 25574 , total =  30830778\n",
      "The GECO3 compression ratio for Coffea_canephora.AUK_PRJEB4211_v1.cds.all.fa is 22.44%\n",
      "The GZIP compression ratio for Coffea_canephora.AUK_PRJEB4211_v1.cds.all.fa is 27.84%\n",
      "Directory or file not found for coffea_canephora in ncrna folder: 550 Failed to change directory.\n",
      "coffea_canephora completed successfully!\n",
      "cds 34557 , total =  42065794\n",
      "The GECO3 compression ratio for Citrus_clementina.Citrus_clementina_v1.0.cds.all.fa is 18.31%\n",
      "The GZIP compression ratio for Citrus_clementina.Citrus_clementina_v1.0.cds.all.fa is 21.24%\n",
      "ncrna 2179 , total =  280037\n",
      "The GECO3 compression ratio for Citrus_clementina.Citrus_clementina_v1.0.ncrna.fa is 9.17%\n",
      "The GZIP compression ratio for Citrus_clementina.Citrus_clementina_v1.0.ncrna.fa is 13.04%\n",
      "citrus_clementina completed successfully!\n",
      "cds 4973 , total =  7504073\n",
      "The GECO3 compression ratio for Cyanidioschyzon_merolae.ASM9120v1.cds.all.fa is 23.65%\n",
      "The GZIP compression ratio for Cyanidioschyzon_merolae.ASM9120v1.cds.all.fa is 28.07%\n",
      "Directory or file not found for cyanidioschyzon_merolae in ncrna folder: 550 Failed to change directory.\n",
      "cyanidioschyzon_merolae completed successfully!\n",
      "cds 29356 , total =  30371299\n",
      "The GECO3 compression ratio for Corchorus_capsularis.CCACVL1_1.0.cds.all.fa is 23.49%\n",
      "The GZIP compression ratio for Corchorus_capsularis.CCACVL1_1.0.cds.all.fa is 27.78%\n",
      "cds 28299 , total =  27850974\n",
      "The GECO3 compression ratio for Cucumis_melo.Melonv4.cds.all.fa is 23.33%\n",
      "The GZIP compression ratio for Cucumis_melo.Melonv4.cds.all.fa is 28.06%\n",
      "Directory or file not found for cucumis_melo in ncrna folder: 550 Failed to change directory.\n",
      "cucumis_melo completed successfully!\n",
      "ncrna 2014 , total =  182659\n",
      "The GECO3 compression ratio for Corchorus_capsularis.CCACVL1_1.0.ncrna.fa is 11.86%\n",
      "The GZIP compression ratio for Corchorus_capsularis.CCACVL1_1.0.ncrna.fa is 15.08%\n",
      "corchorus_capsularis completed successfully!\n",
      "cds 94352 , total =  115365352\n",
      "The GECO3 compression ratio for Camelina_sativa.Cs.cds.all.fa is 18.24%\n",
      "The GZIP compression ratio for Camelina_sativa.Cs.cds.all.fa is 25.62%\n",
      "Directory or file not found for camelina_sativa in ncrna folder: 550 Failed to change directory.\n",
      "camelina_sativa completed successfully!\n",
      "cds 25732 , total =  31078634\n",
      "The GECO3 compression ratio for Corylus_avellana.CavTom2PMs-1.0.cds.all.fa is 21.65%\n",
      "The GZIP compression ratio for Corylus_avellana.CavTom2PMs-1.0.cds.all.fa is 27.97%\n",
      "Directory or file not found for corylus_avellana in ncrna folder: 550 Failed to change directory.\n",
      "corylus_avellana completed successfully!\n",
      "cds 23780 , total =  26295569\n",
      "The GECO3 compression ratio for Cucumis_sativus.ASM407v2.cds.all.fa is 23.76%\n",
      "The GZIP compression ratio for Cucumis_sativus.ASM407v2.cds.all.fa is 28.00%\n",
      "ncrna 682 , total =  80652\n",
      "The GECO3 compression ratio for Cucumis_sativus.ASM407v2.ncrna.fa is 15.10%\n",
      "The GZIP compression ratio for Cucumis_sativus.ASM407v2.ncrna.fa is 19.12%\n",
      "cucumis_sativus completed successfully!\n",
      "cds 45647 , total =  59347281\n",
      "The GECO3 compression ratio for Corymbia_citriodora.Ccitriodora_v2_1.cds.all.fa is 17.86%\n",
      "The GZIP compression ratio for Corymbia_citriodora.Ccitriodora_v2_1.cds.all.fa is 21.76%\n",
      "Directory or file not found for corymbia_citriodora in ncrna folder: 550 Failed to change directory.\n",
      "corymbia_citriodora completed successfully!\n",
      "cds 26505 , total =  32231521\n",
      "The GECO3 compression ratio for Cynara_cardunculus.CcrdV1.cds.all.fa is 23.39%\n",
      "The GZIP compression ratio for Cynara_cardunculus.CcrdV1.cds.all.fa is 27.69%\n",
      "Directory or file not found for cynara_cardunculus in ncrna folder: 550 Failed to change directory.\n",
      "cynara_cardunculus completed successfully!\n",
      "cds 32109 , total =  38138949\n",
      "The GECO3 compression ratio for Daucus_carota.ASM162521v1.cds.all.fa is 23.19%\n",
      "The GZIP compression ratio for Daucus_carota.ASM162521v1.cds.all.fa is 27.88%\n",
      "ncrna 2154 , total =  253038\n",
      "The GECO3 compression ratio for Daucus_carota.ASM162521v1.ncrna.fa is 10.21%\n",
      "The GZIP compression ratio for Daucus_carota.ASM162521v1.ncrna.fa is 14.49%\n",
      "daucus_carota completed successfully!\n",
      "cds 64831 , total =  57149048\n",
      "The GECO3 compression ratio for Dioscorea_rotundata.TDr96_F1_v2_PseudoChromosome.cds.all.fa is 14.45%\n",
      "The GZIP compression ratio for Dioscorea_rotundata.TDr96_F1_v2_PseudoChromosome.cds.all.fa is 16.39%\n",
      "Directory or file not found for dioscorea_rotundata in ncrna folder: 550 Failed to change directory.\n",
      "dioscorea_rotundata completed successfully!\n",
      "cds 257856 , total =  359184088\n",
      "The GECO3 compression ratio for Avena_sativa_ot3098.Oat_OT3098_v2.cds.all.fa is 10.77%\n",
      "The GZIP compression ratio for Avena_sativa_ot3098.Oat_OT3098_v2.cds.all.fa is 9.14%\n",
      "cds 59984 , total =  63229623\n",
      "The GECO3 compression ratio for Digitaria_exilis.Fonio_CM05836.cds.all.fa is 19.98%\n",
      "The GZIP compression ratio for Digitaria_exilis.Fonio_CM05836.cds.all.fa is 27.91%\n",
      "Directory or file not found for digitaria_exilis in ncrna folder: 550 Failed to change directory.\n",
      "digitaria_exilis completed successfully!\n",
      "cds 55182 , total =  62924550\n",
      "The GECO3 compression ratio for Eragrostis_curvula.CERZOS_E.curvula1.0.cds.all.fa is 20.44%\n",
      "The GZIP compression ratio for Eragrostis_curvula.CERZOS_E.curvula1.0.cds.all.fa is 27.75%\n",
      "ncrna 822 , total =  61452\n",
      "The GECO3 compression ratio for Eragrostis_curvula.CERZOS_E.curvula1.0.ncrna.fa is 7.36%\n",
      "The GZIP compression ratio for Eragrostis_curvula.CERZOS_E.curvula1.0.ncrna.fa is 8.40%\n",
      "eragrostis_curvula completed successfully!\n",
      "ncrna 6584 , total =  18585900\n",
      "The GECO3 compression ratio for Avena_sativa_ot3098.Oat_OT3098_v2.ncrna.fa is 12.27%\n",
      "The GZIP compression ratio for Avena_sativa_ot3098.Oat_OT3098_v2.ncrna.fa is 16.47%\n",
      "avena_sativa_ot3098 completed successfully!\n",
      "cds 29485 , total =  36076596\n",
      "The GECO3 compression ratio for Eutrema_salsugineum.Eutsalg1_0.cds.all.fa is 21.65%\n",
      "The GZIP compression ratio for Eutrema_salsugineum.Eutsalg1_0.cds.all.fa is 25.42%\n",
      "Directory or file not found for eutrema_salsugineum in ncrna folder: 550 Failed to change directory.\n",
      "eutrema_salsugineum completed successfully!\n",
      "cds 46920 , total =  53835271\n",
      "The GECO3 compression ratio for Eucalyptus_grandis.Egrandis1_0.cds.all.fa is 18.18%\n",
      "The GZIP compression ratio for Eucalyptus_grandis.Egrandis1_0.cds.all.fa is 22.41%\n",
      "Directory or file not found for eucalyptus_grandis in ncrna folder: 550 Failed to change directory.\n",
      "eucalyptus_grandis completed successfully!\n",
      "cds 7174 , total =  9040469\n",
      "The GECO3 compression ratio for Galdieria_sulphuraria.ASM34128v1.cds.all.fa is 21.36%\n",
      "The GZIP compression ratio for Galdieria_sulphuraria.ASM34128v1.cds.all.fa is 25.94%\n",
      "ncrna 101 , total =  37379\n",
      "The GECO3 compression ratio for Galdieria_sulphuraria.ASM34128v1.ncrna.fa is 10.73%\n",
      "The GZIP compression ratio for Galdieria_sulphuraria.ASM34128v1.ncrna.fa is 14.03%\n",
      "galdieria_sulphuraria completed successfully!\n",
      "cds 37684 , total =  35972786\n",
      "The GECO3 compression ratio for Ficus_carica.UNIPI_FiCari_1.0.cds.all.fa is 21.88%\n",
      "The GZIP compression ratio for Ficus_carica.UNIPI_FiCari_1.0.cds.all.fa is 28.08%\n",
      "Directory or file not found for ficus_carica in ncrna folder: 550 Failed to change directory.\n",
      "ficus_carica completed successfully!\n",
      "cds 65879 , total =  81371768\n",
      "The GECO3 compression ratio for Eragrostis_tef.Salk_teff_dabbi_3.0.cds.all.fa is 21.26%\n",
      "The GZIP compression ratio for Eragrostis_tef.Salk_teff_dabbi_3.0.cds.all.fa is 27.84%\n",
      "Directory or file not found for eragrostis_tef in ncrna folder: 550 Failed to change directory.\n",
      "eragrostis_tef completed successfully!\n",
      "cds 50706 , total =  60670808\n",
      "The GECO3 compression ratio for Fraxinus_excelsior.BATG-0.5.cds.all.fa is 19.17%\n",
      "The GZIP compression ratio for Fraxinus_excelsior.BATG-0.5.cds.all.fa is 21.68%\n",
      "Directory or file not found for fraxinus_excelsior in ncrna folder: 550 Failed to change directory.\n",
      "fraxinus_excelsior completed successfully!\n",
      "cds 37961 , total =  45076731\n",
      "The GECO3 compression ratio for Hordeum_vulgare.MorexV3_pseudomolecules_assembly.cds.all.fa is 20.25%\n",
      "The GZIP compression ratio for Hordeum_vulgare.MorexV3_pseudomolecules_assembly.cds.all.fa is 26.38%\n",
      "Directory or file not found for hordeum_vulgare in ncrna folder: 550 Failed to change directory.cds 103783 , total =  131936856\n",
      "The GECO3 compression ratio for Echinochloa_crusgalli.ec_v3.cds.all.fa is 19.97%\n",
      "The GZIP compression ratio for Echinochloa_crusgalli.ec_v3.cds.all.fa is 27.87%\n",
      "\n",
      "hordeum_vulgare completed successfully!\n",
      "Directory or file not found for echinochloa_crusgalli in ncrna folder: 550 Failed to change directory.\n",
      "echinochloa_crusgalli completed successfully!\n",
      "cds 70864 , total =  65219177\n",
      "The GECO3 compression ratio for Helianthus_annuus.HanXRQr2.0-SUNRISE.cds.all.fa is 21.34%\n",
      "The GZIP compression ratio for Helianthus_annuus.HanXRQr2.0-SUNRISE.cds.all.fa is 28.09%\n",
      "ncrna 12313 , total =  3662611\n",
      "The GECO3 compression ratio for Helianthus_annuus.HanXRQr2.0-SUNRISE.ncrna.fa is 17.78%\n",
      "The GZIP compression ratio for Helianthus_annuus.HanXRQr2.0-SUNRISE.ncrna.fa is 24.80%\n",
      "helianthus_annuus completed successfully!\n",
      "cds 58815 , total =  50589715\n",
      "The GECO3 compression ratio for Hordeum_vulgare_goldenpromise.GPv1.cds.all.fa is 21.37%\n",
      "The GZIP compression ratio for Hordeum_vulgare_goldenpromise.GPv1.cds.all.fa is 28.03%\n",
      "Directory or file not found for hordeum_vulgare_goldenpromise in ncrna folder: 550 Failed to change directory.\n",
      "hordeum_vulgare_goldenpromise completed successfully!\n",
      "cds 69277 , total =  96804094\n",
      "The GECO3 compression ratio for Glycine_soja.ASM419377v2.cds.all.fa is 16.57%\n",
      "The GZIP compression ratio for Glycine_soja.ASM419377v2.cds.all.fa is 18.82%\n",
      "cds 78371 , total =  98939488\n",
      "The GECO3 compression ratio for Gossypium_raimondii.Graimondii2_0_v6.cds.all.fa is 14.57%\n",
      "The GZIP compression ratio for Gossypium_raimondii.Graimondii2_0_v6.cds.all.fa is 14.18%\n",
      "Directory or file not found for gossypium_raimondii in ncrna folder: 550 Failed to change directory.\n",
      "gossypium_raimondii completed successfully!\n",
      "cds 32159 , total =  39472406\n",
      "The GECO3 compression ratio for Hordeum_vulgare_tritex.Morex_V2_scaf.cds.all.fa is 22.58%\n",
      "The GZIP compression ratio for Hordeum_vulgare_tritex.Morex_V2_scaf.cds.all.fa is 28.00%\n",
      "Directory or file not found for hordeum_vulgare_tritex in ncrna folder: 550 Failed to change directory.\n",
      "hordeum_vulgare_tritex completed successfully!\n",
      "ncrna 8927 , total =  9728744\n",
      "The GECO3 compression ratio for Glycine_soja.ASM419377v2.ncrna.fa is 11.17%\n",
      "The GZIP compression ratio for Glycine_soja.ASM419377v2.ncrna.fa is 14.23%\n",
      "glycine_soja completed successfully!\n",
      "cds 88412 , total =  112766699\n",
      "The GECO3 compression ratio for Glycine_max.Glycine_max_v2.1.cds.all.fa is 15.90%\n",
      "The GZIP compression ratio for Glycine_max.Glycine_max_v2.1.cds.all.fa is 17.31%\n",
      "ncrna 1250 , total =  373003\n",
      "The GECO3 compression ratio for Glycine_max.Glycine_max_v2.1.ncrna.fa is 4.26%\n",
      "The GZIP compression ratio for Glycine_max.Glycine_max_v2.1.ncrna.fa is 5.96%\n",
      "glycine_max completed successfully!\n",
      "cds 41077 , total =  54840282\n",
      "The GECO3 compression ratio for Juglans_regia.Walnut_2.0.cds.all.fa is 19.07%\n",
      "The GZIP compression ratio for Juglans_regia.Walnut_2.0.cds.all.fa is 28.08%\n",
      "Directory or file not found for juglans_regia in ncrna folder: 550 Failed to change directory.\n",
      "juglans_regia completed successfully!\n",
      "cds 47019 , total =  62228119\n",
      "The GECO3 compression ratio for Ipomoea_triloba.ASM357664v1.cds.all.fa is 16.99%\n",
      "The GZIP compression ratio for Ipomoea_triloba.ASM357664v1.cds.all.fa is 18.80%\n",
      "Directory or file not found for ipomoea_triloba in ncrna folder: 550 Failed to change directory.\n",
      "ipomoea_triloba completed successfully!\n",
      "cds 38294 , total =  40669811\n",
      "The GECO3 compression ratio for Lactuca_sativa.Lsat_Salinas_v7.cds.all.fa is 22.67%\n",
      "The GZIP compression ratio for Lactuca_sativa.Lsat_Salinas_v7.cds.all.fa is 27.97%\n",
      "Directory or file not found for lactuca_sativa in ncrna folder: 550 Failed to change directory.\n",
      "lactuca_sativa completed successfully!\n",
      "cds 45187 , total =  55524067\n",
      "The GECO3 compression ratio for Kalanchoe_fedtschenkoi.K_fedtschenkoi_M2_v1.cds.all.fa is 16.61%\n",
      "The GZIP compression ratio for Kalanchoe_fedtschenkoi.K_fedtschenkoi_M2_v1.cds.all.fa is 18.85%\n",
      "Directory or file not found for kalanchoe_fedtschenkoi in ncrna folder: 550 Failed to change directory.\n",
      "kalanchoe_fedtschenkoi completed successfully!\n",
      "cds 38960 , total =  52134341\n",
      "The GECO3 compression ratio for Leersia_perrieri.Lperr_V1.4.cds.all.fa is 18.32%\n",
      "The GZIP compression ratio for Leersia_perrieri.Lperr_V1.4.cds.all.fa is 20.43%\n",
      "ncrna 867 , total =  97807\n",
      "The GECO3 compression ratio for Leersia_perrieri.Lperr_V1.4.ncrna.fa is 16.76%\n",
      "The GZIP compression ratio for Leersia_perrieri.Lperr_V1.4.ncrna.fa is 20.89%\n",
      "leersia_perrieri completed successfully!\n",
      "cds 38867 , total =  47191806\n",
      "The GECO3 compression ratio for Lolium_perenne.MPB_Lper_Kyuss_1697.cds.all.fa is 22.43%\n",
      "The GZIP compression ratio for Lolium_perenne.MPB_Lper_Kyuss_1697.cds.all.fa is 27.93%\n",
      "cds 33083 , total =  42631338\n",
      "The GECO3 compression ratio for Lupinus_angustifolius.LupAngTanjil_v1.0.cds.all.fa is 22.62%\n",
      "The GZIP compression ratio for Lupinus_angustifolius.LupAngTanjil_v1.0.cds.all.fa is 27.58%\n",
      "Directory or file not found for lolium_perenne in ncrna folder: 550 Failed to change directory.\n",
      "lolium_perenne completed successfully!\n",
      "ncrna 2096 , total =  311553\n",
      "The GECO3 compression ratio for Lupinus_angustifolius.LupAngTanjil_v1.0.ncrna.fa is 8.70%\n",
      "The GZIP compression ratio for Lupinus_angustifolius.LupAngTanjil_v1.0.ncrna.fa is 13.37%\n",
      "lupinus_angustifolius completed successfully!\n",
      "cds 24674 , total =  32889438\n",
      "The GECO3 compression ratio for Marchantia_polymorpha.Marchanta_polymorpha_v1.cds.all.fa is 17.64%\n",
      "The GZIP compression ratio for Marchantia_polymorpha.Marchanta_polymorpha_v1.cds.all.fa is 20.63%\n",
      "Directory or file not found for marchantia_polymorpha in ncrna folder: 550 Failed to change directory.\n",
      "marchantia_polymorpha completed successfully!\n",
      "cds 40624 , total =  47559880\n",
      "The GECO3 compression ratio for Malus_domestica_golden.ASM211411v1.cds.all.fa is 22.35%\n",
      "The GZIP compression ratio for Malus_domestica_golden.ASM211411v1.cds.all.fa is 28.07%\n",
      "ncrna 1872 , total =  1049284\n",
      "The GECO3 compression ratio for Malus_domestica_golden.ASM211411v1.ncrna.fa is 21.53%\n",
      "The GZIP compression ratio for Malus_domestica_golden.ASM211411v1.ncrna.fa is 26.99%\n",
      "malus_domestica_golden completed successfully!\n",
      "cds 44450 , total =  45297900\n",
      "The GECO3 compression ratio for Medicago_truncatula.MtrunA17r5.0_ANR.cds.all.fa is 22.54%\n",
      "The GZIP compression ratio for Medicago_truncatula.MtrunA17r5.0_ANR.cds.all.fa is 27.87%\n",
      "cds 28438 , total =  33145053\n",
      "The GECO3 compression ratio for Nymphaea_colorata.ASM883128v1.cds.all.fa is 21.66%\n",
      "The GZIP compression ratio for Nymphaea_colorata.ASM883128v1.cds.all.fa is 27.79%\n",
      "ncrna 6097 , total =  4527254\n",
      "The GECO3 compression ratio for Medicago_truncatula.MtrunA17r5.0_ANR.ncrna.fa is 19.12%\n",
      "The GZIP compression ratio for Medicago_truncatula.MtrunA17r5.0_ANR.ncrna.fa is 23.71%\n",
      "medicago_truncatula completed successfully!\n",
      "Directory or file not found for nymphaea_colorata in ncrna folder: 550 Failed to change directory.\n",
      "nymphaea_colorata completed successfully!\n",
      "cds 33320 , total =  37136987\n",
      "The GECO3 compression ratio for Nicotiana_attenuata.NIATTr2.cds.all.fa is 22.96%\n",
      "The GZIP compression ratio for Nicotiana_attenuata.NIATTr2.cds.all.fa is 27.86%\n",
      "ncrna 3084 , total =  696332\n",
      "The GECO3 compression ratio for Nicotiana_attenuata.NIATTr2.ncrna.fa is 6.32%\n",
      "The GZIP compression ratio for Nicotiana_attenuata.NIATTr2.ncrna.fa is 10.15%\n",
      "nicotiana_attenuata completed successfully!\n",
      "cds 45589 , total =  57909714\n",
      "The GECO3 compression ratio for Musa_acuminata.Musa_acuminata_v2.cds.all.fa is 18.52%\n",
      "The GZIP compression ratio for Musa_acuminata.Musa_acuminata_v2.cds.all.fa is 20.86%\n",
      "Directory or file not found for musa_acuminata in ncrna folder: 550 Failed to change directory.\n",
      "musa_acuminata completed successfully!\n",
      "cds 59149 , total =  84551829\n",
      "The GECO3 compression ratio for Manihot_esculenta.M.esculenta_v8.cds.all.fa is 14.22%\n",
      "The GZIP compression ratio for Manihot_esculenta.M.esculenta_v8.cds.all.fa is 14.89%\n",
      "Directory or file not found for manihot_esculenta in ncrna folder: 550 Failed to change directory.\n",
      "manihot_esculenta completed successfully!\n",
      "cds 50678 , total =  52752683\n",
      "The GECO3 compression ratio for Olea_europaea_sylvestris.O_europaea_v1.cds.all.fa is 21.27%\n",
      "The GZIP compression ratio for Olea_europaea_sylvestris.O_europaea_v1.cds.all.fa is 27.84%\n",
      "Directory or file not found for olea_europaea_sylvestris in ncrna folder: 550 Failed to change directory.\n",
      "olea_europaea_sylvestris completed successfully!\n",
      "cds 41595 , total =  50565380\n",
      "The GECO3 compression ratio for Oryza_barthii.O.barthii_v1.cds.all.fa is 19.34%\n",
      "The GZIP compression ratio for Oryza_barthii.O.barthii_v1.cds.all.fa is 21.91%\n",
      "ncrna 916 , total =  100953\n",
      "The GECO3 compression ratio for Oryza_barthii.O.barthii_v1.ncrna.fa is 16.65%\n",
      "The GZIP compression ratio for Oryza_barthii.O.barthii_v1.ncrna.fa is 20.39%\n",
      "oryza_barthii completed successfully!\n",
      "cds 32037 , total =  33411477\n",
      "The GECO3 compression ratio for Oryza_brachyantha.Oryza_brachyantha.v1.4b.cds.all.fa is 23.08%\n",
      "The GZIP compression ratio for Oryza_brachyantha.Oryza_brachyantha.v1.4b.cds.all.fa is 27.60%\n",
      "ncrna 425 , total =  48318\n",
      "The GECO3 compression ratio for Oryza_brachyantha.Oryza_brachyantha.v1.4b.ncrna.fa is 17.14%\n",
      "The GZIP compression ratio for Oryza_brachyantha.Oryza_brachyantha.v1.4b.ncrna.fa is 21.15%\n",
      "oryza_brachyantha completed successfully!\n",
      "cds 33164 , total =  36156010\n",
      "The GECO3 compression ratio for Oryza_glaberrima.Oryza_glaberrima_V1.cds.all.fa is 22.25%\n",
      "The GZIP compression ratio for Oryza_glaberrima.Oryza_glaberrima_V1.cds.all.fa is 27.54%\n",
      "ncrna 966 , total =  110221\n",
      "The GECO3 compression ratio for Oryza_glaberrima.Oryza_glaberrima_V1.ncrna.fa is 13.01%\n",
      "The GZIP compression ratio for Oryza_glaberrima.Oryza_glaberrima_V1.ncrna.fa is 16.22%\n",
      "oryza_glaberrima completed successfully!\n",
      "cds 31686 , total =  34731219\n",
      "The GECO3 compression ratio for Oryza_longistaminata.O_longistaminata_v1.0.cds.all.fa is 22.71%\n",
      "The GZIP compression ratio for Oryza_longistaminata.O_longistaminata_v1.0.cds.all.fa is 27.60%\n",
      "ncrna 1121 , total =  122530\n",
      "The GECO3 compression ratio for Oryza_longistaminata.O_longistaminata_v1.0.ncrna.fa is 15.50%\n",
      "The GZIP compression ratio for Oryza_longistaminata.O_longistaminata_v1.0.ncrna.fa is 19.57%\n",
      "oryza_longistaminata completed successfully!\n",
      "cds 40745 , total =  45442528\n",
      "The GECO3 compression ratio for Oryza_indica.ASM465v1.cds.all.fa is 22.60%\n",
      "The GZIP compression ratio for Oryza_indica.ASM465v1.cds.all.fa is 27.85%\n",
      "ncrna 1286 , total =  186444\n",
      "The GECO3 compression ratio for Oryza_indica.ASM465v1.ncrna.fa is 9.68%\n",
      "The GZIP compression ratio for Oryza_indica.ASM465v1.ncrna.fa is 12.69%\n",
      "oryza_indica completed successfully!\n",
      "cds 46893 , total =  58080667\n",
      "The GECO3 compression ratio for Oryza_glumipatula.Oryza_glumaepatula_v1.5.cds.all.fa is 18.29%\n",
      "The GZIP compression ratio for Oryza_glumipatula.Oryza_glumaepatula_v1.5.cds.all.fa is 20.41%\n",
      "ncrna 644 , total =  68091\n",
      "The GECO3 compression ratio for Oryza_glumipatula.Oryza_glumaepatula_v1.5.ncrna.fa is 15.09%\n",
      "The GZIP compression ratio for Oryza_glumipatula.Oryza_glumaepatula_v1.5.ncrna.fa is 18.76%\n",
      "oryza_glumipatula completed successfully!\n",
      "cds 89176 , total =  106234408\n",
      "The GECO3 compression ratio for Olea_europaea.OLEA9.cds.all.fa is 15.91%\n",
      "The GZIP compression ratio for Olea_europaea.OLEA9.cds.all.fa is 16.67%\n",
      "Directory or file not found for olea_europaea in ncrna folder: 550 Failed to change directory.\n",
      "olea_europaea completed successfully!\n",
      "cds 43455 , total =  55211386\n",
      "The GECO3 compression ratio for Oryza_meridionalis.Oryza_meridionalis_v1.3.cds.all.fa is 16.68%\n",
      "The GZIP compression ratio for Oryza_meridionalis.Oryza_meridionalis_v1.3.cds.all.fa is 18.42%\n",
      "ncrna 933 , total =  104807\n",
      "The GECO3 compression ratio for Oryza_meridionalis.Oryza_meridionalis_v1.3.ncrna.fa is 16.47%\n",
      "The GZIP compression ratio for Oryza_meridionalis.Oryza_meridionalis_v1.3.ncrna.fa is 20.54%\n",
      "oryza_meridionalis completed successfully!\n",
      "cds 48360 , total =  61005857\n",
      "The GECO3 compression ratio for Oryza_nivara.Oryza_nivara_v1.0.cds.all.fa is 18.05%\n",
      "The GZIP compression ratio for Oryza_nivara.Oryza_nivara_v1.0.cds.all.fa is 20.07%\n",
      "ncrna 713 , total =  80695\n",
      "The GECO3 compression ratio for Oryza_nivara.Oryza_nivara_v1.0.ncrna.fa is 15.38%\n",
      "The GZIP compression ratio for Oryza_nivara.Oryza_nivara_v1.0.ncrna.fa is 18.82%\n",
      "oryza_nivara completed successfully!\n",
      "cds 41060 , total =  53482507\n",
      "The GECO3 compression ratio for Oryza_punctata.Oryza_punctata_v1.2.cds.all.fa is 18.38%\n",
      "The GZIP compression ratio for Oryza_punctata.Oryza_punctata_v1.2.cds.all.fa is 20.62%\n",
      "ncrna 788 , total =  95980\n",
      "The GECO3 compression ratio for Oryza_punctata.Oryza_punctata_v1.2.ncrna.fa is 14.07%\n",
      "The GZIP compression ratio for Oryza_punctata.Oryza_punctata_v1.2.ncrna.fa is 17.83%\n",
      "oryza_punctata completed successfully!\n",
      "cds 42582 , total =  42715178\n",
      "The GECO3 compression ratio for Oryza_sativa.IRGSP-1.0.cds.all.fa is 20.22%\n",
      "The GZIP compression ratio for Oryza_sativa.IRGSP-1.0.cds.all.fa is 24.29%\n",
      "ncrna 3184 , total =  1894181\n",
      "The GECO3 compression ratio for Oryza_sativa.IRGSP-1.0.ncrna.fa is 22.32%\n",
      "The GZIP compression ratio for Oryza_sativa.IRGSP-1.0.ncrna.fa is 26.98%\n",
      "oryza_sativa completed successfully!\n",
      "cds 47441 , total =  58900820\n",
      "The GECO3 compression ratio for Oryza_rufipogon.OR_W1943.cds.all.fa is 18.54%\n",
      "The GZIP compression ratio for Oryza_rufipogon.OR_W1943.cds.all.fa is 20.74%\n",
      "ncrna 740 , total =  81221\n",
      "The GECO3 compression ratio for Oryza_rufipogon.OR_W1943.ncrna.fa is 14.61%\n",
      "The GZIP compression ratio for Oryza_rufipogon.OR_W1943.ncrna.fa is 18.20%\n",
      "oryza_rufipogon completed successfully!\n",
      "cds 43438 , total =  51634518\n",
      "The GECO3 compression ratio for Oryza_sativa_arc.Os117425RS1.cds.all.fa is 19.89%\n",
      "The GZIP compression ratio for Oryza_sativa_arc.Os117425RS1.cds.all.fa is 23.39%\n",
      "ncrna 303 , total =  240246\n",
      "The GECO3 compression ratio for Oryza_sativa_arc.Os117425RS1.ncrna.fa is 21.84%\n",
      "The GZIP compression ratio for Oryza_sativa_arc.Os117425RS1.ncrna.fa is 26.65%\n",
      "oryza_sativa_arc completed successfully!\n",
      "cds 44363 , total =  53168424\n",
      "The GECO3 compression ratio for Oryza_sativa_azucena.AzucenaRS1.cds.all.fa is 19.62%\n",
      "The GZIP compression ratio for Oryza_sativa_azucena.AzucenaRS1.cds.all.fa is 23.00%\n",
      "ncrna 307 , total =  238245\n",
      "The GECO3 compression ratio for Oryza_sativa_azucena.AzucenaRS1.ncrna.fa is 20.81%\n",
      "The GZIP compression ratio for Oryza_sativa_azucena.AzucenaRS1.ncrna.fa is 25.54%\n",
      "oryza_sativa_azucena completed successfully!\n",
      "cds 45043 , total =  53801121\n",
      "The GECO3 compression ratio for Oryza_sativa_chaomeo.Os132278RS1.cds.all.fa is 19.58%\n",
      "The GZIP compression ratio for Oryza_sativa_chaomeo.Os132278RS1.cds.all.fa is 22.91%\n",
      "ncrna 283 , total =  209137\n",
      "The GECO3 compression ratio for Oryza_sativa_chaomeo.Os132278RS1.ncrna.fa is 21.60%\n",
      "The GZIP compression ratio for Oryza_sativa_chaomeo.Os132278RS1.ncrna.fa is 26.41%\n",
      "oryza_sativa_chaomeo completed successfully!\n",
      "cds 42379 , total =  49957797\n",
      "The GECO3 compression ratio for Oryza_sativa_gobolsailbalam.Os132424RS1.cds.all.fa is 20.29%\n",
      "The GZIP compression ratio for Oryza_sativa_gobolsailbalam.Os132424RS1.cds.all.fa is 24.04%\n",
      "ncrna 248 , total =  178867\n",
      "The GECO3 compression ratio for Oryza_sativa_gobolsailbalam.Os132424RS1.ncrna.fa is 21.80%\n",
      "The GZIP compression ratio for Oryza_sativa_gobolsailbalam.Os132424RS1.ncrna.fa is 26.53%\n",
      "oryza_sativa_gobolsailbalam completed successfully!\n",
      "cds 43962 , total =  52732658\n",
      "The GECO3 compression ratio for Oryza_sativa_ir64.OsIR64RS1.cds.all.fa is 19.58%\n",
      "The GZIP compression ratio for Oryza_sativa_ir64.OsIR64RS1.cds.all.fa is 22.98%\n",
      "ncrna 268 , total =  215469\n",
      "The GECO3 compression ratio for Oryza_sativa_ir64.OsIR64RS1.ncrna.fa is 22.79%\n",
      "The GZIP compression ratio for Oryza_sativa_ir64.OsIR64RS1.ncrna.fa is 27.43%\n",
      "oryza_sativa_ir64 completed successfully!\n",
      "cds 44021 , total =  52467585\n",
      "The GECO3 compression ratio for Oryza_sativa_ketannangka.Os128077RS1.cds.all.fa is 19.77%\n",
      "The GZIP compression ratio for Oryza_sativa_ketannangka.Os128077RS1.cds.all.fa is 23.20%\n",
      "ncrna 300 , total =  232914\n",
      "The GECO3 compression ratio for Oryza_sativa_ketannangka.Os128077RS1.ncrna.fa is 21.23%\n",
      "The GZIP compression ratio for Oryza_sativa_ketannangka.Os128077RS1.ncrna.fa is 26.33%\n",
      "oryza_sativa_ketannangka completed successfully!\n",
      "cds 43153 , total =  51112803\n",
      "The GECO3 compression ratio for Oryza_sativa_khaoyaiguang.Os127518RS1.cds.all.fa is 20.07%\n",
      "The GZIP compression ratio for Oryza_sativa_khaoyaiguang.Os127518RS1.cds.all.fa is 23.74%\n",
      "ncrna 270 , total =  209565\n",
      "The GECO3 compression ratio for Oryza_sativa_khaoyaiguang.Os127518RS1.ncrna.fa is 21.81%\n",
      "The GZIP compression ratio for Oryza_sativa_khaoyaiguang.Os127518RS1.ncrna.fa is 26.63%\n",
      "oryza_sativa_khaoyaiguang completed successfully!\n",
      "cds 42945 , total =  50586945\n",
      "The GECO3 compression ratio for Oryza_sativa_larhamugad.Os125619RS1.cds.all.fa is 20.12%\n",
      "The GZIP compression ratio for Oryza_sativa_larhamugad.Os125619RS1.cds.all.fa is 23.75%\n",
      "ncrna 258 , total =  212090\n",
      "The GECO3 compression ratio for Oryza_sativa_larhamugad.Os125619RS1.ncrna.fa is 22.42%\n",
      "The GZIP compression ratio for Oryza_sativa_larhamugad.Os125619RS1.ncrna.fa is 27.13%\n",
      "oryza_sativa_larhamugad completed successfully!\n",
      "cds 7603 , total =  9227468\n",
      "The GECO3 compression ratio for Ostreococcus_lucimarinus.ASM9206v1.cds.all.fa is 22.38%\n",
      "The GZIP compression ratio for Ostreococcus_lucimarinus.ASM9206v1.cds.all.fa is 27.25%\n",
      "ncrna 24 , total =  2561\n",
      "The GECO3 compression ratio for Ostreococcus_lucimarinus.ASM9206v1.ncrna.fa is 23.82%\n",
      "The GZIP compression ratio for Ostreococcus_lucimarinus.ASM9206v1.ncrna.fa is 29.13%\n",
      "ostreococcus_lucimarinus completed successfully!\n",
      "cds 43519 , total =  51305166\n",
      "The GECO3 compression ratio for Oryza_sativa_lima.Os127564RS1.cds.all.fa is 20.07%\n",
      "The GZIP compression ratio for Oryza_sativa_lima.Os127564RS1.cds.all.fa is 23.93%\n",
      "ncrna 268 , total =  212127\n",
      "The GECO3 compression ratio for Oryza_sativa_lima.Os127564RS1.ncrna.fa is 22.18%\n",
      "The GZIP compression ratio for Oryza_sativa_lima.Os127564RS1.ncrna.fa is 26.91%\n",
      "oryza_sativa_lima completed successfully!\n",
      "cds 52083 , total =  60798351\n",
      "The GECO3 compression ratio for Oryza_sativa_liuxu.Os125827RS1.cds.all.fa is 18.65%\n",
      "The GZIP compression ratio for Oryza_sativa_liuxu.Os125827RS1.cds.all.fa is 23.60%\n",
      "ncrna 339 , total =  252984\n",
      "The GECO3 compression ratio for Oryza_sativa_liuxu.Os125827RS1.ncrna.fa is 20.74%\n",
      "The GZIP compression ratio for Oryza_sativa_liuxu.Os125827RS1.ncrna.fa is 26.39%\n",
      "oryza_sativa_liuxu completed successfully!\n",
      "cds 52017 , total =  62950386\n",
      "The GECO3 compression ratio for Oryza_sativa_mh63.MH63RS2.cds.all.fa is 17.50%\n",
      "The GZIP compression ratio for Oryza_sativa_mh63.MH63RS2.cds.all.fa is 19.79%\n",
      "cds 43823 , total =  52352898\n",
      "The GECO3 compression ratio for Oryza_sativa_n22.OsN22RS2.cds.all.fa is 19.75%\n",
      "The GZIP compression ratio for Oryza_sativa_n22.OsN22RS2.cds.all.fa is 23.16%\n",
      "ncrna 924 , total =  791677\n",
      "The GECO3 compression ratio for Oryza_sativa_mh63.MH63RS2.ncrna.fa is 19.62%\n",
      "The GZIP compression ratio for Oryza_sativa_mh63.MH63RS2.ncrna.fa is 24.48%\n",
      "oryza_sativa_mh63 completed successfully!\n",
      "ncrna 276 , total =  214411\n",
      "The GECO3 compression ratio for Oryza_sativa_n22.OsN22RS2.ncrna.fa is 20.93%\n",
      "The GZIP compression ratio for Oryza_sativa_n22.OsN22RS2.ncrna.fa is 25.99%\n",
      "oryza_sativa_n22 completed successfully!\n",
      "cds 43214 , total =  51325452\n",
      "The GECO3 compression ratio for Oryza_sativa_natelboro.Os127652RS1.cds.all.fa is 19.90%\n",
      "The GZIP compression ratio for Oryza_sativa_natelboro.Os127652RS1.cds.all.fa is 23.39%\n",
      "ncrna 257 , total =  182112\n",
      "The GECO3 compression ratio for Oryza_sativa_natelboro.Os127652RS1.ncrna.fa is 21.69%\n",
      "The GZIP compression ratio for Oryza_sativa_natelboro.Os127652RS1.ncrna.fa is 26.53%\n",
      "oryza_sativa_natelboro completed successfully!\n",
      "cds 44341 , total =  53067315\n",
      "The GECO3 compression ratio for Oryza_sativa_pr106.Os127742RS1.cds.all.fa is 19.55%\n",
      "The GZIP compression ratio for Oryza_sativa_pr106.Os127742RS1.cds.all.fa is 22.97%\n",
      "ncrna 275 , total =  208684\n",
      "The GECO3 compression ratio for Oryza_sativa_pr106.Os127742RS1.ncrna.fa is 22.15%\n",
      "The GZIP compression ratio for Oryza_sativa_pr106.Os127742RS1.ncrna.fa is 27.05%\n",
      "oryza_sativa_pr106 completed successfully!\n",
      "cds 50680 , total =  62048808\n",
      "The GECO3 compression ratio for Oryza_sativa_zs97.ZS97RS3.cds.all.fa is 17.62%\n",
      "The GZIP compression ratio for Oryza_sativa_zs97.ZS97RS3.cds.all.fa is 19.98%\n",
      "ncrna 948 , total =  789155\n",
      "The GECO3 compression ratio for Oryza_sativa_zs97.ZS97RS3.ncrna.fa is 21.02%\n",
      "The GZIP compression ratio for Oryza_sativa_zs97.ZS97RS3.ncrna.fa is 25.99%\n",
      "oryza_sativa_zs97 completed successfully!\n",
      "cds 42523 , total =  51996270\n",
      "The GECO3 compression ratio for Panicum_hallii.PhalliiHAL_v2.1.cds.all.fa is 18.04%\n",
      "The GZIP compression ratio for Panicum_hallii.PhalliiHAL_v2.1.cds.all.fa is 21.04%\n",
      "Directory or file not found for panicum_hallii in ncrna folder: 550 Failed to change directory.\n",
      "panicum_hallii completed successfully!\n",
      "cds 44192 , total =  54354546\n",
      "The GECO3 compression ratio for Panicum_hallii_fil2.PHallii_v3.1.cds.all.fa is 17.56%\n",
      "The GZIP compression ratio for Panicum_hallii_fil2.PHallii_v3.1.cds.all.fa is 20.56%\n",
      "Directory or file not found for panicum_hallii_fil2 in ncrna folder: 550 Failed to change directory.\n",
      "panicum_hallii_fil2 completed successfully!\n",
      "cds 32720 , total =  41198894\n",
      "The GECO3 compression ratio for Phaseolus_vulgaris.PhaVulg1_0.cds.all.fa is 20.57%\n",
      "The GZIP compression ratio for Phaseolus_vulgaris.PhaVulg1_0.cds.all.fa is 24.01%\n",
      "ncrna 1190 , total =  163586\n",
      "The GECO3 compression ratio for Phaseolus_vulgaris.PhaVulg1_0.ncrna.fa is 11.94%\n",
      "The GZIP compression ratio for Phaseolus_vulgaris.PhaVulg1_0.ncrna.fa is 15.68%\n",
      "phaseolus_vulgaris completed successfully!\n",
      "cds 41770 , total =  47202919\n",
      "The GECO3 compression ratio for Papaver_somniferum.ASM357369v1.cds.all.fa is 21.01%\n",
      "The GZIP compression ratio for Papaver_somniferum.ASM357369v1.cds.all.fa is 27.92%\n",
      "Directory or file not found for papaver_somniferum in ncrna folder: 550 Failed to change directory.\n",
      "papaver_somniferum completed successfully!\n",
      "cds 31784 , total =  38115064\n",
      "The GECO3 compression ratio for Pistacia_vera.PisVer_v2.cds.all.fa is 21.20%\n",
      "The GZIP compression ratio for Pistacia_vera.PisVer_v2.cds.all.fa is 27.97%\n",
      "Directory or file not found for pistacia_vera in ncrna folder: 550 Failed to change directory.\n",
      "pistacia_vera completed successfully!\n",
      "cds 56770 , total =  57926675\n",
      "The GECO3 compression ratio for Pisum_sativum.Pisum_sativum_v1a.cds.all.fa is 18.35%\n",
      "The GZIP compression ratio for Pisum_sativum.Pisum_sativum_v1a.cds.all.fa is 21.21%\n",
      "Directory or file not found for pisum_sativum in ncrna folder: 550 Failed to change directory.\n",
      "pisum_sativum completed successfully!\n",
      "cds 42794 , total =  38760591\n",
      "The GECO3 compression ratio for Prunus_avium.PAV_r1.0.cds.all.fa is 21.22%\n",
      "The GZIP compression ratio for Prunus_avium.PAV_r1.0.cds.all.fa is 27.82%\n",
      "Directory or file not found for prunus_avium in ncrna folder: 550 Failed to change directory.\n",
      "prunus_avium completed successfully!\n",
      "cds 32553 , total =  40354416\n",
      "The GECO3 compression ratio for Prunus_dulcis.ALMONDv2.cds.all.fa is 19.91%\n",
      "The GZIP compression ratio for Prunus_dulcis.ALMONDv2.cds.all.fa is 23.56%\n",
      "ncrna 6455 , total =  4150383\n",
      "The GECO3 compression ratio for Prunus_dulcis.ALMONDv2.ncrna.fa is 18.66%\n",
      "The GZIP compression ratio for Prunus_dulcis.ALMONDv2.ncrna.fa is 25.02%\n",
      "prunus_dulcis completed successfully!\n",
      "cds 39366 , total =  49763784\n",
      "The GECO3 compression ratio for Quercus_lobata.ValleyOak3.0.cds.all.fa is 21.06%\n",
      "The GZIP compression ratio for Quercus_lobata.ValleyOak3.0.cds.all.fa is 27.90%\n",
      "Directory or file not found for quercus_lobata in ncrna folder: 550 Failed to change directory.\n",
      "quercus_lobata completed successfully!\n",
      "cds 52400 , total =  75359469\n",
      "The GECO3 compression ratio for Populus_trichocarpa.Pop_tri_v4.cds.all.fa is 16.00%\n",
      "The GZIP compression ratio for Populus_trichocarpa.Pop_tri_v4.cds.all.fa is 17.36%\n",
      "Directory or file not found for populus_trichocarpa in ncrna folder: 550 Failed to change directory.\n",
      "populus_trichocarpa completed successfully!\n",
      "cds 86669 , total =  116720069\n",
      "The GECO3 compression ratio for Physcomitrium_patens.Phypa_V3.cds.all.fa is 10.99%\n",
      "The GZIP compression ratio for Physcomitrium_patens.Phypa_V3.cds.all.fa is 10.36%\n",
      "ncrna 440 , total =  49193\n",
      "The GECO3 compression ratio for Physcomitrium_patens.Phypa_V3.ncrna.fa is 9.82%\n",
      "The GZIP compression ratio for Physcomitrium_patens.Phypa_V3.ncrna.fa is 12.31%\n",
      "physcomitrium_patens completed successfully!\n",
      "cds 47089 , total =  64914980\n",
      "The GECO3 compression ratio for Prunus_persica.Prunus_persica_NCBIv2.cds.all.fa is 14.32%\n",
      "The GZIP compression ratio for Prunus_persica.Prunus_persica_NCBIv2.cds.all.fa is 15.47%\n",
      "ncrna 976 , total =  188746\n",
      "The GECO3 compression ratio for Prunus_persica.Prunus_persica_NCBIv2.ncrna.fa is 9.40%\n",
      "The GZIP compression ratio for Prunus_persica.Prunus_persica_NCBIv2.ncrna.fa is 12.37%\n",
      "prunus_persica completed successfully!\n",
      "cds 56555 , total =  55334467\n",
      "The GECO3 compression ratio for Quercus_suber.CorkOak1.0.cds.all.fa is 20.27%\n",
      "The GZIP compression ratio for Quercus_suber.CorkOak1.0.cds.all.fa is 25.61%\n",
      "ncrna 367 , total =  3122251\n",
      "The GECO3 compression ratio for Quercus_suber.CorkOak1.0.ncrna.fa is 22.11%\n",
      "The GZIP compression ratio for Quercus_suber.CorkOak1.0.ncrna.fa is 26.80%\n",
      "quercus_suber completed successfully!\n",
      "cds 45464 , total =  43720026\n",
      "The GECO3 compression ratio for Rosa_chinensis.RchiOBHm-V2.cds.all.fa is 21.90%\n",
      "The GZIP compression ratio for Rosa_chinensis.RchiOBHm-V2.cds.all.fa is 27.94%\n",
      "ncrna 4969 , total =  2137516\n",
      "The GECO3 compression ratio for Rosa_chinensis.RchiOBHm-V2.ncrna.fa is 19.56%\n",
      "The GZIP compression ratio for Rosa_chinensis.RchiOBHm-V2.ncrna.fa is 26.65%\n",
      "rosa_chinensis completed successfully!\n",
      "cds 34441 , total =  44180002\n",
      "The GECO3 compression ratio for Secale_cereale.Rye_Lo7_2018_v1p1p1.cds.all.fa is 21.85%\n",
      "The GZIP compression ratio for Secale_cereale.Rye_Lo7_2018_v1p1p1.cds.all.fa is 28.02%\n",
      "Directory or file not found for secale_cereale in ncrna folder: 550 Failed to change directory.\n",
      "secale_cereale completed successfully!\n",
      "cds 25172 , total =  30362225\n",
      "The GECO3 compression ratio for Sesamum_indicum.S_indicum_v1.0.cds.all.fa is 23.54%\n",
      "The GZIP compression ratio for Sesamum_indicum.S_indicum_v1.0.cds.all.fa is 28.12%\n",
      "Directory or file not found for sesamum_indicum in ncrna folder: 550 Failed to change directory.\n",
      "sesamum_indicum completed successfully!\n",
      "cds 53284 , total =  62197261\n",
      "The GECO3 compression ratio for Saccharum_spontaneum.Sspon.HiC_chr_asm.cds.all.fa is 18.12%\n",
      "The GZIP compression ratio for Saccharum_spontaneum.Sspon.HiC_chr_asm.cds.all.fa is 27.74%\n",
      "Directory or file not found for saccharum_spontaneum in ncrna folder: 550 Failed to change directory.\n",
      "saccharum_spontaneum completed successfully!\n",
      "cds 34825 , total =  41180260\n",
      "The GECO3 compression ratio for Selaginella_moellendorffii.v1.0.cds.all.fa is 17.68%\n",
      "The GZIP compression ratio for Selaginella_moellendorffii.v1.0.cds.all.fa is 27.52%\n",
      "ncrna 1221 , total =  275579\n",
      "The GECO3 compression ratio for Selaginella_moellendorffii.v1.0.ncrna.fa is 5.36%\n",
      "The GZIP compression ratio for Selaginella_moellendorffii.v1.0.ncrna.fa is 8.25%\n",
      "selaginella_moellendorffii completed successfully!\n",
      "cds 41023 , total =  45897255\n",
      "The GECO3 compression ratio for Setaria_italica.Setaria_italica_v2.0.cds.all.fa is 20.44%\n",
      "The GZIP compression ratio for Setaria_italica.Setaria_italica_v2.0.cds.all.fa is 24.58%\n",
      "ncrna 779 , total =  117023\n",
      "The GECO3 compression ratio for Setaria_italica.Setaria_italica_v2.0.ncrna.fa is 9.55%\n",
      "The GZIP compression ratio for Setaria_italica.Setaria_italica_v2.0.ncrna.fa is 11.83%\n",
      "setaria_italica completed successfully!\n",
      "cds 34429 , total =  36568052\n",
      "The GECO3 compression ratio for Solanum_lycopersicum.SL3.0.cds.all.fa is 23.19%\n",
      "The GZIP compression ratio for Solanum_lycopersicum.SL3.0.cds.all.fa is 27.83%\n",
      "ncrna 1167 , total =  177422\n",
      "The GECO3 compression ratio for Solanum_lycopersicum.SL3.0.ncrna.fa is 13.18%\n",
      "The GZIP compression ratio for Solanum_lycopersicum.SL3.0.ncrna.fa is 18.03%\n",
      "solanum_lycopersicum completed successfully!\n",
      "cds 52459 , total =  63143046\n",
      "The GECO3 compression ratio for Setaria_viridis.Setaria_viridis_v2.0.cds.all.fa is 17.10%\n",
      "The GZIP compression ratio for Setaria_viridis.Setaria_viridis_v2.0.cds.all.fa is 19.44%\n",
      "Directory or file not found for setaria_viridis in ncrna folder: 550 Failed to change directory.\n",
      "setaria_viridis completed successfully!\n",
      "cds 56210 , total =  50848033\n",
      "The GECO3 compression ratio for Solanum_tuberosum.SolTub_3.0.cds.all.fa is 18.36%\n",
      "The GZIP compression ratio for Solanum_tuberosum.SolTub_3.0.cds.all.fa is 20.54%\n",
      "ncrna 1315 , total =  226548\n",
      "The GECO3 compression ratio for Solanum_tuberosum.SolTub_3.0.ncrna.fa is 10.74%\n",
      "The GZIP compression ratio for Solanum_tuberosum.SolTub_3.0.ncrna.fa is 15.47%\n",
      "solanum_tuberosum completed successfully!\n",
      "cds 36862 , total =  40228452\n",
      "The GECO3 compression ratio for Solanum_tuberosum_rh8903916.ASM1418947v1.cds.all.fa is 21.65%\n",
      "The GZIP compression ratio for Solanum_tuberosum_rh8903916.ASM1418947v1.cds.all.fa is 27.70%\n",
      "Directory or file not found for solanum_tuberosum_rh8903916 in ncrna folder: 550 Failed to change directory.\n",
      "solanum_tuberosum_rh8903916 completed successfully!\n",
      "cds 30439 , total =  44154325\n",
      "The GECO3 compression ratio for Theobroma_cacao_criollo.Criollo_cocoa_genome_V2.cds.all.fa is 17.64%\n",
      "The GZIP compression ratio for Theobroma_cacao_criollo.Criollo_cocoa_genome_V2.cds.all.fa is 19.46%\n",
      "cds 47110 , total =  57859353\n",
      "The GECO3 compression ratio for Sorghum_bicolor.Sorghum_bicolor_NCBIv3.cds.all.fa is 17.57%\n",
      "The GZIP compression ratio for Sorghum_bicolor.Sorghum_bicolor_NCBIv3.cds.all.fa is 20.08%\n",
      "ncrna 1449 , total =  282493\n",
      "The GECO3 compression ratio for Sorghum_bicolor.Sorghum_bicolor_NCBIv3.ncrna.fa is 7.75%\n",
      "The GZIP compression ratio for Sorghum_bicolor.Sorghum_bicolor_NCBIv3.ncrna.fa is 10.47%\n",
      "sorghum_bicolor completed successfully!\n",
      "ncrna 2699 , total =  1930723\n",
      "The GECO3 compression ratio for Theobroma_cacao_criollo.Criollo_cocoa_genome_V2.ncrna.fa is 14.10%\n",
      "The GZIP compression ratio for Theobroma_cacao_criollo.Criollo_cocoa_genome_V2.ncrna.fa is 18.38%\n",
      "theobroma_cacao_criollo completed successfully!\n",
      "cds 41270 , total =  42358498\n",
      "The GECO3 compression ratio for Trifolium_pratense.Trpr.cds.all.fa is 22.31%\n",
      "The GZIP compression ratio for Trifolium_pratense.Trpr.cds.all.fa is 26.29%\n",
      "ncrna 1183 , total =  133547\n",
      "The GECO3 compression ratio for Trifolium_pratense.Trpr.ncrna.fa is 14.21%\n",
      "The GZIP compression ratio for Trifolium_pratense.Trpr.ncrna.fa is 17.96%\n",
      "trifolium_pratense completed successfully!\n",
      "cds 44186 , total =  58263120\n",
      "The GECO3 compression ratio for Theobroma_cacao.Theobroma_cacao_20110822.cds.all.fa is 16.57%\n",
      "The GZIP compression ratio for Theobroma_cacao.Theobroma_cacao_20110822.cds.all.fa is 17.78%\n",
      "ncrna 465 , total =  53557\n",
      "The GECO3 compression ratio for Theobroma_cacao.Theobroma_cacao_20110822.ncrna.fa is 15.84%\n",
      "The GZIP compression ratio for Theobroma_cacao.Theobroma_cacao_20110822.ncrna.fa is 20.00%\n",
      "theobroma_cacao completed successfully!\n",
      "cds 122839 , total =  129198336\n",
      "The GECO3 compression ratio for Triticum_aestivum_cadenza.EIv1.1.cds.all.fa is 19.26%\n",
      "The GZIP compression ratio for Triticum_aestivum_cadenza.EIv1.1.cds.all.fa is 27.97%\n",
      "Directory or file not found for triticum_aestivum_cadenza in ncrna folder: 550 Failed to change directory.\n",
      "triticum_aestivum_cadenza completed successfully!\n",
      "cds 121770 , total =  129241357\n",
      "The GECO3 compression ratio for Triticum_aestivum_claire.EIv1.1.cds.all.fa is 19.27%\n",
      "The GZIP compression ratio for Triticum_aestivum_claire.EIv1.1.cds.all.fa is 27.97%\n",
      "Directory or file not found for triticum_aestivum_claire in ncrna folder: 550 Failed to change directory.\n",
      "triticum_aestivum_claire completed successfully!\n",
      "cds 133346 , total =  177619221\n",
      "The GECO3 compression ratio for Triticum_aestivum.IWGSC.cds.all.fa is 16.91%\n",
      "The GZIP compression ratio for Triticum_aestivum.IWGSC.cds.all.fa is 21.87%\n",
      "ncrna 12853 , total =  1909631\n",
      "The GECO3 compression ratio for Triticum_aestivum.IWGSC.ncrna.fa is 4.34%\n",
      "The GZIP compression ratio for Triticum_aestivum.IWGSC.ncrna.fa is 8.79%\n",
      "triticum_aestivum completed successfully!\n",
      "cds 123395 , total =  180241510\n",
      "The GECO3 compression ratio for Triticum_aestivum_kariega.Tae_Kariega_v1.cds.all.fa is 12.19%\n",
      "The GZIP compression ratio for Triticum_aestivum_kariega.Tae_Kariega_v1.cds.all.fa is 23.99%\n",
      "Directory or file not found for triticum_aestivum_kariega in ncrna folder: 550 Failed to change directory.\n",
      "triticum_aestivum_kariega completed successfully!\n",
      "cds 167830 , total =  218088177\n",
      "The GECO3 compression ratio for Triticum_aestivum_arinalrfor.PGSBv2.1.cds.all.fa is 16.53%\n",
      "The GZIP compression ratio for Triticum_aestivum_arinalrfor.PGSBv2.1.cds.all.fa is 23.45%\n",
      "Directory or file not found for triticum_aestivum_arinalrfor in ncrna folder: 550 Failed to change directory.\n",
      "triticum_aestivum_arinalrfor completed successfully!\n",
      "cds 160615 , total =  206951562\n",
      "The GECO3 compression ratio for Triticum_aestivum_jagger.PGSBv2.1.cds.all.fa is 16.81%\n",
      "The GZIP compression ratio for Triticum_aestivum_jagger.PGSBv2.1.cds.all.fa is 23.95%\n",
      "Directory or file not found for triticum_aestivum_jagger in ncrna folder: 550 Failed to change directory.\n",
      "triticum_aestivum_jagger completed successfully!\n",
      "cds 159592 , total =  206359766\n",
      "The GECO3 compression ratio for Triticum_aestivum_lancer.PGSBv2.1.cds.all.fa is 16.79%\n",
      "The GZIP compression ratio for Triticum_aestivum_lancer.PGSBv2.1.cds.all.fa is 24.11%\n",
      "Directory or file not found for triticum_aestivum_lancer in ncrna folder: 550 Failed to change directory.\n",
      "triticum_aestivum_lancer completed successfully!\n",
      "cds 168985 , total =  221380798\n",
      "The GECO3 compression ratio for Triticum_aestivum_julius.PGSBv2.1.cds.all.fa is 16.44%\n",
      "The GZIP compression ratio for Triticum_aestivum_julius.PGSBv2.1.cds.all.fa is 22.82%\n",
      "Directory or file not found for triticum_aestivum_julius in ncrna folder: 550 Failed to change directory.\n",
      "triticum_aestivum_julius completed successfully!\n",
      "cds 121567 , total =  128747584\n",
      "The GECO3 compression ratio for Triticum_aestivum_paragon.EIv1.1.cds.all.fa is 19.29%\n",
      "The GZIP compression ratio for Triticum_aestivum_paragon.EIv1.1.cds.all.fa is 27.97%\n",
      "Directory or file not found for triticum_aestivum_paragon in ncrna folder: 550 Failed to change directory.\n",
      "triticum_aestivum_paragon completed successfully!\n",
      "cds 159978 , total =  206448906\n",
      "The GECO3 compression ratio for Triticum_aestivum_mace.PGSBv2.1.cds.all.fa is 16.83%\n",
      "The GZIP compression ratio for Triticum_aestivum_mace.PGSBv2.1.cds.all.fa is 24.14%\n",
      "Directory or file not found for triticum_aestivum_mace in ncrna folder: 550 Failed to change directory.\n",
      "triticum_aestivum_mace completed successfully!\n",
      "cds 163650 , total =  210995817\n",
      "The GECO3 compression ratio for Triticum_aestivum_landmark.PGSBv2.1.cds.all.fa is 16.72%\n",
      "The GZIP compression ratio for Triticum_aestivum_landmark.PGSBv2.1.cds.all.fa is 23.54%\n",
      "Directory or file not found for triticum_aestivum_landmark in ncrna folder: 550 Failed to change directory.\n",
      "triticum_aestivum_landmark completed successfully!\n",
      "cds 162401 , total =  209733742\n",
      "The GECO3 compression ratio for Triticum_aestivum_mattis.PGSBv2.1.cds.all.fa is 16.69%\n",
      "The GZIP compression ratio for Triticum_aestivum_mattis.PGSBv2.1.cds.all.fa is 23.81%\n",
      "Directory or file not found for triticum_aestivum_mattis in ncrna folder: 550 Failed to change directory.\n",
      "triticum_aestivum_mattis completed successfully!\n",
      "cds 121524 , total =  129038092\n",
      "The GECO3 compression ratio for Triticum_aestivum_robigus.EIv1.1.cds.all.fa is 19.27%\n",
      "The GZIP compression ratio for Triticum_aestivum_robigus.EIv1.1.cds.all.fa is 27.97%\n",
      "Directory or file not found for triticum_aestivum_robigus in ncrna folder: 550 Failed to change directory.\n",
      "triticum_aestivum_robigus completed successfully!\n",
      "cds 128489 , total =  161905931\n",
      "The GECO3 compression ratio for Triticum_aestivum_renan.Triticum_aestivum_Renan_v2.1.cds.all.fa is 16.51%\n",
      "The GZIP compression ratio for Triticum_aestivum_renan.Triticum_aestivum_Renan_v2.1.cds.all.fa is 22.04%\n",
      "Directory or file not found for triticum_aestivum_renan in ncrna folder: 550 Failed to change directory.\n",
      "triticum_aestivum_renan completed successfully!\n",
      "cds 131813 , total =  175723560\n",
      "The GECO3 compression ratio for Triticum_aestivum_refseqv2.IWGSC_RefSeq_v2.1.cds.all.fa is 16.89%\n",
      "The GZIP compression ratio for Triticum_aestivum_refseqv2.IWGSC_RefSeq_v2.1.cds.all.fa is 21.84%\n",
      "Directory or file not found for triticum_aestivum_refseqv2 in ncrna folder: 550 Failed to change directory.\n",
      "triticum_aestivum_refseqv2 completed successfully!\n",
      "cds 168730 , total =  218131557\n",
      "The GECO3 compression ratio for Triticum_aestivum_norin61.PGSBv2.1.cds.all.fa is 16.54%\n",
      "The GZIP compression ratio for Triticum_aestivum_norin61.PGSBv2.1.cds.all.fa is 23.63%\n",
      "Directory or file not found for triticum_aestivum_norin61 in ncrna folder: 550 Failed to change directory.\n",
      "triticum_aestivum_norin61 completed successfully!\n",
      "cds 37065 , total =  43127952\n",
      "The GECO3 compression ratio for Vicia_faba.Hedin2_genome_v1.cds.all.fa is 19.88%\n",
      "The GZIP compression ratio for Vicia_faba.Hedin2_genome_v1.cds.all.fa is 25.41%\n",
      "Directory or file not found for vicia_faba in ncrna folder: 550 Failed to change directory.\n",
      "vicia_faba completed successfully!\n",
      "cds 61696 , total =  62456501\n",
      "The GECO3 compression ratio for Triticum_urartu.IGDB.cds.all.fa is 16.99%\n",
      "The GZIP compression ratio for Triticum_urartu.IGDB.cds.all.fa is 20.02%\n",
      "Directory or file not found for triticum_urartu in ncrna folder: 550 Failed to change directory.\n",
      "triticum_urartu completed successfully!\n",
      "cds 33860 , total =  37805311\n",
      "The GECO3 compression ratio for Vigna_angularis.Vigan1.1.cds.all.fa is 22.13%\n",
      "The GZIP compression ratio for Vigna_angularis.Vigan1.1.cds.all.fa is 27.48%\n",
      "ncrna 959 , total =  124352\n",
      "The GECO3 compression ratio for Vigna_angularis.Vigan1.1.ncrna.fa is 11.08%\n",
      "The GZIP compression ratio for Vigna_angularis.Vigan1.1.ncrna.fa is 14.78%\n",
      "vigna_angularis completed successfully!\n",
      "cds 23181 , total =  29742927\n",
      "The GECO3 compression ratio for Vigna_radiata.Vradiata_ver6.cds.all.fa is 22.80%\n",
      "The GZIP compression ratio for Vigna_radiata.Vradiata_ver6.cds.all.fa is 26.86%\n",
      "ncrna 796 , total =  84051\n",
      "The GECO3 compression ratio for Vigna_radiata.Vradiata_ver6.ncrna.fa is 13.13%\n",
      "The GZIP compression ratio for Vigna_radiata.Vradiata_ver6.ncrna.fa is 16.61%\n",
      "vigna_radiata completed successfully!\n",
      "cds 122646 , total =  130173324\n",
      "The GECO3 compression ratio for Triticum_aestivum_weebil.WeebilV1.cds.all.fa is 19.22%\n",
      "The GZIP compression ratio for Triticum_aestivum_weebil.WeebilV1.cds.all.fa is 27.97%\n",
      "Directory or file not found for triticum_aestivum_weebil in ncrna folder: 550 Failed to change directory.\n",
      "triticum_aestivum_weebil completed successfully!\n",
      "cds 113301 , total =  130678802\n",
      "The GECO3 compression ratio for Triticum_spelta.PGSBv2.0.cds.all.fa is 19.30%\n",
      "The GZIP compression ratio for Triticum_spelta.PGSBv2.0.cds.all.fa is 28.01%\n",
      "Directory or file not found for triticum_spelta in ncrna folder: 550 Failed to change directory.\n",
      "triticum_spelta completed successfully!\n",
      "cds 41097 , total =  46314728\n",
      "The GECO3 compression ratio for Vitis_vinifera.PN40024.v4.cds.all.fa is 19.69%\n",
      "The GZIP compression ratio for Vitis_vinifera.PN40024.v4.cds.all.fa is 23.42%\n",
      "Directory or file not found for vitis_vinifera in ncrna folder: 550 Failed to change directory.\n",
      "vitis_vinifera completed successfully!\n",
      "cds 54346 , total =  73531804\n",
      "The GECO3 compression ratio for Vigna_unguiculata.ASM411807v1.cds.all.fa is 14.79%\n",
      "The GZIP compression ratio for Vigna_unguiculata.ASM411807v1.cds.all.fa is 15.97%\n",
      "Directory or file not found for vigna_unguiculata in ncrna folder: 550 Failed to change directory.\n",
      "vigna_unguiculata completed successfully!\n",
      "cds 161474 , total =  208832890\n",
      "The GECO3 compression ratio for Triticum_aestivum_stanley.PGSBv2.2.cds.all.fa is 16.77%\n",
      "The GZIP compression ratio for Triticum_aestivum_stanley.PGSBv2.2.cds.all.fa is 23.81%\n",
      "Directory or file not found for triticum_aestivum_stanley in ncrna folder: 550 Failed to change directory.\n",
      "triticum_aestivum_stanley completed successfully!\n",
      "cds 72539 , total =  87511302\n",
      "The GECO3 compression ratio for Zea_mays.Zm-B73-REFERENCE-NAM-5.0.cds.all.fa is 14.43%\n",
      "The GZIP compression ratio for Zea_mays.Zm-B73-REFERENCE-NAM-5.0.cds.all.fa is 15.49%\n",
      "ncrna 4802 , total =  3812673\n",
      "The GECO3 compression ratio for Zea_mays.Zm-B73-REFERENCE-NAM-5.0.ncrna.fa is 18.93%\n",
      "The GZIP compression ratio for Zea_mays.Zm-B73-REFERENCE-NAM-5.0.ncrna.fa is 25.82%\n",
      "zea_mays completed successfully!\n",
      "cds 196105 , total =  271432188\n",
      "The GECO3 compression ratio for Triticum_turgidum.Svevo.v1.cds.all.fa is 12.10%\n",
      "The GZIP compression ratio for Triticum_turgidum.Svevo.v1.cds.all.fa is 10.55%\n",
      "Directory or file not found for triticum_turgidum in ncrna folder: 550 Failed to change directory.\n",
      "triticum_turgidum completed successfully!\n",
      "cds 295286 , total =  302933582\n",
      "The GECO3 compression ratio for Triticum_dicoccoides.WEWSeq_v.1.0.cds.all.fa is 11.41%\n",
      "The GZIP compression ratio for Triticum_dicoccoides.WEWSeq_v.1.0.cds.all.fa is 9.07%\n",
      "ncrna 4731 , total =  605042\n",
      "The GECO3 compression ratio for Triticum_dicoccoides.WEWSeq_v.1.0.ncrna.fa is 7.59%\n",
      "The GZIP compression ratio for Triticum_dicoccoides.WEWSeq_v.1.0.ncrna.fa is 11.74%\n",
      "triticum_dicoccoides completed successfully!\n",
      "Results saved to species_compression_ratios_plants_looptest.csv\n"
     ]
    }
   ],
   "source": [
    "# Usage Plants\n",
    "\n",
    "ftp_server = \"ftp.ebi.ac.uk\" \n",
    "base_ftp_directory = \"/ensemblgenomes/pub/current/plants/fasta/\"\n",
    "output_csv_file = \"compression_ratios_plant.csv\"\n",
    "\n",
    "process_all_organisms(ftp_server, base_ftp_directory, output_csv_file)\n",
    "\n",
    "print(f\"Results saved to {output_csv_file}\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "cds 4776 , total =  7020033\n",
      "The GECO3 compression ratio for Ashbya_gossypii.ASM9102v1.cds.all.fa is 24.18%\n",
      "The GZIP compression ratio for Ashbya_gossypii.ASM9102v1.cds.all.fa is 28.25%\n",
      "ncrna 725 , total =  275179\n",
      "The GECO3 compression ratio for Ashbya_gossypii.ASM9102v1.ncrna.fa is 2.80%\n",
      "The GZIP compression ratio for Ashbya_gossypii.ASM9102v1.ncrna.fa is 4.14%\n",
      "ashbya_gossypii completed successfully!\n",
      "cds 9121 , total =  13531458\n",
      "The GECO3 compression ratio for Aspergillus_clavatus.ASM271v1.cds.all.fa is 24.28%\n",
      "The GZIP compression ratio for Aspergillus_clavatus.ASM271v1.cds.all.fa is 28.17%\n",
      "cds 9623 , total =  14234529\n",
      "The GECO3 compression ratio for Aspergillus_fumigatus.ASM265v1.cds.all.fa is 24.39%\n",
      "The GZIP compression ratio for Aspergillus_fumigatus.ASM265v1.cds.all.fa is 28.24%\n",
      "cds 9929 , total =  14433496\n",
      "The GECO3 compression ratio for Aspergillus_fumigatusa1163.ASM15014v1.cds.all.fa is 24.39%\n",
      "The GZIP compression ratio for Aspergillus_fumigatusa1163.ASM15014v1.cds.all.fa is 28.24%\n",
      "ncrna 229 , total =  18818\n",
      "The GECO3 compression ratio for Aspergillus_fumigatus.ASM265v1.ncrna.fa is 16.66%\n",
      "The GZIP compression ratio for Aspergillus_fumigatus.ASM265v1.ncrna.fa is 20.09%\n",
      "ncrna 180 , total =  14786\n",
      "The GECO3 compression ratio for Aspergillus_fumigatusa1163.ASM15014v1.ncrna.fa is 12.79%\n",
      "The GZIP compression ratio for Aspergillus_fumigatusa1163.ASM15014v1.ncrna.fa is 15.16%\n"
     ]
    }
   ],
   "source": [
    "# Usage Fungi\n",
    "\n",
    "ftp_server = \"ftp.ensemblgenomes.org\" \n",
    "base_ftp_directory = \"/pub/fungi/current/fasta/\"\n",
    "output_csv_file = \"compression_ratios_fungi.csv\"\n",
    "\n",
    "process_all_organisms(ftp_server, base_ftp_directory, output_csv_file)\n",
    "\n",
    "print(f\"Results saved to {output_csv_file}\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "cds 20241 , total =  37975243\n",
      "The GECO3 compression ratio for Acromyrmex_echinatior_gca000204515v1rs.Aech_3.9.cds.all.fa is 13.54%\n",
      "The GZIP compression ratio for Acromyrmex_echinatior_gca000204515v1rs.Aech_3.9.cds.all.fa is 15.30%\n",
      "ncrna 1369 , total =  1175392\n",
      "The GECO3 compression ratio for Acromyrmex_echinatior_gca000204515v1rs.Aech_3.9.ncrna.fa is 17.45%\n",
      "The GZIP compression ratio for Acromyrmex_echinatior_gca000204515v1rs.Aech_3.9.ncrna.fa is 21.20%\n",
      "acromyrmex_echinatior_gca000204515v1rs completed successfully!\n",
      "cds 24967 , total =  44401530\n",
      "The GECO3 compression ratio for Adelges_cooleyi_gca023614345v1rs.UGA_ACOO_1.1.cds.all.fa is 12.35%\n",
      "The GZIP compression ratio for Adelges_cooleyi_gca023614345v1rs.UGA_ACOO_1.1.cds.all.fa is 14.22%\n",
      "ncrna 1983 , total =  1743407\n",
      "The GECO3 compression ratio for Adelges_cooleyi_gca023614345v1rs.UGA_ACOO_1.1.ncrna.fa is 10.64%\n",
      "The GZIP compression ratio for Adelges_cooleyi_gca023614345v1rs.UGA_ACOO_1.1.ncrna.fa is 16.38%\n",
      "adelges_cooleyi_gca023614345v1rs completed successfully!\n",
      "cds 27907 , total =  46265849\n",
      "The GECO3 compression ratio for Acyrthosiphon_pisum_gca005508785v2rs.pea_aphid_22Mar2018_4r6ur_v2.cds.all.fa is 13.94%\n",
      "The GZIP compression ratio for Acyrthosiphon_pisum_gca005508785v2rs.pea_aphid_22Mar2018_4r6ur_v2.cds.all.fa is 16.28%\n",
      "ncrna 3086 , total =  3292824\n",
      "The GECO3 compression ratio for Acyrthosiphon_pisum_gca005508785v2rs.pea_aphid_22Mar2018_4r6ur_v2.ncrna.fa is 14.51%\n",
      "The GZIP compression ratio for Acyrthosiphon_pisum_gca005508785v2rs.pea_aphid_22Mar2018_4r6ur_v2.ncrna.fa is 18.44%\n",
      "acyrthosiphon_pisum_gca005508785v2rs completed successfully!\n",
      "cds 27037 , total =  47053476\n",
      "The GECO3 compression ratio for Actinia_tenebrosa_gca009602425v1.ASM960242v1.cds.all.fa is 17.98%\n",
      "The GZIP compression ratio for Actinia_tenebrosa_gca009602425v1.ASM960242v1.cds.all.fa is 20.40%\n",
      "ncrna 2759 , total =  1882383\n",
      "The GECO3 compression ratio for Actinia_tenebrosa_gca009602425v1.ASM960242v1.ncrna.fa is 19.37%\n",
      "The GZIP compression ratio for Actinia_tenebrosa_gca009602425v1.ASM960242v1.ncrna.fa is 23.57%\n",
      "actinia_tenebrosa_gca009602425v1 completed successfully!\n",
      "cds 33214 , total =  69522977\n",
      "The GECO3 compression ratio for Acanthaster_planci_gca001949145v1.OKI_Apl_1.0.cds.all.fa is 13.35%\n",
      "The GZIP compression ratio for Acanthaster_planci_gca001949145v1.OKI_Apl_1.0.cds.all.fa is 13.71%\n",
      "cds 49300 , total =  56378944\n",
      "The GECO3 compression ratio for Adineta_vaga.AMS_PRJEB1171_v1.cds.all.fa is 20.52%\n",
      "The GZIP compression ratio for Adineta_vaga.AMS_PRJEB1171_v1.cds.all.fa is 27.04%\n",
      "ncrna 737 , total =  77273\n",
      "The GECO3 compression ratio for Adineta_vaga.AMS_PRJEB1171_v1.ncrna.fa is 7.07%\n",
      "The GZIP compression ratio for Adineta_vaga.AMS_PRJEB1171_v1.ncrna.fa is 8.60%\n",
      "adineta_vaga completed successfully!\n",
      "ncrna 2374 , total =  3929598\n",
      "The GECO3 compression ratio for Acanthaster_planci_gca001949145v1.OKI_Apl_1.0.ncrna.fa is 16.40%\n",
      "The GZIP compression ratio for Acanthaster_planci_gca001949145v1.OKI_Apl_1.0.ncrna.fa is 19.99%\n",
      "acanthaster_planci_gca001949145v1 completed successfully!\n",
      "cds 17535 , total =  24179610\n",
      "The GECO3 compression ratio for Aedes_albopictus.AaloF1.cds.all.fa is 21.98%\n",
      "The GZIP compression ratio for Aedes_albopictus.AaloF1.cds.all.fa is 28.06%\n",
      "ncrna 759 , total =  87555\n",
      "The GECO3 compression ratio for Aedes_albopictus.AaloF1.ncrna.fa is 9.38%\n",
      "The GZIP compression ratio for Aedes_albopictus.AaloF1.ncrna.fa is 11.88%\n",
      "aedes_albopictus completed successfully!\n",
      "cds 41860 , total =  69192834\n",
      "The GECO3 compression ratio for Acropora_millepora_gca013753865v1.Amil_v2.1.cds.all.fa is 16.01%\n",
      "The GZIP compression ratio for Acropora_millepora_gca013753865v1.Amil_v2.1.cds.all.fa is 20.06%\n",
      "cds 55607 , total =  75244786\n",
      "The GECO3 compression ratio for Actinia_equina_gca011057435.equina_smartden.arrow4.noredun.cds.all.fa is 17.57%\n",
      "The GZIP compression ratio for Actinia_equina_gca011057435.equina_smartden.arrow4.noredun.cds.all.fa is 22.45%\n",
      "ncrna 8059 , total =  7266283\n",
      "The GECO3 compression ratio for Acropora_millepora_gca013753865v1.Amil_v2.1.ncrna.fa is 13.49%\n",
      "The GZIP compression ratio for Acropora_millepora_gca013753865v1.Amil_v2.1.ncrna.fa is 19.22%\n",
      "acropora_millepora_gca013753865v1 completed successfully!\n",
      "ncrna 7980 , total =  660659\n",
      "The GECO3 compression ratio for Actinia_equina_gca011057435.equina_smartden.arrow4.noredun.ncrna.fa is 6.62%\n",
      "The GZIP compression ratio for Actinia_equina_gca011057435.equina_smartden.arrow4.noredun.ncrna.fa is 10.30%\n",
      "actinia_equina_gca011057435 completed successfully!\n",
      "cds 21401 , total =  40154879\n",
      "The GECO3 compression ratio for Aethina_tumida_gca024364675v1rs.icAetTumi1.1.cds.all.fa is 14.07%\n",
      "The GZIP compression ratio for Aethina_tumida_gca024364675v1rs.icAetTumi1.1.cds.all.fa is 16.45%\n",
      "ncrna 2301 , total =  1520958\n",
      "The GECO3 compression ratio for Aethina_tumida_gca024364675v1rs.icAetTumi1.1.ncrna.fa is 13.62%\n",
      "The GZIP compression ratio for Aethina_tumida_gca024364675v1rs.icAetTumi1.1.ncrna.fa is 18.65%\n",
      "aethina_tumida_gca024364675v1rs completed successfully!\n",
      "cds 22159 , total =  39382127\n",
      "The GECO3 compression ratio for Agrilus_planipennis_gca000699045v2.Apla_2.0.cds.all.fa is 13.74%\n",
      "The GZIP compression ratio for Agrilus_planipennis_gca000699045v2.Apla_2.0.cds.all.fa is 16.37%\n",
      "cds 28391 , total =  60754921\n",
      "The GECO3 compression ratio for Aedes_aegypti_lvpagwg.AaegL5.cds.all.fa is 11.08%\n",
      "The GZIP compression ratio for Aedes_aegypti_lvpagwg.AaegL5.cds.all.fa is 12.50%\n",
      "ncrna 2677 , total =  2764761\n",
      "The GECO3 compression ratio for Agrilus_planipennis_gca000699045v2.Apla_2.0.ncrna.fa is 12.50%\n",
      "The GZIP compression ratio for Agrilus_planipennis_gca000699045v2.Apla_2.0.ncrna.fa is 16.69%\n",
      "agrilus_planipennis_gca000699045v2 completed successfully!\n",
      "ncrna 5068 , total =  3594192\n",
      "The GECO3 compression ratio for Aedes_aegypti_lvpagwg.AaegL5.ncrna.fa is 14.45%\n",
      "The GZIP compression ratio for Aedes_aegypti_lvpagwg.AaegL5.ncrna.fa is 23.52%\n",
      "aedes_aegypti_lvpagwg completed successfully!\n",
      "cds 16819 , total =  28659076\n",
      "The GECO3 compression ratio for Amblyteles_armatorius_gca933228735v1.iyAmbArma2.1.cds.all.fa is 16.77%\n",
      "The GZIP compression ratio for Amblyteles_armatorius_gca933228735v1.iyAmbArma2.1.cds.all.fa is 18.69%\n",
      "ncrna 1453 , total =  790628\n",
      "The GECO3 compression ratio for Amblyteles_armatorius_gca933228735v1.iyAmbArma2.1.ncrna.fa is 11.06%\n",
      "The GZIP compression ratio for Amblyteles_armatorius_gca933228735v1.iyAmbArma2.1.ncrna.fa is 13.94%\n",
      "amblyteles_armatorius_gca933228735v1 completed successfully!\n",
      "cds 14581 , total =  22039878\n",
      "The GECO3 compression ratio for Ancistrocerus_nigricornis_gca916049575v1.iyAncNigr1.1.cds.all.fa is 17.96%\n",
      "The GZIP compression ratio for Ancistrocerus_nigricornis_gca916049575v1.iyAncNigr1.1.cds.all.fa is 20.58%\n",
      "cds 23542 , total =  34625304\n",
      "The GECO3 compression ratio for Amphimedon_queenslandica_gca000090795v2rs.v1.1.cds.all.fa is 19.25%\n",
      "The GZIP compression ratio for Amphimedon_queenslandica_gca000090795v2rs.v1.1.cds.all.fa is 23.55%\n",
      "ncrna 2575 , total =  2485996\n",
      "The GECO3 compression ratio for Ancistrocerus_nigricornis_gca916049575v1.iyAncNigr1.1.ncrna.fa is 6.93%\n",
      "The GZIP compression ratio for Ancistrocerus_nigricornis_gca916049575v1.iyAncNigr1.1.ncrna.fa is 10.49%\n",
      "ancistrocerus_nigricornis_gca916049575v1 completed successfully!\n",
      "cds 18472 , total =  30007884\n",
      "The GECO3 compression ratio for Amyelois_transitella_gca001186105v1rs.ASM118610v1.cds.all.fa is 17.85%\n",
      "The GZIP compression ratio for Amyelois_transitella_gca001186105v1rs.ASM118610v1.cds.all.fa is 21.54%\n",
      "ncrna 772 , total =  243677\n",
      "The GECO3 compression ratio for Amyelois_transitella_gca001186105v1rs.ASM118610v1.ncrna.fa is 18.45%\n",
      "The GZIP compression ratio for Amyelois_transitella_gca001186105v1rs.ASM118610v1.ncrna.fa is 22.77%\n",
      "amyelois_transitella_gca001186105v1rs completed successfully!\n",
      "ncrna 6795 , total =  4078020\n",
      "The GECO3 compression ratio for Amphimedon_queenslandica_gca000090795v2rs.v1.1.ncrna.fa is 13.02%\n",
      "The GZIP compression ratio for Amphimedon_queenslandica_gca000090795v2rs.v1.1.ncrna.fa is 19.44%\n",
      "amphimedon_queenslandica_gca000090795v2rs completed successfully!\n",
      "cds 12571 , total =  21820433\n",
      "The GECO3 compression ratio for Anopheles_albimanus.AalbS2.cds.all.fa is 23.21%\n",
      "The GZIP compression ratio for Anopheles_albimanus.AalbS2.cds.all.fa is 27.15%\n",
      "ncrna 322 , total =  28935\n",
      "The GECO3 compression ratio for Anopheles_albimanus.AalbS2.ncrna.fa is 13.33%\n",
      "The GZIP compression ratio for Anopheles_albimanus.AalbS2.ncrna.fa is 16.48%\n",
      "anopheles_albimanus completed successfully!\n",
      "cds 14037 , total =  23980232\n",
      "The GECO3 compression ratio for Anopheles_arabiensis.AaraD1.cds.all.fa is 21.98%\n",
      "The GZIP compression ratio for Anopheles_arabiensis.AaraD1.cds.all.fa is 25.61%\n",
      "ncrna 378 , total =  35569\n",
      "The GECO3 compression ratio for Anopheles_arabiensis.AaraD1.ncrna.fa is 13.06%\n",
      "The GZIP compression ratio for Anopheles_arabiensis.AaraD1.ncrna.fa is 16.33%\n",
      "anopheles_arabiensis completed successfully!\n",
      "cds 15924 , total =  29088606\n",
      "The GECO3 compression ratio for Anopheles_atroparvus_gca914969975.atroparvus_hifiasm_n225_277Mb.cds.all.fa is 17.99%\n",
      "The GZIP compression ratio for Anopheles_atroparvus_gca914969975.atroparvus_hifiasm_n225_277Mb.cds.all.fa is 20.74%\n",
      "ncrna 1214 , total =  1582588\n",
      "The GECO3 compression ratio for Anopheles_atroparvus_gca914969975.atroparvus_hifiasm_n225_277Mb.ncrna.fa is 1.20%\n",
      "The GZIP compression ratio for Anopheles_atroparvus_gca914969975.atroparvus_hifiasm_n225_277Mb.ncrna.fa is 5.19%\n",
      "anopheles_atroparvus_gca914969975 completed successfully!\n",
      "cds 10960 , total =  18001764\n",
      "The GECO3 compression ratio for Anopheles_christyi.AchrA1.cds.all.fa is 23.93%\n",
      "The GZIP compression ratio for Anopheles_christyi.AchrA1.cds.all.fa is 28.03%\n",
      "ncrna 337 , total =  27764\n",
      "The GECO3 compression ratio for Anopheles_christyi.AchrA1.ncrna.fa is 12.48%\n",
      "The GZIP compression ratio for Anopheles_christyi.AchrA1.ncrna.fa is 15.40%\n",
      "anopheles_christyi completed successfully!\n",
      "cds 14510 , total =  20221507\n",
      "The GECO3 compression ratio for Anopheles_coluzzii.AcolM1.cds.all.fa is 22.83%\n",
      "The GZIP compression ratio for Anopheles_coluzzii.AcolM1.cds.all.fa is 27.63%\n",
      "ncrna 331 , total =  30784\n",
      "The GECO3 compression ratio for Anopheles_coluzzii.AcolM1.ncrna.fa is 12.87%\n",
      "The GZIP compression ratio for Anopheles_coluzzii.AcolM1.ncrna.fa is 15.96%\n",
      "anopheles_coluzzii completed successfully!\n",
      "cds 14694 , total =  26037846\n",
      "The GECO3 compression ratio for Anopheles_coluzzii_ngousso.AcolN1.cds.all.fa is 20.51%\n",
      "The GZIP compression ratio for Anopheles_coluzzii_ngousso.AcolN1.cds.all.fa is 24.13%\n",
      "ncrna 469 , total =  73456\n",
      "The GECO3 compression ratio for Anopheles_coluzzii_ngousso.AcolN1.ncrna.fa is 8.26%\n",
      "The GZIP compression ratio for Anopheles_coluzzii_ngousso.AcolN1.ncrna.fa is 10.78%\n",
      "anopheles_coluzzii_ngousso completed successfully!\n",
      "cds 14385 , total =  21352169\n",
      "The GECO3 compression ratio for Anopheles_culicifacies.AculA1.cds.all.fa is 23.67%\n",
      "The GZIP compression ratio for Anopheles_culicifacies.AculA1.cds.all.fa is 28.06%\n",
      "ncrna 371 , total =  32641\n",
      "The GECO3 compression ratio for Anopheles_culicifacies.AculA1.ncrna.fa is 13.29%\n",
      "The GZIP compression ratio for Anopheles_culicifacies.AculA1.ncrna.fa is 16.29%\n",
      "anopheles_culicifacies completed successfully!\n",
      "cds 10602 , total =  18405614\n",
      "The GECO3 compression ratio for Anopheles_darlingi.AdarC3.cds.all.fa is 23.76%\n",
      "The GZIP compression ratio for Anopheles_darlingi.AdarC3.cds.all.fa is 27.80%\n",
      "ncrna 449 , total =  42531\n",
      "The GECO3 compression ratio for Anopheles_darlingi.AdarC3.ncrna.fa is 13.88%\n",
      "The GZIP compression ratio for Anopheles_darlingi.AdarC3.ncrna.fa is 16.94%\n",
      "anopheles_darlingi completed successfully!\n",
      "cds 32774 , total =  58858667\n",
      "The GECO3 compression ratio for Anneissia_japonica_gca011630105v1.ASM1163010v1.cds.all.fa is 15.13%\n",
      "The GZIP compression ratio for Anneissia_japonica_gca011630105v1.ASM1163010v1.cds.all.fa is 16.71%\n",
      "ncrna 4798 , total =  3738576\n",
      "The GECO3 compression ratio for Anneissia_japonica_gca011630105v1.ASM1163010v1.ncrna.fa is 17.11%\n",
      "The GZIP compression ratio for Anneissia_japonica_gca011630105v1.ASM1163010v1.ncrna.fa is 21.95%\n",
      "anneissia_japonica_gca011630105v1 completed successfully!\n",
      "cds 11998 , total =  20396424\n",
      "The GECO3 compression ratio for Anopheles_epiroticus.AepiE1.cds.all.fa is 23.76%\n",
      "The GZIP compression ratio for Anopheles_epiroticus.AepiE1.cds.all.fa is 27.84%\n",
      "cds 13172 , total =  21575446\n",
      "The GECO3 compression ratio for Anopheles_dirus.AdirW1.cds.all.fa is 23.16%\n",
      "The GZIP compression ratio for Anopheles_dirus.AdirW1.cds.all.fa is 27.30%\n",
      "ncrna 366 , total =  33153\n",
      "The GECO3 compression ratio for Anopheles_epiroticus.AepiE1.ncrna.fa is 12.88%\n",
      "The GZIP compression ratio for Anopheles_epiroticus.AepiE1.ncrna.fa is 15.85%\n",
      "anopheles_epiroticus completed successfully!\n",
      "ncrna 325 , total =  32250\n",
      "The GECO3 compression ratio for Anopheles_dirus.AdirW1.ncrna.fa is 13.62%\n",
      "The GZIP compression ratio for Anopheles_dirus.AdirW1.ncrna.fa is 17.27%\n",
      "anopheles_dirus completed successfully!\n",
      "cds 13029 , total =  21247970\n",
      "The GECO3 compression ratio for Anopheles_farauti.AfarF2.cds.all.fa is 23.22%\n",
      "The GZIP compression ratio for Anopheles_farauti.AfarF2.cds.all.fa is 27.73%\n",
      "ncrna 361 , total =  34187\n",
      "The GECO3 compression ratio for Anopheles_farauti.AfarF2.ncrna.fa is 12.79%\n",
      "The GZIP compression ratio for Anopheles_farauti.AfarF2.ncrna.fa is 15.90%\n",
      "anopheles_farauti completed successfully!\n",
      "cds 14865 , total =  16497483\n",
      "The GECO3 compression ratio for Anopheles_maculatus.AmacM1.cds.all.fa is 23.96%\n",
      "The GZIP compression ratio for Anopheles_maculatus.AmacM1.cds.all.fa is 28.01%\n",
      "ncrna 177 , total =  15673\n",
      "The GECO3 compression ratio for Anopheles_maculatus.AmacM1.ncrna.fa is 18.61%\n",
      "The GZIP compression ratio for Anopheles_maculatus.AmacM1.ncrna.fa is 23.08%\n",
      "anopheles_maculatus completed successfully!\n",
      "cds 14621 , total =  25656806\n",
      "The GECO3 compression ratio for Anopheles_funestus.AfunF3.cds.all.fa is 20.54%\n",
      "The GZIP compression ratio for Anopheles_funestus.AfunF3.cds.all.fa is 24.55%\n",
      "cds 59724 , total =  109091320\n",
      "The GECO3 compression ratio for Amphibalanus_amphitrite_gca019059575v1.NRLGWU_Aamphi_draft.cds.all.fa is 12.84%\n",
      "The GZIP compression ratio for Amphibalanus_amphitrite_gca019059575v1.NRLGWU_Aamphi_draft.cds.all.fa is 13.32%\n",
      "ncrna 1082 , total =  560948\n",
      "The GECO3 compression ratio for Anopheles_funestus.AfunF3.ncrna.fa is 2.66%\n",
      "The GZIP compression ratio for Anopheles_funestus.AfunF3.ncrna.fa is 4.88%\n",
      "anopheles_funestus completed successfully!\n",
      "cds 15125 , total =  26719060\n",
      "The GECO3 compression ratio for Anopheles_gambiae.AgamP4.cds.all.fa is 20.25%\n",
      "The GZIP compression ratio for Anopheles_gambiae.AgamP4.cds.all.fa is 23.76%\n",
      "ncrna 729 , total =  139452\n",
      "The GECO3 compression ratio for Anopheles_gambiae.AgamP4.ncrna.fa is 6.33%\n",
      "The GZIP compression ratio for Anopheles_gambiae.AgamP4.ncrna.fa is 8.82%\n",
      "anopheles_gambiae completed successfully!\n",
      "ncrna 4647 , total =  4566067\n",
      "The GECO3 compression ratio for Amphibalanus_amphitrite_gca019059575v1.NRLGWU_Aamphi_draft.ncrna.fa is 9.19%\n",
      "The GZIP compression ratio for Amphibalanus_amphitrite_gca019059575v1.NRLGWU_Aamphi_draft.ncrna.fa is 20.87%\n",
      "amphibalanus_amphitrite_gca019059575v1 completed successfully!\n",
      "cds 15389 , total =  21261962\n",
      "The GECO3 compression ratio for Anopheles_melas.AmelC2.cds.all.fa is 22.45%\n",
      "The GZIP compression ratio for Anopheles_melas.AmelC2.cds.all.fa is 27.72%\n",
      "ncrna 410 , total =  35581\n",
      "The GECO3 compression ratio for Anopheles_melas.AmelC2.ncrna.fa is 12.75%\n",
      "The GZIP compression ratio for Anopheles_melas.AmelC2.ncrna.fa is 15.58%\n",
      "anopheles_melas completed successfully!\n",
      "cds 13343 , total =  21088146\n",
      "The GECO3 compression ratio for Anopheles_merus.AmerM2.cds.all.fa is 22.55%\n",
      "The GZIP compression ratio for Anopheles_merus.AmerM2.cds.all.fa is 27.04%\n",
      "ncrna 428 , total =  39174\n",
      "The GECO3 compression ratio for Anopheles_merus.AmerM2.ncrna.fa is 12.49%\n",
      "The GZIP compression ratio for Anopheles_merus.AmerM2.ncrna.fa is 15.29%\n",
      "anopheles_merus completed successfully!\n",
      "cds 13056 , total =  21347690\n",
      "The GECO3 compression ratio for Anopheles_minimus.AminM1.cds.all.fa is 23.78%\n",
      "The GZIP compression ratio for Anopheles_minimus.AminM1.cds.all.fa is 27.68%\n",
      "ncrna 328 , total =  30340\n",
      "The GECO3 compression ratio for Anopheles_minimus.AminM1.ncrna.fa is 14.05%\n",
      "The GZIP compression ratio for Anopheles_minimus.AminM1.ncrna.fa is 17.53%\n",
      "anopheles_minimus completed successfully!\n",
      "cds 12950 , total =  17993758\n",
      "The GECO3 compression ratio for Anopheles_sinensis.AsinS2.cds.all.fa is 23.26%\n",
      "The GZIP compression ratio for Anopheles_sinensis.AsinS2.cds.all.fa is 27.86%\n",
      "ncrna 301 , total =  27035\n",
      "The GECO3 compression ratio for Anopheles_sinensis.AsinS2.ncrna.fa is 13.96%\n",
      "The GZIP compression ratio for Anopheles_sinensis.AsinS2.ncrna.fa is 17.19%\n",
      "anopheles_sinensis completed successfully!\n",
      "cds 13900 , total =  22714672\n",
      "The GECO3 compression ratio for Anopheles_quadriannulatus.AquaS1.cds.all.fa is 22.45%\n",
      "The GZIP compression ratio for Anopheles_quadriannulatus.AquaS1.cds.all.fa is 26.25%\n",
      "ncrna 404 , total =  36462\n",
      "The GECO3 compression ratio for Anopheles_quadriannulatus.AquaS1.ncrna.fa is 12.43%\n",
      "The GZIP compression ratio for Anopheles_quadriannulatus.AquaS1.ncrna.fa is 15.51%\n",
      "anopheles_quadriannulatus completed successfully!\n",
      "cds 19352 , total =  21892085\n",
      "The GECO3 compression ratio for Anopheles_sinensis_china.AsinC2.cds.all.fa is 23.77%\n",
      "The GZIP compression ratio for Anopheles_sinensis_china.AsinC2.cds.all.fa is 27.83%\n",
      "ncrna 342 , total =  33564\n",
      "The GECO3 compression ratio for Anopheles_sinensis_china.AsinC2.ncrna.fa is 12.94%\n",
      "The GZIP compression ratio for Anopheles_sinensis_china.AsinC2.ncrna.fa is 16.24%\n",
      "anopheles_sinensis_china completed successfully!\n",
      "cds 13561 , total =  21385271\n",
      "The GECO3 compression ratio for Anopheles_stephensi.AsteS1.cds.all.fa is 23.56%\n",
      "The GZIP compression ratio for Anopheles_stephensi.AsteS1.cds.all.fa is 27.67%\n",
      "ncrna 275 , total =  28335\n",
      "The GECO3 compression ratio for Anopheles_stephensi.AsteS1.ncrna.fa is 14.77%\n",
      "The GZIP compression ratio for Anopheles_stephensi.AsteS1.ncrna.fa is 18.63%\n",
      "anopheles_stephensi completed successfully!\n",
      "cds 11789 , total =  20279728\n",
      "The GECO3 compression ratio for Anopheles_stephensi_indian.AsteI2.cds.all.fa is 23.76%\n",
      "The GZIP compression ratio for Anopheles_stephensi_indian.AsteI2.cds.all.fa is 27.82%\n",
      "ncrna 400 , total =  37118\n",
      "The GECO3 compression ratio for Anopheles_stephensi_indian.AsteI2.ncrna.fa is 13.24%\n",
      "The GZIP compression ratio for Anopheles_stephensi_indian.AsteI2.ncrna.fa is 16.29%\n",
      "anopheles_stephensi_indian completed successfully!\n",
      "cds 20632 , total =  35920184\n",
      "The GECO3 compression ratio for Anoplophora_glabripennis.Agla_2.0.cds.all.fa is 15.82%\n",
      "The GZIP compression ratio for Anoplophora_glabripennis.Agla_2.0.cds.all.fa is 18.92%\n",
      "ncrna 1267 , total =  745344\n",
      "The GECO3 compression ratio for Anoplophora_glabripennis.Agla_2.0.ncrna.fa is 11.47%\n",
      "The GZIP compression ratio for Anoplophora_glabripennis.Agla_2.0.ncrna.fa is 16.77%\n",
      "anoplophora_glabripennis completed successfully!\n",
      "cds 18628 , total =  34325624\n",
      "The GECO3 compression ratio for Apis_florea_gca000184785v2rs.Aflo_1.0.cds.all.fa is 13.85%\n",
      "The GZIP compression ratio for Apis_florea_gca000184785v2rs.Aflo_1.0.cds.all.fa is 15.66%\n",
      "cds 20617 , total =  42166592\n",
      "The GECO3 compression ratio for Aphidius_gifuensis_gca014905175v1.ASM1490517v1.cds.all.fa is 14.75%\n",
      "The GZIP compression ratio for Aphidius_gifuensis_gca014905175v1.ASM1490517v1.cds.all.fa is 14.55%\n",
      "ncrna 1126 , total =  887492\n",
      "The GECO3 compression ratio for Aphidius_gifuensis_gca014905175v1.ASM1490517v1.ncrna.fa is 14.24%\n",
      "The GZIP compression ratio for Aphidius_gifuensis_gca014905175v1.ASM1490517v1.ncrna.fa is 18.45%\n",
      "aphidius_gifuensis_gca014905175v1 completed successfully!\n",
      "cds 20012 , total =  40673629\n",
      "The GECO3 compression ratio for Apis_dorsata_gca000469605v1rs.Apis_dorsata_1.3.cds.all.fa is 12.39%\n",
      "The GZIP compression ratio for Apis_dorsata_gca000469605v1rs.Apis_dorsata_1.3.cds.all.fa is 13.67%\n",
      "ncrna 2501 , total =  2679861\n",
      "The GECO3 compression ratio for Apis_florea_gca000184785v2rs.Aflo_1.0.ncrna.fa is 18.89%\n",
      "The GZIP compression ratio for Apis_florea_gca000184785v2rs.Aflo_1.0.ncrna.fa is 22.63%\n",
      "apis_florea_gca000184785v2rs completed successfully!\n",
      "cds 23731 , total =  49784692\n",
      "The GECO3 compression ratio for Anthonomus_grandis_gca022605725v3rs.icAntGran1.3.cds.all.fa is 11.90%\n",
      "The GZIP compression ratio for Anthonomus_grandis_gca022605725v3rs.icAntGran1.3.cds.all.fa is 13.81%\n",
      "ncrna 2557 , total =  2415779\n",
      "The GECO3 compression ratio for Apis_dorsata_gca000469605v1rs.Apis_dorsata_1.3.ncrna.fa is 18.16%\n",
      "The GZIP compression ratio for Apis_dorsata_gca000469605v1rs.Apis_dorsata_1.3.ncrna.fa is 21.85%\n",
      "apis_dorsata_gca000469605v1rs completed successfully!\n",
      "ncrna 5544 , total =  8470618\n",
      "The GECO3 compression ratio for Anthonomus_grandis_gca022605725v3rs.icAntGran1.3.ncrna.fa is 10.56%\n",
      "The GZIP compression ratio for Anthonomus_grandis_gca022605725v3rs.icAntGran1.3.ncrna.fa is 15.38%\n",
      "anthonomus_grandis_gca022605725v3rs completed successfully!\n",
      "cds 26656 , total =  45552316\n",
      "The GECO3 compression ratio for Aplysia_californica_gca000002075v2.AplCal3.0.cds.all.fa is 17.62%\n",
      "The GZIP compression ratio for Aplysia_californica_gca000002075v2.AplCal3.0.cds.all.fa is 19.80%\n",
      "ncrna 2348 , total =  2033633\n",
      "The GECO3 compression ratio for Aplysia_californica_gca000002075v2.AplCal3.0.ncrna.fa is 18.55%\n",
      "The GZIP compression ratio for Aplysia_californica_gca000002075v2.AplCal3.0.ncrna.fa is 23.19%\n",
      "aplysia_californica_gca000002075v2 completed successfully!\n",
      "cds 10657 , total =  18276831\n",
      "The GECO3 compression ratio for Atta_cephalotes.Attacep1.0.cds.all.fa is 23.83%\n",
      "The GZIP compression ratio for Atta_cephalotes.Attacep1.0.cds.all.fa is 27.80%\n",
      "cds 23470 , total =  57396111\n",
      "The GECO3 compression ratio for Apis_mellifera.Amel_HAv3.1.cds.all.fa is 9.72%\n",
      "The GZIP compression ratio for Apis_mellifera.Amel_HAv3.1.cds.all.fa is 11.08%\n",
      "ncrna 449 , total =  57908\n",
      "The GECO3 compression ratio for Atta_cephalotes.Attacep1.0.ncrna.fa is 17.03%\n",
      "The GZIP compression ratio for Atta_cephalotes.Attacep1.0.ncrna.fa is 20.94%\n",
      "atta_cephalotes completed successfully!\n",
      "ncrna 3948 , total =  8108030\n",
      "The GECO3 compression ratio for Apis_mellifera.Amel_HAv3.1.ncrna.fa is 13.14%\n",
      "The GZIP compression ratio for Apis_mellifera.Amel_HAv3.1.ncrna.fa is 15.44%\n",
      "apis_mellifera completed successfully!\n",
      "cds 22003 , total =  46090043\n",
      "The GECO3 compression ratio for Asterias_rubens_gca902459465v3.eAstRub1.3.cds.all.fa is 14.82%\n",
      "The GZIP compression ratio for Asterias_rubens_gca902459465v3.eAstRub1.3.cds.all.fa is 17.50%\n",
      "cds 20571 , total =  36366230\n",
      "The GECO3 compression ratio for Athalia_rosae_gca917208135v1.iyAthRosa1.1.cds.all.fa is 14.52%\n",
      "The GZIP compression ratio for Athalia_rosae_gca917208135v1.iyAthRosa1.1.cds.all.fa is 16.34%\n",
      "cds 57968 , total =  69439114\n",
      "The GECO3 compression ratio for Ascaris_suum.ASM18702v3.cds.all.fa is 9.84%\n",
      "The GZIP compression ratio for Ascaris_suum.ASM18702v3.cds.all.fa is 10.00%\n",
      "ncrna 2373 , total =  1797923\n",
      "The GECO3 compression ratio for Athalia_rosae_gca917208135v1.iyAthRosa1.1.ncrna.fa is 12.51%\n",
      "The GZIP compression ratio for Athalia_rosae_gca917208135v1.iyAthRosa1.1.ncrna.fa is 16.50%\n",
      "athalia_rosae_gca917208135v1 completed successfully!\n",
      "ncrna 1750 , total =  3630527\n",
      "The GECO3 compression ratio for Asterias_rubens_gca902459465v3.eAstRub1.3.ncrna.fa is 14.06%\n",
      "The GZIP compression ratio for Asterias_rubens_gca902459465v3.eAstRub1.3.ncrna.fa is 19.05%\n",
      "asterias_rubens_gca902459465v3 completed successfully!\n",
      "ncrna 1579 , total =  6399310\n",
      "The GECO3 compression ratio for Ascaris_suum.ASM18702v3.ncrna.fa is 11.32%\n",
      "The GZIP compression ratio for Ascaris_suum.ASM18702v3.ncrna.fa is 15.94%\n",
      "ascaris_suum completed successfully!\n",
      "cds 22113 , total =  49572306\n",
      "The GECO3 compression ratio for Athalia_rosae_gca000344095v2.Aros_2.0.cds.all.fa is 11.27%\n",
      "The GZIP compression ratio for Athalia_rosae_gca000344095v2.Aros_2.0.cds.all.fa is 12.94%\n",
      "ncrna 857 , total =  847938\n",
      "The GECO3 compression ratio for Athalia_rosae_gca000344095v2.Aros_2.0.ncrna.fa is 14.44%\n",
      "The GZIP compression ratio for Athalia_rosae_gca000344095v2.Aros_2.0.ncrna.fa is 17.95%\n",
      "athalia_rosae_gca000344095v2 completed successfully!\n",
      "cds 20833 , total =  41022404\n",
      "The GECO3 compression ratio for Bactrocera_dorsalis_gca000789215v2.ASM78921v2.cds.all.fa is 14.06%\n",
      "The GZIP compression ratio for Bactrocera_dorsalis_gca000789215v2.ASM78921v2.cds.all.fa is 15.82%\n",
      "ncrna 1032 , total =  852436\n",
      "The GECO3 compression ratio for Bactrocera_dorsalis_gca000789215v2.ASM78921v2.ncrna.fa is 19.45%\n",
      "The GZIP compression ratio for Bactrocera_dorsalis_gca000789215v2.ASM78921v2.ncrna.fa is 23.27%\n",
      "bactrocera_dorsalis_gca000789215v2 completed successfully!\n",
      "cds 13510 , total =  18726334\n",
      "The GECO3 compression ratio for Belgica_antarctica.ASM77530v1.cds.all.fa is 23.81%\n",
      "The GZIP compression ratio for Belgica_antarctica.ASM77530v1.cds.all.fa is 27.92%\n",
      "ncrna 293 , total =  26327\n",
      "The GECO3 compression ratio for Belgica_antarctica.ASM77530v1.ncrna.fa is 15.38%\n",
      "The GZIP compression ratio for Belgica_antarctica.ASM77530v1.ncrna.fa is 21.35%\n",
      "belgica_antarctica completed successfully!\n",
      "cds 22638 , total =  48949840\n",
      "The GECO3 compression ratio for Bactrocera_latifrons_gca001853355v1.ASM185335v1.cds.all.fa is 12.65%\n",
      "The GZIP compression ratio for Bactrocera_latifrons_gca001853355v1.ASM185335v1.cds.all.fa is 14.04%\n",
      "ncrna 1166 , total =  798947\n",
      "The GECO3 compression ratio for Bactrocera_latifrons_gca001853355v1.ASM185335v1.ncrna.fa is 18.42%\n",
      "The GZIP compression ratio for Bactrocera_latifrons_gca001853355v1.ASM185335v1.ncrna.fa is 22.37%\n",
      "bactrocera_latifrons_gca001853355v1 completed successfully!\n",
      "cds 25557 , total =  55044358\n",
      "The GECO3 compression ratio for Bactrocera_neohumeralis_gca024586455v2rs.APGP_CSIRO_Bneo_wtdbg2_racon_allhic_juicebox.fasta_v2.cds.all.fa is 12.10%\n",
      "The GZIP compression ratio for Bactrocera_neohumeralis_gca024586455v2rs.APGP_CSIRO_Bneo_wtdbg2_racon_allhic_juicebox.fasta_v2.cds.all.fa is 13.45%\n",
      "cds 23950 , total =  49534721\n",
      "The GECO3 compression ratio for Bactrocera_tryoni_gca016617805v2.CSIRO_BtryS06_freeze1.cds.all.fa is 13.14%\n",
      "The GZIP compression ratio for Bactrocera_tryoni_gca016617805v2.CSIRO_BtryS06_freeze1.cds.all.fa is 14.83%\n",
      "cds 26928 , total =  43505800\n",
      "The GECO3 compression ratio for Bemisia_tabaci_asiaii5.ASIAII5_n227_616Mb.cds.all.fa is 11.87%\n",
      "The GZIP compression ratio for Bemisia_tabaci_asiaii5.ASIAII5_n227_616Mb.cds.all.fa is 13.66%\n",
      "ncrnacds 27923 , total =  39709166\n",
      "The GECO3 compression ratio for Bemisia_tabaci_ssa1nig.SSA1SG1_Nigeria_n4493_656Mb.cds.all.fa is 12.28%\n",
      "The GZIP compression ratio for Bemisia_tabaci_ssa1nig.SSA1SG1_Nigeria_n4493_656Mb.cds.all.fa is 14.82%\n",
      " 3110 , total =  4172929\n",
      "The GECO3 compression ratio for Bactrocera_neohumeralis_gca024586455v2rs.APGP_CSIRO_Bneo_wtdbg2_racon_allhic_juicebox.fasta_v2.ncrna.fa is 14.55%\n",
      "The GZIP compression ratio for Bactrocera_neohumeralis_gca024586455v2rs.APGP_CSIRO_Bneo_wtdbg2_racon_allhic_juicebox.fasta_v2.ncrna.fa is 19.29%\n",
      "bactrocera_neohumeralis_gca024586455v2rs completed successfully!\n",
      "cds 26825 , total =  31819405\n",
      "The GECO3 compression ratio for Bemisia_tabaci_ssa2nig.SSA2_Nigeria_n785_625mb.cds.all.fa is 12.83%\n",
      "The GZIP compression ratio for Bemisia_tabaci_ssa2nig.SSA2_Nigeria_n785_625mb.cds.all.fa is 15.11%\n",
      "ncrna 1722 , total =  1314770\n",
      "The GECO3 compression ratio for Bactrocera_tryoni_gca016617805v2.CSIRO_BtryS06_freeze1.ncrna.fa is 15.62%\n",
      "The GZIP compression ratio for Bactrocera_tryoni_gca016617805v2.CSIRO_BtryS06_freeze1.ncrna.fa is 19.87%\n",
      "bactrocera_tryoni_gca016617805v2 completed successfully!\n",
      "ncrna 1898 , total =  3376376\n",
      "The GECO3 compression ratio for Bemisia_tabaci_asiaii5.ASIAII5_n227_616Mb.ncrna.fa is 17.62%\n",
      "The GZIP compression ratio for Bemisia_tabaci_asiaii5.ASIAII5_n227_616Mb.ncrna.fa is 21.65%\n",
      "bemisia_tabaci_asiaii5 completed successfully!\n",
      "ncrna 1937 , total =  3349695\n",
      "The GECO3 compression ratio for Bemisia_tabaci_ssa1nig.SSA1SG1_Nigeria_n4493_656Mb.ncrna.fa is 17.18%\n",
      "The GZIP compression ratio for Bemisia_tabaci_ssa1nig.SSA1SG1_Nigeria_n4493_656Mb.ncrna.fa is 21.25%\n",
      "bemisia_tabaci_ssa1nig completed successfully!\n",
      "ncrna 2102 , total =  3421436\n",
      "The GECO3 compression ratio for Bemisia_tabaci_ssa2nig.SSA2_Nigeria_n785_625mb.ncrna.fa is 18.06%\n",
      "The GZIP compression ratio for Bemisia_tabaci_ssa2nig.SSA2_Nigeria_n785_625mb.ncrna.fa is 22.05%\n",
      "bemisia_tabaci_ssa2nig completed successfully!\n",
      "cds 28022 , total =  46418084\n",
      "The GECO3 compression ratio for Bemisia_tabaci_ssa1ug.SSA1_SG1_Uganda_n955_657Mb.cds.all.fa is 11.18%\n",
      "The GZIP compression ratio for Bemisia_tabaci_ssa1ug.SSA1_SG1_Uganda_n955_657Mb.cds.all.fa is 13.46%\n",
      "cds 27844 , total =  32104054\n",
      "The GECO3 compression ratio for Bemisia_tabaci_ssa3nig.SSA3_Nigeria_2278n_648mb.cds.all.fa is 12.75%\n",
      "The GZIP compression ratio for Bemisia_tabaci_ssa3nig.SSA3_Nigeria_2278n_648mb.cds.all.fa is 15.11%\n",
      "cds 23614 , total =  22384459\n",
      "The GECO3 compression ratio for Bemisia_tabaci_uganda1.Uganda1_n5713_630Mb.cds.all.fa is 13.22%\n",
      "The GZIP compression ratio for Bemisia_tabaci_uganda1.Uganda1_n5713_630Mb.cds.all.fa is 16.32%\n",
      "ncrna 1710 , total =  3276655\n",
      "The GECO3 compression ratio for Bemisia_tabaci_ssa1ug.SSA1_SG1_Uganda_n955_657Mb.ncrna.fa is 17.28%\n",
      "The GZIP compression ratio for Bemisia_tabaci_ssa1ug.SSA1_SG1_Uganda_n955_657Mb.ncrna.fa is 21.43%\n",
      "bemisia_tabaci_ssa1ug completed successfully!\n",
      "ncrna 1402 , total =  2351860\n",
      "The GECO3 compression ratio for Bemisia_tabaci_uganda1.Uganda1_n5713_630Mb.ncrna.fa is 18.23%\n",
      "The GZIP compression ratio for Bemisia_tabaci_uganda1.Uganda1_n5713_630Mb.ncrna.fa is 22.32%\n",
      "bemisia_tabaci_uganda1 completed successfully!\n",
      "ncrna 2138 , total =  3579593\n",
      "The GECO3 compression ratio for Bemisia_tabaci_ssa3nig.SSA3_Nigeria_2278n_648mb.ncrna.fa is 18.16%\n",
      "The GZIP compression ratio for Bemisia_tabaci_ssa3nig.SSA3_Nigeria_2278n_648mb.ncrna.fa is 22.15%\n",
      "bemisia_tabaci_ssa3nig completed successfully!\n",
      "cds 21603 , total =  37635259\n",
      "The GECO3 compression ratio for Bicyclus_anynana_gca900239965v1rs.Bicyclus_anynana_v1.2.cds.all.fa is 15.49%\n",
      "The GZIP compression ratio for Bicyclus_anynana_gca900239965v1rs.Bicyclus_anynana_v1.2.cds.all.fa is 17.96%\n",
      "ncrna 1329 , total =  671409\n",
      "The GECO3 compression ratio for Bicyclus_anynana_gca900239965v1rs.Bicyclus_anynana_v1.2.ncrna.fa is 18.16%\n",
      "The GZIP compression ratio for Bicyclus_anynana_gca900239965v1rs.Bicyclus_anynana_v1.2.ncrna.fa is 23.36%\n",
      "bicyclus_anynana_gca900239965v1rs completed successfully!\n",
      "cds 22091 , total =  44640897\n",
      "The GECO3 compression ratio for Bombus_terrestris.Bter_1.0.cds.all.fa is 11.70%\n",
      "The GZIP compression ratio for Bombus_terrestris.Bter_1.0.cds.all.fa is 12.98%\n",
      "cds 19224 , total =  36486884\n",
      "The GECO3 compression ratio for Bombyx_mandarina_gca003987935v1rs.ASM398793v1.cds.all.fa is 14.56%\n",
      "The GZIP compression ratio for Bombyx_mandarina_gca003987935v1rs.ASM398793v1.cds.all.fa is 17.00%\n",
      "ncrna 1857 , total =  3149194\n",
      "The GECO3 compression ratio for Bombus_terrestris.Bter_1.0.ncrna.fa is 14.52%\n",
      "The GZIP compression ratio for Bombus_terrestris.Bter_1.0.ncrna.fa is 17.59%\n",
      "bombus_terrestris completed successfully!\n",
      "ncrna 996 , total =  316581\n",
      "The GECO3 compression ratio for Bombyx_mandarina_gca003987935v1rs.ASM398793v1.ncrna.fa is 14.85%\n",
      "The GZIP compression ratio for Bombyx_mandarina_gca003987935v1rs.ASM398793v1.ncrna.fa is 19.77%\n",
      "bombyx_mandarina_gca003987935v1rs completed successfully!\n",
      "cds 24471 , total =  53430089\n",
      "The GECO3 compression ratio for Bombus_impatiens.BIMP_2.2.cds.all.fa is 10.40%\n",
      "The GZIP compression ratio for Bombus_impatiens.BIMP_2.2.cds.all.fa is 11.14%\n",
      "cds 36838 , total =  54573566\n",
      "The GECO3 compression ratio for Biomphalaria_glabrata.BglaB1.cds.all.fa is 16.42%\n",
      "The GZIP compression ratio for Biomphalaria_glabrata.BglaB1.cds.all.fa is 17.90%\n",
      "ncrna 2768 , total =  3874671\n",
      "The GECO3 compression ratio for Bombus_impatiens.BIMP_2.2.ncrna.fa is 18.11%\n",
      "The GZIP compression ratio for Bombus_impatiens.BIMP_2.2.ncrna.fa is 21.62%\n",
      "bombus_impatiens completed successfully!\n",
      "cds 27776 , total =  60699996\n",
      "The GECO3 compression ratio for Bombus_huntii_gca024542735v1rs.iyBomHunt1.1.cds.all.fa is 9.43%\n",
      "The GZIP compression ratio for Bombus_huntii_gca024542735v1rs.iyBomHunt1.1.cds.all.fa is 10.48%\n",
      "ncrna 6823 , total =  3569419\n",
      "The GECO3 compression ratio for Biomphalaria_glabrata.BglaB1.ncrna.fa is 19.68%\n",
      "The GZIP compression ratio for Biomphalaria_glabrata.BglaB1.ncrna.fa is 23.68%\n",
      "biomphalaria_glabrata completed successfully!\n",
      "cds 35682 , total =  53474451\n",
      "The GECO3 compression ratio for Bombus_terrestris_gca910591885v2.iyBomTerr1.2.cds.all.fa is 11.67%\n",
      "The GZIP compression ratio for Bombus_terrestris_gca910591885v2.iyBomTerr1.2.cds.all.fa is 12.19%\n",
      "cds 24285 , total =  54048579\n",
      "The GECO3 compression ratio for Bombus_vancouverensis_gca011952275v1rs.Bvanc_JDL1245.cds.all.fa is 10.33%\n",
      "The GZIP compression ratio for Bombus_vancouverensis_gca011952275v1rs.Bvanc_JDL1245.cds.all.fa is 11.91%\n",
      "ncrna 2212 , total =  2380971\n",
      "The GECO3 compression ratio for Bombus_vancouverensis_gca011952275v1rs.Bvanc_JDL1245.ncrna.fa is 15.79%\n",
      "The GZIP compression ratio for Bombus_vancouverensis_gca011952275v1rs.Bvanc_JDL1245.ncrna.fa is 20.83%\n",
      "bombus_vancouverensis_gca011952275v1rs completed successfully!\n",
      "ncrna 6790 , total =  4360819\n",
      "The GECO3 compression ratio for Bombus_terrestris_gca910591885v2.iyBomTerr1.2.ncrna.fa is 19.68%\n",
      "The GZIP compression ratio for Bombus_terrestris_gca910591885v2.iyBomTerr1.2.ncrna.fa is 24.23%\n",
      "bombus_terrestris_gca910591885v2 completed successfully!\n",
      "ncrna 5134 , total =  15683957\n",
      "The GECO3 compression ratio for Bombus_huntii_gca024542735v1rs.iyBomHunt1.1.ncrna.fa is 10.80%\n",
      "The GZIP compression ratio for Bombus_huntii_gca024542735v1rs.iyBomHunt1.1.ncrna.fa is 14.42%\n",
      "bombus_huntii_gca024542735v1rs completed successfully!\n",
      "cds 15918 , total =  23154050\n",
      "The GECO3 compression ratio for Brugia_malayi.Bmal-4.0.cds.all.fa is 16.23%\n",
      "The GZIP compression ratio for Brugia_malayi.Bmal-4.0.cds.all.fa is 18.55%\n",
      "ncrna 573 , total =  59255\n",
      "The GECO3 compression ratio for Brugia_malayi.Bmal-4.0.ncrna.fa is 20.37%\n",
      "The GZIP compression ratio for Brugia_malayi.Bmal-4.0.ncrna.fa is 25.42%\n",
      "brugia_malayi completed successfully!\n",
      "cds 36011 , total =  36442285\n",
      "The GECO3 compression ratio for Branchiostoma_lanceolatum.BraLan2.cds.all.fa is 21.93%\n",
      "The GZIP compression ratio for Branchiostoma_lanceolatum.BraLan2.cds.all.fa is 27.38%\n",
      "cds 27309 , total =  56574854\n",
      "The GECO3 compression ratio for Bombyx_mori.Bmori_2016v1.0.cds.all.fa is 11.17%\n",
      "The GZIP compression ratio for Bombyx_mori.Bmori_2016v1.0.cds.all.fa is 13.38%\n",
      "Directory or file not found for branchiostoma_lanceolatum in ncrna folder: 550 Failed to change directory.\n",
      "branchiostoma_lanceolatum completed successfully!\n",
      "cds 24602 , total =  32481317\n",
      "The GECO3 compression ratio for Caenorhabditis_briggsae.CB4.cds.all.fa is 19.22%\n",
      "The GZIP compression ratio for Caenorhabditis_briggsae.CB4.cds.all.fa is 21.98%\n",
      "ncrna 3762 , total =  3189998\n",
      "The GECO3 compression ratio for Bombyx_mori.Bmori_2016v1.0.ncrna.fa is 13.68%\n",
      "The GZIP compression ratio for Bombyx_mori.Bmori_2016v1.0.ncrna.fa is 18.35%\n",
      "bombyx_mori completed successfully!\n",
      "ncrna 1285 , total =  134841\n",
      "The GECO3 compression ratio for Caenorhabditis_briggsae.CB4.ncrna.fa is 11.47%\n",
      "The GZIP compression ratio for Caenorhabditis_briggsae.CB4.ncrna.fa is 14.38%\n",
      "caenorhabditis_briggsae completed successfully!\n",
      "cds 30800 , total =  36814015\n",
      "The GECO3 compression ratio for Caenorhabditis_brenneri.C_brenneri-6.0.1b.cds.all.fa is 22.17%\n",
      "The GZIP compression ratio for Caenorhabditis_brenneri.C_brenneri-6.0.1b.cds.all.fa is 27.78%\n",
      "ncrna 2174 , total =  204320\n",
      "The GECO3 compression ratio for Caenorhabditis_brenneri.C_brenneri-6.0.1b.ncrna.fa is 10.06%\n",
      "The GZIP compression ratio for Caenorhabditis_brenneri.C_brenneri-6.0.1b.ncrna.fa is 12.64%\n",
      "caenorhabditis_brenneri completed successfully!\n",
      "cds 36373 , total =  32627264\n",
      "The GECO3 compression ratio for Caenorhabditis_japonica.C_japonica-7.0.1.cds.all.fa is 17.38%\n",
      "The GZIP compression ratio for Caenorhabditis_japonica.C_japonica-7.0.1.cds.all.fa is 22.54%\n",
      "ncrna 1876 , total =  212603\n",
      "The GECO3 compression ratio for Caenorhabditis_japonica.C_japonica-7.0.1.ncrna.fa is 6.43%\n",
      "The GZIP compression ratio for Caenorhabditis_japonica.C_japonica-7.0.1.ncrna.fa is 8.10%\n",
      "caenorhabditis_japonica completed successfully!\n",
      "cds 31865 , total =  45006717\n",
      "The GECO3 compression ratio for Caenorhabditis_elegans.WBcel235.cds.all.fa is 14.89%\n",
      "The GZIP compression ratio for Caenorhabditis_elegans.WBcel235.cds.all.fa is 16.38%\n",
      "ncrna 25311 , total =  1886078\n",
      "The GECO3 compression ratio for Caenorhabditis_elegans.WBcel235.ncrna.fa is 20.61%\n",
      "The GZIP compression ratio for Caenorhabditis_elegans.WBcel235.ncrna.fa is 26.49%\n",
      "caenorhabditis_elegans completed successfully!\n",
      "cds 31533 , total =  37959977\n",
      "The GECO3 compression ratio for Caenorhabditis_remanei.C_remanei-15.0.1.cds.all.fa is 22.13%\n",
      "The GZIP compression ratio for Caenorhabditis_remanei.C_remanei-15.0.1.cds.all.fa is 27.09%\n",
      "ncrna 1339 , total =  126444\n",
      "The GECO3 compression ratio for Caenorhabditis_remanei.C_remanei-15.0.1.ncrna.fa is 11.45%\n",
      "The GZIP compression ratio for Caenorhabditis_remanei.C_remanei-15.0.1.ncrna.fa is 14.23%\n",
      "caenorhabditis_remanei completed successfully!\n",
      "cds 23971 , total =  52248203\n",
      "The GECO3 compression ratio for Camponotus_floridanus_gca003227725v1rs.Cflo_v7.5.cds.all.fa is 11.23%\n",
      "The GZIP compression ratio for Camponotus_floridanus_gca003227725v1rs.Cflo_v7.5.cds.all.fa is 13.14%\n",
      "ncrna 1542 , total =  1398300\n",
      "The GECO3 compression ratio for Camponotus_floridanus_gca003227725v1rs.Cflo_v7.5.ncrna.fa is 12.64%\n",
      "The GZIP compression ratio for Camponotus_floridanus_gca003227725v1rs.Cflo_v7.5.ncrna.fa is 16.73%\n",
      "camponotus_floridanus_gca003227725v1rs completed successfully!\n",
      "cds 32175 , total =  34378213\n",
      "The GECO3 compression ratio for Capitella_teleta.Capitella_teleta_v1.0.cds.all.fa is 22.00%\n",
      "The GZIP compression ratio for Capitella_teleta.Capitella_teleta_v1.0.cds.all.fa is 27.56%\n",
      "ncrna 1341 , total =  119838\n",
      "The GECO3 compression ratio for Capitella_teleta.Capitella_teleta_v1.0.ncrna.fa is 6.19%\n",
      "The GZIP compression ratio for Capitella_teleta.Capitella_teleta_v1.0.ncrna.fa is 7.88%\n",
      "capitella_teleta completed successfully!\n",
      "cds 22395 , total =  47853471\n",
      "The GECO3 compression ratio for Cataglyphis_hispanica_gca021464435v1rs.ULB_Chis1_1.0.cds.all.fa is 11.31%\n",
      "The GZIP compression ratio for Cataglyphis_hispanica_gca021464435v1rs.ULB_Chis1_1.0.cds.all.fa is 12.54%\n",
      "ncrna 1120 , total =  1394893\n",
      "The GECO3 compression ratio for Cataglyphis_hispanica_gca021464435v1rs.ULB_Chis1_1.0.ncrna.fa is 13.67%\n",
      "The GZIP compression ratio for Cataglyphis_hispanica_gca021464435v1rs.ULB_Chis1_1.0.ncrna.fa is 17.16%\n",
      "cataglyphis_hispanica_gca021464435v1rs completed successfully!\n",
      "cds 19220 , total =  41711664\n",
      "The GECO3 compression ratio for Chelonus_insularis_gca013357705v1rs.ASM1335770v1.cds.all.fa is 12.69%\n",
      "The GZIP compression ratio for Chelonus_insularis_gca013357705v1rs.ASM1335770v1.cds.all.fa is 13.96%\n",
      "ncrna 1059 , total =  1147445\n",
      "The GECO3 compression ratio for Chelonus_insularis_gca013357705v1rs.ASM1335770v1.ncrna.fa is 15.42%\n",
      "The GZIP compression ratio for Chelonus_insularis_gca013357705v1rs.ASM1335770v1.ncrna.fa is 19.24%\n",
      "chelonus_insularis_gca013357705v1rs completed successfully!\n",
      "cds 35529 , total =  50015527\n",
      "The GECO3 compression ratio for Centruroides_sculpturatus_gca000671375v2.Cexi_2.0.cds.all.fa is 16.18%\n",
      "The GZIP compression ratio for Centruroides_sculpturatus_gca000671375v2.Cexi_2.0.cds.all.fa is 18.91%\n",
      "cds 22949 , total =  52263057\n",
      "The GECO3 compression ratio for Ceratitis_capitata_gca000347755v4.Ccap_2.0.cds.all.fa is 12.27%\n",
      "The GZIP compression ratio for Ceratitis_capitata_gca000347755v4.Ccap_2.0.cds.all.fa is 13.87%\n",
      "ncrna 4903 , total =  3845006\n",
      "The GECO3 compression ratio for Centruroides_sculpturatus_gca000671375v2.Cexi_2.0.ncrna.fa is 17.17%\n",
      "The GZIP compression ratio for Centruroides_sculpturatus_gca000671375v2.Cexi_2.0.ncrna.fa is 21.81%\n",
      "centruroides_sculpturatus_gca000671375v2 completed successfully!\n",
      "ncrna 1944 , total =  1747205\n",
      "The GECO3 compression ratio for Ceratitis_capitata_gca000347755v4.Ccap_2.0.ncrna.fa is 17.42%\n",
      "The GZIP compression ratio for Ceratitis_capitata_gca000347755v4.Ccap_2.0.ncrna.fa is 21.06%\n",
      "ceratitis_capitata_gca000347755v4 completed successfully!\n",
      "cds 24194 , total =  45828155\n",
      "The GECO3 compression ratio for Cimex_lectularius.Clec_2.0.cds.all.fa is 11.97%\n",
      "The GZIP compression ratio for Cimex_lectularius.Clec_2.0.cds.all.fa is 13.10%\n",
      "cds 17308 , total =  30427811\n",
      "The GECO3 compression ratio for Copidosoma_floridanum_gca000648655v2.Cflo_2.0.cds.all.fa is 16.04%\n",
      "The GZIP compression ratio for Copidosoma_floridanum_gca000648655v2.Cflo_2.0.cds.all.fa is 19.00%\n",
      "ncrna 641 , total =  230342\n",
      "The GECO3 compression ratio for Copidosoma_floridanum_gca000648655v2.Cflo_2.0.ncrna.fa is 18.43%\n",
      "The GZIP compression ratio for Copidosoma_floridanum_gca000648655v2.Cflo_2.0.ncrna.fa is 22.89%\n",
      "copidosoma_floridanum_gca000648655v2 completed successfully!\n",
      "ncrna 2846 , total =  1600799\n",
      "The GECO3 compression ratio for Cimex_lectularius.Clec_2.0.ncrna.fa is 17.18%\n",
      "The GZIP compression ratio for Cimex_lectularius.Clec_2.0.ncrna.fa is 20.86%\n",
      "cimex_lectularius completed successfully!\n",
      "cds 31450 , total =  53765972\n",
      "The GECO3 compression ratio for Cherax_quadricarinatus_gca026875155v2rs.ASM2687515v2.cds.all.fa is 14.17%\n",
      "The GZIP compression ratio for Cherax_quadricarinatus_gca026875155v2rs.ASM2687515v2.cds.all.fa is 15.72%\n",
      "ncrna 3558 , total =  3783844\n",
      "The GECO3 compression ratio for Cherax_quadricarinatus_gca026875155v2rs.ASM2687515v2.ncrna.fa is 15.92%\n",
      "The GZIP compression ratio for Cherax_quadricarinatus_gca026875155v2rs.ASM2687515v2.ncrna.fa is 20.93%\n",
      "cherax_quadricarinatus_gca026875155v2rs completed successfully!\n",
      "cds 25895 , total =  41780650\n",
      "The GECO3 compression ratio for Clytia_hemisphaerica_gca902728285.Clytia_hemisphaerica_genome_assembly.cds.all.fa is 18.75%\n",
      "The GZIP compression ratio for Clytia_hemisphaerica_gca902728285.Clytia_hemisphaerica_genome_assembly.cds.all.fa is 21.67%\n",
      "ncrna 2771 , total =  261306\n",
      "The GECO3 compression ratio for Clytia_hemisphaerica_gca902728285.Clytia_hemisphaerica_genome_assembly.ncrna.fa is 3.29%\n",
      "The GZIP compression ratio for Clytia_hemisphaerica_gca902728285.Clytia_hemisphaerica_genome_assembly.ncrna.fa is 4.65%\n",
      "clytia_hemisphaerica_gca902728285 completed successfully!\n",
      "cds 19745 , total =  32043025\n",
      "The GECO3 compression ratio for Coremacera_marginata_gca914767935v1.idCorMarg1.1.cds.all.fa is 18.05%\n",
      "The GZIP compression ratio for Coremacera_marginata_gca914767935v1.idCorMarg1.1.cds.all.fa is 20.35%\n",
      "ncrna 2198 , total =  937562\n",
      "The GECO3 compression ratio for Coremacera_marginata_gca914767935v1.idCorMarg1.1.ncrna.fa is 19.69%\n",
      "The GZIP compression ratio for Coremacera_marginata_gca914767935v1.idCorMarg1.1.ncrna.fa is 24.10%\n",
      "coremacera_marginata_gca914767935v1 completed successfully!\n",
      "cds 25331 , total =  52988956\n",
      "The GECO3 compression ratio for Cotesia_glomerata_gca020080835v1.MPM_Cglom_v2.3.cds.all.fa is 11.82%\n",
      "The GZIP compression ratio for Cotesia_glomerata_gca020080835v1.MPM_Cglom_v2.3.cds.all.fa is 13.94%\n",
      "ncrna 3375 , total =  2456075\n",
      "The GECO3 compression ratio for Cotesia_glomerata_gca020080835v1.MPM_Cglom_v2.3.ncrna.fa is 15.52%\n",
      "The GZIP compression ratio for Cotesia_glomerata_gca020080835v1.MPM_Cglom_v2.3.ncrna.fa is 20.54%\n",
      "cotesia_glomerata_gca020080835v1 completed successfully!\n",
      "cds 19038 , total =  25074001\n",
      "The GECO3 compression ratio for Culex_quinquefasciatus.CpipJ2.cds.all.fa is 22.45%\n",
      "The GZIP compression ratio for Culex_quinquefasciatus.CpipJ2.cds.all.fa is 27.64%\n",
      "ncrna 828 , total =  101088\n",
      "The GECO3 compression ratio for Culex_quinquefasciatus.CpipJ2.ncrna.fa is 8.79%\n",
      "The GZIP compression ratio for Culex_quinquefasciatus.CpipJ2.ncrna.fa is 11.34%\n",
      "culex_quinquefasciatus completed successfully!\n",
      "cds 21954 , total =  33330136\n",
      "The GECO3 compression ratio for Ctenocephalides_felis_gca003426905v1rs.ASM342690v1.cds.all.fa is 16.56%\n",
      "The GZIP compression ratio for Ctenocephalides_felis_gca003426905v1rs.ASM342690v1.cds.all.fa is 23.51%\n",
      "ncrna 8183 , total =  1769098\n",
      "The GECO3 compression ratio for Ctenocephalides_felis_gca003426905v1rs.ASM342690v1.ncrna.fa is 5.51%\n",
      "The GZIP compression ratio for Ctenocephalides_felis_gca003426905v1rs.ASM342690v1.ncrna.fa is 9.87%\n",
      "ctenocephalides_felis_gca003426905v1rs completed successfully!\n",
      "cds 29285 , total =  57308556\n",
      "The GECO3 compression ratio for Cryptotermes_secundus_gca002891405v2rs.Csec_1.0.cds.all.fa is 11.92%\n",
      "The GZIP compression ratio for Cryptotermes_secundus_gca002891405v2rs.Csec_1.0.cds.all.fa is 12.63%\n",
      "ncrna 2655 , total =  3197218\n",
      "The GECO3 compression ratio for Cryptotermes_secundus_gca002891405v2rs.Csec_1.0.ncrna.fa is 16.01%\n",
      "The GZIP compression ratio for Cryptotermes_secundus_gca002891405v2rs.Csec_1.0.ncrna.fa is 20.22%\n",
      "cryptotermes_secundus_gca002891405v2rs completed successfully!\n",
      "cds 24531 , total =  49261437\n",
      "The GECO3 compression ratio for Culex_quinquefasciatus_gca015732765v1.VPISU_Cqui_1.0_pri_paternal.cds.all.fa is 13.77%\n",
      "The GZIP compression ratio for Culex_quinquefasciatus_gca015732765v1.VPISU_Cqui_1.0_pri_paternal.cds.all.fa is 15.33%\n",
      "ncrna 2923 , total =  1739364\n",
      "The GECO3 compression ratio for Culex_quinquefasciatus_gca015732765v1.VPISU_Cqui_1.0_pri_paternal.ncrna.fa is 16.10%\n",
      "The GZIP compression ratio for Culex_quinquefasciatus_gca015732765v1.VPISU_Cqui_1.0_pri_paternal.ncrna.fa is 24.34%\n",
      "culex_quinquefasciatus_gca015732765v1 completed successfully!\n",
      "cds 21241 , total =  36483939\n",
      "The GECO3 compression ratio for Culicoides_sonorensis.Cson1.cds.all.fa is 16.65%\n",
      "The GZIP compression ratio for Culicoides_sonorensis.Cson1.cds.all.fa is 20.28%\n",
      "Directory or file not found for culicoides_sonorensis in ncrna folder: 550 Failed to change directory.\n",
      "culicoides_sonorensis completed successfully!\n",
      "cds 19762 , total =  34624270\n",
      "The GECO3 compression ratio for Danaus_plexippus.Dplex_v4.cds.all.fa is 15.21%\n",
      "The GZIP compression ratio for Danaus_plexippus.Dplex_v4.cds.all.fa is 18.05%\n",
      "ncrna 1452 , total =  582485\n",
      "The GECO3 compression ratio for Danaus_plexippus.Dplex_v4.ncrna.fa is 17.80%\n",
      "The GZIP compression ratio for Danaus_plexippus.Dplex_v4.ncrna.fa is 21.78%\n",
      "danaus_plexippus completed successfully!\n",
      "cds 49950 , total =  102219678\n",
      "The GECO3 compression ratio for Crassostrea_angulata_gca025612915v2rs.ASM2561291v2.cds.all.fa is 13.78%\n",
      "The GZIP compression ratio for Crassostrea_angulata_gca025612915v2rs.ASM2561291v2.cds.all.fa is 14.72%\n",
      "cds 79948 , total =  98808058\n",
      "The GECO3 compression ratio for Crassostrea_gigas.cgigas_uk_roslin_v1.cds.all.fa is 14.10%\n",
      "The GZIP compression ratio for Crassostrea_gigas.cgigas_uk_roslin_v1.cds.all.fa is 14.43%\n",
      "cds 17509 , total =  29442529\n",
      "The GECO3 compression ratio for Daphnia_carinata_gca022539665v3rs.CSIRO_AGI_Dcar_v0.2.cds.all.fa is 18.78%\n",
      "The GZIP compression ratio for Daphnia_carinata_gca022539665v3rs.CSIRO_AGI_Dcar_v0.2.cds.all.fa is 22.23%\n",
      "cds 29510 , total =  53321080\n",
      "The GECO3 compression ratio for Daktulosphaira_vitifoliae_gca025091365v1rs.ASM2509136v1.cds.all.fa is 12.04%\n",
      "The GZIP compression ratio for Daktulosphaira_vitifoliae_gca025091365v1rs.ASM2509136v1.cds.all.fa is 13.41%\n",
      "ncrna 4875 , total =  4679909\n",
      "The GECO3 compression ratio for Crassostrea_angulata_gca025612915v2rs.ASM2561291v2.ncrna.fa is 13.79%\n",
      "The GZIP compression ratio for Crassostrea_angulata_gca025612915v2rs.ASM2561291v2.ncrna.fa is 21.37%\n",
      "crassostrea_angulata_gca025612915v2rs completed successfully!\n",
      "ncrna 6757 , total =  5934275\n",
      "The GECO3 compression ratio for Daphnia_carinata_gca022539665v3rs.CSIRO_AGI_Dcar_v0.2.ncrna.fa is 8.50%\n",
      "The GZIP compression ratio for Daphnia_carinata_gca022539665v3rs.CSIRO_AGI_Dcar_v0.2.ncrna.fa is 14.21%\n",
      "daphnia_carinata_gca022539665v3rs completed successfully!\n",
      "ncrna 3137 , total =  3698814\n",
      "The GECO3 compression ratio for Daktulosphaira_vitifoliae_gca025091365v1rs.ASM2509136v1.ncrna.fa is 13.66%\n",
      "The GZIP compression ratio for Daktulosphaira_vitifoliae_gca025091365v1rs.ASM2509136v1.ncrna.fa is 17.50%\n",
      "daktulosphaira_vitifoliae_gca025091365v1rs completed successfully!\n",
      "ncrna 9320 , total =  9564044\n",
      "The GECO3 compression ratio for Crassostrea_gigas.cgigas_uk_roslin_v1.ncrna.fa is 15.18%\n",
      "The GZIP compression ratio for Crassostrea_gigas.cgigas_uk_roslin_v1.ncrna.fa is 18.78%\n",
      "crassostrea_gigas completed successfully!\n",
      "cds 60213 , total =  118005401\n",
      "The GECO3 compression ratio for Crassostrea_virginica_gca002022765v4.C_virginica_3.0.cds.all.fa is 12.92%\n",
      "The GZIP compression ratio for Crassostrea_virginica_gca002022765v4.C_virginica_3.0.cds.all.fa is 14.59%\n",
      "cds 28221 , total =  50980762\n",
      "The GECO3 compression ratio for Daphnia_magna_gca020631705v2.ASM2063170v1.1.cds.all.fa is 13.23%\n",
      "The GZIP compression ratio for Daphnia_magna_gca020631705v2.ASM2063170v1.1.cds.all.fa is 16.27%\n",
      "ncrna 5339 , total =  4786369\n",
      "The GECO3 compression ratio for Crassostrea_virginica_gca002022765v4.C_virginica_3.0.ncrna.fa is 12.59%\n",
      "The GZIP compression ratio for Crassostrea_virginica_gca002022765v4.C_virginica_3.0.ncrna.fa is 18.99%\n",
      "crassostrea_virginica_gca002022765v4 completed successfully!\n",
      "cds 20551 , total =  36168564\n",
      "The GECO3 compression ratio for Dendroctonus_ponderosae_gca000355655v1.DendPond_male_gca000355655v1.0.cds.all.fa is 13.88%\n",
      "The GZIP compression ratio for Dendroctonus_ponderosae_gca000355655v1.DendPond_male_gca000355655v1.0.cds.all.fa is 16.60%\n",
      "ncrna 1273 , total =  983447\n",
      "The GECO3 compression ratio for Dendroctonus_ponderosae_gca000355655v1.DendPond_male_gca000355655v1.0.ncrna.fa is 19.22%\n",
      "The GZIP compression ratio for Dendroctonus_ponderosae_gca000355655v1.DendPond_male_gca000355655v1.0.ncrna.fa is 24.28%\n",
      "dendroctonus_ponderosae_gca000355655v1 completed successfully!\n",
      "cds 27985 , total =  58465972\n",
      "The GECO3 compression ratio for Daphnia_pulex_gca021134715v1rs.ASM2113471v1.cds.all.fa is 12.01%\n",
      "The GZIP compression ratio for Daphnia_pulex_gca021134715v1rs.ASM2113471v1.cds.all.fa is 14.07%\n",
      "cds 27597 , total =  52433472\n",
      "The GECO3 compression ratio for Daphnia_pulicaria_gca021234035v2rs.SC_F0_13B.cds.all.fa is 13.74%\n",
      "The GZIP compression ratio for Daphnia_pulicaria_gca021234035v2rs.SC_F0_13B.cds.all.fa is 16.35%\n",
      "ncrna 15634 , total =  24537950\n",
      "The GECO3 compression ratio for Daphnia_magna_gca020631705v2.ASM2063170v1.1.ncrna.fa is 6.69%\n",
      "The GZIP compression ratio for Daphnia_magna_gca020631705v2.ASM2063170v1.1.ncrna.fa is 8.93%\n",
      "daphnia_magna_gca020631705v2 completed successfully!\n",
      "ncrna 10502 , total =  12144053\n",
      "The GECO3 compression ratio for Daphnia_pulex_gca021134715v1rs.ASM2113471v1.ncrna.fa is 9.43%\n",
      "The GZIP compression ratio for Daphnia_pulex_gca021134715v1rs.ASM2113471v1.ncrna.fa is 13.25%\n",
      "daphnia_pulex_gca021134715v1rs completed successfully!\n",
      "cds 12837 , total =  21901559\n",
      "The GECO3 compression ratio for Dermatophagoides_pteronyssinus_gca001901225v2.ASM190122v2.cds.all.fa is 20.12%\n",
      "The GZIP compression ratio for Dermatophagoides_pteronyssinus_gca001901225v2.ASM190122v2.cds.all.fa is 23.12%\n",
      "ncrna 325 , total =  67088\n",
      "The GECO3 compression ratio for Dermatophagoides_pteronyssinus_gca001901225v2.ASM190122v2.ncrna.fa is 20.78%\n",
      "The GZIP compression ratio for Dermatophagoides_pteronyssinus_gca001901225v2.ASM190122v2.ncrna.fa is 25.58%\n",
      "dermatophagoides_pteronyssinus_gca001901225v2 completed successfully!\n",
      "cds 30415 , total =  51170014\n",
      "The GECO3 compression ratio for Dermacentor_andersoni_gca023375885v2rs.qqDerAnde1.2.cds.all.fa is 15.26%\n",
      "The GZIP compression ratio for Dermacentor_andersoni_gca023375885v2rs.qqDerAnde1.2.cds.all.fa is 17.61%\n",
      "ncrna 17961 , total =  11403761\n",
      "The GECO3 compression ratio for Daphnia_pulicaria_gca021234035v2rs.SC_F0_13B.ncrna.fa is 8.78%\n",
      "The GZIP compression ratio for Daphnia_pulicaria_gca021234035v2rs.SC_F0_13B.ncrna.fa is 12.58%\n",
      "daphnia_pulicaria_gca021234035v2rs completed successfully!\n",
      "cds 28741 , total =  53238744\n",
      "The GECO3 compression ratio for Dendronephthya_gigantea_gca004324835v1.DenGig_1.0.cds.all.fa is 16.49%\n",
      "The GZIP compression ratio for Dendronephthya_gigantea_gca004324835v1.DenGig_1.0.cds.all.fa is 20.15%\n",
      "ncrna 10850 , total =  4204623\n",
      "The GECO3 compression ratio for Dermacentor_andersoni_gca023375885v2rs.qqDerAnde1.2.ncrna.fa is 12.42%\n",
      "The GZIP compression ratio for Dermacentor_andersoni_gca023375885v2rs.qqDerAnde1.2.ncrna.fa is 18.84%\n",
      "dermacentor_andersoni_gca023375885v2rs completed successfully!\n",
      "ncrna 7062 , total =  2736572\n",
      "The GECO3 compression ratio for Dendronephthya_gigantea_gca004324835v1.DenGig_1.0.ncrna.fa is 11.68%\n",
      "The GZIP compression ratio for Dendronephthya_gigantea_gca004324835v1.DenGig_1.0.ncrna.fa is 18.09%\n",
      "dendronephthya_gigantea_gca004324835v1 completed successfully!\n",
      "cds 36397 , total =  58450826\n",
      "The GECO3 compression ratio for Dermacentor_silvarum_gca013339745v2rs.BIME_Dsil_1.4.cds.all.fa is 14.23%\n",
      "The GZIP compression ratio for Dermacentor_silvarum_gca013339745v2rs.BIME_Dsil_1.4.cds.all.fa is 16.68%\n",
      "ncrna 11712 , total =  7118383\n",
      "The GECO3 compression ratio for Dermacentor_silvarum_gca013339745v2rs.BIME_Dsil_1.4.ncrna.fa is 13.71%\n",
      "The GZIP compression ratio for Dermacentor_silvarum_gca013339745v2rs.BIME_Dsil_1.4.ncrna.fa is 20.68%\n",
      "dermacentor_silvarum_gca013339745v2rs completed successfully!\n",
      "cds 24727 , total =  26933375\n",
      "The GECO3 compression ratio for Diaphorina_citri_gca000475195v1rs.Diaci_psyllid_genome_assembly_version_1.1.cds.all.fa is 18.21%\n",
      "The GZIP compression ratio for Diaphorina_citri_gca000475195v1rs.Diaci_psyllid_genome_assembly_version_1.1.cds.all.fa is 22.93%\n",
      "cds 28061 , total =  40892712\n",
      "The GECO3 compression ratio for Diabrotica_virgifera_gca003013835v2.Dvir_v2.0.cds.all.fa is 15.57%\n",
      "The GZIP compression ratio for Diabrotica_virgifera_gca003013835v2.Dvir_v2.0.cds.all.fa is 18.22%\n",
      "cds 16101 , total =  26947275\n",
      "The GECO3 compression ratio for Dimorphilus_gyrociliatus_gca904063045v1.Dgyrociliatus_assembly.cds.all.fa is 21.30%\n",
      "The GZIP compression ratio for Dimorphilus_gyrociliatus_gca904063045v1.Dgyrociliatus_assembly.cds.all.fa is 24.66%\n",
      "Directory or file not found for dimorphilus_gyrociliatus_gca904063045v1 in ncrna folder: No file ending with 'ncrna.fa.gz' found in the directory.\n",
      "dimorphilus_gyrociliatus_gca904063045v1 completed successfully!\n",
      "cds 19024 , total =  24837678\n",
      "The GECO3 compression ratio for Dinothrombium_tinctorium.DtinU1.cds.all.fa is 17.00%\n",
      "The GZIP compression ratio for Dinothrombium_tinctorium.DtinU1.cds.all.fa is 27.91%\n",
      "Directory or file not found for dinothrombium_tinctorium in ncrna folder: 550 Failed to change directory.\n",
      "dinothrombium_tinctorium completed successfully!\n",
      "ncrna 6418 , total =  3134104\n",
      "The GECO3 compression ratio for Diaphorina_citri_gca000475195v1rs.Diaci_psyllid_genome_assembly_version_1.1.ncrna.fa is 21.17%\n",
      "The GZIP compression ratio for Diaphorina_citri_gca000475195v1rs.Diaci_psyllid_genome_assembly_version_1.1.ncrna.fa is 25.99%\n",
      "diaphorina_citri_gca000475195v1rs completed successfully!\n",
      "ncrna 4627 , total =  2451801\n",
      "The GECO3 compression ratio for Diabrotica_virgifera_gca003013835v2.Dvir_v2.0.ncrna.fa is 17.84%\n",
      "The GZIP compression ratio for Diabrotica_virgifera_gca003013835v2.Dvir_v2.0.ncrna.fa is 25.67%\n",
      "diabrotica_virgifera_gca003013835v2 completed successfully!\n",
      "cds 17476 , total =  28590609\n",
      "The GECO3 compression ratio for Diuraphis_noxia_gca001186385v1.Dnoxia_1.0.cds.all.fa is 16.12%\n",
      "The GZIP compression ratio for Diuraphis_noxia_gca001186385v1.Dnoxia_1.0.cds.all.fa is 18.28%\n",
      "ncrna 545 , total =  239705\n",
      "The GECO3 compression ratio for Diuraphis_noxia_gca001186385v1.Dnoxia_1.0.ncrna.fa is 17.87%\n",
      "The GZIP compression ratio for Diuraphis_noxia_gca001186385v1.Dnoxia_1.0.ncrna.fa is 21.92%\n",
      "diuraphis_noxia_gca001186385v1 completed successfully!\n",
      "cds 31900 , total =  68835329\n",
      "The GECO3 compression ratio for Diabrotica_virgifera_gca917563875v2rs.PGI_DIABVI_V3a.cds.all.fa is 10.91%\n",
      "The GZIP compression ratio for Diabrotica_virgifera_gca917563875v2rs.PGI_DIABVI_V3a.cds.all.fa is 15.75%\n",
      "cds 24248 , total =  53706042\n",
      "The GECO3 compression ratio for Drosophila_albomicans_gca009650485v2rs.ASM965048v2.cds.all.fa is 12.40%\n",
      "The GZIP compression ratio for Drosophila_albomicans_gca009650485v2rs.ASM965048v2.cds.all.fa is 14.13%\n",
      "ncrna 1799 , total =  1360773\n",
      "The GECO3 compression ratio for Drosophila_albomicans_gca009650485v2rs.ASM965048v2.ncrna.fa is 11.83%\n",
      "The GZIP compression ratio for Drosophila_albomicans_gca009650485v2rs.ASM965048v2.ncrna.fa is 15.19%\n",
      "drosophila_albomicans_gca009650485v2rs completed successfully!\n",
      "cds 21191 , total =  43009950\n",
      "The GECO3 compression ratio for Drosophila_ananassae.dana_caf1.cds.all.fa is 14.30%\n",
      "The GZIP compression ratio for Drosophila_ananassae.dana_caf1.cds.all.fa is 16.59%\n",
      "cds 19500 , total =  41324697\n",
      "The GECO3 compression ratio for Drosophila_arizonae_gca001654025v1rs.ASM165402v1.cds.all.fa is 13.85%\n",
      "The GZIP compression ratio for Drosophila_arizonae_gca001654025v1rs.ASM165402v1.cds.all.fa is 15.75%\n",
      "ncrna 5864 , total =  12785453\n",
      "The GECO3 compression ratio for Diabrotica_virgifera_gca917563875v2rs.PGI_DIABVI_V3a.ncrna.fa is 5.38%\n",
      "The GZIP compression ratio for Diabrotica_virgifera_gca917563875v2rs.PGI_DIABVI_V3a.ncrna.fa is 15.96%\n",
      "diabrotica_virgifera_gca917563875v2rs completed successfully!\n",
      "ncrna 945 , total =  381756\n",
      "The GECO3 compression ratio for Drosophila_arizonae_gca001654025v1rs.ASM165402v1.ncrna.fa is 19.14%\n",
      "The GZIP compression ratio for Drosophila_arizonae_gca001654025v1rs.ASM165402v1.ncrna.fa is 23.06%\n",
      "drosophila_arizonae_gca001654025v1rs completed successfully!\n",
      "ncrna 1466 , total =  615335\n",
      "The GECO3 compression ratio for Drosophila_ananassae.dana_caf1.ncrna.fa is 14.72%\n",
      "The GZIP compression ratio for Drosophila_ananassae.dana_caf1.ncrna.fa is 18.63%\n",
      "drosophila_ananassae completed successfully!\n",
      "cds 23478 , total =  49509231\n",
      "The GECO3 compression ratio for Drosophila_bipectinata_gca000236285v2rs.Dbip_2.0.cds.all.fa is 13.35%\n",
      "The GZIP compression ratio for Drosophila_bipectinata_gca000236285v2rs.Dbip_2.0.cds.all.fa is 15.35%\n",
      "ncrna 949 , total =  582045\n",
      "The GECO3 compression ratio for Drosophila_bipectinata_gca000236285v2rs.Dbip_2.0.ncrna.fa is 16.09%\n",
      "The GZIP compression ratio for Drosophila_bipectinata_gca000236285v2rs.Dbip_2.0.ncrna.fa is 19.92%\n",
      "drosophila_bipectinata_gca000236285v2rs completed successfully!\n",
      "cds 25915 , total =  60971076\n",
      "The GECO3 compression ratio for Drosophila_ananassae_gca017639315v2rs.ASM1763931v2.cds.all.fa is 11.44%\n",
      "The GZIP compression ratio for Drosophila_ananassae_gca017639315v2rs.ASM1763931v2.cds.all.fa is 13.62%\n",
      "cds 26154 , total =  58968363\n",
      "The GECO3 compression ratio for Drosophila_biarmipes_gca025231255v1rs.RU_DBia_V1.1.cds.all.fa is 11.67%\n",
      "The GZIP compression ratio for Drosophila_biarmipes_gca025231255v1rs.RU_DBia_V1.1.cds.all.fa is 13.53%\n",
      "ncrna 2478 , total =  2245718\n",
      "The GECO3 compression ratio for Drosophila_ananassae_gca017639315v2rs.ASM1763931v2.ncrna.fa is 15.03%\n",
      "The GZIP compression ratio for Drosophila_ananassae_gca017639315v2rs.ASM1763931v2.ncrna.fa is 19.91%\n",
      "drosophila_ananassae_gca017639315v2rs completed successfully!\n",
      "ncrna 2925 , total =  3379805\n",
      "The GECO3 compression ratio for Drosophila_biarmipes_gca025231255v1rs.RU_DBia_V1.1.ncrna.fa is 6.74%\n",
      "The GZIP compression ratio for Drosophila_biarmipes_gca025231255v1rs.RU_DBia_V1.1.ncrna.fa is 9.32%\n",
      "drosophila_biarmipes_gca025231255v1rs completed successfully!\n",
      "cds 19109 , total =  37460212\n",
      "The GECO3 compression ratio for Drosophila_busckii_gca011750605v1rs.ASM1175060v1.cds.all.fa is 15.04%\n",
      "The GZIP compression ratio for Drosophila_busckii_gca011750605v1rs.ASM1175060v1.cds.all.fa is 16.66%\n",
      "ncrna 628 , total =  260821\n",
      "The GECO3 compression ratio for Drosophila_busckii_gca011750605v1rs.ASM1175060v1.ncrna.fa is 17.42%\n",
      "The GZIP compression ratio for Drosophila_busckii_gca011750605v1rs.ASM1175060v1.ncrna.fa is 21.13%\n",
      "drosophila_busckii_gca011750605v1rs completed successfully!\n",
      "cds 19592 , total =  39469592\n",
      "The GECO3 compression ratio for Drosophila_erecta.dere_caf1.cds.all.fa is 14.92%\n",
      "The GZIP compression ratio for Drosophila_erecta.dere_caf1.cds.all.fa is 17.21%\n",
      "ncrna 1074 , total =  151866\n",
      "The GECO3 compression ratio for Drosophila_erecta.dere_caf1.ncrna.fa is 17.56%\n",
      "The GZIP compression ratio for Drosophila_erecta.dere_caf1.ncrna.fa is 21.18%\n",
      "drosophila_erecta completed successfully!\n",
      "cds 23135 , total =  47089453\n",
      "The GECO3 compression ratio for Drosophila_erecta_gca003286155v2rs.DereRS2.cds.all.fa is 13.40%\n",
      "The GZIP compression ratio for Drosophila_erecta_gca003286155v2rs.DereRS2.cds.all.fa is 14.58%\n",
      "ncrna 951 , total =  416463\n",
      "The GECO3 compression ratio for Drosophila_erecta_gca003286155v2rs.DereRS2.ncrna.fa is 13.05%\n",
      "The GZIP compression ratio for Drosophila_erecta_gca003286155v2rs.DereRS2.ncrna.fa is 16.31%\n",
      "drosophila_erecta_gca003286155v2rs completed successfully!\n",
      "cds 24259 , total =  52785559\n",
      "The GECO3 compression ratio for Drosophila_elegans_gca000224195v2rs.Dele_2.0.cds.all.fa is 13.07%\n",
      "The GZIP compression ratio for Drosophila_elegans_gca000224195v2rs.Dele_2.0.cds.all.fa is 15.40%\n",
      "ncrna 883 , total =  539785\n",
      "The GECO3 compression ratio for Drosophila_elegans_gca000224195v2rs.Dele_2.0.ncrna.fa is 16.74%\n",
      "The GZIP compression ratio for Drosophila_elegans_gca000224195v2rs.Dele_2.0.ncrna.fa is 20.97%\n",
      "drosophila_elegans_gca000224195v2rs completed successfully!\n",
      "cds 19429 , total =  37808833\n",
      "The GECO3 compression ratio for Drosophila_grimshawi.dgri_caf1.cds.all.fa is 15.09%\n",
      "The GZIP compression ratio for Drosophila_grimshawi.dgri_caf1.cds.all.fa is 17.83%\n",
      "cds 22460 , total =  46537694\n",
      "The GECO3 compression ratio for Drosophila_eugracilis_gca018153835v1rs.ASM1815383v1.cds.all.fa is 13.41%\n",
      "The GZIP compression ratio for Drosophila_eugracilis_gca018153835v1rs.ASM1815383v1.cds.all.fa is 15.10%\n",
      "ncrna 1223 , total =  449266\n",
      "The GECO3 compression ratio for Drosophila_grimshawi.dgri_caf1.ncrna.fa is 15.72%\n",
      "The GZIP compression ratio for Drosophila_grimshawi.dgri_caf1.ncrna.fa is 19.91%\n",
      "drosophila_grimshawi completed successfully!\n",
      "ncrna 926 , total =  635519\n",
      "The GECO3 compression ratio for Drosophila_eugracilis_gca018153835v1rs.ASM1815383v1.ncrna.fa is 13.42%\n",
      "The GZIP compression ratio for Drosophila_eugracilis_gca018153835v1rs.ASM1815383v1.ncrna.fa is 16.80%\n",
      "drosophila_eugracilis_gca018153835v1rs completed successfully!\n",
      "cds 22110 , total =  43262263\n",
      "The GECO3 compression ratio for Drosophila_ficusphila_gca018152265v1rs.ASM1815226v1.cds.all.fa is 14.50%\n",
      "The GZIP compression ratio for Drosophila_ficusphila_gca018152265v1rs.ASM1815226v1.cds.all.fa is 16.15%\n",
      "ncrna 1342 , total =  1002063\n",
      "The GECO3 compression ratio for Drosophila_ficusphila_gca018152265v1rs.ASM1815226v1.ncrna.fa is 12.66%\n",
      "The GZIP compression ratio for Drosophila_ficusphila_gca018152265v1rs.ASM1815226v1.ncrna.fa is 16.11%\n",
      "drosophila_ficusphila_gca018152265v1rs completed successfully!\n",
      "cds 75288 , total =  139805476\n",
      "The GECO3 compression ratio for Dreissena_polymorpha_gca020536995v1rs.UMN_Dpol_1.0.cds.all.fa is 11.75%\n",
      "The GZIP compression ratio for Dreissena_polymorpha_gca020536995v1rs.UMN_Dpol_1.0.cds.all.fa is 12.84%\n",
      "cds 19405 , total =  39503153\n",
      "The GECO3 compression ratio for Drosophila_grimshawi_gca018153295v1rs.ASM1815329v1.cds.all.fa is 14.68%\n",
      "The GZIP compression ratio for Drosophila_grimshawi_gca018153295v1rs.ASM1815329v1.cds.all.fa is 16.86%\n",
      "ncrna 1017 , total =  678644\n",
      "The GECO3 compression ratio for Drosophila_grimshawi_gca018153295v1rs.ASM1815329v1.ncrna.fa is 10.00%\n",
      "The GZIP compression ratio for Drosophila_grimshawi_gca018153295v1rs.ASM1815329v1.ncrna.fa is 12.49%\n",
      "drosophila_grimshawi_gca018153295v1rs completed successfully!\n",
      "cds 22824 , total =  52258470\n",
      "The GECO3 compression ratio for Drosophila_guanche_gca900245975v1rs.DGUA_6.cds.all.fa is 12.48%\n",
      "The GZIP compression ratio for Drosophila_guanche_gca900245975v1rs.DGUA_6.cds.all.fa is 13.44%\n",
      "ncrna 1099 , total =  630222\n",
      "The GECO3 compression ratio for Drosophila_guanche_gca900245975v1rs.DGUA_6.ncrna.fa is 16.49%\n",
      "The GZIP compression ratio for Drosophila_guanche_gca900245975v1rs.DGUA_6.ncrna.fa is 20.62%\n",
      "drosophila_guanche_gca900245975v1rs completed successfully!\n",
      "cds 19534 , total =  36273754\n",
      "The GECO3 compression ratio for Drosophila_innubila_gca004354385v1rs.UK_Dinn_1.0.cds.all.fa is 16.21%\n",
      "The GZIP compression ratio for Drosophila_innubila_gca004354385v1rs.UK_Dinn_1.0.cds.all.fa is 18.34%\n",
      "cds 21429 , total =  45513905\n",
      "The GECO3 compression ratio for Drosophila_hydei_gca003285905v2rs.DhydRS2.cds.all.fa is 13.28%\n",
      "The GZIP compression ratio for Drosophila_hydei_gca003285905v2rs.DhydRS2.cds.all.fa is 15.15%\n",
      "ncrna 1522 , total =  644293\n",
      "The GECO3 compression ratio for Drosophila_innubila_gca004354385v1rs.UK_Dinn_1.0.ncrna.fa is 16.36%\n",
      "The GZIP compression ratio for Drosophila_innubila_gca004354385v1rs.UK_Dinn_1.0.ncrna.fa is 20.18%\n",
      "drosophila_innubila_gca004354385v1rs completed successfully!\n",
      "ncrna 1147 , total =  523943\n",
      "The GECO3 compression ratio for Drosophila_hydei_gca003285905v2rs.DhydRS2.ncrna.fa is 13.90%\n",
      "The GZIP compression ratio for Drosophila_hydei_gca003285905v2rs.DhydRS2.ncrna.fa is 17.15%\n",
      "drosophila_hydei_gca003285905v2rs completed successfully!\n",
      "cds 24987 , total =  53654876\n",
      "The GECO3 compression ratio for Drosophila_gunungcola_gca025200985v1rs.Dgunungcola_SK_2.cds.all.fa is 12.20%\n",
      "The GZIP compression ratio for Drosophila_gunungcola_gca025200985v1rs.Dgunungcola_SK_2.cds.all.fa is 14.35%\n",
      "ncrna 1552 , total =  1135214\n",
      "The GECO3 compression ratio for Drosophila_gunungcola_gca025200985v1rs.Dgunungcola_SK_2.ncrna.fa is 14.87%\n",
      "The GZIP compression ratio for Drosophila_gunungcola_gca025200985v1rs.Dgunungcola_SK_2.ncrna.fa is 19.03%\n",
      "drosophila_gunungcola_gca025200985v1rs completed successfully!\n",
      "ncrna 20578 , total =  35598809\n",
      "The GECO3 compression ratio for Dreissena_polymorpha_gca020536995v1rs.UMN_Dpol_1.0.ncrna.fa is 7.46%\n",
      "The GZIP compression ratio for Dreissena_polymorpha_gca020536995v1rs.UMN_Dpol_1.0.ncrna.fa is 13.46%\n",
      "dreissena_polymorpha_gca020536995v1rs completed successfully!\n",
      "cds 23239 , total =  48762174\n",
      "The GECO3 compression ratio for Drosophila_mauritiana_gca004382145v1rs.ASM438214v1.cds.all.fa is 13.10%\n",
      "The GZIP compression ratio for Drosophila_mauritiana_gca004382145v1rs.ASM438214v1.cds.all.fa is 14.54%\n",
      "cds 22799 , total =  47090529\n",
      "The GECO3 compression ratio for Drosophila_kikkawai_gca018152535v1rs.ASM1815253v1.cds.all.fa is 13.96%\n",
      "The GZIP compression ratio for Drosophila_kikkawai_gca018152535v1rs.ASM1815253v1.cds.all.fa is 15.79%\n",
      "ncrna 1100 , total =  628869\n",
      "The GECO3 compression ratio for Drosophila_kikkawai_gca018152535v1rs.ASM1815253v1.ncrna.fa is 12.51%\n",
      "The GZIP compression ratio for Drosophila_kikkawai_gca018152535v1rs.ASM1815253v1.ncrna.fa is 16.56%\n",
      "drosophila_kikkawai_gca018152535v1rs completed successfully!\n",
      "ncrna 1803 , total =  1846686\n",
      "The GECO3 compression ratio for Drosophila_mauritiana_gca004382145v1rs.ASM438214v1.ncrna.fa is 5.74%\n",
      "The GZIP compression ratio for Drosophila_mauritiana_gca004382145v1rs.ASM438214v1.ncrna.fa is 7.93%\n",
      "drosophila_mauritiana_gca004382145v1rs completed successfully!\n",
      "cds 30799 , total =  61218290\n",
      "The GECO3 compression ratio for Drosophila_melanogaster.BDGP6.46.cds.all.fa is 11.60%\n",
      "The GZIP compression ratio for Drosophila_melanogaster.BDGP6.46.cds.all.fa is 12.37%\n",
      "cds 16878 , total =  21612551\n",
      "The GECO3 compression ratio for Drosophila_persimilis.dper_caf1.cds.all.fa is 23.22%\n",
      "The GZIP compression ratio for Drosophila_persimilis.dper_caf1.cds.all.fa is 27.89%\n",
      "cds 20110 , total =  39041981\n",
      "The GECO3 compression ratio for Drosophila_mojavensis.dmoj_caf1.cds.all.fa is 14.75%\n",
      "The GZIP compression ratio for Drosophila_mojavensis.dmoj_caf1.cds.all.fa is 16.62%\n",
      "ncrna 779 , total =  80830\n",
      "The GECO3 compression ratio for Drosophila_persimilis.dper_caf1.ncrna.fa is 13.18%\n",
      "The GZIP compression ratio for Drosophila_persimilis.dper_caf1.ncrna.fa is 15.97%\n",
      "drosophila_persimilis completed successfully!\n",
      "ncrna 1386 , total =  538961\n",
      "The GECO3 compression ratio for Drosophila_mojavensis.dmoj_caf1.ncrna.fa is 14.25%\n",
      "The GZIP compression ratio for Drosophila_mojavensis.dmoj_caf1.ncrna.fa is 18.60%\n",
      "drosophila_mojavensis completed successfully!\n",
      "ncrna 4557 , total =  3469581\n",
      "The GECO3 compression ratio for Drosophila_melanogaster.BDGP6.46.ncrna.fa is 18.15%\n",
      "The GZIP compression ratio for Drosophila_melanogaster.BDGP6.46.ncrna.fa is 21.96%\n",
      "drosophila_melanogaster completed successfully!\n",
      "cds 20344 , total =  40765283\n",
      "The GECO3 compression ratio for Drosophila_navojoa_gca001654015v2rs.UFRJ_Dnav_4.2.cds.all.fa is 14.35%\n",
      "The GZIP compression ratio for Drosophila_navojoa_gca001654015v2rs.UFRJ_Dnav_4.2.cds.all.fa is 15.79%\n",
      "ncrna 1302 , total =  509937\n",
      "The GECO3 compression ratio for Drosophila_navojoa_gca001654015v2rs.UFRJ_Dnav_4.2.ncrna.fa is 18.97%\n",
      "The GZIP compression ratio for Drosophila_navojoa_gca001654015v2rs.UFRJ_Dnav_4.2.ncrna.fa is 22.71%\n",
      "drosophila_navojoa_gca001654015v2rs completed successfully!\n",
      "cds 23284 , total =  44857842\n",
      "The GECO3 compression ratio for Drosophila_obscura_gca018151105v1rs.ASM1815110v1.cds.all.fa is 14.25%\n",
      "The GZIP compression ratio for Drosophila_obscura_gca018151105v1rs.ASM1815110v1.cds.all.fa is 15.89%\n",
      "cds 23305 , total =  51239950\n",
      "The GECO3 compression ratio for Drosophila_mojavensis_gca018153725v1rs.ASM1815372v1.cds.all.fa is 12.55%\n",
      "The GZIP compression ratio for Drosophila_mojavensis_gca018153725v1rs.ASM1815372v1.cds.all.fa is 13.97%\n",
      "ncrna 1781 , total =  1216947\n",
      "The GECO3 compression ratio for Drosophila_obscura_gca018151105v1rs.ASM1815110v1.ncrna.fa is 14.68%\n",
      "The GZIP compression ratio for Drosophila_obscura_gca018151105v1rs.ASM1815110v1.ncrna.fa is 18.97%\n",
      "drosophila_obscura_gca018151105v1rs completed successfully!\n",
      "ncrna 2366 , total =  2457890\n",
      "The GECO3 compression ratio for Drosophila_mojavensis_gca018153725v1rs.ASM1815372v1.ncrna.fa is 14.37%\n",
      "The GZIP compression ratio for Drosophila_mojavensis_gca018153725v1rs.ASM1815372v1.ncrna.fa is 18.36%\n",
      "drosophila_mojavensis_gca018153725v1rs completed successfully!\n",
      "cds 20342 , total =  38658510\n",
      "The GECO3 compression ratio for Drosophila_persimilis_gca003286085v2rs.DperRS2.cds.all.fa is 15.72%\n",
      "The GZIP compression ratio for Drosophila_persimilis_gca003286085v2rs.DperRS2.cds.all.fa is 18.14%\n",
      "ncrna 1266 , total =  716414\n",
      "The GECO3 compression ratio for Drosophila_persimilis_gca003286085v2rs.DperRS2.ncrna.fa is 11.88%\n",
      "The GZIP compression ratio for Drosophila_persimilis_gca003286085v2rs.DperRS2.ncrna.fa is 16.44%\n",
      "drosophila_persimilis_gca003286085v2rs completed successfully!\n",
      "cds 32149 , total =  67180216\n",
      "The GECO3 compression ratio for Drosophila_miranda_gca003369915v2rs.D.miranda_PacBio2.1.cds.all.fa is 11.39%\n",
      "The GZIP compression ratio for Drosophila_miranda_gca003369915v2rs.D.miranda_PacBio2.1.cds.all.fa is 14.23%\n",
      "ncrna 4065 , total =  4717584\n",
      "The GECO3 compression ratio for Drosophila_miranda_gca003369915v2rs.D.miranda_PacBio2.1.ncrna.fa is 8.82%\n",
      "The GZIP compression ratio for Drosophila_miranda_gca003369915v2rs.D.miranda_PacBio2.1.ncrna.fa is 15.85%\n",
      "drosophila_miranda_gca003369915v2rs completed successfully!\n",
      "cds 16469 , total =  21471225\n",
      "The GECO3 compression ratio for Drosophila_sechellia.dsec_caf1.cds.all.fa is 23.46%\n",
      "The GZIP compression ratio for Drosophila_sechellia.dsec_caf1.cds.all.fa is 28.01%\n",
      "ncrna 878 , total =  101116\n",
      "The GECO3 compression ratio for Drosophila_sechellia.dsec_caf1.ncrna.fa is 12.88%\n",
      "The GZIP compression ratio for Drosophila_sechellia.dsec_caf1.ncrna.fa is 15.64%\n",
      "drosophila_sechellia completed successfully!\n",
      "cds 23456 , total =  46708547\n",
      "The GECO3 compression ratio for Drosophila_pseudoobscura.Dpse_3.0.cds.all.fa is 13.60%\n",
      "The GZIP compression ratio for Drosophila_pseudoobscura.Dpse_3.0.cds.all.fa is 15.11%\n",
      "ncrna 3492 , total =  2616928\n",
      "The GECO3 compression ratio for Drosophila_pseudoobscura.Dpse_3.0.ncrna.fa is 13.16%\n",
      "The GZIP compression ratio for Drosophila_pseudoobscura.Dpse_3.0.ncrna.fa is 16.33%\n",
      "drosophila_pseudoobscura completed successfully!\n",
      "cds 23428 , total =  51057238\n",
      "The GECO3 compression ratio for Drosophila_rhopaloa_gca000236305v2rs.ASM1815211v1.cds.all.fa is 12.81%\n",
      "The GZIP compression ratio for Drosophila_rhopaloa_gca000236305v2rs.ASM1815211v1.cds.all.fa is 14.79%\n",
      "ncrna 965 , total =  573241\n",
      "The GECO3 compression ratio for Drosophila_rhopaloa_gca000236305v2rs.ASM1815211v1.ncrna.fa is 11.77%\n",
      "The GZIP compression ratio for Drosophila_rhopaloa_gca000236305v2rs.ASM1815211v1.ncrna.fa is 15.63%\n",
      "drosophila_rhopaloa_gca000236305v2rs completed successfully!\n",
      "cds 24461 , total =  54905971\n",
      "The GECO3 compression ratio for Drosophila_santomea_gca016746245v2rs.Prin_Dsan_1.1.cds.all.fa is 12.01%\n",
      "The GZIP compression ratio for Drosophila_santomea_gca016746245v2rs.Prin_Dsan_1.1.cds.all.fa is 13.89%\n",
      "cds 26574 , total =  62342145\n",
      "The GECO3 compression ratio for Drosophila_pseudoobscura_gca009870125v2rs.UCI_Dpse_MV25.cds.all.fa is 11.39%\n",
      "The GZIP compression ratio for Drosophila_pseudoobscura_gca009870125v2rs.UCI_Dpse_MV25.cds.all.fa is 12.66%\n",
      "ncrna 1429 , total =  1073066\n",
      "The GECO3 compression ratio for Drosophila_santomea_gca016746245v2rs.Prin_Dsan_1.1.ncrna.fa is 13.91%\n",
      "The GZIP compression ratio for Drosophila_santomea_gca016746245v2rs.Prin_Dsan_1.1.ncrna.fa is 17.88%\n",
      "drosophila_santomea_gca016746245v2rs completed successfully!\n",
      "cds 23935 , total =  50765624\n",
      "The GECO3 compression ratio for Drosophila_sechellia_gca004382195v2rs.ASM438219v2.cds.all.fa is 12.70%\n",
      "The GZIP compression ratio for Drosophila_sechellia_gca004382195v2rs.ASM438219v2.cds.all.fa is 14.35%\n",
      "ncrna 3230 , total =  3187348\n",
      "The GECO3 compression ratio for Drosophila_pseudoobscura_gca009870125v2rs.UCI_Dpse_MV25.ncrna.fa is 13.36%\n",
      "The GZIP compression ratio for Drosophila_pseudoobscura_gca009870125v2rs.UCI_Dpse_MV25.ncrna.fa is 17.58%\n",
      "drosophila_pseudoobscura_gca009870125v2rs completed successfully!\n",
      "cds 24119 , total =  49458036\n",
      "The GECO3 compression ratio for Drosophila_simulans.ASM75419v3.cds.all.fa is 12.58%\n",
      "The GZIP compression ratio for Drosophila_simulans.ASM75419v3.cds.all.fa is 14.15%\n",
      "ncrna 2468 , total =  2535012\n",
      "The GECO3 compression ratio for Drosophila_sechellia_gca004382195v2rs.ASM438219v2.ncrna.fa is 8.54%\n",
      "The GZIP compression ratio for Drosophila_sechellia_gca004382195v2rs.ASM438219v2.ncrna.fa is 12.19%\n",
      "drosophila_sechellia_gca004382195v2rs completed successfully!\n",
      "ncrna 1521 , total =  2268672\n",
      "The GECO3 compression ratio for Drosophila_simulans.ASM75419v3.ncrna.fa is 17.02%\n",
      "The GZIP compression ratio for Drosophila_simulans.ASM75419v3.ncrna.fa is 20.57%\n",
      "drosophila_simulans completed successfully!\n",
      "cds 26225 , total =  58699620\n",
      "The GECO3 compression ratio for Drosophila_simulans_gca016746395v2rs.Prin_Dsim_3.1.cds.all.fa is 11.51%\n",
      "The GZIP compression ratio for Drosophila_simulans_gca016746395v2rs.Prin_Dsim_3.1.cds.all.fa is 12.85%\n",
      "ncrna 1914 , total =  2636001\n",
      "The GECO3 compression ratio for Drosophila_simulans_gca016746395v2rs.Prin_Dsim_3.1.ncrna.fa is 12.64%\n",
      "The GZIP compression ratio for Drosophila_simulans_gca016746395v2rs.Prin_Dsim_3.1.ncrna.fa is 15.88%\n",
      "drosophila_simulans_gca016746395v2rs completed successfully!\n",
      "cds 22475 , total =  49279245\n",
      "The GECO3 compression ratio for Drosophila_subobscura_gca008121235v1rs.UCBerk_Dsub_1.0.cds.all.fa is 13.07%\n",
      "The GZIP compression ratio for Drosophila_subobscura_gca008121235v1rs.UCBerk_Dsub_1.0.cds.all.fa is 14.01%\n",
      "ncrna 991 , total =  581746\n",
      "The GECO3 compression ratio for Drosophila_subobscura_gca008121235v1rs.UCBerk_Dsub_1.0.ncrna.fa is 14.92%\n",
      "The GZIP compression ratio for Drosophila_subobscura_gca008121235v1rs.UCBerk_Dsub_1.0.ncrna.fa is 18.77%\n",
      "drosophila_subobscura_gca008121235v1rs completed successfully!\n",
      "cds 20007 , total =  36198403\n",
      "The GECO3 compression ratio for Drosophila_takahashii_gca018152695v1rs.ASM1815269v1.cds.all.fa is 16.70%\n",
      "The GZIP compression ratio for Drosophila_takahashii_gca018152695v1rs.ASM1815269v1.cds.all.fa is 19.05%\n",
      "ncrna 881 , total =  244887\n",
      "The GECO3 compression ratio for Drosophila_takahashii_gca018152695v1rs.ASM1815269v1.ncrna.fa is 15.55%\n",
      "The GZIP compression ratio for Drosophila_takahashii_gca018152695v1rs.ASM1815269v1.ncrna.fa is 19.14%\n",
      "drosophila_takahashii_gca018152695v1rs completed successfully!\n",
      "cds 25688 , total =  57003523\n",
      "The GECO3 compression ratio for Drosophila_subpulchrella_gca014743375v2rs.RU_Dsub_v1.1.cds.all.fa is 11.99%\n",
      "The GZIP compression ratio for Drosophila_subpulchrella_gca014743375v2rs.RU_Dsub_v1.1.cds.all.fa is 13.55%\n",
      "cds 20548 , total =  39716432\n",
      "The GECO3 compression ratio for Drosophila_virilis.dvir_caf1.cds.all.fa is 14.87%\n",
      "The GZIP compression ratio for Drosophila_virilis.dvir_caf1.cds.all.fa is 16.72%\n",
      "cds 25540 , total =  54271942\n",
      "The GECO3 compression ratio for Drosophila_suzukii_gca013340165v1rs.LBDM_Dsuz_2.1.pri.cds.all.fa is 12.47%\n",
      "The GZIP compression ratio for Drosophila_suzukii_gca013340165v1rs.LBDM_Dsuz_2.1.pri.cds.all.fa is 14.60%\n",
      "ncrna 2751 , total =  2675418\n",
      "The GECO3 compression ratio for Drosophila_subpulchrella_gca014743375v2rs.RU_Dsub_v1.1.ncrna.fa is 5.02%\n",
      "The GZIP compression ratio for Drosophila_subpulchrella_gca014743375v2rs.RU_Dsub_v1.1.ncrna.fa is 7.32%\n",
      "drosophila_subpulchrella_gca014743375v2rs completed successfully!\n",
      "ncrna 1828 , total =  608910\n",
      "The GECO3 compression ratio for Drosophila_virilis.dvir_caf1.ncrna.fa is 13.24%\n",
      "The GZIP compression ratio for Drosophila_virilis.dvir_caf1.ncrna.fa is 16.49%\n",
      "drosophila_virilis completed successfully!\n",
      "ncrna 2419 , total =  2297994\n",
      "The GECO3 compression ratio for Drosophila_suzukii_gca013340165v1rs.LBDM_Dsuz_2.1.pri.ncrna.fa is 9.01%\n",
      "The GZIP compression ratio for Drosophila_suzukii_gca013340165v1rs.LBDM_Dsuz_2.1.pri.ncrna.fa is 13.29%\n",
      "drosophila_suzukii_gca013340165v1rs completed successfully!\n",
      "cds 22853 , total =  46612915\n",
      "The GECO3 compression ratio for Drosophila_teissieri_gca016746235v2rs.Prin_Dtei_1.1.cds.all.fa is 13.69%\n",
      "The GZIP compression ratio for Drosophila_teissieri_gca016746235v2rs.Prin_Dtei_1.1.cds.all.fa is 15.25%\n",
      "cds 15182 , total =  25108260\n",
      "The GECO3 compression ratio for Drosophila_willistoni.dwil_caf1.cds.all.fa is 21.11%\n",
      "The GZIP compression ratio for Drosophila_willistoni.dwil_caf1.cds.all.fa is 24.71%\n",
      "ncrna 1033 , total =  690515\n",
      "The GECO3 compression ratio for Drosophila_teissieri_gca016746235v2rs.Prin_Dtei_1.1.ncrna.fa is 12.35%\n",
      "The GZIP compression ratio for Drosophila_teissieri_gca016746235v2rs.Prin_Dtei_1.1.ncrna.fa is 15.78%\n",
      "drosophila_teissieri_gca016746235v2rs completed successfully!\n",
      "ncrna 892 , total =  117323\n",
      "The GECO3 compression ratio for Drosophila_willistoni.dwil_caf1.ncrna.fa is 13.25%\n",
      "The GZIP compression ratio for Drosophila_willistoni.dwil_caf1.ncrna.fa is 16.31%\n",
      "drosophila_willistoni completed successfully!\n",
      "cds 23429 , total =  56415288\n",
      "The GECO3 compression ratio for Drosophila_virilis_gca003285735v2rs.DvirRS2.cds.all.fa is 11.61%\n",
      "The GZIP compression ratio for Drosophila_virilis_gca003285735v2rs.DvirRS2.cds.all.fa is 14.47%\n",
      "ncrna 2992 , total =  3111193\n",
      "The GECO3 compression ratio for Drosophila_virilis_gca003285735v2rs.DvirRS2.ncrna.fa is 14.78%\n",
      "The GZIP compression ratio for Drosophila_virilis_gca003285735v2rs.DvirRS2.ncrna.fa is 18.71%\n",
      "drosophila_virilis_gca003285735v2rs completed successfully!\n",
      "cds 21170 , total =  40975136\n",
      "The GECO3 compression ratio for Drosophila_willistoni_gca018902025v2rs.UCI_dwil_1.1.cds.all.fa is 15.15%\n",
      "The GZIP compression ratio for Drosophila_willistoni_gca018902025v2rs.UCI_dwil_1.1.cds.all.fa is 17.45%\n",
      "cds 11319 , total =  15867689\n",
      "The GECO3 compression ratio for Echinococcus_granulosus_gca000524195v1rs.ASM52419v1.cds.all.fa is 24.30%\n",
      "The GZIP compression ratio for Echinococcus_granulosus_gca000524195v1rs.ASM52419v1.cds.all.fa is 28.28%\n",
      "Directory or file not found for echinococcus_granulosus_gca000524195v1rs in ncrna folder: 550 Failed to change directory.\n",
      "echinococcus_granulosus_gca000524195v1rs completed successfully!\n",
      "ncrna 2224 , total =  2124423\n",
      "The GECO3 compression ratio for Drosophila_willistoni_gca018902025v2rs.UCI_dwil_1.1.ncrna.fa is 8.67%\n",
      "The GZIP compression ratio for Drosophila_willistoni_gca018902025v2rs.UCI_dwil_1.1.ncrna.fa is 12.28%\n",
      "drosophila_willistoni_gca018902025v2rs completed successfully!\n",
      "cds 12157 , total =  22032406\n",
      "The GECO3 compression ratio for Dufourea_novaeangliae_gca001272555v1rs.ASM127255v1.cds.all.fa is 19.85%\n",
      "The GZIP compression ratio for Dufourea_novaeangliae_gca001272555v1rs.ASM127255v1.cds.all.fa is 23.14%\n",
      "ncrna 334 , total =  87341\n",
      "The GECO3 compression ratio for Dufourea_novaeangliae_gca001272555v1rs.ASM127255v1.ncrna.fa is 21.14%\n",
      "The GZIP compression ratio for Dufourea_novaeangliae_gca001272555v1rs.ASM127255v1.ncrna.fa is 25.30%\n",
      "dufourea_novaeangliae_gca001272555v1rs completed successfully!\n",
      "cds 15659 , total =  30287632\n",
      "The GECO3 compression ratio for Eufriesea_mexicana_gca001483705v1rs.ASM148370v1.cds.all.fa is 15.38%\n",
      "The GZIP compression ratio for Eufriesea_mexicana_gca001483705v1rs.ASM148370v1.cds.all.fa is 17.90%\n",
      "cds 23304 , total =  47723542\n",
      "The GECO3 compression ratio for Drosophila_yakuba.dyak_caf1.cds.all.fa is 13.10%\n",
      "The GZIP compression ratio for Drosophila_yakuba.dyak_caf1.cds.all.fa is 15.23%\n",
      "ncrna 749 , total =  556361\n",
      "The GECO3 compression ratio for Eufriesea_mexicana_gca001483705v1rs.ASM148370v1.ncrna.fa is 18.01%\n",
      "The GZIP compression ratio for Eufriesea_mexicana_gca001483705v1rs.ASM148370v1.ncrna.fa is 21.87%\n",
      "eufriesea_mexicana_gca001483705v1rs completed successfully!\n",
      "ncrna 1325 , total =  468659\n",
      "The GECO3 compression ratio for Drosophila_yakuba.dyak_caf1.ncrna.fa is 15.45%\n",
      "The GZIP compression ratio for Drosophila_yakuba.dyak_caf1.ncrna.fa is 19.40%\n",
      "drosophila_yakuba completed successfully!\n",
      "cds 11873 , total =  18289869\n",
      "The GECO3 compression ratio for Galendromus_occidentalis_gca000255335v2rs.Mocc_1.0.cds.all.fa is 23.48%\n",
      "The GZIP compression ratio for Galendromus_occidentalis_gca000255335v2rs.Mocc_1.0.cds.all.fa is 27.85%\n",
      "ncrna 440 , total =  210855\n",
      "The GECO3 compression ratio for Galendromus_occidentalis_gca000255335v2rs.Mocc_1.0.ncrna.fa is 21.47%\n",
      "The GZIP compression ratio for Galendromus_occidentalis_gca000255335v2rs.Mocc_1.0.ncrna.fa is 25.22%\n",
      "galendromus_occidentalis_gca000255335v2rs completed successfully!\n",
      "cds 25398 , total =  59764222\n",
      "The GECO3 compression ratio for Drosophila_yakuba_gca016746365v2rs.Prin_Dyak_Tai18E2_2.1.cds.all.fa is 11.30%\n",
      "The GZIP compression ratio for Drosophila_yakuba_gca016746365v2rs.Prin_Dyak_Tai18E2_2.1.cds.all.fa is 13.47%\n",
      "ncrna 2209 , total =  3767980\n",
      "The GECO3 compression ratio for Drosophila_yakuba_gca016746365v2rs.Prin_Dyak_Tai18E2_2.1.ncrna.fa is 17.68%\n",
      "The GZIP compression ratio for Drosophila_yakuba_gca016746365v2rs.Prin_Dyak_Tai18E2_2.1.ncrna.fa is 21.32%\n",
      "drosophila_yakuba_gca016746365v2rs completed successfully!\n",
      "cds 30425 , total =  47331137\n",
      "The GECO3 compression ratio for Eurytemora_affinis_gca000591075v2.Eaff_2.0.cds.all.fa is 16.74%\n",
      "The GZIP compression ratio for Eurytemora_affinis_gca000591075v2.Eaff_2.0.cds.all.fa is 18.20%\n",
      "cds 27753 , total =  44113473\n",
      "The GECO3 compression ratio for Exaiptasia_diaphana_gca001417965v1.Aiptasia_genome_1.1.cds.all.fa is 19.43%\n",
      "The GZIP compression ratio for Exaiptasia_diaphana_gca001417965v1.Aiptasia_genome_1.1.cds.all.fa is 22.52%\n",
      "ncrna 3487 , total =  2131086\n",
      "The GECO3 compression ratio for Eurytemora_affinis_gca000591075v2.Eaff_2.0.ncrna.fa is 18.25%\n",
      "The GZIP compression ratio for Eurytemora_affinis_gca000591075v2.Eaff_2.0.ncrna.fa is 22.59%\n",
      "eurytemora_affinis_gca000591075v2 completed successfully!\n",
      "ncrna 2119 , total =  1437120\n",
      "The GECO3 compression ratio for Exaiptasia_diaphana_gca001417965v1.Aiptasia_genome_1.1.ncrna.fa is 20.50%\n",
      "The GZIP compression ratio for Exaiptasia_diaphana_gca001417965v1.Aiptasia_genome_1.1.ncrna.fa is 25.11%\n",
      "exaiptasia_diaphana_gca001417965v1 completed successfully!\n",
      "cds 28732 , total =  42707084\n",
      "The GECO3 compression ratio for Folsomia_candida.ASM221717v1.cds.all.fa is 23.07%\n",
      "The GZIP compression ratio for Folsomia_candida.ASM221717v1.cds.all.fa is 27.87%\n",
      "Directory or file not found for folsomia_candida in ncrna folder: 550 Failed to change directory.\n",
      "folsomia_candida completed successfully!\n",
      "cds 20166 , total =  32800432\n",
      "The GECO3 compression ratio for Galleria_mellonella_gca003640425v2rs.ASM364042v2.cds.all.fa is 15.36%\n",
      "The GZIP compression ratio for Galleria_mellonella_gca003640425v2rs.ASM364042v2.cds.all.fa is 17.71%\n",
      "ncrna 766 , total =  525578\n",
      "The GECO3 compression ratio for Galleria_mellonella_gca003640425v2rs.ASM364042v2.ncrna.fa is 14.79%\n",
      "The GZIP compression ratio for Galleria_mellonella_gca003640425v2rs.ASM364042v2.ncrna.fa is 18.08%\n",
      "galleria_mellonella_gca003640425v2rs completed successfully!\n",
      "cds 19731 , total =  24524936\n",
      "The GECO3 compression ratio for Glossina_austeni.GausT1.cds.all.fa is 23.50%\n",
      "The GZIP compression ratio for Glossina_austeni.GausT1.cds.all.fa is 27.95%\n",
      "ncrna 378 , total =  38553\n",
      "The GECO3 compression ratio for Glossina_austeni.GausT1.ncrna.fa is 16.46%\n",
      "The GZIP compression ratio for Glossina_austeni.GausT1.ncrna.fa is 20.37%\n",
      "glossina_austeni completed successfully!\n",
      "cds 14659 , total =  20688660\n",
      "The GECO3 compression ratio for Glossina_brevipalpis.GbreI1.cds.all.fa is 23.65%\n",
      "The GZIP compression ratio for Glossina_brevipalpis.GbreI1.cds.all.fa is 27.88%\n",
      "ncrna 335 , total =  32537\n",
      "The GECO3 compression ratio for Glossina_brevipalpis.GbreI1.ncrna.fa is 15.73%\n",
      "The GZIP compression ratio for Glossina_brevipalpis.GbreI1.ncrna.fa is 19.68%\n",
      "glossina_brevipalpis completed successfully!\n",
      "cds 12653 , total =  20848169\n",
      "The GECO3 compression ratio for Glossina_morsitans.GmorY1.cds.all.fa is 23.58%\n",
      "The GZIP compression ratio for Glossina_morsitans.GmorY1.cds.all.fa is 27.74%\n",
      "ncrna 376 , total =  42748\n",
      "The GECO3 compression ratio for Glossina_morsitans.GmorY1.ncrna.fa is 16.71%\n",
      "The GZIP compression ratio for Glossina_morsitans.GmorY1.ncrna.fa is 20.78%\n",
      "glossina_morsitans completed successfully!\n",
      "cds 33249 , total =  58645315\n",
      "The GECO3 compression ratio for Gigantopelta_aegis_gca016097555v1.Gae_host_genome.cds.all.fa is 16.19%\n",
      "The GZIP compression ratio for Gigantopelta_aegis_gca016097555v1.Gae_host_genome.cds.all.fa is 17.94%\n",
      "cds 19291 , total =  23444585\n",
      "The GECO3 compression ratio for Glossina_pallidipes.GpalI1.cds.all.fa is 23.52%\n",
      "The GZIP compression ratio for Glossina_pallidipes.GpalI1.cds.all.fa is 27.95%\n",
      "ncrna 360 , total =  35069\n",
      "The GECO3 compression ratio for Glossina_pallidipes.GpalI1.ncrna.fa is 16.02%\n",
      "The GZIP compression ratio for Glossina_pallidipes.GpalI1.ncrna.fa is 19.62%\n",
      "glossina_pallidipes completed successfully!\n",
      "ncrna 3516 , total =  2581078\n",
      "The GECO3 compression ratio for Gigantopelta_aegis_gca016097555v1.Gae_host_genome.ncrna.fa is 17.47%\n",
      "The GZIP compression ratio for Gigantopelta_aegis_gca016097555v1.Gae_host_genome.ncrna.fa is 22.60%\n",
      "gigantopelta_aegis_gca016097555v1 completed successfully!\n",
      "cds 20168 , total =  24693467\n",
      "The GECO3 compression ratio for Glossina_palpalis.GpapI1.cds.all.fa is 22.69%\n",
      "The GZIP compression ratio for Glossina_palpalis.GpapI1.cds.all.fa is 27.78%\n",
      "ncrna 373 , total =  38495\n",
      "The GECO3 compression ratio for Glossina_palpalis.GpapI1.ncrna.fa is 16.39%\n",
      "The GZIP compression ratio for Glossina_palpalis.GpapI1.ncrna.fa is 20.29%\n",
      "glossina_palpalis completed successfully!\n",
      "cds 19683 , total =  29283118\n",
      "The GECO3 compression ratio for Glyphotaelius_pellucidus_gca936435175v1.iiGlyPell1.1.cds.all.fa is 17.81%\n",
      "The GZIP compression ratio for Glyphotaelius_pellucidus_gca936435175v1.iiGlyPell1.1.cds.all.fa is 20.48%\n",
      "ncrna 2187 , total =  888020\n",
      "The GECO3 compression ratio for Glyphotaelius_pellucidus_gca936435175v1.iiGlyPell1.1.ncrna.fa is 17.81%\n",
      "The GZIP compression ratio for Glyphotaelius_pellucidus_gca936435175v1.iiGlyPell1.1.ncrna.fa is 23.27%\n",
      "glyphotaelius_pellucidus_gca936435175v1 completed successfully!\n",
      "cds 12256 , total =  21852051\n",
      "The GECO3 compression ratio for Habropoda_laboriosa_gca001263275v1rs.ASM126327v1.cds.all.fa is 20.20%\n",
      "The GZIP compression ratio for Habropoda_laboriosa_gca001263275v1rs.ASM126327v1.cds.all.fa is 23.70%\n",
      "cds 54884 , total =  118347399\n",
      "The GECO3 compression ratio for Eriocheir_sinensis_gca024679095v1rs.ASM2467909v1.cds.all.fa is 9.60%\n",
      "The GZIP compression ratio for Eriocheir_sinensis_gca024679095v1rs.ASM2467909v1.cds.all.fa is 9.49%\n",
      "ncrna 264 , total =  48893\n",
      "The GECO3 compression ratio for Habropoda_laboriosa_gca001263275v1rs.ASM126327v1.ncrna.fa is 19.93%\n",
      "The GZIP compression ratio for Habropoda_laboriosa_gca001263275v1rs.ASM126327v1.ncrna.fa is 24.32%\n",
      "habropoda_laboriosa_gca001263275v1rs completed successfully!\n",
      "cds 22320 , total =  49331002\n",
      "The GECO3 compression ratio for Glossina_fuscipes.Yale_Gfus_2.cds.all.fa is 12.21%\n",
      "The GZIP compression ratio for Glossina_fuscipes.Yale_Gfus_2.cds.all.fa is 13.96%\n",
      "ncrna 1666 , total =  1993462\n",
      "The GECO3 compression ratio for Glossina_fuscipes.Yale_Gfus_2.ncrna.fa is 16.79%\n",
      "The GZIP compression ratio for Glossina_fuscipes.Yale_Gfus_2.ncrna.fa is 20.58%\n",
      "glossina_fuscipes completed successfully!\n",
      "cds 27144 , total =  24208923\n",
      "The GECO3 compression ratio for Haemaphysalis_longicornis_gca013339765v1.ASM1333976v1.cds.all.fa is 21.90%\n",
      "The GZIP compression ratio for Haemaphysalis_longicornis_gca013339765v1.ASM1333976v1.cds.all.fa is 28.04%\n",
      "ncrna 6980 , total =  591093\n",
      "The GECO3 compression ratio for Haemaphysalis_longicornis_gca013339765v1.ASM1333976v1.ncrna.fa is 4.66%\n",
      "The GZIP compression ratio for Haemaphysalis_longicornis_gca013339765v1.ASM1333976v1.ncrna.fa is 7.28%\n",
      "haemaphysalis_longicornis_gca013339765v1 completed successfully!\n",
      "cds 12829 , total =  17467287\n",
      "The GECO3 compression ratio for Heliconius_melpomene.Hmel1.cds.all.fa is 23.62%\n",
      "The GZIP compression ratio for Heliconius_melpomene.Hmel1.cds.all.fa is 27.79%\n",
      "ncrna 4408 , total =  347482\n",
      "The GECO3 compression ratio for Heliconius_melpomene.Hmel1.ncrna.fa is 5.70%\n",
      "The GZIP compression ratio for Heliconius_melpomene.Hmel1.ncrna.fa is 8.69%\n",
      "heliconius_melpomene completed successfully!\n",
      "cds 25026 , total =  42733868\n",
      "The GECO3 compression ratio for Halyomorpha_halys_gca000696795v2rs.Hhal_2.0.cds.all.fa is 13.76%\n",
      "The GZIP compression ratio for Halyomorpha_halys_gca000696795v2rs.Hhal_2.0.cds.all.fa is 15.23%\n",
      "ncrna 2593 , total =  1298867\n",
      "The GECO3 compression ratio for Halyomorpha_halys_gca000696795v2rs.Hhal_2.0.ncrna.fa is 18.06%\n",
      "The GZIP compression ratio for Halyomorpha_halys_gca000696795v2rs.Hhal_2.0.ncrna.fa is 21.75%\n",
      "halyomorpha_halys_gca000696795v2rs completed successfully!\n",
      "ncrna 23706 , total =  47597522\n",
      "The GECO3 compression ratio for Eriocheir_sinensis_gca024679095v1rs.ASM2467909v1.ncrna.fa is 7.19%\n",
      "The GZIP compression ratio for Eriocheir_sinensis_gca024679095v1rs.ASM2467909v1.ncrna.fa is 8.96%\n",
      "eriocheir_sinensis_gca024679095v1rs completed successfully!\n",
      "cds 26772 , total =  55379877\n",
      "The GECO3 compression ratio for Harpegnathos_saltator_gca003227715v2rs.Hsal_v8.5.cds.all.fa is 10.76%\n",
      "The GZIP compression ratio for Harpegnathos_saltator_gca003227715v2rs.Hsal_v8.5.cds.all.fa is 12.46%\n",
      "ncrna 1735 , total =  1605911\n",
      "The GECO3 compression ratio for Harpegnathos_saltator_gca003227715v2rs.Hsal_v8.5.ncrna.fa is 15.40%\n",
      "The GZIP compression ratio for Harpegnathos_saltator_gca003227715v2rs.Hsal_v8.5.ncrna.fa is 19.76%\n",
      "harpegnathos_saltator_gca003227715v2rs completed successfully!\n",
      "cds 23432 , total =  26411046\n",
      "The GECO3 compression ratio for Helobdella_robusta.Helro1.cds.all.fa is 22.37%\n",
      "The GZIP compression ratio for Helobdella_robusta.Helro1.cds.all.fa is 27.66%\n",
      "ncrna 437 , total =  52682\n",
      "The GECO3 compression ratio for Helobdella_robusta.Helro1.ncrna.fa is 8.76%\n",
      "The GZIP compression ratio for Helobdella_robusta.Helro1.ncrna.fa is 11.54%\n",
      "helobdella_robusta completed successfully!\n",
      "cds 43422 , total =  66451013\n",
      "The GECO3 compression ratio for Haliotis_rubra_gca003918875v1rs.ASM391887v1.cds.all.fa is 16.81%\n",
      "The GZIP compression ratio for Haliotis_rubra_gca003918875v1rs.ASM391887v1.cds.all.fa is 19.31%\n",
      "cds 23696 , total =  47328844\n",
      "The GECO3 compression ratio for Helicoverpa_zea_gca022581195v1rs.ilHelZeax1.1.cds.all.fa is 13.25%\n",
      "The GZIP compression ratio for Helicoverpa_zea_gca022581195v1rs.ilHelZeax1.1.cds.all.fa is 15.97%\n",
      "ncrna 1948 , total =  1174899\n",
      "The GECO3 compression ratio for Helicoverpa_zea_gca022581195v1rs.ilHelZeax1.1.ncrna.fa is 16.09%\n",
      "The GZIP compression ratio for Helicoverpa_zea_gca022581195v1rs.ilHelZeax1.1.ncrna.fa is 23.59%\n",
      "helicoverpa_zea_gca022581195v1rs completed successfully!\n",
      "ncrna 10950 , total =  6406570\n",
      "The GECO3 compression ratio for Haliotis_rubra_gca003918875v1rs.ASM391887v1.ncrna.fa is 16.86%\n",
      "The GZIP compression ratio for Haliotis_rubra_gca003918875v1rs.ASM391887v1.ncrna.fa is 21.11%\n",
      "haliotis_rubra_gca003918875v1rs completed successfully!\n",
      "cds 28010 , total =  61201765\n",
      "The GECO3 compression ratio for Helicoverpa_armigera_gca023701775v1rs.HaSCD2.cds.all.fa is 11.13%\n",
      "The GZIP compression ratio for Helicoverpa_armigera_gca023701775v1rs.HaSCD2.cds.all.fa is 13.03%\n",
      "ncrna 2352 , total =  2342220\n",
      "The GECO3 compression ratio for Helicoverpa_armigera_gca023701775v1rs.HaSCD2.ncrna.fa is 9.65%\n",
      "The GZIP compression ratio for Helicoverpa_armigera_gca023701775v1rs.HaSCD2.ncrna.fa is 16.30%\n",
      "helicoverpa_armigera_gca023701775v1rs completed successfully!\n",
      "cds 22454 , total =  26825574\n",
      "The GECO3 compression ratio for Hofstenia_miamia.HmiaM1.cds.all.fa is 22.74%\n",
      "The GZIP compression ratio for Hofstenia_miamia.HmiaM1.cds.all.fa is 27.70%\n",
      "ncrna 849 , total =  73021\n",
      "The GECO3 compression ratio for Hofstenia_miamia.HmiaM1.ncrna.fa is 8.91%\n",
      "The GZIP compression ratio for Hofstenia_miamia.HmiaM1.ncrna.fa is 11.75%\n",
      "hofstenia_miamia completed successfully!\n",
      "cds 24828 , total =  51779013\n",
      "The GECO3 compression ratio for Hermetia_illucens_gca905115235v1.iHerIll2.2.curated.20191125.cds.all.fa is 12.30%\n",
      "The GZIP compression ratio for Hermetia_illucens_gca905115235v1.iHerIll2.2.curated.20191125.cds.all.fa is 14.21%\n",
      "ncrna 1785 , total =  1338885\n",
      "The GECO3 compression ratio for Hermetia_illucens_gca905115235v1.iHerIll2.2.curated.20191125.ncrna.fa is 17.94%\n",
      "The GZIP compression ratio for Hermetia_illucens_gca905115235v1.iHerIll2.2.curated.20191125.ncrna.fa is 22.29%\n",
      "hermetia_illucens_gca905115235v1 completed successfully!\n",
      "cds 55609 , total =  113216253\n",
      "The GECO3 compression ratio for Haliotis_rufescens_gca023055435v1rs.xgHalRufe1.0.p.cds.all.fa is 13.18%\n",
      "The GZIP compression ratio for Haliotis_rufescens_gca023055435v1rs.xgHalRufe1.0.p.cds.all.fa is 13.56%\n",
      "cds 31163 , total =  50440740\n",
      "The GECO3 compression ratio for Homalodisca_vitripennis_gca021130785v2rs.UT_GWSS_2.1.cds.all.fa is 13.21%\n",
      "The GZIP compression ratio for Homalodisca_vitripennis_gca021130785v2rs.UT_GWSS_2.1.cds.all.fa is 16.22%\n",
      "cds 29644 , total =  28460148\n",
      "The GECO3 compression ratio for Hyalomma_asiaticum_gca013339685v1.ASM1333968v1.cds.all.fa is 21.54%\n",
      "The GZIP compression ratio for Hyalomma_asiaticum_gca013339685v1.ASM1333968v1.cds.all.fa is 28.04%\n",
      "ncrna 1468 , total =  157699\n",
      "The GECO3 compression ratio for Hyalomma_asiaticum_gca013339685v1.ASM1333968v1.ncrna.fa is 9.74%\n",
      "The GZIP compression ratio for Hyalomma_asiaticum_gca013339685v1.ASM1333968v1.ncrna.fa is 12.98%\n",
      "hyalomma_asiaticum_gca013339685v1 completed successfully!\n",
      "ncrna 2240 , total =  1984426\n",
      "The GECO3 compression ratio for Homalodisca_vitripennis_gca021130785v2rs.UT_GWSS_2.1.ncrna.fa is 16.86%\n",
      "The GZIP compression ratio for Homalodisca_vitripennis_gca021130785v2rs.UT_GWSS_2.1.ncrna.fa is 21.10%\n",
      "homalodisca_vitripennis_gca021130785v2rs completed successfully!\n",
      "ncrna 12345 , total =  6341950\n",
      "The GECO3 compression ratio for Haliotis_rufescens_gca023055435v1rs.xgHalRufe1.0.p.ncrna.fa is 10.89%\n",
      "The GZIP compression ratio for Haliotis_rufescens_gca023055435v1rs.xgHalRufe1.0.p.ncrna.fa is 15.78%\n",
      "haliotis_rufescens_gca023055435v1rs completed successfully!\n",
      "cds 22749 , total =  38652570\n",
      "The GECO3 compression ratio for Hyalella_azteca_gca000764305v2.Hazt_2.0.cds.all.fa is 19.12%\n",
      "The GZIP compression ratio for Hyalella_azteca_gca000764305v2.Hazt_2.0.cds.all.fa is 22.12%\n",
      "ncrna 1401 , total =  935918\n",
      "The GECO3 compression ratio for Hyalella_azteca_gca000764305v2.Hazt_2.0.ncrna.fa is 20.80%\n",
      "The GZIP compression ratio for Hyalella_azteca_gca000764305v2.Hazt_2.0.ncrna.fa is 25.05%\n",
      "hyalella_azteca_gca000764305v2 completed successfully!\n",
      "cds 11429 , total =  19766113\n",
      "The GECO3 compression ratio for Hymenolepis_microstoma.HMN_v3.cds.all.fa is 19.77%\n",
      "The GZIP compression ratio for Hymenolepis_microstoma.HMN_v3.cds.all.fa is 23.82%\n",
      "Directory or file not found for hymenolepis_microstoma in ncrna folder: 550 Failed to change directory.\n",
      "hymenolepis_microstoma completed successfully!\n",
      "cds 40732 , total =  77359310\n",
      "The GECO3 compression ratio for Homarus_americanus_gca018991925v1.GMGI_Hamer_2.0.cds.all.fa is 12.29%\n",
      "The GZIP compression ratio for Homarus_americanus_gca018991925v1.GMGI_Hamer_2.0.cds.all.fa is 13.38%\n",
      "cds 15859 , total =  23062512\n",
      "The GECO3 compression ratio for Ichneumon_xanthorius_gca917499995v1.iyIchXant1.1.cds.all.fa is 18.73%\n",
      "The GZIP compression ratio for Ichneumon_xanthorius_gca917499995v1.iyIchXant1.1.cds.all.fa is 21.25%\n",
      "ncrna 1628 , total =  579849\n",
      "The GECO3 compression ratio for Ichneumon_xanthorius_gca917499995v1.iyIchXant1.1.ncrna.fa is 19.58%\n",
      "The GZIP compression ratio for Ichneumon_xanthorius_gca917499995v1.iyIchXant1.1.ncrna.fa is 23.35%\n",
      "ichneumon_xanthorius_gca917499995v1 completed successfully!\n",
      "cds 32703 , total =  56983009\n",
      "The GECO3 compression ratio for Hydra_vulgaris_gca022113875v1rs.Hydra_105_v3.cds.all.fa is 14.68%\n",
      "The GZIP compression ratio for Hydra_vulgaris_gca022113875v1rs.Hydra_105_v3.cds.all.fa is 16.91%\n",
      "ncrna 5358 , total =  5885919\n",
      "The GECO3 compression ratio for Homarus_americanus_gca018991925v1.GMGI_Hamer_2.0.ncrna.fa is 11.30%\n",
      "The GZIP compression ratio for Homarus_americanus_gca018991925v1.GMGI_Hamer_2.0.ncrna.fa is 16.04%\n",
      "homarus_americanus_gca018991925v1 completed successfully!\n",
      "cds 20853 , total =  27076196\n",
      "The GECO3 compression ratio for Hypsibius_exemplaris_gca002082055v1.nHd_3.1.cds.all.fa is 22.39%\n",
      "The GZIP compression ratio for Hypsibius_exemplaris_gca002082055v1.nHd_3.1.cds.all.fa is 26.37%\n",
      "ncrna 138 , total =  22718\n",
      "The GECO3 compression ratio for Hypsibius_exemplaris_gca002082055v1.nHd_3.1.ncrna.fa is 16.59%\n",
      "The GZIP compression ratio for Hypsibius_exemplaris_gca002082055v1.nHd_3.1.ncrna.fa is 20.56%\n",
      "hypsibius_exemplaris_gca002082055v1 completed successfully!\n",
      "cds 54006 , total =  77219386\n",
      "The GECO3 compression ratio for Homarus_gammarus_gca958450375v1.Homarus_gammarus.cds.all.fa is 12.98%\n",
      "The GZIP compression ratio for Homarus_gammarus_gca958450375v1.Homarus_gammarus.cds.all.fa is 12.99%\n",
      "cds 26883 , total =  29005395\n",
      "The GECO3 compression ratio for Ixodes_persulcatus_gca013358835v1.BIME_Iper_1.3.cds.all.fa is 20.48%\n",
      "The GZIP compression ratio for Ixodes_persulcatus_gca013358835v1.BIME_Iper_1.3.cds.all.fa is 27.98%\n",
      "ncrna 5258 , total =  598079\n",
      "The GECO3 compression ratio for Ixodes_persulcatus_gca013358835v1.BIME_Iper_1.3.ncrna.fa is 5.74%\n",
      "The GZIP compression ratio for Ixodes_persulcatus_gca013358835v1.BIME_Iper_1.3.ncrna.fa is 9.40%\n",
      "ixodes_persulcatus_gca013358835v1 completed successfully!\n",
      "ncrna 9213 , total =  7624517\n",
      "The GECO3 compression ratio for Hydra_vulgaris_gca022113875v1rs.Hydra_105_v3.ncrna.fa is 12.89%\n",
      "The GZIP compression ratio for Hydra_vulgaris_gca022113875v1rs.Hydra_105_v3.ncrna.fa is 19.63%\n",
      "hydra_vulgaris_gca022113875v1rs completed successfully!\n",
      "cds 28497 , total =  51694594\n",
      "The GECO3 compression ratio for Hydractinia_symbiolongicarpus_gca029227915v2rs.HSymV2.1.cds.all.fa is 15.87%\n",
      "The GZIP compression ratio for Hydractinia_symbiolongicarpus_gca029227915v2rs.HSymV2.1.cds.all.fa is 19.31%\n",
      "cds 20488 , total =  17510164\n",
      "The GECO3 compression ratio for Ixodes_scapularis.IscaW1.cds.all.fa is 23.66%\n",
      "The GZIP compression ratio for Ixodes_scapularis.IscaW1.cds.all.fa is 27.81%\n",
      "ncrna 2854 , total =  279304\n",
      "The GECO3 compression ratio for Ixodes_scapularis.IscaW1.ncrna.fa is 9.71%\n",
      "The GZIP compression ratio for Ixodes_scapularis.IscaW1.ncrna.fa is 14.03%\n",
      "ixodes_scapularis completed successfully!\n",
      "ncrna 35635 , total =  16347740\n",
      "The GECO3 compression ratio for Homarus_gammarus_gca958450375v1.Homarus_gammarus.ncrna.fa is 12.86%\n",
      "The GZIP compression ratio for Homarus_gammarus_gca958450375v1.Homarus_gammarus.ncrna.fa is 18.33%\n",
      "homarus_gammarus_gca958450375v1 completed successfully!\n",
      "cds 19062 , total =  16008358\n",
      "The GECO3 compression ratio for Ixodes_scapularis_ise6.IscaI1.cds.all.fa is 23.72%\n",
      "The GZIP compression ratio for Ixodes_scapularis_ise6.IscaI1.cds.all.fa is 27.99%\n",
      "ncrna 29959 , total =  9776389\n",
      "The GECO3 compression ratio for Hydractinia_symbiolongicarpus_gca029227915v2rs.HSymV2.1.ncrna.fa is 4.50%\n",
      "The GZIP compression ratio for Hydractinia_symbiolongicarpus_gca029227915v2rs.HSymV2.1.ncrna.fa is 7.51%\n",
      "hydractinia_symbiolongicarpus_gca029227915v2rs completed successfully!\n",
      "ncrna 4408 , total =  456593\n",
      "The GECO3 compression ratio for Ixodes_scapularis_ise6.IscaI1.ncrna.fa is 6.86%\n",
      "The GZIP compression ratio for Ixodes_scapularis_ise6.IscaI1.ncrna.fa is 10.30%\n",
      "ixodes_scapularis_ise6 completed successfully!\n",
      "cds 14667 , total =  15139752\n",
      "The GECO3 compression ratio for Leptotrombidium_deliense.LdelU1.cds.all.fa is 23.79%\n",
      "The GZIP compression ratio for Leptotrombidium_deliense.LdelU1.cds.all.fa is 27.90%\n",
      "ncrna 348 , total =  32381\n",
      "The GECO3 compression ratio for Leptotrombidium_deliense.LdelU1.ncrna.fa is 15.93%\n",
      "The GZIP compression ratio for Leptotrombidium_deliense.LdelU1.ncrna.fa is 20.64%\n",
      "leptotrombidium_deliense completed successfully!\n",
      "cds 19038 , total =  29668304\n",
      "The GECO3 compression ratio for Leptinotarsa_decemlineata_gca000500325v2.Ldec_2.0.cds.all.fa is 16.96%\n",
      "The GZIP compression ratio for Leptinotarsa_decemlineata_gca000500325v2.Ldec_2.0.cds.all.fa is 19.81%\n",
      "cds 23735 , total =  39631702\n",
      "The GECO3 compression ratio for Leguminivora_glycinivorella_gca023078275v1rs.LegGlyc_1.1.cds.all.fa is 14.91%\n",
      "The GZIP compression ratio for Leguminivora_glycinivorella_gca023078275v1rs.LegGlyc_1.1.cds.all.fa is 18.42%\n",
      "ncrna 2358 , total =  1587880\n",
      "The GECO3 compression ratio for Leptinotarsa_decemlineata_gca000500325v2.Ldec_2.0.ncrna.fa is 19.30%\n",
      "The GZIP compression ratio for Leptinotarsa_decemlineata_gca000500325v2.Ldec_2.0.ncrna.fa is 23.72%\n",
      "leptinotarsa_decemlineata_gca000500325v2 completed successfully!\n",
      "ncrna 2532 , total =  1664577\n",
      "The GECO3 compression ratio for Leguminivora_glycinivorella_gca023078275v1rs.LegGlyc_1.1.ncrna.fa is 8.50%\n",
      "The GZIP compression ratio for Leguminivora_glycinivorella_gca023078275v1rs.LegGlyc_1.1.ncrna.fa is 20.52%\n",
      "leguminivora_glycinivorella_gca023078275v1rs completed successfully!\n",
      "cds 20654 , total =  34670499\n",
      "The GECO3 compression ratio for Lepeophtheirus_salmonis_gca016086655v3rs.UVic_Lsal_1.0.cds.all.fa is 15.89%\n",
      "The GZIP compression ratio for Lepeophtheirus_salmonis_gca016086655v3rs.UVic_Lsal_1.0.cds.all.fa is 18.24%\n",
      "cds 19679 , total =  26895987\n",
      "The GECO3 compression ratio for Limnephilus_marmoratus_gca917880885v1.iiLimMarm1.1.cds.all.fa is 18.48%\n",
      "The GZIP compression ratio for Limnephilus_marmoratus_gca917880885v1.iiLimMarm1.1.cds.all.fa is 21.48%\n",
      "cds 19016 , total =  26254015\n",
      "The GECO3 compression ratio for Limnephilus_rhombicus_gca929108145v2.iiLimRhom1.2.cds.all.fa is 18.54%\n",
      "The GZIP compression ratio for Limnephilus_rhombicus_gca929108145v2.iiLimRhom1.2.cds.all.fa is 21.50%\n",
      "cds 19302 , total =  27578645\n",
      "The GECO3 compression ratio for Limnephilus_lunatus_gca917563855v2.iiLimLuna2.2.cds.all.fa is 18.26%\n",
      "The GZIP compression ratio for Limnephilus_lunatus_gca917563855v2.iiLimLuna2.2.cds.all.fa is 21.20%\n",
      "ncrna 3290 , total =  993179\n",
      "The GECO3 compression ratio for Limnephilus_marmoratus_gca917880885v1.iiLimMarm1.1.ncrna.fa is 17.52%\n",
      "The GZIP compression ratio for Limnephilus_marmoratus_gca917880885v1.iiLimMarm1.1.ncrna.fa is 22.81%\n",
      "limnephilus_marmoratus_gca917880885v1 completed successfully!\n",
      "cds 34235 , total =  50886965\n",
      "The GECO3 compression ratio for Ixodes_scapularis_gca016920785v2.ASM1692078v2.cds.all.fa is 15.39%\n",
      "The GZIP compression ratio for Ixodes_scapularis_gca016920785v2.ASM1692078v2.cds.all.fa is 20.65%\n",
      "ncrna 3712 , total =  1017417\n",
      "The GECO3 compression ratio for Limnephilus_rhombicus_gca929108145v2.iiLimRhom1.2.ncrna.fa is 16.23%\n",
      "The GZIP compression ratio for Limnephilus_rhombicus_gca929108145v2.iiLimRhom1.2.ncrna.fa is 20.87%\n",
      "limnephilus_rhombicus_gca929108145v2 completed successfully!\n",
      "ncrna 2712 , total =  1024025\n",
      "The GECO3 compression ratio for Limnephilus_lunatus_gca917563855v2.iiLimLuna2.2.ncrna.fa is 18.13%\n",
      "The GZIP compression ratio for Limnephilus_lunatus_gca917563855v2.iiLimLuna2.2.ncrna.fa is 23.42%\n",
      "limnephilus_lunatus_gca917563855v2 completed successfully!\n",
      "ncrna 5813 , total =  7613695\n",
      "The GECO3 compression ratio for Lepeophtheirus_salmonis_gca016086655v3rs.UVic_Lsal_1.0.ncrna.fa is 15.17%\n",
      "The GZIP compression ratio for Lepeophtheirus_salmonis_gca016086655v3rs.UVic_Lsal_1.0.ncrna.fa is 19.36%\n",
      "lepeophtheirus_salmonis_gca016086655v3rs completed successfully!\n",
      "ncrna 6331 , total =  2930145\n",
      "The GECO3 compression ratio for Ixodes_scapularis_gca016920785v2.ASM1692078v2.ncrna.fa is 11.05%\n",
      "The GZIP compression ratio for Ixodes_scapularis_gca016920785v2.ASM1692078v2.ncrna.fa is 18.97%\n",
      "ixodes_scapularis_gca016920785v2 completed successfully!\n",
      "cds 15445 , total =  15532952\n",
      "The GECO3 compression ratio for Loa_loa.Loa_loa_V3.cds.all.fa is 22.80%\n",
      "The GZIP compression ratio for Loa_loa.Loa_loa_V3.cds.all.fa is 26.65%\n",
      "ncrna 221 , total =  27449\n",
      "The GECO3 compression ratio for Loa_loa.Loa_loa_V3.ncrna.fa is 17.15%\n",
      "The GZIP compression ratio for Loa_loa.Loa_loa_V3.ncrna.fa is 21.93%\n",
      "loa_loa completed successfully!\n",
      "cds 14452 , total =  22505748\n",
      "The GECO3 compression ratio for Lucilia_cuprina.ASM118794v1.cds.all.fa is 23.21%\n",
      "The GZIP compression ratio for Lucilia_cuprina.ASM118794v1.cds.all.fa is 27.76%\n",
      "ncrna 767 , total =  65626\n",
      "The GECO3 compression ratio for Lucilia_cuprina.ASM118794v1.ncrna.fa is 8.64%\n",
      "The GZIP compression ratio for Lucilia_cuprina.ASM118794v1.ncrna.fa is 10.43%\n",
      "lucilia_cuprina completed successfully!\n",
      "cds 23340 , total =  26846023\n",
      "The GECO3 compression ratio for Lottia_gigantea.Lotgi1.cds.all.fa is 21.74%\n",
      "The GZIP compression ratio for Lottia_gigantea.Lotgi1.cds.all.fa is 26.82%\n",
      "cds 21674 , total =  43462176\n",
      "The GECO3 compression ratio for Linepithema_humile_gca000217595v1rs.Lhum_UMD_V04.cds.all.fa is 12.71%\n",
      "The GZIP compression ratio for Linepithema_humile_gca000217595v1rs.Lhum_UMD_V04.cds.all.fa is 14.48%\n",
      "ncrna 1151 , total =  115702\n",
      "The GECO3 compression ratio for Lottia_gigantea.Lotgi1.ncrna.fa is 6.46%\n",
      "The GZIP compression ratio for Lottia_gigantea.Lotgi1.ncrna.fa is 8.20%\n",
      "lottia_gigantea completed successfully!\n",
      "ncrna 1589 , total =  1651657\n",
      "The GECO3 compression ratio for Linepithema_humile_gca000217595v1rs.Lhum_UMD_V04.ncrna.fa is 16.85%\n",
      "The GZIP compression ratio for Linepithema_humile_gca000217595v1rs.Lhum_UMD_V04.ncrna.fa is 20.69%\n",
      "linepithema_humile_gca000217595v1rs completed successfully!\n",
      "cds 10457 , total =  15548804\n",
      "The GECO3 compression ratio for Lutzomyia_longipalpis.LlonJ1.cds.all.fa is 22.36%\n",
      "The GZIP compression ratio for Lutzomyia_longipalpis.LlonJ1.cds.all.fa is 26.66%\n",
      "ncrna 338 , total =  34773\n",
      "The GECO3 compression ratio for Lutzomyia_longipalpis.LlonJ1.ncrna.fa is 13.98%\n",
      "The GZIP compression ratio for Lutzomyia_longipalpis.LlonJ1.ncrna.fa is 17.60%\n",
      "lutzomyia_longipalpis completed successfully!\n",
      "cds 34594 , total =  51874135\n",
      "The GECO3 compression ratio for Lineus_longissimus_gca910592395v2.tnLinLong1.2.cds.all.fa is 16.52%\n",
      "The GZIP compression ratio for Lineus_longissimus_gca910592395v2.tnLinLong1.2.cds.all.fa is 17.85%\n",
      "cds 38676 , total =  64050038\n",
      "The GECO3 compression ratio for Limulus_polyphemus_gca000517525v1.Limulus_polyphemus_2.1.2.cds.all.fa is 15.72%\n",
      "The GZIP compression ratio for Limulus_polyphemus_gca000517525v1.Limulus_polyphemus_2.1.2.cds.all.fa is 16.16%\n",
      "ncrna 7306 , total =  2566131\n",
      "The GECO3 compression ratio for Lineus_longissimus_gca910592395v2.tnLinLong1.2.ncrna.fa is 16.09%\n",
      "The GZIP compression ratio for Lineus_longissimus_gca910592395v2.tnLinLong1.2.ncrna.fa is 21.96%\n",
      "lineus_longissimus_gca910592395v2 completed successfully!\n",
      "ncrna 4928 , total =  5054400\n",
      "The GECO3 compression ratio for Limulus_polyphemus_gca000517525v1.Limulus_polyphemus_2.1.2.ncrna.fa is 17.48%\n",
      "The GZIP compression ratio for Limulus_polyphemus_gca000517525v1.Limulus_polyphemus_2.1.2.ncrna.fa is 20.74%\n",
      "limulus_polyphemus_gca000517525v1 completed successfully!\n",
      "cds 22082 , total =  47417549\n",
      "The GECO3 compression ratio for Lucilia_cuprina_gca022045245v1rs.ASM2204524v1.cds.all.fa is 13.51%\n",
      "The GZIP compression ratio for Lucilia_cuprina_gca022045245v1rs.ASM2204524v1.cds.all.fa is 15.33%\n",
      "ncrna 2035 , total =  1378692\n",
      "The GECO3 compression ratio for Lucilia_cuprina_gca022045245v1rs.ASM2204524v1.ncrna.fa is 17.02%\n",
      "The GZIP compression ratio for Lucilia_cuprina_gca022045245v1rs.ASM2204524v1.ncrna.fa is 21.08%\n",
      "lucilia_cuprina_gca022045245v1rs completed successfully!\n",
      "cds 41392 , total =  73904219\n",
      "The GECO3 compression ratio for Lingula_anatina_gca001039355v2.LinAna2.0.cds.all.fa is 15.09%\n",
      "The GZIP compression ratio for Lingula_anatina_gca001039355v2.LinAna2.0.cds.all.fa is 17.45%\n",
      "cds 68899 , total =  95694974\n",
      "The GECO3 compression ratio for Limnoperna_fortunei_gca944474755v1.xbLimFort5.1.cds.all.fa is 15.24%\n",
      "The GZIP compression ratio for Limnoperna_fortunei_gca944474755v1.xbLimFort5.1.cds.all.fa is 14.39%\n",
      "ncrna 3285 , total =  2015021\n",
      "The GECO3 compression ratio for Lingula_anatina_gca001039355v2.LinAna2.0.ncrna.fa is 17.93%\n",
      "The GZIP compression ratio for Lingula_anatina_gca001039355v2.LinAna2.0.ncrna.fa is 23.54%\n",
      "lingula_anatina_gca001039355v2 completed successfully!\n",
      "cds 16754 , total =  26532634\n",
      "The GECO3 compression ratio for Machimus_atricapillus_gca933228815v1.idMacAtri3.1.cds.all.fa is 17.95%\n",
      "The GZIP compression ratio for Machimus_atricapillus_gca933228815v1.idMacAtri3.1.cds.all.fa is 20.41%\n",
      "ncrna 708 , total =  197693\n",
      "The GECO3 compression ratio for Machimus_atricapillus_gca933228815v1.idMacAtri3.1.ncrna.fa is 15.89%\n",
      "The GZIP compression ratio for Machimus_atricapillus_gca933228815v1.idMacAtri3.1.ncrna.fa is 19.26%\n",
      "machimus_atricapillus_gca933228815v1 completed successfully!\n",
      "cds 27757 , total =  45523637\n",
      "The GECO3 compression ratio for Lytechinus_pictus_gca015342785v2rs.UCSD_Lpic_2.1.cds.all.fa is 19.15%\n",
      "The GZIP compression ratio for Lytechinus_pictus_gca015342785v2rs.UCSD_Lpic_2.1.cds.all.fa is 23.72%\n",
      "cds 11463 , total =  7617449\n",
      "The GECO3 compression ratio for Megaselia_scalaris.Msca1.cds.all.fa is 23.63%\n",
      "The GZIP compression ratio for Megaselia_scalaris.Msca1.cds.all.fa is 27.81%\n",
      "ncrna 236 , total =  22628\n",
      "The GECO3 compression ratio for Megaselia_scalaris.Msca1.ncrna.fa is 17.70%\n",
      "The GZIP compression ratio for Megaselia_scalaris.Msca1.ncrna.fa is 22.24%\n",
      "megaselia_scalaris completed successfully!\n",
      "ncrna 4495 , total =  2576296\n",
      "The GECO3 compression ratio for Lytechinus_pictus_gca015342785v2rs.UCSD_Lpic_2.1.ncrna.fa is 17.01%\n",
      "The GZIP compression ratio for Lytechinus_pictus_gca015342785v2rs.UCSD_Lpic_2.1.ncrna.fa is 23.99%\n",
      "lytechinus_pictus_gca015342785v2rs completed successfully!\n",
      "cds 22410 , total =  27884220\n",
      "The GECO3 compression ratio for Mayetiola_destructor.Mdes_1.0.cds.all.fa is 19.93%\n",
      "The GZIP compression ratio for Mayetiola_destructor.Mdes_1.0.cds.all.fa is 22.84%\n",
      "ncrna 250 , total =  23644\n",
      "The GECO3 compression ratio for Mayetiola_destructor.Mdes_1.0.ncrna.fa is 17.41%\n",
      "The GZIP compression ratio for Mayetiola_destructor.Mdes_1.0.ncrna.fa is 22.04%\n",
      "mayetiola_destructor completed successfully!\n",
      "ncrna 73900 , total =  28679934\n",
      "The GECO3 compression ratio for Limnoperna_fortunei_gca944474755v1.xbLimFort5.1.ncrna.fa is 12.08%\n",
      "The GZIP compression ratio for Limnoperna_fortunei_gca944474755v1.xbLimFort5.1.ncrna.fa is 17.15%\n",
      "limnoperna_fortunei_gca944474755v1 completed successfully!\n",
      "cds 24854 , total =  45328183\n",
      "The GECO3 compression ratio for Manduca_sexta_gca014839805v1rs.JHU_Msex_v1.0.cds.all.fa is 13.73%\n",
      "The GZIP compression ratio for Manduca_sexta_gca014839805v1rs.JHU_Msex_v1.0.cds.all.fa is 16.50%\n",
      "cds 16668 , total =  15977918\n",
      "The GECO3 compression ratio for Melitaea_cinxia.MelCinx1.0.cds.all.fa is 23.82%\n",
      "The GZIP compression ratio for Melitaea_cinxia.MelCinx1.0.cds.all.fa is 28.07%\n",
      "ncrna 566 , total =  71150\n",
      "The GECO3 compression ratio for Melitaea_cinxia.MelCinx1.0.ncrna.fa is 12.68%\n",
      "The GZIP compression ratio for Melitaea_cinxia.MelCinx1.0.ncrna.fa is 16.97%\n",
      "melitaea_cinxia completed successfully!\n",
      "ncrna 3332 , total =  2446303\n",
      "The GECO3 compression ratio for Manduca_sexta_gca014839805v1rs.JHU_Msex_v1.0.ncrna.fa is 16.50%\n",
      "The GZIP compression ratio for Manduca_sexta_gca014839805v1rs.JHU_Msex_v1.0.ncrna.fa is 22.42%\n",
      "manduca_sexta_gca014839805v1rs completed successfully!\n",
      "cds 33682 , total =  73993884\n",
      "The GECO3 compression ratio for Lytechinus_variegatus_gca018143015v1.Lvar_3.0.cds.all.fa is 14.00%\n",
      "The GZIP compression ratio for Lytechinus_variegatus_gca018143015v1.Lvar_3.0.cds.all.fa is 16.60%\n",
      "ncrna 4037 , total =  3647433\n",
      "The GECO3 compression ratio for Lytechinus_variegatus_gca018143015v1.Lvar_3.0.ncrna.fa is 13.75%\n",
      "The GZIP compression ratio for Lytechinus_variegatus_gca018143015v1.Lvar_3.0.ncrna.fa is 22.22%\n",
      "lytechinus_variegatus_gca018143015v1 completed successfully!\n",
      "cds 34269 , total =  64226157\n",
      "The GECO3 compression ratio for Macrosteles_quadrilineatus_gca028750875v1rs.UCM_ALF_1.0.cds.all.fa is 12.16%\n",
      "The GZIP compression ratio for Macrosteles_quadrilineatus_gca028750875v1rs.UCM_ALF_1.0.cds.all.fa is 16.73%\n",
      "ncrna 2330 , total =  1141426\n",
      "The GECO3 compression ratio for Macrosteles_quadrilineatus_gca028750875v1rs.UCM_ALF_1.0.ncrna.fa is 12.32%\n",
      "The GZIP compression ratio for Macrosteles_quadrilineatus_gca028750875v1rs.UCM_ALF_1.0.ncrna.fa is 20.66%\n",
      "macrosteles_quadrilineatus_gca028750875v1rs completed successfully!\n",
      "cds 19074 , total =  34047605\n",
      "The GECO3 compression ratio for Melanaphis_sacchari_gca002803265v2rs.SCAv2.0.cds.all.fa is 15.19%\n",
      "The GZIP compression ratio for Melanaphis_sacchari_gca002803265v2rs.SCAv2.0.cds.all.fa is 16.97%\n",
      "ncrna 759 , total =  368392\n",
      "The GECO3 compression ratio for Melanaphis_sacchari_gca002803265v2rs.SCAv2.0.ncrna.fa is 16.93%\n",
      "The GZIP compression ratio for Melanaphis_sacchari_gca002803265v2rs.SCAv2.0.ncrna.fa is 20.91%\n",
      "melanaphis_sacchari_gca002803265v2rs completed successfully!\n",
      "cds 26024 , total =  54251315\n",
      "The GECO3 compression ratio for Megachile_rotundata_gca000220905v1rs.MROT_1.0.cds.all.fa is 10.47%\n",
      "The GZIP compression ratio for Megachile_rotundata_gca000220905v1rs.MROT_1.0.cds.all.fa is 11.33%\n",
      "ncrna 1768 , total =  4227361\n",
      "The GECO3 compression ratio for Megachile_rotundata_gca000220905v1rs.MROT_1.0.ncrna.fa is 9.45%\n",
      "The GZIP compression ratio for Megachile_rotundata_gca000220905v1rs.MROT_1.0.ncrna.fa is 11.76%\n",
      "megachile_rotundata_gca000220905v1rs completed successfully!\n",
      "cds 14471 , total =  19572019\n",
      "The GECO3 compression ratio for Microctonus_aethiopoides_gca030272655v1cm.UoO_Maeth_MO.cds.all.fa is 23.35%\n",
      "The GZIP compression ratio for Microctonus_aethiopoides_gca030272655v1cm.UoO_Maeth_MO.cds.all.fa is 27.69%\n",
      "Directory or file not found for microctonus_aethiopoides_gca030272655v1cm in ncrna folder: 550 Failed to change directory.\n",
      "microctonus_aethiopoides_gca030272655v1cm completed successfully!\n",
      "cds 14172 , total =  19810469\n",
      "The GECO3 compression ratio for Microctonus_aethiopoides_gca030272935v1cm.UoO_Maeth_FR.cds.all.fa is 23.36%\n",
      "The GZIP compression ratio for Microctonus_aethiopoides_gca030272935v1cm.UoO_Maeth_FR.cds.all.fa is 27.67%\n",
      "Directory or file not found for microctonus_aethiopoides_gca030272935v1cm in ncrna folder: 550 Failed to change directory.\n",
      "microctonus_aethiopoides_gca030272935v1cm completed successfully!\n",
      "cds 13959 , total =  22275348\n",
      "The GECO3 compression ratio for Microctonus_aethiopoides_gca030347275v1cm.UoO_Maeth_IR.cds.all.fa is 21.03%\n",
      "The GZIP compression ratio for Microctonus_aethiopoides_gca030347275v1cm.UoO_Maeth_IR.cds.all.fa is 24.36%\n",
      "Directory or file not found for microctonus_aethiopoides_gca030347275v1cm in ncrna folder: 550 Failed to change directory.\n",
      "microctonus_aethiopoides_gca030347275v1cm completed successfully!\n",
      "cds 12921 , total =  19814555\n",
      "The GECO3 compression ratio for Microctonus_hyperodae_gca030347285v1cm.UoO_Mhyp.cds.all.fa is 21.52%\n",
      "The GZIP compression ratio for Microctonus_hyperodae_gca030347285v1cm.UoO_Mhyp.cds.all.fa is 24.91%\n",
      "cds 29593 , total =  44071088\n",
      "The GECO3 compression ratio for Melitaea_cinxia_gca905220565v1.ilMelCinx1.1.cds.all.fa is 13.47%\n",
      "The GZIP compression ratio for Melitaea_cinxia_gca905220565v1.ilMelCinx1.1.cds.all.fa is 15.08%\n",
      "Directory or file not found for microctonus_hyperodae_gca030347285v1cm in ncrna folder: 550 Failed to change directory.\n",
      "microctonus_hyperodae_gca030347285v1cm completed successfully!\n",
      "ncrna 5642 , total =  3227493\n",
      "The GECO3 compression ratio for Melitaea_cinxia_gca905220565v1.ilMelCinx1.1.ncrna.fa is 14.36%\n",
      "The GZIP compression ratio for Melitaea_cinxia_gca905220565v1.ilMelCinx1.1.ncrna.fa is 21.14%\n",
      "melitaea_cinxia_gca905220565v1 completed successfully!\n",
      "cds 16058 , total =  22253965\n",
      "The GECO3 compression ratio for Mnemiopsis_leidyi.MneLei_Aug2011.cds.all.fa is 23.84%\n",
      "The GZIP compression ratio for Mnemiopsis_leidyi.MneLei_Aug2011.cds.all.fa is 27.88%\n",
      "ncrna 239 , total =  24238\n",
      "The GECO3 compression ratio for Mnemiopsis_leidyi.MneLei_Aug2011.ncrna.fa is 12.44%\n",
      "The GZIP compression ratio for Mnemiopsis_leidyi.MneLei_Aug2011.ncrna.fa is 15.60%\n",
      "mnemiopsis_leidyi completed successfully!\n",
      "cds 19550 , total =  37499400\n",
      "The GECO3 compression ratio for Musca_domestica.MdomA1.cds.all.fa is 16.28%\n",
      "The GZIP compression ratio for Musca_domestica.MdomA1.cds.all.fa is 18.59%\n",
      "ncrna 1663 , total =  493444\n",
      "The GECO3 compression ratio for Musca_domestica.MdomA1.ncrna.fa is 17.59%\n",
      "The GZIP compression ratio for Musca_domestica.MdomA1.ncrna.fa is 21.62%\n",
      "musca_domestica completed successfully!\n",
      "cds 19153 , total =  15461420\n",
      "The GECO3 compression ratio for Necator_americanus.N_americanus_v1.cds.all.fa is 23.90%\n",
      "The GZIP compression ratio for Necator_americanus.N_americanus_v1.cds.all.fa is 28.04%\n",
      "Directory or file not found for necator_americanus in ncrna folder: 550 Failed to change directory.\n",
      "necator_americanus completed successfully!\n",
      "cds 18723 , total =  32791069\n",
      "The GECO3 compression ratio for Myopa_tessellatipennis_gca943737955v1.idMyoTess1.1.cds.all.fa is 16.58%\n",
      "The GZIP compression ratio for Myopa_tessellatipennis_gca943737955v1.idMyoTess1.1.cds.all.fa is 18.82%\n",
      "ncrna 1541 , total =  783329\n",
      "The GECO3 compression ratio for Myopa_tessellatipennis_gca943737955v1.idMyoTess1.1.ncrna.fa is 18.63%\n",
      "The GZIP compression ratio for Myopa_tessellatipennis_gca943737955v1.idMyoTess1.1.ncrna.fa is 23.01%\n",
      "myopa_tessellatipennis_gca943737955v1 completed successfully!\n",
      "cds 29165 , total =  60322059\n",
      "The GECO3 compression ratio for Monomorium_pharaonis_gca013373865v2.ASM1337386v2.cds.all.fa is 10.59%\n",
      "The GZIP compression ratio for Monomorium_pharaonis_gca013373865v2.ASM1337386v2.cds.all.fa is 12.69%\n",
      "ncrna 3516 , total =  9386422\n",
      "The GECO3 compression ratio for Monomorium_pharaonis_gca013373865v2.ASM1337386v2.ncrna.fa is 17.98%\n",
      "The GZIP compression ratio for Monomorium_pharaonis_gca013373865v2.ASM1337386v2.ncrna.fa is 22.12%\n",
      "monomorium_pharaonis_gca013373865v2 completed successfully!\n",
      "cds 24780 , total =  24934797\n",
      "The GECO3 compression ratio for Nematostella_vectensis.ASM20922v1.cds.all.fa is 21.29%\n",
      "The GZIP compression ratio for Nematostella_vectensis.ASM20922v1.cds.all.fa is 26.34%\n",
      "ncrna 7915 , total =  803374\n",
      "The GECO3 compression ratio for Nematostella_vectensis.ASM20922v1.ncrna.fa is 3.29%\n",
      "The GZIP compression ratio for Nematostella_vectensis.ASM20922v1.ncrna.fa is 5.48%\n",
      "nematostella_vectensis completed successfully!\n",
      "cds 41567 , total =  81916912\n",
      "The GECO3 compression ratio for Mizuhopecten_yessoensis_gca002113885v2.ASM211388v2.cds.all.fa is 14.12%\n",
      "The GZIP compression ratio for Mizuhopecten_yessoensis_gca002113885v2.ASM211388v2.cds.all.fa is 14.95%\n",
      "ncrna 2865 , total =  2836672\n",
      "The GECO3 compression ratio for Mizuhopecten_yessoensis_gca002113885v2.ASM211388v2.ncrna.fa is 17.07%\n",
      "The GZIP compression ratio for Mizuhopecten_yessoensis_gca002113885v2.ASM211388v2.ncrna.fa is 21.56%\n",
      "mizuhopecten_yessoensis_gca002113885v2 completed successfully!\n",
      "cds 60841 , total =  104650955\n",
      "The GECO3 compression ratio for Mercenaria_mercenaria_gca014805675v2.ASM1480567v1.1.cds.all.fa is 15.46%\n",
      "The GZIP compression ratio for Mercenaria_mercenaria_gca014805675v2.ASM1480567v1.1.cds.all.fa is 16.39%\n",
      "cds 34173 , total =  71230144\n",
      "The GECO3 compression ratio for Nasonia_vitripennis.Nvit_psr_1.1.cds.all.fa is 9.57%\n",
      "The GZIP compression ratio for Nasonia_vitripennis.Nvit_psr_1.1.cds.all.fa is 10.50%\n",
      "ncrna 2015 , total =  2585474\n",
      "The GECO3 compression ratio for Nasonia_vitripennis.Nvit_psr_1.1.ncrna.fa is 14.75%\n",
      "The GZIP compression ratio for Nasonia_vitripennis.Nvit_psr_1.1.ncrna.fa is 19.04%\n",
      "nasonia_vitripennis completed successfully!\n",
      "ncrna 7752 , total =  10272304\n",
      "The GECO3 compression ratio for Mercenaria_mercenaria_gca014805675v2.ASM1480567v1.1.ncrna.fa is 13.87%\n",
      "The GZIP compression ratio for Mercenaria_mercenaria_gca014805675v2.ASM1480567v1.1.ncrna.fa is 19.59%\n",
      "mercenaria_mercenaria_gca014805675v2 completed successfully!\n",
      "cds 28581 , total =  62154110\n",
      "The GECO3 compression ratio for Neodiprion_lecontei_gca021901455v1rs.iyNeoLeco1.1.cds.all.fa is 10.10%\n",
      "The GZIP compression ratio for Neodiprion_lecontei_gca021901455v1rs.iyNeoLeco1.1.cds.all.fa is 11.34%\n",
      "cds 13436 , total =  16537685\n",
      "The GECO3 compression ratio for Onchocerca_volvulus.ASM49940v2.cds.all.fa is 21.85%\n",
      "The GZIP compression ratio for Onchocerca_volvulus.ASM49940v2.cds.all.fa is 25.36%\n",
      "ncrna 458 , total =  48603\n",
      "The GECO3 compression ratio for Onchocerca_volvulus.ASM49940v2.ncrna.fa is 9.71%\n",
      "The GZIP compression ratio for Onchocerca_volvulus.ASM49940v2.ncrna.fa is 12.48%\n",
      "onchocerca_volvulus completed successfully!\n",
      "cds 28705 , total =  63037633\n",
      "The GECO3 compression ratio for Neodiprion_pinetum_gca021155775v1rs.iyNeoPine1.1.cds.all.fa is 10.01%\n",
      "The GZIP compression ratio for Neodiprion_pinetum_gca021155775v1rs.iyNeoPine1.1.cds.all.fa is 11.23%\n",
      "ncrna 3079 , total =  3942292\n",
      "The GECO3 compression ratio for Neodiprion_lecontei_gca021901455v1rs.iyNeoLeco1.1.ncrna.fa is 10.08%\n",
      "The GZIP compression ratio for Neodiprion_lecontei_gca021901455v1rs.iyNeoLeco1.1.ncrna.fa is 12.99%\n",
      "neodiprion_lecontei_gca021901455v1rs completed successfully!\n",
      "ncrna 2746 , total =  3240419\n",
      "The GECO3 compression ratio for Neodiprion_pinetum_gca021155775v1rs.iyNeoPine1.1.ncrna.fa is 12.45%\n",
      "The GZIP compression ratio for Neodiprion_pinetum_gca021155775v1rs.iyNeoPine1.1.ncrna.fa is 15.92%\n",
      "neodiprion_pinetum_gca021155775v1rs completed successfully!\n",
      "cds 32923 , total =  52548995\n",
      "The GECO3 compression ratio for Nilaparvata_lugens_gca014356525v1rs.ASM1435652v1.cds.all.fa is 13.02%\n",
      "The GZIP compression ratio for Nilaparvata_lugens_gca014356525v1rs.ASM1435652v1.cds.all.fa is 15.43%\n",
      "ncrna 3172 , total =  3245592\n",
      "The GECO3 compression ratio for Nilaparvata_lugens_gca014356525v1rs.ASM1435652v1.ncrna.fa is 13.55%\n",
      "The GZIP compression ratio for Nilaparvata_lugens_gca014356525v1rs.ASM1435652v1.ncrna.fa is 20.10%\n",
      "nilaparvata_lugens_gca014356525v1rs completed successfully!\n",
      "cds 29037 , total =  53913719\n",
      "The GECO3 compression ratio for Octopus_bimaculoides_gca001194135v2rs.ASM119413v2.cds.all.fa is 13.54%\n",
      "The GZIP compression ratio for Octopus_bimaculoides_gca001194135v2rs.ASM119413v2.cds.all.fa is 14.44%\n",
      "cds 21668 , total =  41132500\n",
      "The GECO3 compression ratio for Onthophagus_taurus_gca000648695v2.Otau_2.0.cds.all.fa is 14.44%\n",
      "The GZIP compression ratio for Onthophagus_taurus_gca000648695v2.Otau_2.0.cds.all.fa is 16.83%\n",
      "ncrna 4688 , total =  4999808\n",
      "The GECO3 compression ratio for Octopus_bimaculoides_gca001194135v2rs.ASM119413v2.ncrna.fa is 18.01%\n",
      "The GZIP compression ratio for Octopus_bimaculoides_gca001194135v2rs.ASM119413v2.ncrna.fa is 21.96%\n",
      "octopus_bimaculoides_gca001194135v2rs completed successfully!\n",
      "cds 61919 , total =  125925494\n",
      "The GECO3 compression ratio for Mya_arenaria_gca026914265v1rs.ASM2691426v1.cds.all.fa is 13.06%\n",
      "The GZIP compression ratio for Mya_arenaria_gca026914265v1rs.ASM2691426v1.cds.all.fa is 14.62%\n",
      "ncrna 2193 , total =  1871185\n",
      "The GECO3 compression ratio for Onthophagus_taurus_gca000648695v2.Otau_2.0.ncrna.fa is 15.65%\n",
      "The GZIP compression ratio for Onthophagus_taurus_gca000648695v2.Otau_2.0.ncrna.fa is 20.43%\n",
      "onthophagus_taurus_gca000648695v2 completed successfully!\n",
      "cds 16969 , total =  26944737\n",
      "The GECO3 compression ratio for Oppia_nitens_gca028296485v1rs.nextdenovo_longstitch_racon4_pilon3.cds.all.fa is 19.59%\n",
      "The GZIP compression ratio for Oppia_nitens_gca028296485v1rs.nextdenovo_longstitch_racon4_pilon3.cds.all.fa is 23.24%\n",
      "ncrna 366 , total =  124034\n",
      "The GECO3 compression ratio for Oppia_nitens_gca028296485v1rs.nextdenovo_longstitch_racon4_pilon3.ncrna.fa is 14.19%\n",
      "The GZIP compression ratio for Oppia_nitens_gca028296485v1rs.nextdenovo_longstitch_racon4_pilon3.ncrna.fa is 17.61%\n",
      "oppia_nitens_gca028296485v1rs completed successfully!\n",
      "ncrna 8379 , total =  8947261\n",
      "The GECO3 compression ratio for Mya_arenaria_gca026914265v1rs.ASM2691426v1.ncrna.fa is 9.17%\n",
      "The GZIP compression ratio for Mya_arenaria_gca026914265v1rs.ASM2691426v1.ncrna.fa is 18.44%\n",
      "mya_arenaria_gca026914265v1rs completed successfully!\n",
      "cds 36112 , total =  72111929\n",
      "The GECO3 compression ratio for Octopus_sinensis_gca006345805v1.ASM634580v1.cds.all.fa is 12.13%\n",
      "The GZIP compression ratio for Octopus_sinensis_gca006345805v1.ASM634580v1.cds.all.fa is 12.68%\n",
      "ncrna 10088 , total =  4479456\n",
      "The GECO3 compression ratio for Octopus_sinensis_gca006345805v1.ASM634580v1.ncrna.fa is 19.13%\n",
      "The GZIP compression ratio for Octopus_sinensis_gca006345805v1.ASM634580v1.ncrna.fa is 23.63%\n",
      "octopus_sinensis_gca006345805v1 completed successfully!\n",
      "cds 24215 , total =  50585078\n",
      "The GECO3 compression ratio for Ooceraea_biroi_gca003672135v1.Obir_v5.4.cds.all.fa is 11.28%\n",
      "The GZIP compression ratio for Ooceraea_biroi_gca003672135v1.Obir_v5.4.cds.all.fa is 13.05%\n",
      "cds 20247 , total =  25788956\n",
      "The GECO3 compression ratio for Orchesella_cincta.ASM171814v1.cds.all.fa is 23.74%\n",
      "The GZIP compression ratio for Orchesella_cincta.ASM171814v1.cds.all.fa is 27.98%\n",
      "Directory or file not found for orchesella_cincta in ncrna folder: 550 Failed to change directory.\n",
      "orchesella_cincta completed successfully!\n",
      "ncrna 2154 , total =  4699758\n",
      "The GECO3 compression ratio for Ooceraea_biroi_gca003672135v1.Obir_v5.4.ncrna.fa is 18.97%\n",
      "The GZIP compression ratio for Ooceraea_biroi_gca003672135v1.Obir_v5.4.ncrna.fa is 23.00%\n",
      "ooceraea_biroi_gca003672135v1 completed successfully!\n",
      "cds 32587 , total =  49759987\n",
      "The GECO3 compression ratio for Orbicella_faveolata_gca002042975v1.ofav_dov_v1.cds.all.fa is 18.39%\n",
      "The GZIP compression ratio for Orbicella_faveolata_gca002042975v1.ofav_dov_v1.cds.all.fa is 21.50%\n",
      "cds 14862 , total =  25128785\n",
      "The GECO3 compression ratio for Panonychus_citri_gca014898815v1rs.ASM1489881v1.cds.all.fa is 17.11%\n",
      "The GZIP compression ratio for Panonychus_citri_gca014898815v1rs.ASM1489881v1.cds.all.fa is 20.80%\n",
      "ncrna 3673 , total =  4794618\n",
      "The GECO3 compression ratio for Orbicella_faveolata_gca002042975v1.ofav_dov_v1.ncrna.fa is 16.99%\n",
      "The GZIP compression ratio for Orbicella_faveolata_gca002042975v1.ofav_dov_v1.ncrna.fa is 21.00%\n",
      "orbicella_faveolata_gca002042975v1 completed successfully!\n",
      "cds 19696 , total =  41173599\n",
      "The GECO3 compression ratio for Orussus_abietinus_gca000612105v2.Oabi_2.0.cds.all.fa is 12.22%\n",
      "The GZIP compression ratio for Orussus_abietinus_gca000612105v2.Oabi_2.0.cds.all.fa is 13.99%\n",
      "ncrna 704 , total =  712136\n",
      "The GECO3 compression ratio for Orussus_abietinus_gca000612105v2.Oabi_2.0.ncrna.fa is 17.61%\n",
      "The GZIP compression ratio for Orussus_abietinus_gca000612105v2.Oabi_2.0.ncrna.fa is 21.12%\n",
      "orussus_abietinus_gca000612105v2 completed successfully!\n",
      "ncrna 2286 , total =  5604828\n",
      "The GECO3 compression ratio for Panonychus_citri_gca014898815v1rs.ASM1489881v1.ncrna.fa is 8.33%\n",
      "The GZIP compression ratio for Panonychus_citri_gca014898815v1rs.ASM1489881v1.ncrna.fa is 9.31%\n",
      "panonychus_citri_gca014898815v1rs completed successfully!\n",
      "cds 34633 , total =  49659222\n",
      "The GECO3 compression ratio for Owenia_fusiformis_gca903813345v1.Owenia_assembly_annotated.cds.all.fa is 20.63%\n",
      "The GZIP compression ratio for Owenia_fusiformis_gca903813345v1.Owenia_assembly_annotated.cds.all.fa is 24.26%\n",
      "Directory or file not found for owenia_fusiformis_gca903813345v1 in ncrna folder: No file ending with 'ncrna.fa.gz' found in the directory.\n",
      "owenia_fusiformis_gca903813345v1 completed successfully!\n",
      "cds 30867 , total =  47281905\n",
      "The GECO3 compression ratio for Paramacrobiotus_metropolitanus_gca019649055v1rs.Prichtersi_v1.0.cds.all.fa is 16.55%\n",
      "The GZIP compression ratio for Paramacrobiotus_metropolitanus_gca019649055v1rs.Prichtersi_v1.0.cds.all.fa is 18.77%\n",
      "ncrna 2059 , total =  2600103\n",
      "The GECO3 compression ratio for Paramacrobiotus_metropolitanus_gca019649055v1rs.Prichtersi_v1.0.ncrna.fa is 14.67%\n",
      "The GZIP compression ratio for Paramacrobiotus_metropolitanus_gca019649055v1rs.Prichtersi_v1.0.ncrna.fa is 18.37%\n",
      "paramacrobiotus_metropolitanus_gca019649055v1rs completed successfully!\n",
      "cds 28775 , total =  42733272\n",
      "The GECO3 compression ratio for Patella_pellucida_gca917208275v1.xgPatPell1.1.cds.all.fa is 17.46%\n",
      "The GZIP compression ratio for Patella_pellucida_gca917208275v1.xgPatPell1.1.cds.all.fa is 18.76%\n",
      "ncrna 25114 , total =  4717792\n",
      "The GECO3 compression ratio for Patella_pellucida_gca917208275v1.xgPatPell1.1.ncrna.fa is 9.11%\n",
      "The GZIP compression ratio for Patella_pellucida_gca917208275v1.xgPatPell1.1.ncrna.fa is 14.21%\n",
      "patella_pellucida_gca917208275v1 completed successfully!\n",
      "cds 33089 , total =  57825563\n",
      "The GECO3 compression ratio for Parasteatoda_tepidariorum_gca000365465v3.Ptep_3.0.cds.all.fa is 14.71%\n",
      "The GZIP compression ratio for Parasteatoda_tepidariorum_gca000365465v3.Ptep_3.0.cds.all.fa is 15.54%\n",
      "ncrna 4318 , total =  3256104\n",
      "The GECO3 compression ratio for Parasteatoda_tepidariorum_gca000365465v3.Ptep_3.0.ncrna.fa is 17.58%\n",
      "The GZIP compression ratio for Parasteatoda_tepidariorum_gca000365465v3.Ptep_3.0.ncrna.fa is 21.67%\n",
      "parasteatoda_tepidariorum_gca000365465v3 completed successfully!\n",
      "cds 31491 , total =  46161336\n",
      "The GECO3 compression ratio for Patella_vulgata_gca932274485v1.xgPatVulg1.1.cds.all.fa is 17.15%\n",
      "The GZIP compression ratio for Patella_vulgata_gca932274485v1.xgPatVulg1.1.cds.all.fa is 18.25%\n",
      "cds 10788 , total =  16670755\n",
      "The GECO3 compression ratio for Pediculus_humanus.PhumU2.cds.all.fa is 23.24%\n",
      "The GZIP compression ratio for Pediculus_humanus.PhumU2.cds.all.fa is 27.51%\n",
      "ncrna 15792 , total =  4105782\n",
      "The GECO3 compression ratio for Patella_vulgata_gca932274485v1.xgPatVulg1.1.ncrna.fa is 12.60%\n",
      "The GZIP compression ratio for Patella_vulgata_gca932274485v1.xgPatVulg1.1.ncrna.fa is 18.69%\n",
      "patella_vulgata_gca932274485v1 completed successfully!\n",
      "ncrna 927 , total =  123475\n",
      "The GECO3 compression ratio for Pediculus_humanus.PhumU2.ncrna.fa is 7.71%\n",
      "The GZIP compression ratio for Pediculus_humanus.PhumU2.ncrna.fa is 10.02%\n",
      "pediculus_humanus completed successfully!\n",
      "cds 54607 , total =  102967959\n",
      "The GECO3 compression ratio for Ostrea_edulis_gca023158985v1rs.OEROSLIN_1.1.cds.all.fa is 12.54%\n",
      "The GZIP compression ratio for Ostrea_edulis_gca023158985v1rs.OEROSLIN_1.1.cds.all.fa is 13.83%\n",
      "cds 23452 , total =  51449910\n",
      "The GECO3 compression ratio for Pectinophora_gossypiella_gca024362695v1rs.ilPecGoss1.1.cds.all.fa is 12.09%\n",
      "The GZIP compression ratio for Pectinophora_gossypiella_gca024362695v1rs.ilPecGoss1.1.cds.all.fa is 15.61%\n",
      "ncrna 2618 , total =  2191392\n",
      "The GECO3 compression ratio for Pectinophora_gossypiella_gca024362695v1rs.ilPecGoss1.1.ncrna.fa is 9.14%\n",
      "The GZIP compression ratio for Pectinophora_gossypiella_gca024362695v1rs.ilPecGoss1.1.ncrna.fa is 18.77%\n",
      "pectinophora_gossypiella_gca024362695v1rs completed successfully!\n",
      "cds 39918 , total =  72445846\n",
      "The GECO3 compression ratio for Pecten_maximus_gca902652985v1rs.xPecMax1.1.cds.all.fa is 15.11%\n",
      "The GZIP compression ratio for Pecten_maximus_gca902652985v1rs.xPecMax1.1.cds.all.fa is 16.97%\n",
      "cds 35403 , total =  77365328\n",
      "The GECO3 compression ratio for Patiria_miniata_gca015706575v1.Pmin_3.0.cds.all.fa is 13.64%\n",
      "The GZIP compression ratio for Patiria_miniata_gca015706575v1.Pmin_3.0.cds.all.fa is 15.88%\n",
      "ncrna 10947 , total =  16956585\n",
      "The GECO3 compression ratio for Ostrea_edulis_gca023158985v1rs.OEROSLIN_1.1.ncrna.fa is 9.69%\n",
      "The GZIP compression ratio for Ostrea_edulis_gca023158985v1rs.OEROSLIN_1.1.ncrna.fa is 16.60%\n",
      "ostrea_edulis_gca023158985v1rs completed successfully!\n",
      "ncrna 4413 , total =  4010713\n",
      "The GECO3 compression ratio for Patiria_miniata_gca015706575v1.Pmin_3.0.ncrna.fa is 12.35%\n",
      "The GZIP compression ratio for Patiria_miniata_gca015706575v1.Pmin_3.0.ncrna.fa is 20.31%\n",
      "patiria_miniata_gca015706575v1 completed successfully!\n",
      "ncrna 5578 , total =  7467193\n",
      "The GECO3 compression ratio for Pecten_maximus_gca902652985v1rs.xPecMax1.1.ncrna.fa is 10.83%\n",
      "The GZIP compression ratio for Pecten_maximus_gca902652985v1rs.xPecMax1.1.ncrna.fa is 15.36%\n",
      "pecten_maximus_gca902652985v1rs completed successfully!\n",
      "cds 8813 , total =  5818998\n",
      "The GECO3 compression ratio for Phlebotomus_perniciosus_gca918844115v2.pperniciosus_asm_v2.0.cds.all.fa is 17.58%\n",
      "The GZIP compression ratio for Phlebotomus_perniciosus_gca918844115v2.pperniciosus_asm_v2.0.cds.all.fa is 21.24%\n",
      "ncrna 106 , total =  17692\n",
      "The GECO3 compression ratio for Phlebotomus_perniciosus_gca918844115v2.pperniciosus_asm_v2.0.ncrna.fa is 13.88%\n",
      "The GZIP compression ratio for Phlebotomus_perniciosus_gca918844115v2.pperniciosus_asm_v2.0.ncrna.fa is 17.86%\n",
      "phlebotomus_perniciosus_gca918844115v2 completed successfully!\n",
      "cds 11405 , total =  12644617\n",
      "The GECO3 compression ratio for Phlebotomus_papatasi.PpapI1.cds.all.fa is 23.63%\n",
      "The GZIP compression ratio for Phlebotomus_papatasi.PpapI1.cds.all.fa is 27.92%\n",
      "ncrna 444 , total =  44070\n",
      "The GECO3 compression ratio for Phlebotomus_papatasi.PpapI1.ncrna.fa is 17.04%\n",
      "The GZIP compression ratio for Phlebotomus_papatasi.PpapI1.ncrna.fa is 22.08%\n",
      "phlebotomus_papatasi completed successfully!\n",
      "cds 33273 , total =  47733038\n",
      "The GECO3 compression ratio for Penaeus_vannamei_gca003789085v1.ASM378908v1.cds.all.fa is 16.33%\n",
      "The GZIP compression ratio for Penaeus_vannamei_gca003789085v1.ASM378908v1.cds.all.fa is 19.84%\n",
      "cds 34830 , total =  66136475\n",
      "The GECO3 compression ratio for Penaeus_chinensis_gca019202785v2rs.ASM1920278v2.cds.all.fa is 12.96%\n",
      "The GZIP compression ratio for Penaeus_chinensis_gca019202785v2rs.ASM1920278v2.cds.all.fa is 14.24%\n",
      "ncrna 4611 , total =  1755368\n",
      "The GECO3 compression ratio for Penaeus_vannamei_gca003789085v1.ASM378908v1.ncrna.fa is 16.69%\n",
      "The GZIP compression ratio for Penaeus_vannamei_gca003789085v1.ASM378908v1.ncrna.fa is 21.00%\n",
      "penaeus_vannamei_gca003789085v1 completed successfully!\n",
      "ncrna 3672 , total =  1287936\n",
      "The GECO3 compression ratio for Penaeus_chinensis_gca019202785v2rs.ASM1920278v2.ncrna.fa is 17.46%\n",
      "The GZIP compression ratio for Penaeus_chinensis_gca019202785v2rs.ASM1920278v2.ncrna.fa is 21.38%\n",
      "penaeus_chinensis_gca019202785v2rs completed successfully!\n",
      "cds 32900 , total =  51055571\n",
      "The GECO3 compression ratio for Penaeus_monodon_gca015228065v1.NSTDA_Pmon_1.cds.all.fa is 15.38%\n",
      "The GZIP compression ratio for Penaeus_monodon_gca015228065v1.NSTDA_Pmon_1.cds.all.fa is 18.48%\n",
      "ncrna 4491 , total =  1606589\n",
      "The GECO3 compression ratio for Penaeus_monodon_gca015228065v1.NSTDA_Pmon_1.ncrna.fa is 17.65%\n",
      "The GZIP compression ratio for Penaeus_monodon_gca015228065v1.NSTDA_Pmon_1.ncrna.fa is 21.89%\n",
      "penaeus_monodon_gca015228065v1 completed successfully!\n",
      "cds 19128 , total =  33961350\n",
      "The GECO3 compression ratio for Pogonomyrmex_barbatus_gca000187915v1rs.Pbar_UMD_V03.cds.all.fa is 14.58%\n",
      "The GZIP compression ratio for Pogonomyrmex_barbatus_gca000187915v1rs.Pbar_UMD_V03.cds.all.fa is 16.68%\n",
      "ncrna 1391 , total =  1099477\n",
      "The GECO3 compression ratio for Pogonomyrmex_barbatus_gca000187915v1rs.Pbar_UMD_V03.ncrna.fa is 19.34%\n",
      "The GZIP compression ratio for Pogonomyrmex_barbatus_gca000187915v1rs.Pbar_UMD_V03.ncrna.fa is 23.24%\n",
      "pogonomyrmex_barbatus_gca000187915v1rs completed successfully!\n",
      "cds 39224 , total =  83142953\n",
      "The GECO3 compression ratio for Penaeus_japonicus_gca017312705v1.Mj_TUMSAT_v1.0.cds.all.fa is 11.74%\n",
      "The GZIP compression ratio for Penaeus_japonicus_gca017312705v1.Mj_TUMSAT_v1.0.cds.all.fa is 13.06%\n",
      "ncrna 3596 , total =  1396522\n",
      "The GECO3 compression ratio for Penaeus_japonicus_gca017312705v1.Mj_TUMSAT_v1.0.ncrna.fa is 17.08%\n",
      "The GZIP compression ratio for Penaeus_japonicus_gca017312705v1.Mj_TUMSAT_v1.0.ncrna.fa is 21.52%\n",
      "penaeus_japonicus_gca017312705v1 completed successfully!\n",
      "cds 25183 , total =  40912417\n",
      "The GECO3 compression ratio for Pocillopora_damicornis_gca003704095v1.ASM370409v1.cds.all.fa is 20.02%\n",
      "The GZIP compression ratio for Pocillopora_damicornis_gca003704095v1.ASM370409v1.cds.all.fa is 22.64%\n",
      "cds 19297 , total =  42703108\n",
      "The GECO3 compression ratio for Polistes_canadensis_gca001313835v1rs.ASM131383v1.cds.all.fa is 11.92%\n",
      "The GZIP compression ratio for Polistes_canadensis_gca001313835v1rs.ASM131383v1.cds.all.fa is 12.90%\n",
      "ncrna 2153 , total =  1713963\n",
      "The GECO3 compression ratio for Pocillopora_damicornis_gca003704095v1.ASM370409v1.ncrna.fa is 19.80%\n",
      "The GZIP compression ratio for Pocillopora_damicornis_gca003704095v1.ASM370409v1.ncrna.fa is 23.76%\n",
      "pocillopora_damicornis_gca003704095v1 completed successfully!\n",
      "ncrna 993 , total =  971901\n",
      "The GECO3 compression ratio for Polistes_canadensis_gca001313835v1rs.ASM131383v1.ncrna.fa is 17.16%\n",
      "The GZIP compression ratio for Polistes_canadensis_gca001313835v1rs.ASM131383v1.ncrna.fa is 20.91%\n",
      "polistes_canadensis_gca001313835v1rs completed successfully!\n",
      "cds 20855 , total =  46428168\n",
      "The GECO3 compression ratio for Polistes_dominula_gca001465965v1rs.Pdom_r1.2.cds.all.fa is 11.57%\n",
      "The GZIP compression ratio for Polistes_dominula_gca001465965v1rs.Pdom_r1.2.cds.all.fa is 12.45%\n",
      "cds 13055 , total =  14012669\n",
      "The GECO3 compression ratio for Pomphorhynchus_laevis_gca012934845v2gb.ASM1293484v2.cds.all.fa is 19.24%\n",
      "The GZIP compression ratio for Pomphorhynchus_laevis_gca012934845v2gb.ASM1293484v2.cds.all.fa is 24.88%\n",
      "Directory or file not found for pomphorhynchus_laevis_gca012934845v2gb in ncrna folder: 550 Failed to change directory.\n",
      "pomphorhynchus_laevis_gca012934845v2gb completed successfully!\n",
      "ncrna 1319 , total =  1247179\n",
      "The GECO3 compression ratio for Polistes_dominula_gca001465965v1rs.Pdom_r1.2.ncrna.fa is 15.30%\n",
      "The GZIP compression ratio for Polistes_dominula_gca001465965v1rs.Pdom_r1.2.ncrna.fa is 19.13%\n",
      "polistes_dominula_gca001465965v1rs completed successfully!\n",
      "cds 27080 , total =  36179916\n",
      "The GECO3 compression ratio for Pollicipes_pollicipes_gca011947565v2.Ppol_2.cds.all.fa is 16.39%\n",
      "The GZIP compression ratio for Pollicipes_pollicipes_gca011947565v2.Ppol_2.cds.all.fa is 20.97%\n",
      "ncrna 4523 , total =  1847286\n",
      "The GECO3 compression ratio for Pollicipes_pollicipes_gca011947565v2.Ppol_2.ncrna.fa is 16.55%\n",
      "The GZIP compression ratio for Pollicipes_pollicipes_gca011947565v2.Ppol_2.ncrna.fa is 22.82%\n",
      "pollicipes_pollicipes_gca011947565v2 completed successfully!\n",
      "cds 20897 , total =  44442468\n",
      "The GECO3 compression ratio for Polistes_fuscatus_gca010416935v1rs.CU_Pfus_HIC.cds.all.fa is 11.96%\n",
      "The GZIP compression ratio for Polistes_fuscatus_gca010416935v1rs.CU_Pfus_HIC.cds.all.fa is 13.33%\n",
      "ncrna 3199 , total =  2206470\n",
      "The GECO3 compression ratio for Polistes_fuscatus_gca010416935v1rs.CU_Pfus_HIC.ncrna.fa is 16.60%\n",
      "The GZIP compression ratio for Polistes_fuscatus_gca010416935v1rs.CU_Pfus_HIC.ncrna.fa is 21.02%\n",
      "polistes_fuscatus_gca010416935v1rs completed successfully!\n",
      "cds 20692 , total =  29658657\n",
      "The GECO3 compression ratio for Priapulus_caudatus_gca000485595v2.Priapulus_caudatus-5.0.1.cds.all.fa is 17.16%\n",
      "The GZIP compression ratio for Priapulus_caudatus_gca000485595v2.Priapulus_caudatus-5.0.1.cds.all.fa is 19.98%\n",
      "ncrna 2495 , total =  1384185\n",
      "The GECO3 compression ratio for Priapulus_caudatus_gca000485595v2.Priapulus_caudatus-5.0.1.ncrna.fa is 20.03%\n",
      "The GZIP compression ratio for Priapulus_caudatus_gca000485595v2.Priapulus_caudatus-5.0.1.ncrna.fa is 24.31%\n",
      "priapulus_caudatus_gca000485595v2 completed successfully!\n",
      "cds 26462 , total =  35069716\n",
      "The GECO3 compression ratio for Pristionchus_pacificus.El_Paco_fix.cds.all.fa is 21.82%\n",
      "The GZIP compression ratio for Pristionchus_pacificus.El_Paco_fix.cds.all.fa is 26.96%\n",
      "Directory or file not found for pristionchus_pacificus in ncrna folder: 550 Failed to change directory.\n",
      "pristionchus_pacificus completed successfully!\n",
      "cds 40391 , total =  75070715\n",
      "The GECO3 compression ratio for Pomacea_canaliculata_gca003073045v1.ASM307304v1.cds.all.fa is 13.67%\n",
      "The GZIP compression ratio for Pomacea_canaliculata_gca003073045v1.ASM307304v1.cds.all.fa is 14.46%\n",
      "cds 15106 , total =  16361860\n",
      "The GECO3 compression ratio for Rhodnius_prolixus.RproC3.cds.all.fa is 23.21%\n",
      "The GZIP compression ratio for Rhodnius_prolixus.RproC3.cds.all.fa is 27.75%\n",
      "ncrna 623 , total =  79157\n",
      "The GECO3 compression ratio for Rhodnius_prolixus.RproC3.ncrna.fa is 14.49%\n",
      "The GZIP compression ratio for Rhodnius_prolixus.RproC3.ncrna.fa is 18.25%\n",
      "rhodnius_prolixus completed successfully!\n",
      "cds 24211 , total =  36223281\n",
      "The GECO3 compression ratio for Rhipicephalus_microplus_gca013339725v1.ASM1333972v1.cds.all.fa is 16.92%\n",
      "The GZIP compression ratio for Rhipicephalus_microplus_gca013339725v1.ASM1333972v1.cds.all.fa is 21.12%\n",
      "ncrna 3891 , total =  5761164\n",
      "The GECO3 compression ratio for Pomacea_canaliculata_gca003073045v1.ASM307304v1.ncrna.fa is 16.59%\n",
      "The GZIP compression ratio for Pomacea_canaliculata_gca003073045v1.ASM307304v1.ncrna.fa is 20.56%\n",
      "pomacea_canaliculata_gca003073045v1 completed successfully!\n",
      "ncrna 9569 , total =  2849034\n",
      "The GECO3 compression ratio for Rhipicephalus_microplus_gca013339725v1.ASM1333972v1.ncrna.fa is 17.03%\n",
      "The GZIP compression ratio for Rhipicephalus_microplus_gca013339725v1.ASM1333972v1.ncrna.fa is 22.20%\n",
      "rhipicephalus_microplus_gca013339725v1 completed successfully!\n",
      "cds 30758 , total =  50975095\n",
      "The GECO3 compression ratio for Rhagoletis_pomonella_gca013731165v1.Rhpom_1.0.cds.all.fa is 14.23%\n",
      "The GZIP compression ratio for Rhagoletis_pomonella_gca013731165v1.Rhpom_1.0.cds.all.fa is 18.60%\n",
      "ncrna 2931 , total =  2224552\n",
      "The GECO3 compression ratio for Rhagoletis_pomonella_gca013731165v1.Rhpom_1.0.ncrna.fa is 15.56%\n",
      "The GZIP compression ratio for Rhagoletis_pomonella_gca013731165v1.Rhpom_1.0.ncrna.fa is 22.39%\n",
      "rhagoletis_pomonella_gca013731165v1 completed successfully!\n",
      "cds 30609 , total =  46526018\n",
      "The GECO3 compression ratio for Rhipicephalus_sanguineus_gca013339695v2rs.BIME_Rsan_1.4.cds.all.fa is 16.28%\n",
      "The GZIP compression ratio for Rhipicephalus_sanguineus_gca013339695v2rs.BIME_Rsan_1.4.cds.all.fa is 19.75%\n",
      "cds 38519 , total =  78718057\n",
      "The GECO3 compression ratio for Portunus_trituberculatus_gca017591435v1.ASM1759143v1.cds.all.fa is 11.55%\n",
      "The GZIP compression ratio for Portunus_trituberculatus_gca017591435v1.ASM1759143v1.cds.all.fa is 11.70%\n",
      "ncrna 10078 , total =  4913421\n",
      "The GECO3 compression ratio for Rhipicephalus_sanguineus_gca013339695v2rs.BIME_Rsan_1.4.ncrna.fa is 11.63%\n",
      "The GZIP compression ratio for Rhipicephalus_sanguineus_gca013339695v2rs.BIME_Rsan_1.4.ncrna.fa is 18.91%\n",
      "rhipicephalus_sanguineus_gca013339695v2rs completed successfully!\n",
      "ncrna 6538 , total =  5563962\n",
      "The GECO3 compression ratio for Portunus_trituberculatus_gca017591435v1.ASM1759143v1.ncrna.fa is 14.56%\n",
      "The GZIP compression ratio for Portunus_trituberculatus_gca017591435v1.ASM1759143v1.ncrna.fa is 19.23%\n",
      "cds 19503 , total =  35527311\n",
      "The GECO3 compression ratio for Rhopalosiphum_maidis_gca003676215v3.ASM367621v3.cds.all.fa is 14.34%\n",
      "The GZIP compression ratio for Rhopalosiphum_maidis_gca003676215v3.ASM367621v3.cds.all.fa is 16.30%\n",
      "portunus_trituberculatus_gca017591435v1 completed successfully!\n",
      "cds 9174 , total =  15863969\n",
      "The GECO3 compression ratio for Sarcoptes_scabiei.ASM1459567v1.cds.all.fa is 22.84%\n",
      "The GZIP compression ratio for Sarcoptes_scabiei.ASM1459567v1.cds.all.fa is 27.33%\n",
      "Directory or file not found for sarcoptes_scabiei in ncrna folder: 550 Failed to change directory.\n",
      "sarcoptes_scabiei completed successfully!\n",
      "ncrna 1116 , total =  1019888\n",
      "The GECO3 compression ratio for Rhopalosiphum_maidis_gca003676215v3.ASM367621v3.ncrna.fa is 7.14%\n",
      "The GZIP compression ratio for Rhopalosiphum_maidis_gca003676215v3.ASM367621v3.ncrna.fa is 10.27%\n",
      "rhopalosiphum_maidis_gca003676215v3 completed successfully!\n",
      "cds 22111 , total =  32808539\n",
      "The GECO3 compression ratio for Saccoglossus_kowalevskii_gca000003605v1.Skow_1.1.cds.all.fa is 22.23%\n",
      "The GZIP compression ratio for Saccoglossus_kowalevskii_gca000003605v1.Skow_1.1.cds.all.fa is 26.33%\n",
      "cds 45538 , total =  89677812\n",
      "The GECO3 compression ratio for Procambarus_clarkii_gca020424385v2.ASM2042438v2.cds.all.fa is 11.65%\n",
      "The GZIP compression ratio for Procambarus_clarkii_gca020424385v2.ASM2042438v2.cds.all.fa is 12.15%\n",
      "ncrna 10734 , total =  1175689\n",
      "The GECO3 compression ratio for Saccoglossus_kowalevskii_gca000003605v1.Skow_1.1.ncrna.fa is 11.39%\n",
      "The GZIP compression ratio for Saccoglossus_kowalevskii_gca000003605v1.Skow_1.1.ncrna.fa is 15.32%\n",
      "saccoglossus_kowalevskii_gca000003605v1 completed successfully!\n",
      "ncrna 4066 , total =  3101749\n",
      "The GECO3 compression ratio for Procambarus_clarkii_gca020424385v2.ASM2042438v2.ncrna.fa is 14.95%\n",
      "The GZIP compression ratio for Procambarus_clarkii_gca020424385v2.ASM2042438v2.ncrna.fa is 20.23%\n",
      "procambarus_clarkii_gca020424385v2 completed successfully!\n",
      "cds 26125 , total =  43136207\n",
      "The GECO3 compression ratio for Schistocerca_americana_gca021461395v2rs.iqSchAmer2.1.cds.all.fa is 15.67%\n",
      "The GZIP compression ratio for Schistocerca_americana_gca021461395v2rs.iqSchAmer2.1.cds.all.fa is 18.44%\n",
      "cds 26362 , total =  44908157\n",
      "The GECO3 compression ratio for Schistocerca_cancellata_gca023864275v2rs.iqSchCanc2.1.cds.all.fa is 14.98%\n",
      "The GZIP compression ratio for Schistocerca_cancellata_gca023864275v2rs.iqSchCanc2.1.cds.all.fa is 17.32%\n",
      "cds 8934 , total =  17449921\n",
      "The GECO3 compression ratio for Schistosoma_haematobium_gca000699445v2rs.SchHae_2.0.cds.all.fa is 22.24%\n",
      "The GZIP compression ratio for Schistosoma_haematobium_gca000699445v2rs.SchHae_2.0.cds.all.fa is 27.55%\n",
      "Directory or file not found for schistosoma_haematobium_gca000699445v2rs in ncrna folder: 550 Failed to change directory.\n",
      "schistosoma_haematobium_gca000699445v2rs completed successfully!\n",
      "ncrna 56595 , total =  17793534\n",
      "The GECO3 compression ratio for Schistocerca_americana_gca021461395v2rs.iqSchAmer2.1.ncrna.fa is 2.03%\n",
      "The GZIP compression ratio for Schistocerca_americana_gca021461395v2rs.iqSchAmer2.1.ncrna.fa is 3.53%\n",
      "schistocerca_americana_gca021461395v2rs completed successfully!\n",
      "cds 25717 , total =  42649946\n",
      "The GECO3 compression ratio for Schistocerca_piceifrons_gca021461385v2rs.iqSchPice1.1.cds.all.fa is 15.62%\n",
      "The GZIP compression ratio for Schistocerca_piceifrons_gca021461385v2rs.iqSchPice1.1.cds.all.fa is 18.52%\n",
      "cds 14499 , total =  25733850\n",
      "The GECO3 compression ratio for Schistosoma_mansoni.Smansoni_v7.cds.all.fa is 17.78%\n",
      "The GZIP compression ratio for Schistosoma_mansoni.Smansoni_v7.cds.all.fa is 19.95%\n",
      "ncrna 80283 , total =  17533327\n",
      "The GECO3 compression ratio for Schistocerca_cancellata_gca023864275v2rs.iqSchCanc2.1.ncrna.fa is 2.76%\n",
      "The GZIP compression ratio for Schistocerca_cancellata_gca023864275v2rs.iqSchCanc2.1.ncrna.fa is 4.59%\n",
      "schistocerca_cancellata_gca023864275v2rs completed successfully!\n",
      "Directory or file not found for schistosoma_mansoni in ncrna folder: 550 Failed to change directory.\n",
      "schistosoma_mansoni completed successfully!\n",
      "cds 28445 , total =  48842992\n",
      "The GECO3 compression ratio for Schistocerca_nitens_gca023898315v2rs.iqSchNite1.1.cds.all.fa is 14.33%\n",
      "The GZIP compression ratio for Schistocerca_nitens_gca023898315v2rs.iqSchNite1.1.cds.all.fa is 16.53%\n",
      "cds 27654 , total =  47533616\n",
      "The GECO3 compression ratio for Schistocerca_serialis_gca023864345v3rs.iqSchSeri2.2.cds.all.fa is 14.47%\n",
      "The GZIP compression ratio for Schistocerca_serialis_gca023864345v3rs.iqSchSeri2.2.cds.all.fa is 16.85%\n",
      "ncrna 69524 , total =  13781171\n",
      "The GECO3 compression ratio for Schistocerca_piceifrons_gca021461385v2rs.iqSchPice1.1.ncrna.fa is 2.98%\n",
      "The GZIP compression ratio for Schistocerca_piceifrons_gca021461385v2rs.iqSchPice1.1.ncrna.fa is 5.45%\n",
      "schistocerca_piceifrons_gca021461385v2rs completed successfully!\n",
      "ncrna 49216 , total =  9248428\n",
      "The GECO3 compression ratio for Schistocerca_nitens_gca023898315v2rs.iqSchNite1.1.ncrna.fa is 4.40%\n",
      "The GZIP compression ratio for Schistocerca_nitens_gca023898315v2rs.iqSchNite1.1.ncrna.fa is 6.89%\n",
      "schistocerca_nitens_gca023898315v2rs completed successfully!\n",
      "ncrna 51523 , total =  17492152\n",
      "The GECO3 compression ratio for Schistocerca_serialis_gca023864345v3rs.iqSchSeri2.2.ncrna.fa is 2.68%\n",
      "The GZIP compression ratio for Schistocerca_serialis_gca023864345v3rs.iqSchSeri2.2.ncrna.fa is 4.41%\n",
      "schistocerca_serialis_gca023864345v3rs completed successfully!\n",
      "cds 37988 , total =  76811065\n",
      "The GECO3 compression ratio for Schistocerca_gregaria_gca023897955v2rs.iqSchGreg1.2.cds.all.fa is 12.02%\n",
      "The GZIP compression ratio for Schistocerca_gregaria_gca023897955v2rs.iqSchGreg1.2.cds.all.fa is 13.01%\n",
      "cds 21316 , total =  36026849\n",
      "The GECO3 compression ratio for Sipha_flava_gca003268045v1.YSA_version1.cds.all.fa is 14.59%\n",
      "The GZIP compression ratio for Sipha_flava_gca003268045v1.YSA_version1.cds.all.fa is 16.47%\n",
      "ncrna 1203 , total =  1032862\n",
      "The GECO3 compression ratio for Sipha_flava_gca003268045v1.YSA_version1.ncrna.fa is 16.56%\n",
      "The GZIP compression ratio for Sipha_flava_gca003268045v1.YSA_version1.ncrna.fa is 20.50%\n",
      "sipha_flava_gca003268045v1 completed successfully!\n",
      "cds 27135 , total =  23187054\n",
      "The GECO3 compression ratio for Stegodyphus_mimosarum.Stegodyphus_mimosarum_v1.cds.all.fa is 21.44%\n",
      "The GZIP compression ratio for Stegodyphus_mimosarum.Stegodyphus_mimosarum_v1.cds.all.fa is 26.73%\n",
      "ncrna 2036 , total =  250580\n",
      "The GECO3 compression ratio for Stegodyphus_mimosarum.Stegodyphus_mimosarum_v1.ncrna.fa is 11.56%\n",
      "The GZIP compression ratio for Stegodyphus_mimosarum.Stegodyphus_mimosarum_v1.ncrna.fa is 16.43%\n",
      "stegodyphus_mimosarum completed successfully!\n",
      "cds 23498 , total =  42948759\n",
      "The GECO3 compression ratio for Sitophilus_oryzae_gca002938485v2rs.Soryzae_2.0.cds.all.fa is 13.99%\n",
      "The GZIP compression ratio for Sitophilus_oryzae_gca002938485v2rs.Soryzae_2.0.cds.all.fa is 16.64%\n",
      "cds 15008 , total =  21719417\n",
      "The GECO3 compression ratio for Strigamia_maritima.Smar1.cds.all.fa is 23.28%\n",
      "The GZIP compression ratio for Strigamia_maritima.Smar1.cds.all.fa is 27.75%\n",
      "ncrna 3107 , total =  2597146\n",
      "The GECO3 compression ratio for Sitophilus_oryzae_gca002938485v2rs.Soryzae_2.0.ncrna.fa is 9.73%\n",
      "The GZIP compression ratio for Sitophilus_oryzae_gca002938485v2rs.Soryzae_2.0.ncrna.fa is 18.54%\n",
      "sitophilus_oryzae_gca002938485v2rs completed successfully!\n",
      "ncrna 469 , total =  50068\n",
      "The GECO3 compression ratio for Strigamia_maritima.Smar1.ncrna.fa is 13.38%\n",
      "The GZIP compression ratio for Strigamia_maritima.Smar1.ncrna.fa is 17.21%\n",
      "strigamia_maritima completed successfully!\n",
      "ncrna 76299 , total =  24535445\n",
      "The GECO3 compression ratio for Schistocerca_gregaria_gca023897955v2rs.iqSchGreg1.2.ncrna.fa is 3.04%\n",
      "The GZIP compression ratio for Schistocerca_gregaria_gca023897955v2rs.iqSchGreg1.2.ncrna.fa is 4.95%\n",
      "schistocerca_gregaria_gca023897955v2rs completed successfully!\n",
      "cds 29836 , total =  43172530\n",
      "The GECO3 compression ratio for Stegodyphus_dumicola_gca010614865v2rs.ASM1061486v2.cds.all.fa is 16.35%\n",
      "The GZIP compression ratio for Stegodyphus_dumicola_gca010614865v2rs.ASM1061486v2.cds.all.fa is 18.71%\n",
      "ncrna 6252 , total =  4091775\n",
      "The GECO3 compression ratio for Stegodyphus_dumicola_gca010614865v2rs.ASM1061486v2.ncrna.fa is 17.87%\n",
      "The GZIP compression ratio for Stegodyphus_dumicola_gca010614865v2rs.ASM1061486v2.ncrna.fa is 22.40%\n",
      "stegodyphus_dumicola_gca010614865v2rs completed successfully!\n",
      "cds 22501 , total =  47090644\n",
      "The GECO3 compression ratio for Stomoxys_calcitrans.ScalU1.cds.all.fa is 13.95%\n",
      "The GZIP compression ratio for Stomoxys_calcitrans.ScalU1.cds.all.fa is 15.75%\n",
      "cds 12561 , total =  17651124\n",
      "The GECO3 compression ratio for Strongyloides_ratti.S_ratti_ED321_v5_0_4.cds.all.fa is 21.83%\n",
      "The GZIP compression ratio for Strongyloides_ratti.S_ratti_ED321_v5_0_4.cds.all.fa is 26.10%\n",
      "ncrna 3167 , total =  2283816\n",
      "The GECO3 compression ratio for Stomoxys_calcitrans.ScalU1.ncrna.fa is 18.37%\n",
      "The GZIP compression ratio for Stomoxys_calcitrans.ScalU1.ncrna.fa is 22.27%\n",
      "stomoxys_calcitrans completed successfully!\n",
      "cds 30910 , total =  63726679\n",
      "The GECO3 compression ratio for Solenopsis_invicta.UNIL_Sinv_3.0.cds.all.fa is 10.60%\n",
      "The GZIP compression ratio for Solenopsis_invicta.UNIL_Sinv_3.0.cds.all.fa is 12.27%\n",
      "ncrna 509 , total =  48960\n",
      "The GECO3 compression ratio for Strongyloides_ratti.S_ratti_ED321_v5_0_4.ncrna.fa is 14.42%\n",
      "The GZIP compression ratio for Strongyloides_ratti.S_ratti_ED321_v5_0_4.ncrna.fa is 18.15%\n",
      "strongyloides_ratti completed successfully!\n",
      "cds 15020 , total =  10804662\n",
      "The GECO3 compression ratio for Thelohanellus_kitauei.ASM82789v1.cds.all.fa is 22.26%\n",
      "The GZIP compression ratio for Thelohanellus_kitauei.ASM82789v1.cds.all.fa is 27.72%\n",
      "ncrna 89 , total =  8011\n",
      "The GECO3 compression ratio for Thelohanellus_kitauei.ASM82789v1.ncrna.fa is 18.87%\n",
      "The GZIP compression ratio for Thelohanellus_kitauei.ASM82789v1.ncrna.fa is 23.37%\n",
      "thelohanellus_kitauei completed successfully!\n",
      "ncrna 2224 , total =  4396178\n",
      "The GECO3 compression ratio for Solenopsis_invicta.UNIL_Sinv_3.0.ncrna.fa is 17.58%\n",
      "The GZIP compression ratio for Solenopsis_invicta.UNIL_Sinv_3.0.ncrna.fa is 21.98%\n",
      "solenopsis_invicta completed successfully!\n",
      "cds 17671 , total =  20477229\n",
      "The GECO3 compression ratio for Tetranychus_urticae.ASM23943v1.cds.all.fa is 22.32%\n",
      "The GZIP compression ratio for Tetranychus_urticae.ASM23943v1.cds.all.fa is 27.68%\n",
      "ncrna 771 , total =  111548\n",
      "The GECO3 compression ratio for Tetranychus_urticae.ASM23943v1.ncrna.fa is 11.31%\n",
      "The GZIP compression ratio for Tetranychus_urticae.ASM23943v1.ncrna.fa is 14.27%\n",
      "tetranychus_urticae completed successfully!\n",
      "cds 15577 , total =  24287112\n",
      "The GECO3 compression ratio for Tigriopus_californicus_gca007210705.Tcal_SD_v2.1.cds.all.fa is 23.90%\n",
      "The GZIP compression ratio for Tigriopus_californicus_gca007210705.Tcal_SD_v2.1.cds.all.fa is 28.00%\n",
      "ncrna 304 , total =  30193\n",
      "The GECO3 compression ratio for Tigriopus_californicus_gca007210705.Tcal_SD_v2.1.ncrna.fa is 11.85%\n",
      "The GZIP compression ratio for Tigriopus_californicus_gca007210705.Tcal_SD_v2.1.ncrna.fa is 14.63%\n",
      "tigriopus_californicus_gca007210705 completed successfully!\n",
      "cds 18273 , total =  24596754\n",
      "The GECO3 compression ratio for Trialeurodes_vaporariorum_gca011764245.ASM1176424v1.cds.all.fa is 21.21%\n",
      "The GZIP compression ratio for Trialeurodes_vaporariorum_gca011764245.ASM1176424v1.cds.all.fa is 27.95%\n",
      "ncrna 994 , total =  84502\n",
      "The GECO3 compression ratio for Trialeurodes_vaporariorum_gca011764245.ASM1176424v1.ncrna.fa is 7.05%\n",
      "The GZIP compression ratio for Trialeurodes_vaporariorum_gca011764245.ASM1176424v1.ncrna.fa is 9.26%\n",
      "trialeurodes_vaporariorum_gca011764245 completed successfully!\n",
      "cds 18534 , total =  27313633\n",
      "The GECO3 compression ratio for Tribolium_castaneum.Tcas5.2.cds.all.fa is 20.12%\n",
      "The GZIP compression ratio for Tribolium_castaneum.Tcas5.2.cds.all.fa is 23.95%\n",
      "ncrna 460 , total =  47444\n",
      "The GECO3 compression ratio for Tribolium_castaneum.Tcas5.2.ncrna.fa is 8.86%\n",
      "The GZIP compression ratio for Tribolium_castaneum.Tcas5.2.ncrna.fa is 11.22%\n",
      "tribolium_castaneum completed successfully!\n",
      "cds 33252 , total =  54607350\n",
      "The GECO3 compression ratio for Stylophora_pistillata_gca002571385v1.Stylophora_pistillata_v1.cds.all.fa is 18.38%\n",
      "The GZIP compression ratio for Stylophora_pistillata_gca002571385v1.Stylophora_pistillata_v1.cds.all.fa is 20.75%\n",
      "ncrna 3014 , total =  2439388\n",
      "The GECO3 compression ratio for Stylophora_pistillata_gca002571385v1.Stylophora_pistillata_v1.ncrna.fa is 18.62%\n",
      "The GZIP compression ratio for Stylophora_pistillata_gca002571385v1.Stylophora_pistillata_v1.ncrna.fa is 22.56%\n",
      "stylophora_pistillata_gca002571385v1 completed successfully!\n",
      "cds 34029 , total =  57919441\n",
      "The GECO3 compression ratio for Teleopsis_dalmanni.ASM223713v2.cds.all.fa is 13.80%\n",
      "The GZIP compression ratio for Teleopsis_dalmanni.ASM223713v2.cds.all.fa is 19.06%\n",
      "ncrna 2998 , total =  2261973\n",
      "The GECO3 compression ratio for Teleopsis_dalmanni.ASM223713v2.ncrna.fa is 9.59%\n",
      "The GZIP compression ratio for Teleopsis_dalmanni.ASM223713v2.ncrna.fa is 14.77%\n",
      "teleopsis_dalmanni completed successfully!\n",
      "cds 16380 , total =  15650962\n",
      "The GECO3 compression ratio for Trichinella_spiralis.Tspiralis1.cds.all.fa is 21.01%\n",
      "The GZIP compression ratio for Trichinella_spiralis.Tspiralis1.cds.all.fa is 27.78%\n",
      "cds 38437 , total =  79075021\n",
      "The GECO3 compression ratio for Strongylocentrotus_purpuratus.Spur_5.0.cds.all.fa is 14.24%\n",
      "The GZIP compression ratio for Strongylocentrotus_purpuratus.Spur_5.0.cds.all.fa is 17.56%\n",
      "Directory or file not found for trichinella_spiralis in ncrna folder: 550 Failed to change directory.\n",
      "trichinella_spiralis completed successfully!\n",
      "cds 27217 , total =  50694226\n",
      "The GECO3 compression ratio for Thrips_palmi_gca012932325v1rs.TpBJ_2018v1.cds.all.fa is 13.10%\n",
      "The GZIP compression ratio for Thrips_palmi_gca012932325v1rs.TpBJ_2018v1.cds.all.fa is 14.59%\n",
      "cds 20571 , total =  40562002\n",
      "The GECO3 compression ratio for Tribolium_madens_gca015345945v1rs.Tmad_KSU_1.1.cds.all.fa is 13.34%\n",
      "The GZIP compression ratio for Tribolium_madens_gca015345945v1rs.Tmad_KSU_1.1.cds.all.fa is 15.44%\n",
      "ncrna 1378 , total =  790356\n",
      "The GECO3 compression ratio for Thrips_palmi_gca012932325v1rs.TpBJ_2018v1.ncrna.fa is 14.07%\n",
      "The GZIP compression ratio for Thrips_palmi_gca012932325v1rs.TpBJ_2018v1.ncrna.fa is 18.12%\n",
      "thrips_palmi_gca012932325v1rs completed successfully!\n",
      "ncrna 708 , total =  528611\n",
      "The GECO3 compression ratio for Tribolium_madens_gca015345945v1rs.Tmad_KSU_1.1.ncrna.fa is 17.14%\n",
      "The GZIP compression ratio for Tribolium_madens_gca015345945v1rs.Tmad_KSU_1.1.ncrna.fa is 21.71%\n",
      "tribolium_madens_gca015345945v1rs completed successfully!\n",
      "ncrna 6114 , total =  5071032\n",
      "The GECO3 compression ratio for Strongylocentrotus_purpuratus.Spur_5.0.ncrna.fa is 12.17%\n",
      "The GZIP compression ratio for Strongylocentrotus_purpuratus.Spur_5.0.ncrna.fa is 21.87%\n",
      "strongylocentrotus_purpuratus completed successfully!\n",
      "cds 11520 , total =  15689617\n",
      "The GECO3 compression ratio for Trichoplax_adhaerens.ASM15027v1.cds.all.fa is 23.59%\n",
      "The GZIP compression ratio for Trichoplax_adhaerens.ASM15027v1.cds.all.fa is 27.91%\n",
      "ncrna 79 , total =  9249\n",
      "The GECO3 compression ratio for Trichoplax_adhaerens.ASM15027v1.ncrna.fa is 21.03%\n",
      "The GZIP compression ratio for Trichoplax_adhaerens.ASM15027v1.ncrna.fa is 26.21%\n",
      "trichoplax_adhaerens completed successfully!\n",
      "cds 17441 , total =  22858049\n",
      "The GECO3 compression ratio for Trichuris_muris.TMUE3.0.cds.all.fa is 16.96%\n",
      "The GZIP compression ratio for Trichuris_muris.TMUE3.0.cds.all.fa is 23.50%\n",
      "ncrna 759 , total =  163590\n",
      "The GECO3 compression ratio for Trichuris_muris.TMUE3.0.ncrna.fa is 9.44%\n",
      "The GZIP compression ratio for Trichuris_muris.TMUE3.0.ncrna.fa is 13.85%\n",
      "trichuris_muris completed successfully!\n",
      "cds 21161 , total =  42765047\n",
      "The GECO3 compression ratio for Trichogramma_pretiosum_gca000599845v3.Tpre_2.0.cds.all.fa is 13.60%\n",
      "The GZIP compression ratio for Trichogramma_pretiosum_gca000599845v3.Tpre_2.0.cds.all.fa is 15.43%\n",
      "cds 14610 , total =  18203019\n",
      "The GECO3 compression ratio for Zootermopsis_nevadensis.ZooNev1.0.cds.all.fa is 23.73%\n",
      "The GZIP compression ratio for Zootermopsis_nevadensis.ZooNev1.0.cds.all.fa is 27.88%\n",
      "ncrna 559 , total =  221289\n",
      "The GECO3 compression ratio for Trichogramma_pretiosum_gca000599845v3.Tpre_2.0.ncrna.fa is 17.71%\n",
      "The GZIP compression ratio for Trichogramma_pretiosum_gca000599845v3.Tpre_2.0.ncrna.fa is 21.66%\n",
      "trichogramma_pretiosum_gca000599845v3 completed successfully!\n",
      "ncrna 357 , total =  39977\n",
      "The GECO3 compression ratio for Zootermopsis_nevadensis.ZooNev1.0.ncrna.fa is 16.19%\n",
      "The GZIP compression ratio for Zootermopsis_nevadensis.ZooNev1.0.ncrna.fa is 20.18%\n",
      "zootermopsis_nevadensis completed successfully!\n",
      "cds 18993 , total =  31537632\n",
      "The GECO3 compression ratio for Uloborus_diversus_gca026930045v1rs.Udiv.v.3.1.cds.all.fa is 18.73%\n",
      "The GZIP compression ratio for Uloborus_diversus_gca026930045v1rs.Udiv.v.3.1.cds.all.fa is 23.00%\n",
      "ncrna 3507 , total =  1000733\n",
      "The GECO3 compression ratio for Uloborus_diversus_gca026930045v1rs.Udiv.v.3.1.ncrna.fa is 14.11%\n",
      "The GZIP compression ratio for Uloborus_diversus_gca026930045v1rs.Udiv.v.3.1.ncrna.fa is 19.70%\n",
      "uloborus_diversus_gca026930045v1rs completed successfully!\n",
      "cds 17465 , total =  33851601\n",
      "The GECO3 compression ratio for Zerene_cesonia_gca012273895v2rs.Zerene_cesonia_1.1.cds.all.fa is 14.91%\n",
      "The GZIP compression ratio for Zerene_cesonia_gca012273895v2rs.Zerene_cesonia_1.1.cds.all.fa is 17.94%\n",
      "ncrna 850 , total =  306645\n",
      "The GECO3 compression ratio for Zerene_cesonia_gca012273895v2rs.Zerene_cesonia_1.1.ncrna.fa is 19.03%\n",
      "The GZIP compression ratio for Zerene_cesonia_gca012273895v2rs.Zerene_cesonia_1.1.ncrna.fa is 23.35%\n",
      "zerene_cesonia_gca012273895v2rs completed successfully!\n",
      "cds 26214 , total =  44207712\n",
      "The GECO3 compression ratio for Varroa_jacobsoni_gca002532875v1rs.vjacob_1.0.cds.all.fa is 11.40%\n",
      "The GZIP compression ratio for Varroa_jacobsoni_gca002532875v1rs.vjacob_1.0.cds.all.fa is 13.02%\n",
      "cds 23831 , total =  51811776\n",
      "The GECO3 compression ratio for Venturia_canescens_gca019457755v1rs.ASM1945775v1.cds.all.fa is 11.61%\n",
      "The GZIP compression ratio for Venturia_canescens_gca019457755v1rs.ASM1945775v1.cds.all.fa is 13.13%\n",
      "cds 30219 , total =  58990268\n",
      "The GECO3 compression ratio for Varroa_destructor_gca002443255.Vdes_3.0.cds.all.fa is 9.04%\n",
      "The GZIP compression ratio for Varroa_destructor_gca002443255.Vdes_3.0.cds.all.fa is 10.08%\n",
      "ncrna 2667 , total =  4062039\n",
      "The GECO3 compression ratio for Venturia_canescens_gca019457755v1rs.ASM1945775v1.ncrna.fa is 12.98%\n",
      "The GZIP compression ratio for Venturia_canescens_gca019457755v1rs.ASM1945775v1.ncrna.fa is 16.29%\n",
      "venturia_canescens_gca019457755v1rs completed successfully!\n",
      "ncrna 7053 , total =  12445539\n",
      "The GECO3 compression ratio for Varroa_jacobsoni_gca002532875v1rs.vjacob_1.0.ncrna.fa is 16.28%\n",
      "The GZIP compression ratio for Varroa_jacobsoni_gca002532875v1rs.vjacob_1.0.ncrna.fa is 19.08%\n",
      "varroa_jacobsoni_gca002532875v1rs completed successfully!\n",
      "ncrna 3677 , total =  6754022\n",
      "The GECO3 compression ratio for Varroa_destructor_gca002443255.Vdes_3.0.ncrna.fa is 15.07%\n",
      "The GZIP compression ratio for Varroa_destructor_gca002443255.Vdes_3.0.ncrna.fa is 17.91%\n",
      "varroa_destructor_gca002443255 completed successfully!\n",
      "Results saved to compression_ratios_metazoa.csv\n"
     ]
    }
   ],
   "source": [
    "# Usage Metazoa\n",
    "\n",
    "ftp_server = \"ftp.ensemblgenomes.org\" \n",
    "base_ftp_directory = \"/pub/metazoa/current/fasta/\"\n",
    "output_csv_file = \"compression_ratios_metazoa.csv\"\n",
    "\n",
    "process_all_organisms(ftp_server, base_ftp_directory, output_csv_file)\n",
    "\n",
    "print(f\"Results saved to {output_csv_file}\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "cds 7364 , total =  9778297\n",
      "The GECO3 compression ratio for Giardia_lamblia.GL2.cds.all.fa is 22.81%\n",
      "The GZIP compression ratio for Giardia_lamblia.GL2.cds.all.fa is 28.16%\n",
      "ncrna 90 , total =  19645\n",
      "The GECO3 compression ratio for Giardia_lamblia.GL2.ncrna.fa is 13.05%\n",
      "The GZIP compression ratio for Giardia_lamblia.GL2.ncrna.fa is 16.13%\n",
      "giardia_lamblia completed successfully!\n",
      "cds 8113 , total =  10237239\n",
      "The GECO3 compression ratio for Entamoeba_histolytica.JCVI-ESG2-1.0.cds.all.fa is 20.55%\n",
      "The GZIP compression ratio for Entamoeba_histolytica.JCVI-ESG2-1.0.cds.all.fa is 26.22%\n",
      "ncrna 97 , total =  11011\n",
      "The GECO3 compression ratio for Entamoeba_histolytica.JCVI-ESG2-1.0.ncrna.fa is 17.86%\n",
      "The GZIP compression ratio for Entamoeba_histolytica.JCVI-ESG2-1.0.ncrna.fa is 22.12%\n",
      "entamoeba_histolytica completed successfully!\n",
      "cds 14321 , total =  13918876\n",
      "The GECO3 compression ratio for Hyaloperonospora_arabidopsidis.HyaAraEmoy2_2.0.cds.all.fa is 22.23%\n",
      "The GZIP compression ratio for Hyaloperonospora_arabidopsidis.HyaAraEmoy2_2.0.cds.all.fa is 28.11%\n",
      "ncrna 257 , total =  21110\n",
      "The GECO3 compression ratio for Hyaloperonospora_arabidopsidis.HyaAraEmoy2_2.0.ncrna.fa is 13.91%\n",
      "The GZIP compression ratio for Hyaloperonospora_arabidopsidis.HyaAraEmoy2_2.0.ncrna.fa is 17.32%\n",
      "hyaloperonospora_arabidopsidis completed successfully!\n",
      "cds 13804 , total =  18932306\n",
      "The GECO3 compression ratio for Albugo_laibachii.ENA1.cds.all.fa is 19.03%\n",
      "The GZIP compression ratio for Albugo_laibachii.ENA1.cds.all.fa is 24.50%\n",
      "ncrna 171 , total =  24914\n",
      "The GECO3 compression ratio for Albugo_laibachii.ENA1.ncrna.fa is 19.39%\n",
      "The GZIP compression ratio for Albugo_laibachii.ENA1.ncrna.fa is 24.06%\n",
      "albugo_laibachii completed successfully!\n",
      "cds 13233 , total =  21031869\n",
      "The GECO3 compression ratio for Dictyostelium_discoideum.dicty_2.7.cds.all.fa is 19.94%\n",
      "The GZIP compression ratio for Dictyostelium_discoideum.dicty_2.7.cds.all.fa is 24.75%\n",
      "ncrna 488 , total =  40109\n",
      "The GECO3 compression ratio for Dictyostelium_discoideum.dicty_2.7.ncrna.fa is 8.94%\n",
      "The GZIP compression ratio for Dictyostelium_discoideum.dicty_2.7.ncrna.fa is 10.68%\n",
      "dictyostelium_discoideum completed successfully!\n",
      "cds 8315 , total =  15717316\n",
      "The GECO3 compression ratio for Leishmania_major.ASM272v2.cds.all.fa is 22.55%\n",
      "The GZIP compression ratio for Leishmania_major.ASM272v2.cds.all.fa is 27.04%\n",
      "ncrna 1621 , total =  262601\n",
      "The GECO3 compression ratio for Leishmania_major.ASM272v2.ncrna.fa is 6.15%\n",
      "The GZIP compression ratio for Leishmania_major.ASM272v2.ncrna.fa is 9.38%\n",
      "leishmania_major completed successfully!\n",
      "cds 21706 , total =  28817865\n",
      "The GECO3 compression ratio for Bigelowiella_natans.Bigna1.cds.all.fa is 23.61%\n",
      "The GZIP compression ratio for Bigelowiella_natans.Bigna1.cds.all.fa is 27.95%\n",
      "cds 24969 , total =  30671093\n",
      "The GECO3 compression ratio for Guillardia_theta.Guith1.cds.all.fa is 22.74%\n",
      "The GZIP compression ratio for Guillardia_theta.Guith1.cds.all.fa is 27.30%\n",
      "ncrna 608 , total =  114245\n",
      "The GECO3 compression ratio for Bigelowiella_natans.Bigna1.ncrna.fa is 11.50%\n",
      "The GZIP compression ratio for Bigelowiella_natans.Bigna1.ncrna.fa is 16.20%\n",
      "bigelowiella_natans completed successfully!\n",
      "ncrna 279 , total =  56075\n",
      "The GECO3 compression ratio for Guillardia_theta.Guith1.ncrna.fa is 12.71%\n",
      "The GZIP compression ratio for Guillardia_theta.Guith1.ncrna.fa is 16.11%\n",
      "guillardia_theta completed successfully!\n",
      "cds 12178 , total =  18113227\n",
      "The GECO3 compression ratio for Phaeodactylum_tricornutum.ASM15095v2.cds.all.fa is 22.93%\n",
      "The GZIP compression ratio for Phaeodactylum_tricornutum.ASM15095v2.cds.all.fa is 28.06%\n",
      "ncrna 177 , total =  27659\n",
      "The GECO3 compression ratio for Phaeodactylum_tricornutum.ASM15095v2.ncrna.fa is 19.57%\n",
      "The GZIP compression ratio for Phaeodactylum_tricornutum.ASM15095v2.ncrna.fa is 24.38%\n",
      "phaeodactylum_tricornutum completed successfully!\n",
      "cds 10650 , total =  11344308\n",
      "The GECO3 compression ratio for Phytophthora_kernoviae.PhyKer238_432v1.cds.all.fa is 24.06%\n",
      "The GZIP compression ratio for Phytophthora_kernoviae.PhyKer238_432v1.cds.all.fa is 28.05%\n",
      "ncrna 87 , total =  12080\n",
      "The GECO3 compression ratio for Phytophthora_kernoviae.PhyKer238_432v1.ncrna.fa is 22.01%\n",
      "The GZIP compression ratio for Phytophthora_kernoviae.PhyKer238_432v1.ncrna.fa is 27.48%\n",
      "phytophthora_kernoviae completed successfully!\n",
      "cds 17787 , total =  22423430\n",
      "The GECO3 compression ratio for Phytophthora_infestans.ASM14294v1.cds.all.fa is 22.28%\n",
      "The GZIP compression ratio for Phytophthora_infestans.ASM14294v1.cds.all.fa is 28.05%\n",
      "cds 11635 , total =  14242768\n",
      "The GECO3 compression ratio for Phytophthora_lateralis.MPF4_v1.0.cds.all.fa is 23.55%\n",
      "The GZIP compression ratio for Phytophthora_lateralis.MPF4_v1.0.cds.all.fa is 27.87%\n",
      "ncrna 105 , total =  10840\n",
      "The GECO3 compression ratio for Phytophthora_lateralis.MPF4_v1.0.ncrna.fa is 19.54%\n",
      "The GZIP compression ratio for Phytophthora_lateralis.MPF4_v1.0.ncrna.fa is 24.35%\n",
      "phytophthora_lateralis completed successfully!\n",
      "ncrna 1668 , total =  554476\n",
      "The GECO3 compression ratio for Phytophthora_infestans.ASM14294v1.ncrna.fa is 4.42%\n",
      "The GZIP compression ratio for Phytophthora_infestans.ASM14294v1.ncrna.fa is 6.36%\n",
      "phytophthora_infestans completed successfully!\n",
      "cds 38554 , total =  39762434\n",
      "The GECO3 compression ratio for Emiliania_huxleyi.Emiliana_huxleyi_CCMP1516_main_genome_assembly_v1.0.cds.all.fa is 19.16%\n",
      "The GZIP compression ratio for Emiliania_huxleyi.Emiliana_huxleyi_CCMP1516_main_genome_assembly_v1.0.cds.all.fa is 26.23%\n",
      "ncrna 199 , total =  169587\n",
      "The GECO3 compression ratio for Emiliania_huxleyi.Emiliana_huxleyi_CCMP1516_main_genome_assembly_v1.0.ncrna.fa is 7.13%\n",
      "The GZIP compression ratio for Emiliania_huxleyi.Emiliana_huxleyi_CCMP1516_main_genome_assembly_v1.0.ncrna.fa is 10.72%\n",
      "emiliania_huxleyi completed successfully!\n",
      "cds 4903 , total =  10154092\n",
      "The GECO3 compression ratio for Plasmodium_berghei.PBANKA01.cds.all.fa is 21.31%\n",
      "The GZIP compression ratio for Plasmodium_berghei.PBANKA01.cds.all.fa is 25.75%\n",
      "ncrna 141 , total =  29527\n",
      "The GECO3 compression ratio for Plasmodium_berghei.PBANKA01.ncrna.fa is 16.48%\n",
      "The GZIP compression ratio for Plasmodium_berghei.PBANKA01.ncrna.fa is 21.05%\n",
      "plasmodium_berghei completed successfully!\n",
      "cds 15605 , total =  22377332\n",
      "The GECO3 compression ratio for Phytophthora_ramorum.ASM14973v1.cds.all.fa is 21.01%\n",
      "The GZIP compression ratio for Phytophthora_ramorum.ASM14973v1.cds.all.fa is 27.75%\n",
      "ncrna 198 , total =  22127\n",
      "The GECO3 compression ratio for Phytophthora_ramorum.ASM14973v1.ncrna.fa is 12.34%\n",
      "The GZIP compression ratio for Phytophthora_ramorum.ASM14973v1.ncrna.fa is 14.64%\n",
      "phytophthora_ramorum completed successfully!\n",
      "Directory or file not found for protists_alveolata1_collection in cds folder: 550 Failed to change directory.\n",
      "Directory or file not found for protists_alveolata1_collection in ncrna folder: 550 Failed to change directory.\n",
      "protists_alveolata1_collection completed successfully!\n",
      "Directory or file not found for protists_amoebozoa1_collection in cds folder: 550 Failed to change directory.\n",
      "Directory or file not found for protists_amoebozoa1_collection in ncrna folder: 550 Failed to change directory.\n",
      "protists_amoebozoa1_collection completed successfully!\n",
      "Directory or file not found for protists_apusozoa1_collection in cds folder: 550 Failed to change directory.\n",
      "Directory or file not found for protists_apusozoa1_collection in ncrna folder: 550 Failed to change directory.\n",
      "protists_apusozoa1_collection completed successfully!\n",
      "cds 5157 , total =  10444556\n",
      "The GECO3 compression ratio for Plasmodium_chabaudi.PCHAS01.cds.all.fa is 21.61%\n",
      "The GZIP compression ratio for Plasmodium_chabaudi.PCHAS01.cds.all.fa is 26.13%\n",
      "Directory or file not found for protists_choanoflagellida1_collection in cds folder: 550 Failed to change directory.\n",
      "Directory or file not found for protists_choanoflagellida1_collection in ncrna folder: 550 Failed to change directory.\n",
      "protists_choanoflagellida1_collection completed successfully!\n",
      "Directory or file not found for protists_cryptophyta1_collection in cds folder: 550 Failed to change directory.\n",
      "Directory or file not found for protists_cryptophyta1_collection in ncrna folder: 550 Failed to change directory.\n",
      "protists_cryptophyta1_collection completed successfully!\n",
      "ncrna 132 , total =  22618\n",
      "The GECO3 compression ratio for Plasmodium_chabaudi.PCHAS01.ncrna.fa is 21.06%\n",
      "The GZIP compression ratio for Plasmodium_chabaudi.PCHAS01.ncrna.fa is 30.12%\n",
      "plasmodium_chabaudi completed successfully!\n",
      "Directory or file not found for protists_euglenozoa1_collection in cds folder: 550 Failed to change directory.\n",
      "Directory or file not found for protists_fornicata1_collection in cds folder: 550 Failed to change directory.\n",
      "Directory or file not found for protists_euglenozoa1_collection in ncrna folder: 550 Failed to change directory.\n",
      "protists_euglenozoa1_collection completed successfully!\n",
      "Directory or file not found for protists_fornicata1_collection in ncrna folder: 550 Failed to change directory.\n",
      "protists_fornicata1_collection completed successfully!\n",
      "Directory or file not found for protists_heterolobosea1_collection in cds folder: 550 Failed to change directory.\n",
      "Directory or file not found for protists_ichthyosporea1_collection in cds folder: 550 Failed to change directory.\n",
      "Directory or file not found for protists_heterolobosea1_collection in ncrna folder: 550 Failed to change directory.\n",
      "protists_heterolobosea1_collection completed successfully!\n",
      "Directory or file not found for protists_ichthyosporea1_collection in ncrna folder: 550 Failed to change directory.\n",
      "protists_ichthyosporea1_collection completed successfully!\n",
      "Directory or file not found for protists_nucleariidaeandfonticulagroup1_collection in cds folder: 550 Failed to change directory.\n",
      "Directory or file not found for protists_parabasalia1_collection in cds folder: 550 Failed to change directory.\n",
      "Directory or file not found for protists_nucleariidaeandfonticulagroup1_collection in ncrna folder: 550 Failed to change directory.cds 5102 , total =  11131968\n",
      "The GECO3 compression ratio for Plasmodium_knowlesi.ASM635v1.cds.all.fa is 23.29%\n",
      "The GZIP compression ratio for Plasmodium_knowlesi.ASM635v1.cds.all.fa is 27.74%\n",
      "\n",
      "protists_nucleariidaeandfonticulagroup1_collection completed successfully!\n",
      "Directory or file not found for protists_parabasalia1_collection in ncrna folder: 550 Failed to change directory.\n",
      "protists_parabasalia1_collection completed successfully!\n",
      "Directory or file not found for protists_rhizaria1_collection in cds folder: 550 Failed to change directory.\n",
      "Directory or file not found for protists_stramenopiles1_collection in cds folder: 550 Failed to change directory.\n",
      "Directory or file not found for protists_rhizaria1_collection in ncrna folder: 550 Failed to change directory.\n",
      "protists_rhizaria1_collection completed successfully!\n",
      "Directory or file not found for protists_stramenopiles1_collection in ncrna folder: 550 Failed to change directory.\n",
      "protists_stramenopiles1_collection completed successfully!\n",
      "ncrna 108 , total =  50182\n",
      "The GECO3 compression ratio for Plasmodium_knowlesi.ASM635v1.ncrna.fa is 13.99%\n",
      "The GZIP compression ratio for Plasmodium_knowlesi.ASM635v1.ncrna.fa is 21.45%\n",
      "plasmodium_knowlesi completed successfully!\n",
      "cds 5358 , total =  12309897\n",
      "The GECO3 compression ratio for Plasmodium_falciparum.ASM276v2.cds.all.fa is 20.56%\n",
      "The GZIP compression ratio for Plasmodium_falciparum.ASM276v2.cds.all.fa is 25.30%\n",
      "ncrna 252 , total =  139527\n",
      "The GECO3 compression ratio for Plasmodium_falciparum.ASM276v2.ncrna.fa is 12.19%\n",
      "The GZIP compression ratio for Plasmodium_falciparum.ASM276v2.ncrna.fa is 19.91%\n",
      "plasmodium_falciparum completed successfully!\n",
      "cds 5389 , total =  11259778\n",
      "The GECO3 compression ratio for Plasmodium_vivax.ASM241v2.cds.all.fa is 23.31%\n",
      "The GZIP compression ratio for Plasmodium_vivax.ASM241v2.cds.all.fa is 27.61%\n",
      "ncrna 110 , total =  46728\n",
      "The GECO3 compression ratio for Plasmodium_vivax.ASM241v2.ncrna.fa is 17.12%\n",
      "The GZIP compression ratio for Plasmodium_vivax.ASM241v2.ncrna.fa is 21.69%\n",
      "plasmodium_vivax completed successfully!\n",
      "cds 26489 , total =  30130606\n",
      "The GECO3 compression ratio for Phytophthora_sojae.P_sojae_V3_0.cds.all.fa is 21.17%\n",
      "The GZIP compression ratio for Phytophthora_sojae.P_sojae_V3_0.cds.all.fa is 26.48%\n",
      "cds 28117 , total =  35835288\n",
      "The GECO3 compression ratio for Phytophthora_parasitica.Phyt_para_P1569_V1.cds.all.fa is 19.08%\n",
      "The GZIP compression ratio for Phytophthora_parasitica.Phyt_para_P1569_V1.cds.all.fa is 22.21%\n",
      "ncrna 1631 , total =  119980\n",
      "The GECO3 compression ratio for Phytophthora_sojae.P_sojae_V3_0.ncrna.fa is 1.85%\n",
      "The GZIP compression ratio for Phytophthora_sojae.P_sojae_V3_0.ncrna.fa is 2.72%\n",
      "phytophthora_sojae completed successfully!\n",
      "ncrna 106 , total =  11526\n",
      "The GECO3 compression ratio for Phytophthora_parasitica.Phyt_para_P1569_V1.ncrna.fa is 13.46%\n",
      "The GZIP compression ratio for Phytophthora_parasitica.Phyt_para_P1569_V1.ncrna.fa is 17.05%\n",
      "phytophthora_parasitica completed successfully!\n",
      "cds 39642 , total =  53860110\n",
      "The GECO3 compression ratio for Paramecium_tetraurelia.ASM16542v1.cds.all.fa is 22.63%\n",
      "The GZIP compression ratio for Paramecium_tetraurelia.ASM16542v1.cds.all.fa is 27.06%\n",
      "ncrna 262 , total =  21969\n",
      "The GECO3 compression ratio for Paramecium_tetraurelia.ASM16542v1.ncrna.fa is 10.51%\n",
      "The GZIP compression ratio for Paramecium_tetraurelia.ASM16542v1.ncrna.fa is 12.30%\n",
      "paramecium_tetraurelia completed successfully!\n",
      "cds 12312 , total =  18096718\n",
      "The GECO3 compression ratio for Pythium_aphanidermatum.pag1_scaffolds_v1.cds.all.fa is 23.93%\n",
      "The GZIP compression ratio for Pythium_aphanidermatum.pag1_scaffolds_v1.cds.all.fa is 27.97%\n",
      "ncrna 52 , total =  11660\n",
      "The GECO3 compression ratio for Pythium_aphanidermatum.pag1_scaffolds_v1.ncrna.fa is 21.90%\n",
      "The GZIP compression ratio for Pythium_aphanidermatum.pag1_scaffolds_v1.ncrna.fa is 27.32%\n",
      "pythium_aphanidermatum completed successfully!\n",
      "cds 12039 , total =  19122633\n",
      "The GECO3 compression ratio for Pseudonitzschia_multistriata.ASM90066040v1.cds.all.fa is 23.29%\n",
      "The GZIP compression ratio for Pseudonitzschia_multistriata.ASM90066040v1.cds.all.fa is 27.70%\n",
      "Directory or file not found for pseudonitzschia_multistriata in ncrna folder: 550 Failed to change directory.\n",
      "pseudonitzschia_multistriata completed successfully!\n",
      "cds 13805 , total =  18468799\n",
      "The GECO3 compression ratio for Pythium_arrhenomanes.par_scaffolds_v1.cds.all.fa is 23.30%\n",
      "The GZIP compression ratio for Pythium_arrhenomanes.par_scaffolds_v1.cds.all.fa is 27.52%\n",
      "ncrna 50 , total =  4700\n",
      "The GECO3 compression ratio for Pythium_arrhenomanes.par_scaffolds_v1.ncrna.fa is 23.38%\n",
      "The GZIP compression ratio for Pythium_arrhenomanes.par_scaffolds_v1.ncrna.fa is 28.98%\n",
      "pythium_arrhenomanes completed successfully!\n",
      "cds 13805 , total =  20628453\n",
      "The GECO3 compression ratio for Pythium_irregulare.pir_scaffolds_v1.cds.all.fa is 23.37%\n",
      "The GZIP compression ratio for Pythium_irregulare.pir_scaffolds_v1.cds.all.fa is 27.60%\n",
      "ncrna 59 , total =  7263\n",
      "The GECO3 compression ratio for Pythium_irregulare.pir_scaffolds_v1.ncrna.fa is 17.13%\n",
      "The GZIP compression ratio for Pythium_irregulare.pir_scaffolds_v1.ncrna.fa is 21.09%\n",
      "pythium_irregulare completed successfully!\n",
      "cds 15249 , total =  20382747\n",
      "The GECO3 compression ratio for Pythium_iwayamai.piw_scaffolds_v1.cds.all.fa is 22.35%\n",
      "The GZIP compression ratio for Pythium_iwayamai.piw_scaffolds_v1.cds.all.fa is 27.43%\n",
      "ncrna 87 , total =  9328\n",
      "The GECO3 compression ratio for Pythium_iwayamai.piw_scaffolds_v1.ncrna.fa is 20.69%\n",
      "The GZIP compression ratio for Pythium_iwayamai.piw_scaffolds_v1.ncrna.fa is 25.80%\n",
      "pythium_iwayamai completed successfully!\n",
      "cds 11958 , total =  18819600\n",
      "The GECO3 compression ratio for Pythium_vexans.pve_scaffolds_v1.cds.all.fa is 22.95%\n",
      "The GZIP compression ratio for Pythium_vexans.pve_scaffolds_v1.cds.all.fa is 27.38%\n",
      "cds 15322 , total =  19899935\n",
      "The GECO3 compression ratio for Pythium_ultimum.pug.cds.all.fa is 22.87%\n",
      "The GZIP compression ratio for Pythium_ultimum.pug.cds.all.fa is 27.75%\n",
      "ncrna 536 , total =  49695\n",
      "The GECO3 compression ratio for Pythium_ultimum.pug.ncrna.fa is 8.91%\n",
      "The GZIP compression ratio for Pythium_ultimum.pug.ncrna.fa is 10.34%\n",
      "pythium_ultimum completed successfully!\n",
      "ncrna 34 , total =  5490\n",
      "The GECO3 compression ratio for Pythium_vexans.pve_scaffolds_v1.ncrna.fa is 25.56%\n",
      "The GZIP compression ratio for Pythium_vexans.pve_scaffolds_v1.ncrna.fa is 31.71%\n",
      "pythium_vexans completed successfully!\n",
      "cds 8747 , total =  13171678\n",
      "The GECO3 compression ratio for Trypanosoma_brucei.TryBru_Apr2005_chr11.cds.all.fa is 22.96%\n",
      "The GZIP compression ratio for Trypanosoma_brucei.TryBru_Apr2005_chr11.cds.all.fa is 27.75%\n",
      "ncrna 683 , total =  211130\n",
      "The GECO3 compression ratio for Trypanosoma_brucei.TryBru_Apr2005_chr11.ncrna.fa is 4.79%\n",
      "The GZIP compression ratio for Trypanosoma_brucei.TryBru_Apr2005_chr11.ncrna.fa is 6.55%\n",
      "trypanosoma_brucei completed successfully!\n",
      "cds 11673 , total =  17491542\n",
      "The GECO3 compression ratio for Thalassiosira_pseudonana.ASM14940v2.cds.all.fa is 23.50%\n",
      "The GZIP compression ratio for Thalassiosira_pseudonana.ASM14940v2.cds.all.fa is 27.72%\n",
      "ncrna 98 , total =  8090\n",
      "The GECO3 compression ratio for Thalassiosira_pseudonana.ASM14940v2.ncrna.fa is 18.20%\n",
      "The GZIP compression ratio for Thalassiosira_pseudonana.ASM14940v2.ncrna.fa is 22.51%\n",
      "thalassiosira_pseudonana completed successfully!\n",
      "cds 8303 , total =  20013826\n",
      "The GECO3 compression ratio for Toxoplasma_gondii.TGA4.cds.all.fa is 23.42%\n",
      "The GZIP compression ratio for Toxoplasma_gondii.TGA4.cds.all.fa is 27.68%\n",
      "ncrna 598 , total =  373401\n",
      "The GECO3 compression ratio for Toxoplasma_gondii.TGA4.ncrna.fa is 2.04%\n",
      "The GZIP compression ratio for Toxoplasma_gondii.TGA4.ncrna.fa is 4.90%\n",
      "toxoplasma_gondii completed successfully!\n",
      "cds 24725 , total =  48191182\n",
      "The GECO3 compression ratio for Tetrahymena_thermophila.JCVI-TTA1-2.2.cds.all.fa is 22.19%\n",
      "The GZIP compression ratio for Tetrahymena_thermophila.JCVI-TTA1-2.2.cds.all.fa is 26.38%\n",
      "ncrna 927 , total =  80119\n",
      "The GECO3 compression ratio for Tetrahymena_thermophila.JCVI-TTA1-2.2.ncrna.fa is 5.30%\n",
      "The GZIP compression ratio for Tetrahymena_thermophila.JCVI-TTA1-2.2.ncrna.fa is 6.53%\n",
      "tetrahymena_thermophila completed successfully!\n",
      "Results saved to compression_ratios_protist.csv\n"
     ]
    }
   ],
   "source": [
    "# Usage Protist\n",
    "\n",
    "ftp_server = \"ftp.ensemblgenomes.org\" \n",
    "base_ftp_directory = \"/pub/protists/current/fasta/\"\n",
    "output_csv_file = \"compression_ratios_protist.csv\"\n",
    "\n",
    "process_all_organisms(ftp_server, base_ftp_directory, output_csv_file)\n",
    "\n",
    "print(f\"Results saved to {output_csv_file}\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.9.18"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
